// @ark-ui/react/dist/components/file-upload/use-file-upload.js
"use client";
import { useEnvironmentContext } from "../../providers/environment/use-environment-context.js";
import { useLocaleContext } from "../../providers/locale/use-locale-context.js";
import { useFieldContext } from "../field/use-field-context.js";
import * as fileUpload from "@zag-js/file-upload";
import { useId } from "react";
import { normalizeProps, useMachine } from "@zag-js/react";
//#region src/components/file-upload/use-file-upload.ts
var useFileUpload = (props) => {
	const id = useId();
	const { getRootNode } = useEnvironmentContext();
	const { dir, locale } = useLocaleContext();
	const field = useFieldContext();
	const machineProps = {
		id,
		ids: {
			label: field?.ids.label,
			hiddenInput: field?.ids.control
		},
		dir,
		disabled: field?.disabled,
		locale,
		required: field?.required,
		invalid: field?.invalid,
		getRootNode,
		...props
	};
	const service = useMachine(fileUpload.machine, machineProps);
	return fileUpload.connect(service, normalizeProps);
};
//#endregion
export { useFileUpload };
