// @zag-js/drawer/dist/drawer.stack.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/drawer.stack.ts
var drawer_stack_exports = {};
__export(drawer_stack_exports, {
  connectStack: () => connectStack,
  createStack: () => createStack
});
module.exports = __toCommonJS(drawer_stack_exports);
var import_utils = require("@zag-js/utils");
function resolveSnapshot(entries) {
  let openCount = 0;
  let frontmostHeight = 0;
  let swipeProgress = 0;
  let frontmostOrder = -1;
  entries.forEach((entry) => {
    if (!entry.open) return;
    openCount += 1;
    if (entry.order < frontmostOrder) return;
    frontmostOrder = entry.order;
    frontmostHeight = entry.height > 0 ? entry.height : 0;
    swipeProgress = entry.swiping ? (0, import_utils.clampValue)(entry.swipeProgress, 0, 1) : 0;
  });
  return {
    active: openCount > 0,
    openCount,
    swipeProgress,
    frontmostHeight
  };
}
function createStack() {
  const entries = /* @__PURE__ */ new Map();
  const listeners = /* @__PURE__ */ new Set();
  let order = 0;
  let pendingNotify = false;
  let snapshot = Object.freeze({
    active: false,
    openCount: 0,
    swipeProgress: 0,
    frontmostHeight: 0
  });
  const scheduleNotify = () => {
    if (pendingNotify) return;
    pendingNotify = true;
    queueMicrotask(() => {
      pendingNotify = false;
      listeners.forEach((listener) => listener());
    });
  };
  const notify = () => {
    const nextSnapshot = resolveSnapshot(entries);
    if (snapshot.active === nextSnapshot.active && snapshot.openCount === nextSnapshot.openCount && snapshot.swipeProgress === nextSnapshot.swipeProgress && snapshot.frontmostHeight === nextSnapshot.frontmostHeight) {
      return;
    }
    snapshot = Object.freeze(nextSnapshot);
    scheduleNotify();
  };
  const ensureEntry = (id) => {
    let entry = entries.get(id);
    if (!entry) {
      entry = { order: ++order, open: false, height: 0, swiping: false, swipeProgress: 0 };
      entries.set(id, entry);
    }
    return entry;
  };
  return {
    getSnapshot() {
      return snapshot;
    },
    subscribe(listener) {
      listeners.add(listener);
      return () => {
        listeners.delete(listener);
      };
    },
    register(id) {
      ensureEntry(id);
      notify();
    },
    unregister(id) {
      if (!entries.delete(id)) return;
      notify();
    },
    setOpen(id, open) {
      const entry = ensureEntry(id);
      if (entry.open === open) return;
      entry.open = open;
      if (!open) {
        entry.swiping = false;
        entry.swipeProgress = 0;
      }
      notify();
    },
    setHeight(id, height) {
      const entry = ensureEntry(id);
      const nextHeight = Number.isFinite(height) && height > 0 ? height : 0;
      if (entry.height === nextHeight) return;
      entry.height = nextHeight;
      notify();
    },
    setSwipe(id, swiping, progress) {
      const entry = ensureEntry(id);
      const nextProgress = swiping ? (0, import_utils.clampValue)(Number.isFinite(progress) ? progress : 0, 0, 1) : 0;
      if (entry.swiping === swiping && entry.swipeProgress === nextProgress) return;
      entry.swiping = swiping;
      entry.swipeProgress = nextProgress;
      notify();
    }
  };
}
function connectStack(snapshot, normalize) {
  const getIndentProps = () => {
    return normalize.element({
      "data-active": snapshot.active ? "" : void 0,
      "data-inactive": snapshot.active ? void 0 : "",
      style: {
        "--drawer-swipe-progress": `${(0, import_utils.clampValue)(snapshot.swipeProgress, 0, 1)}`,
        "--drawer-frontmost-height": snapshot.frontmostHeight > 0 ? `${snapshot.frontmostHeight}px` : void 0
      }
    });
  };
  return {
    getIndentProps,
    getIndentBackgroundProps: getIndentProps
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  connectStack,
  createStack
});
