// @zag-js/drawer/dist/drawer.connect.mjs
import "./chunk-QZ7TP4HQ.mjs";

// src/drawer.connect.ts
import { dataAttr, getEventPoint, isLeftClick } from "@zag-js/dom-query";
import { clampValue, compact, toPx } from "@zag-js/utils";
import { parts } from "./drawer.anatomy.mjs";
import * as dom from "./drawer.dom.mjs";
import { oppositeSwipeDirection, resolveSwipeDirection } from "./utils/drawer-session.mjs";
import { isNegativeSwipeDirection, isVerticalSwipeDirection } from "./utils/session.mjs";
var SWIPE_OPEN_HIDDEN_OFFSET = 9999;
function getSwipeOpenOffset(swipingOpen, dragOffset, contentSize) {
  if (!swipingOpen || dragOffset !== null) return null;
  return contentSize ?? SWIPE_OPEN_HIDDEN_OFFSET;
}
function connect(service, normalize) {
  const { state, send, context, scope, prop, refs } = service;
  const open = state.hasTag("open");
  const closed = state.matches("closed");
  const closing = state.matches("closing");
  const swipingOpen = state.matches("swiping-open");
  const dragOffset = context.get("dragOffset");
  const dragging = dragOffset !== null;
  const triggerValue = context.get("triggerValue");
  const snapPoint = context.get("snapPoint");
  const swipeDirection = prop("swipeDirection");
  const physicalDirection = resolveSwipeDirection(swipeDirection, prop("dir"));
  const contentSize = context.get("contentSize");
  const swipeStrength = context.get("swipeStrength");
  const resolvedActiveSnapPoint = context.get("resolvedActiveSnapPoint");
  const snapPointOffset = resolvedActiveSnapPoint?.offset ?? 0;
  const nestedMetrics = context.get("nestedMetrics");
  const swipeOpenOffset = getSwipeOpenOffset(swipingOpen, dragOffset, contentSize);
  const currentOffset = swipeOpenOffset ?? dragOffset ?? snapPointOffset;
  const signedSnapPointOffset = isNegativeSwipeDirection(physicalDirection) ? -snapPointOffset : snapPointOffset;
  const isActivelySwiping = dragging || swipingOpen;
  const swipeMovement = dragging || swipingOpen ? currentOffset - snapPointOffset : 0;
  const signedMovement = isNegativeSwipeDirection(physicalDirection) ? -swipeMovement : swipeMovement;
  const swipeProgress = isActivelySwiping && contentSize && contentSize > 0 ? clampValue(Math.abs(signedMovement) / contentSize, 0, 1) : swipingOpen ? 1 : 0;
  const signedCurrentOffset = isNegativeSwipeDirection(physicalDirection) ? -currentOffset : currentOffset;
  const translateX = isVerticalSwipeDirection(physicalDirection) ? 0 : signedCurrentOffset;
  const translateY = isVerticalSwipeDirection(physicalDirection) ? signedCurrentOffset : 0;
  function onContentPointerDown(event) {
    refs.get("swipeSession").contentPointerDown({
      event,
      getDoc: () => scope.getDoc(),
      getContentEl: () => dom.getContentEl(scope),
      getWin: () => scope.getWin(),
      swipeDirection: physicalDirection,
      canCommit: () => state.hasTag("open") && !state.matches("closing"),
      onCommit(point) {
        send({ type: "POINTER_DOWN", point });
      }
    });
  }
  function onGrabberPointerDown(event) {
    refs.get("swipeSession").grabberPointerDown({
      event,
      point: getEventPoint(event),
      canCommit: () => state.hasTag("open") && !state.matches("closing"),
      onCommit(point) {
        send({ type: "POINTER_DOWN", point });
      }
    });
  }
  return {
    open,
    dragging,
    setOpen(nextOpen) {
      const open2 = state.hasTag("open");
      if (open2 === nextOpen) return;
      send({ type: nextOpen ? "OPEN" : "CLOSE" });
    },
    snapPoints: prop("snapPoints"),
    swipeDirection,
    snapPoint,
    setSnapPoint(snapPoint2) {
      const currentSnapPoint = context.get("snapPoint");
      if (currentSnapPoint === snapPoint2) return;
      send({ type: "SNAP_POINT.SET", snapPoint: snapPoint2 });
    },
    getOpenPercentage() {
      if (!open || !contentSize) return 0;
      return clampValue(1 - currentOffset / contentSize, 0, 1);
    },
    getSnapPointIndex() {
      if (snapPoint === null) return -1;
      return prop("snapPoints").indexOf(snapPoint);
    },
    getContentSize() {
      return contentSize;
    },
    getPositionerProps() {
      return normalize.element({
        ...parts.positioner.attrs,
        id: dom.getPositionerId(scope),
        dir: prop("dir"),
        hidden: closed,
        "data-state": open ? "open" : "closed",
        "data-swipe-direction": physicalDirection,
        style: compact({
          pointerEvents: closing || !prop("modal") ? "none" : void 0
        })
      });
    },
    getContentProps(props = { draggable: true }) {
      const movementX = isVerticalSwipeDirection(physicalDirection) ? 0 : signedMovement;
      const movementY = isVerticalSwipeDirection(physicalDirection) ? signedMovement : 0;
      const rendered = context.get("rendered");
      return normalize.element({
        ...parts.content.attrs,
        dir: prop("dir"),
        id: dom.getContentId(scope),
        tabIndex: -1,
        role: prop("role"),
        "aria-modal": prop("modal"),
        "aria-labelledby": rendered.title ? dom.getTitleId(scope) : void 0,
        "aria-describedby": rendered.description ? dom.getDescriptionId(scope) : void 0,
        hidden: !open,
        "data-state": open ? "open" : "closed",
        "data-expanded": resolvedActiveSnapPoint?.offset === 0 ? "" : void 0,
        "data-swipe-direction": physicalDirection,
        "data-swiping": dragging || swipingOpen ? "" : void 0,
        "data-dragging": dragging ? "" : void 0,
        "data-nested-drawer-open": nestedMetrics.open ? "" : void 0,
        "data-nested-drawer-swiping": nestedMetrics.swiping ? "" : void 0,
        style: compact({
          pointerEvents: prop("modal") ? void 0 : "auto",
          visibility: swipingOpen && dragOffset === null ? "hidden" : void 0,
          transform: "translate3d(var(--drawer-translate-x, 0px), var(--drawer-translate-y, 0px), 0)",
          transitionDuration: dragging || swipingOpen ? "0s" : void 0,
          "--drawer-translate": toPx(translateY),
          "--drawer-translate-x": toPx(translateX),
          "--drawer-translate-y": toPx(translateY),
          "--drawer-snap-point-offset-x": isVerticalSwipeDirection(physicalDirection) ? "0px" : toPx(signedSnapPointOffset),
          "--drawer-snap-point-offset-y": isVerticalSwipeDirection(physicalDirection) ? toPx(signedSnapPointOffset) : "0px",
          "--drawer-swipe-movement-x": toPx(movementX),
          "--drawer-swipe-movement-y": toPx(movementY),
          "--drawer-swipe-strength": `${swipeStrength}`,
          "--nested-drawers": `${nestedMetrics.count}`,
          "--drawer-height": nestedMetrics.height > 0 ? toPx(nestedMetrics.height) : void 0,
          "--drawer-frontmost-height": nestedMetrics.frontmostHeight > 0 ? toPx(nestedMetrics.frontmostHeight) : void 0,
          willChange: "transform"
        }),
        onPointerDown(event) {
          if (!props.draggable) return;
          onContentPointerDown(event);
        }
      });
    },
    getTitleProps() {
      return normalize.element({
        ...parts.title.attrs,
        id: dom.getTitleId(scope),
        dir: prop("dir")
      });
    },
    getDescriptionProps() {
      return normalize.element({
        ...parts.description.attrs,
        id: dom.getDescriptionId(scope),
        dir: prop("dir")
      });
    },
    triggerValue,
    setTriggerValue(value) {
      send({ type: "OPEN", value: value ?? void 0 });
    },
    getTriggerProps(props = {}) {
      const { value } = props;
      const current = value == null ? false : triggerValue === value;
      return normalize.button({
        ...parts.trigger.attrs,
        dir: prop("dir"),
        id: dom.getTriggerId(scope, value),
        "data-ownedby": scope.id,
        "data-value": value,
        "aria-haspopup": "dialog",
        type: "button",
        "aria-expanded": value == null ? open : open && current,
        "data-state": open ? "open" : "closed",
        "aria-controls": dom.getContentId(scope),
        "data-current": dataAttr(current),
        onClick(event) {
          if (event.defaultPrevented) return;
          const shouldSwitch = open && value != null && !current;
          send({ type: shouldSwitch ? "TRIGGER_VALUE.SET" : open ? "CLOSE" : "OPEN", value });
        }
      });
    },
    getBackdropProps() {
      return normalize.element({
        ...parts.backdrop.attrs,
        id: dom.getBackdropId(scope),
        hidden: !open || swipingOpen && dragOffset === null,
        "data-state": open ? "open" : "closed",
        "data-swiping": dragging || swipingOpen ? "" : void 0,
        style: {
          willChange: "opacity",
          pointerEvents: closing ? "none" : void 0,
          "--drawer-swipe-progress": `${swipeProgress}`,
          "--drawer-swipe-strength": `${swipeStrength}`
        }
      });
    },
    getGrabberProps() {
      return normalize.element({
        ...parts.grabber.attrs,
        id: dom.getGrabberId(scope),
        onPointerDown(event) {
          onGrabberPointerDown(event);
        },
        style: {
          touchAction: "none"
        }
      });
    },
    getGrabberIndicatorProps() {
      return normalize.element({
        ...parts.grabberIndicator.attrs,
        id: dom.getGrabberIndicatorId(scope)
      });
    },
    getCloseTriggerProps() {
      return normalize.button({
        ...parts.closeTrigger.attrs,
        id: dom.getCloseTriggerId(scope),
        onClick() {
          send({ type: "CLOSE" });
        }
      });
    },
    getSwipeAreaProps(props = {}) {
      const disabled = props.disabled ?? false;
      const openDirection = props.swipeDirection ?? oppositeSwipeDirection[swipeDirection];
      const physicalOpenDirection = resolveSwipeDirection(openDirection, prop("dir"));
      return normalize.element({
        ...parts.swipeArea.attrs,
        id: dom.getSwipeAreaId(scope),
        role: "presentation",
        "aria-hidden": true,
        "data-state": open ? "open" : "closed",
        "data-swiping": swipingOpen ? "" : void 0,
        "data-swipe-direction": physicalOpenDirection,
        "data-disabled": disabled ? "" : void 0,
        style: {
          touchAction: isVerticalSwipeDirection(physicalOpenDirection) ? "pan-x" : "pan-y",
          pointerEvents: disabled || open && !swipingOpen ? "none" : void 0
        },
        onPointerDown(event) {
          if (disabled) return;
          if (!isLeftClick(event)) return;
          if (event.pointerType === "touch") return;
          if (open && !swipingOpen) return;
          send({ type: "SWIPE_AREA.START", point: getEventPoint(event) });
          if (event.cancelable) event.preventDefault();
        },
        onTouchStart(event) {
          if (disabled) return;
          if (open && !swipingOpen) return;
          const touch = event.touches[0];
          if (!touch) return;
          send({ type: "SWIPE_AREA.START", point: { x: touch.clientX, y: touch.clientY } });
        }
      });
    }
  };
}
export {
  connect
};
