// @zag-js/drawer/dist/drawer.registry.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var __publicField = (obj, key, value) => __defNormalProp(obj, typeof key !== "symbol" ? key + "" : key, value);

// src/drawer.registry.ts
var drawer_registry_exports = {};
__export(drawer_registry_exports, {
  drawerRegistry: () => drawerRegistry
});
module.exports = __toCommonJS(drawer_registry_exports);
var DrawerRegistry = class {
  constructor() {
    __publicField(this, "elements", /* @__PURE__ */ new Map());
    __publicField(this, "swipingIds", /* @__PURE__ */ new Set());
    __publicField(this, "swipeProgress", /* @__PURE__ */ new Map());
    __publicField(this, "listeners", /* @__PURE__ */ new Set());
  }
  notify() {
    this.listeners.forEach((fn) => fn());
  }
  register(id, el) {
    this.elements.set(id, el);
    this.notify();
  }
  unregister(id) {
    this.swipingIds.delete(id);
    this.swipeProgress.delete(id);
    if (!this.elements.delete(id)) return;
    this.notify();
  }
  setSwiping(id, swiping) {
    const changed = swiping ? !this.swipingIds.has(id) : this.swipingIds.has(id);
    if (!changed && swiping) return;
    if (swiping) {
      this.swipingIds.add(id);
    } else {
      this.swipingIds.delete(id);
      this.swipeProgress.delete(id);
    }
    this.notify();
  }
  setSwipeProgress(id, progress) {
    this.swipeProgress.set(id, progress);
    this.notify();
  }
  getSwipeProgressAfter(id) {
    const keys = [...this.elements.keys()];
    const myIndex = keys.indexOf(id);
    if (myIndex === -1) return 0;
    for (let i = keys.length - 1; i > myIndex; i -= 1) {
      if (this.swipingIds.has(keys[i])) {
        return this.swipeProgress.get(keys[i]) ?? 0;
      }
    }
    return 0;
  }
  hasSwipingAfter(id) {
    const keys = [...this.elements.keys()];
    const myIndex = keys.indexOf(id);
    if (myIndex === -1) return false;
    return keys.slice(myIndex + 1).some((key) => this.swipingIds.has(key));
  }
  getEntries() {
    return this.elements;
  }
  subscribe(fn) {
    this.listeners.add(fn);
    return () => {
      this.listeners.delete(fn);
    };
  }
};
var drawerRegistry = new DrawerRegistry();
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  drawerRegistry
});
