// @zag-js/drawer/dist/drawer.dom.mjs
import "./chunk-QZ7TP4HQ.mjs";

// src/drawer.dom.ts
import { isHTMLElement, queryAll } from "@zag-js/dom-query";
import { isFunction } from "@zag-js/utils";
var getContentId = (ctx) => ctx.ids?.content ?? `drawer:${ctx.id}:content`;
var getPositionerId = (ctx) => ctx.ids?.positioner ?? `drawer:${ctx.id}:positioner`;
var getTitleId = (ctx) => ctx.ids?.title ?? `drawer:${ctx.id}:title`;
var getDescriptionId = (ctx) => ctx.ids?.description ?? `drawer:${ctx.id}:description`;
var getTriggerId = (ctx, value) => {
  const customId = ctx.ids?.trigger;
  if (customId != null) return isFunction(customId) ? customId(value) : customId;
  return value ? `drawer:${ctx.id}:trigger:${value}` : `drawer:${ctx.id}:trigger`;
};
var getTriggerEls = (ctx) => queryAll(ctx.getRootNode(), `[data-scope="drawer"][data-part="trigger"][data-ownedby="${ctx.id}"]`);
var getActiveTriggerEl = (ctx, value) => {
  if (value == null) return getTriggerEl(ctx) ?? getTriggerEls(ctx)[0];
  return ctx.getById(getTriggerId(ctx, value));
};
var getBackdropId = (ctx) => ctx.ids?.backdrop ?? `drawer:${ctx.id}:backdrop`;
var getHeaderId = (ctx) => ctx.ids?.header ?? `drawer:${ctx.id}:header`;
var getGrabberId = (ctx) => ctx.ids?.grabber ?? `drawer:${ctx.id}:grabber`;
var getGrabberIndicatorId = (ctx) => ctx.ids?.grabberIndicator ?? `drawer:${ctx.id}:grabber-indicator`;
var getCloseTriggerId = (ctx) => ctx.ids?.closeTrigger ?? `drawer:${ctx.id}:close-trigger`;
var getSwipeAreaId = (ctx) => ctx.ids?.swipeArea ?? `drawer:${ctx.id}:swipe-area`;
var getContentEl = (ctx) => ctx.getById(getContentId(ctx));
var getPositionerEl = (ctx) => ctx.getById(getPositionerId(ctx));
var getTitleEl = (ctx) => ctx.getById(getTitleId(ctx));
var getDescriptionEl = (ctx) => ctx.getById(getDescriptionId(ctx));
var getTriggerEl = (ctx) => ctx.getById(getTriggerId(ctx));
var getBackdropEl = (ctx) => ctx.getById(getBackdropId(ctx));
var getHeaderEl = (ctx) => ctx.getById(getHeaderId(ctx));
var getGrabberEl = (ctx) => ctx.getById(getGrabberId(ctx));
var getGrabberIndicatorEl = (ctx) => ctx.getById(getGrabberIndicatorId(ctx));
var getCloseTriggerEl = (ctx) => ctx.getById(getCloseTriggerId(ctx));
var getSwipeAreaEl = (ctx) => ctx.getById(getSwipeAreaId(ctx));
function isPointerWithinContentOrSwipeArea(target, content, swipeArea) {
  if (!isHTMLElement(target)) return false;
  return Boolean(content && content.contains(target) || swipeArea && swipeArea.contains(target));
}
var getScrollEls = (scope) => {
  const els = { x: [], y: [] };
  const contentEl = getContentEl(scope);
  if (!contentEl) return els;
  const nodes = queryAll(contentEl, "*");
  nodes.forEach((node) => {
    const y = node.scrollHeight > node.clientHeight;
    if (y) els.y.push(node);
    const x = node.scrollWidth > node.clientWidth;
    if (x) els.x.push(node);
  });
  return els;
};
export {
  getActiveTriggerEl,
  getBackdropEl,
  getBackdropId,
  getCloseTriggerEl,
  getCloseTriggerId,
  getContentEl,
  getContentId,
  getDescriptionEl,
  getDescriptionId,
  getGrabberEl,
  getGrabberId,
  getGrabberIndicatorEl,
  getGrabberIndicatorId,
  getHeaderEl,
  getHeaderId,
  getPositionerEl,
  getPositionerId,
  getScrollEls,
  getSwipeAreaEl,
  getSwipeAreaId,
  getTitleEl,
  getTitleId,
  getTriggerEl,
  getTriggerEls,
  getTriggerId,
  isPointerWithinContentOrSwipeArea
};
