// @zag-js/drawer/dist/drawer.dom.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/drawer.dom.ts
var drawer_dom_exports = {};
__export(drawer_dom_exports, {
  getActiveTriggerEl: () => getActiveTriggerEl,
  getBackdropEl: () => getBackdropEl,
  getBackdropId: () => getBackdropId,
  getCloseTriggerEl: () => getCloseTriggerEl,
  getCloseTriggerId: () => getCloseTriggerId,
  getContentEl: () => getContentEl,
  getContentId: () => getContentId,
  getDescriptionEl: () => getDescriptionEl,
  getDescriptionId: () => getDescriptionId,
  getGrabberEl: () => getGrabberEl,
  getGrabberId: () => getGrabberId,
  getGrabberIndicatorEl: () => getGrabberIndicatorEl,
  getGrabberIndicatorId: () => getGrabberIndicatorId,
  getHeaderEl: () => getHeaderEl,
  getHeaderId: () => getHeaderId,
  getPositionerEl: () => getPositionerEl,
  getPositionerId: () => getPositionerId,
  getScrollEls: () => getScrollEls,
  getSwipeAreaEl: () => getSwipeAreaEl,
  getSwipeAreaId: () => getSwipeAreaId,
  getTitleEl: () => getTitleEl,
  getTitleId: () => getTitleId,
  getTriggerEl: () => getTriggerEl,
  getTriggerEls: () => getTriggerEls,
  getTriggerId: () => getTriggerId,
  isPointerWithinContentOrSwipeArea: () => isPointerWithinContentOrSwipeArea
});
module.exports = __toCommonJS(drawer_dom_exports);
var import_dom_query = require("@zag-js/dom-query");
var import_utils = require("@zag-js/utils");
var getContentId = (ctx) => ctx.ids?.content ?? `drawer:${ctx.id}:content`;
var getPositionerId = (ctx) => ctx.ids?.positioner ?? `drawer:${ctx.id}:positioner`;
var getTitleId = (ctx) => ctx.ids?.title ?? `drawer:${ctx.id}:title`;
var getDescriptionId = (ctx) => ctx.ids?.description ?? `drawer:${ctx.id}:description`;
var getTriggerId = (ctx, value) => {
  const customId = ctx.ids?.trigger;
  if (customId != null) return (0, import_utils.isFunction)(customId) ? customId(value) : customId;
  return value ? `drawer:${ctx.id}:trigger:${value}` : `drawer:${ctx.id}:trigger`;
};
var getTriggerEls = (ctx) => (0, import_dom_query.queryAll)(ctx.getRootNode(), `[data-scope="drawer"][data-part="trigger"][data-ownedby="${ctx.id}"]`);
var getActiveTriggerEl = (ctx, value) => {
  if (value == null) return getTriggerEl(ctx) ?? getTriggerEls(ctx)[0];
  return ctx.getById(getTriggerId(ctx, value));
};
var getBackdropId = (ctx) => ctx.ids?.backdrop ?? `drawer:${ctx.id}:backdrop`;
var getHeaderId = (ctx) => ctx.ids?.header ?? `drawer:${ctx.id}:header`;
var getGrabberId = (ctx) => ctx.ids?.grabber ?? `drawer:${ctx.id}:grabber`;
var getGrabberIndicatorId = (ctx) => ctx.ids?.grabberIndicator ?? `drawer:${ctx.id}:grabber-indicator`;
var getCloseTriggerId = (ctx) => ctx.ids?.closeTrigger ?? `drawer:${ctx.id}:close-trigger`;
var getSwipeAreaId = (ctx) => ctx.ids?.swipeArea ?? `drawer:${ctx.id}:swipe-area`;
var getContentEl = (ctx) => ctx.getById(getContentId(ctx));
var getPositionerEl = (ctx) => ctx.getById(getPositionerId(ctx));
var getTitleEl = (ctx) => ctx.getById(getTitleId(ctx));
var getDescriptionEl = (ctx) => ctx.getById(getDescriptionId(ctx));
var getTriggerEl = (ctx) => ctx.getById(getTriggerId(ctx));
var getBackdropEl = (ctx) => ctx.getById(getBackdropId(ctx));
var getHeaderEl = (ctx) => ctx.getById(getHeaderId(ctx));
var getGrabberEl = (ctx) => ctx.getById(getGrabberId(ctx));
var getGrabberIndicatorEl = (ctx) => ctx.getById(getGrabberIndicatorId(ctx));
var getCloseTriggerEl = (ctx) => ctx.getById(getCloseTriggerId(ctx));
var getSwipeAreaEl = (ctx) => ctx.getById(getSwipeAreaId(ctx));
function isPointerWithinContentOrSwipeArea(target, content, swipeArea) {
  if (!(0, import_dom_query.isHTMLElement)(target)) return false;
  return Boolean(content && content.contains(target) || swipeArea && swipeArea.contains(target));
}
var getScrollEls = (scope) => {
  const els = { x: [], y: [] };
  const contentEl = getContentEl(scope);
  if (!contentEl) return els;
  const nodes = (0, import_dom_query.queryAll)(contentEl, "*");
  nodes.forEach((node) => {
    const y = node.scrollHeight > node.clientHeight;
    if (y) els.y.push(node);
    const x = node.scrollWidth > node.clientWidth;
    if (x) els.x.push(node);
  });
  return els;
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getActiveTriggerEl,
  getBackdropEl,
  getBackdropId,
  getCloseTriggerEl,
  getCloseTriggerId,
  getContentEl,
  getContentId,
  getDescriptionEl,
  getDescriptionId,
  getGrabberEl,
  getGrabberId,
  getGrabberIndicatorEl,
  getGrabberIndicatorId,
  getHeaderEl,
  getHeaderId,
  getPositionerEl,
  getPositionerId,
  getScrollEls,
  getSwipeAreaEl,
  getSwipeAreaId,
  getTitleEl,
  getTitleId,
  getTriggerEl,
  getTriggerEls,
  getTriggerId,
  isPointerWithinContentOrSwipeArea
});
