// @zag-js/drawer/dist/drawer.machine.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/drawer.machine.ts
var drawer_machine_exports = {};
__export(drawer_machine_exports, {
  machine: () => machine
});
module.exports = __toCommonJS(drawer_machine_exports);
var import_aria_hidden = require("@zag-js/aria-hidden");
var import_core = require("@zag-js/core");
var import_dismissable = require("@zag-js/dismissable");
var import_dom_query = require("@zag-js/dom-query");
var import_focus_trap = require("@zag-js/focus-trap");
var import_remove_scroll = require("@zag-js/remove-scroll");
var import_utils = require("@zag-js/utils");
var dom = __toESM(require("./drawer.dom.js"));
var import_drawer = require("./drawer.registry.js");
var import_drawer_session = require("./utils/drawer-session.js");
var import_session = require("./utils/session.js");
var import_snap_point = require("./utils/snap-point.js");
var { and } = (0, import_core.createGuards)();
var getActiveSnapOffset = (context) => context.get("resolvedActiveSnapPoint")?.offset ?? 0;
var hasRemSnapPoints = (snapPoints) => snapPoints.some((snapPoint) => typeof snapPoint === "string" && snapPoint.trim().endsWith("rem"));
var DEFAULT_SNAP_POINTS = [1];
var machine = (0, import_core.createMachine)({
  props({ props, scope }) {
    const alertDialog = props.role === "alertdialog";
    const initialFocusEl = alertDialog ? () => dom.getCloseTriggerEl(scope) : void 0;
    const modal = typeof props.modal === "boolean" ? props.modal : true;
    const snapPoints = props.snapPoints ?? DEFAULT_SNAP_POINTS;
    return {
      modal,
      trapFocus: modal,
      preventScroll: modal,
      closeOnInteractOutside: true,
      closeOnEscape: true,
      restoreFocus: true,
      role: "dialog",
      initialFocusEl,
      snapPoints,
      defaultSnapPoint: props.defaultSnapPoint ?? snapPoints[0] ?? null,
      swipeDirection: "down",
      snapToSequentialPoints: false,
      swipeVelocityThreshold: 500,
      closeThreshold: 0.5,
      preventDragOnScroll: true,
      ...props
    };
  },
  context({ bindable, prop, scope }) {
    return {
      triggerValue: bindable(() => ({
        defaultValue: prop("defaultTriggerValue") ?? null,
        value: prop("triggerValue"),
        onChange(value) {
          const onTriggerValueChange = prop("onTriggerValueChange");
          if (!onTriggerValueChange) return;
          const triggerElement = dom.getActiveTriggerEl(scope, value);
          onTriggerValueChange({ value, triggerElement });
        }
      })),
      dragOffset: bindable(() => ({
        defaultValue: null
      })),
      snapPoint: bindable(() => ({
        defaultValue: prop("defaultSnapPoint"),
        value: prop("snapPoint"),
        onChange(snapPoint) {
          return prop("onSnapPointChange")?.({ snapPoint });
        }
      })),
      resolvedActiveSnapPoint: bindable(() => ({
        defaultValue: null
      })),
      contentSize: bindable(() => ({
        defaultValue: null
      })),
      viewportSize: bindable(() => ({
        defaultValue: 0
      })),
      rootFontSize: bindable(() => ({
        defaultValue: 16
      })),
      swipeStrength: bindable(() => ({
        defaultValue: 1
      })),
      rendered: bindable(() => ({
        defaultValue: { title: true, description: true }
      })),
      nestedMetrics: bindable(() => ({
        defaultValue: { count: 0, height: 0, frontmostHeight: 0, open: false, swiping: false },
        isEqual: import_utils.isEqual
      }))
    };
  },
  refs({ prop }) {
    return {
      swipeSession: new import_drawer_session.DrawerSwipeSession({
        preventDragOnScroll: () => prop("preventDragOnScroll")
      }),
      snapBackFrame: import_dom_query.AnimationFrame.create()
    };
  },
  computed: {
    drawerId({ prop, scope }) {
      return String(prop("id") ?? scope.id);
    },
    physicalSwipeDirection({ prop }) {
      return (0, import_drawer_session.resolveSwipeDirection)(prop("swipeDirection"), prop("dir"));
    },
    resolvedSnapPoints({ context, prop }) {
      const contentSize = context.get("contentSize");
      const viewportSize = context.get("viewportSize");
      const rootFontSize = context.get("rootFontSize");
      if (contentSize === null) return [];
      const points = prop("snapPoints").map((snapPoint) => (0, import_snap_point.resolveSnapPoint)(snapPoint, { contentSize, viewportSize, rootFontSize })).filter((point) => point !== null);
      return (0, import_snap_point.dedupeSnapPoints)(points);
    }
  },
  watch({ track, context, prop, action, computed }) {
    track(
      [
        () => context.get("snapPoint"),
        () => context.get("contentSize"),
        () => context.get("viewportSize"),
        () => context.get("rootFontSize"),
        () => prop("snapPoints").join("|")
      ],
      () => {
        const snapPoint = context.get("snapPoint");
        const contentSize = context.get("contentSize");
        const viewportSize = context.get("viewportSize");
        const rootFontSize = context.get("rootFontSize");
        if (snapPoint === null || contentSize === null) {
          context.set("resolvedActiveSnapPoint", null);
          return;
        }
        const resolvedPoints = computed("resolvedSnapPoints");
        const matchedPoint = resolvedPoints.find((point) => Object.is(point.value, snapPoint));
        if (matchedPoint) {
          context.set("resolvedActiveSnapPoint", matchedPoint);
          return;
        }
        const resolvedActiveSnapPoint = (0, import_snap_point.resolveSnapPoint)(snapPoint, { contentSize, viewportSize, rootFontSize });
        if (resolvedActiveSnapPoint) {
          context.set("resolvedActiveSnapPoint", resolvedActiveSnapPoint);
          return;
        }
        const fallbackPoint = resolvedPoints[0];
        if (!fallbackPoint) {
          context.set("resolvedActiveSnapPoint", null);
          return;
        }
        context.set("snapPoint", fallbackPoint.value);
        context.set("resolvedActiveSnapPoint", fallbackPoint);
      }
    );
    track([() => prop("open")], () => {
      action(["toggleVisibility"]);
    });
    track(
      [() => context.get("dragOffset"), () => context.get("contentSize"), () => getActiveSnapOffset(context)],
      () => {
        action(["syncDrawerStack"]);
      }
    );
  },
  initialState({ prop }) {
    const open = prop("open") || prop("defaultOpen");
    return open ? "open" : "closed";
  },
  on: {
    "SNAP_POINT.SET": {
      actions: ["setSnapPoint"]
    }
  },
  states: {
    open: {
      tags: ["open"],
      entry: ["checkRenderedElements", "setInitialFocus", "deferClearDragOffset"],
      effects: [
        "trackDismissableElement",
        "preventScroll",
        "trapFocus",
        "hideContentBelow",
        "trackPointerMove",
        "trackSizeMeasurements",
        "trackNestedDrawerMetrics",
        "trackDrawerStack"
      ],
      on: {
        "TRIGGER_VALUE.SET": {
          actions: ["setTriggerValue"]
        },
        "CONTROLLED.CLOSE": {
          target: "closing",
          actions: ["clearSwipeOpenAnimation", "cancelSnapBack"]
        },
        POINTER_DOWN: {
          actions: ["setPointerStart", "cancelSnapBack"]
        },
        POINTER_MOVE: [
          {
            guard: "isDragging",
            actions: ["setDragOffset"]
          },
          {
            guard: "shouldStartDragging",
            actions: ["setRegistrySwiping", "setDragOffset"]
          }
        ],
        SNAP_BACK: {
          guard: "isDragging",
          actions: ["deferClearDragOffset", "resetSwipeStrength"]
        },
        POINTER_UP: [
          {
            guard: and("shouldCloseOnSwipe", "isOpenControlled"),
            actions: [
              "clearSwipeOpenAnimation",
              "clearRegistrySwiping",
              "clearPointerStart",
              "setDismissSwipeStrength",
              "invokeOnClose",
              "scheduleSnapBack"
            ]
          },
          {
            guard: "shouldCloseOnSwipe",
            target: "closing",
            actions: ["clearSwipeOpenAnimation", "clearRegistrySwiping", "setDismissSwipeStrength"]
          },
          {
            guard: "isDragging",
            actions: [
              "clearRegistrySwiping",
              "suppressBackdropAnimation",
              "setSnapSwipeStrength",
              "setClosestSnapPoint",
              "clearPointerStart",
              "clearDragOffset",
              "clearVelocityTracking"
            ]
          },
          {
            actions: ["clearRegistrySwiping", "clearPointerStart", "clearDragOffset", "clearVelocityTracking"]
          }
        ],
        POINTER_CANCEL: [
          {
            guard: "isDragging",
            actions: ["clearRegistrySwiping", "clearPointerStart", "clearDragOffset", "clearVelocityTracking"]
          },
          {
            actions: ["clearRegistrySwiping", "clearPointerStart", "clearVelocityTracking"]
          }
        ],
        CLOSE: [
          {
            guard: "isOpenControlled",
            actions: ["clearSwipeOpenAnimation", "invokeOnClose"]
          },
          {
            target: "closing",
            actions: ["clearSwipeOpenAnimation", "resetSwipeStrength", "invokeOnClose"]
          }
        ]
      }
    },
    closing: {
      entry: ["cancelSnapBack"],
      effects: ["trackExitAnimation"],
      on: {
        "CONTROLLED.OPEN": {
          target: "open"
        },
        OPEN: [
          {
            guard: "isOpenControlled",
            actions: ["setTriggerValue", "invokeOnOpen"]
          },
          {
            target: "open",
            actions: ["setTriggerValue", "invokeOnOpen"]
          }
        ],
        "TRIGGER_VALUE.SET": {
          target: "open",
          actions: ["setTriggerValue", "invokeOnOpen"]
        },
        ANIMATION_END: {
          target: "closed",
          actions: [
            "invokeOnClose",
            "clearPointerStart",
            "clearDragOffset",
            "clearActiveSnapPoint",
            "clearResolvedActiveSnapPoint",
            "clearSizeMeasurements",
            "clearVelocityTracking"
          ]
        }
      }
    },
    "swipe-area-dragging": {
      tags: ["closed"],
      effects: ["trackSwipeOpenPointerMove"],
      on: {
        POINTER_MOVE: {
          guard: "hasSwipeIntent",
          target: "swiping-open"
        },
        POINTER_UP: {
          target: "closed",
          actions: ["clearPointerStart", "clearVelocityTracking"]
        },
        POINTER_CANCEL: {
          target: "closed",
          actions: ["clearPointerStart", "clearVelocityTracking"]
        }
      }
    },
    "swiping-open": {
      tags: ["open"],
      effects: ["trackSwipeOpenPointerMove", "trackSizeMeasurements"],
      on: {
        POINTER_MOVE: {
          actions: ["setSwipeOpenDragOffset"]
        },
        POINTER_UP: [
          {
            guard: and("shouldOpenOnSwipe", "isOpenControlled"),
            actions: ["clearPointerStart", "invokeOnOpen"]
          },
          {
            guard: "shouldOpenOnSwipe",
            target: "open",
            actions: ["clearPointerStart", "invokeOnOpen"]
          },
          {
            target: "closed",
            actions: ["clearPointerStart", "clearDragOffset", "clearSizeMeasurements"]
          }
        ],
        POINTER_CANCEL: {
          target: "closed",
          actions: ["clearPointerStart", "clearDragOffset", "clearSizeMeasurements", "clearVelocityTracking"]
        },
        "CONTROLLED.OPEN": {
          target: "open"
        },
        CLOSE: {
          target: "closed",
          actions: ["clearPointerStart", "clearDragOffset", "clearSizeMeasurements"]
        }
      }
    },
    closed: {
      tags: ["closed"],
      on: {
        "CONTROLLED.OPEN": {
          target: "open"
        },
        OPEN: [
          {
            guard: "isOpenControlled",
            actions: ["setTriggerValue", "invokeOnOpen"]
          },
          {
            target: "open",
            actions: ["setTriggerValue", "invokeOnOpen"]
          }
        ],
        "SWIPE_AREA.START": {
          target: "swipe-area-dragging",
          actions: ["setPointerStart"]
        }
      }
    }
  },
  implementations: {
    guards: {
      isOpenControlled: ({ prop }) => prop("open") !== void 0,
      isDragging({ context }) {
        return context.get("dragOffset") !== null;
      },
      shouldStartDragging({ computed, prop, refs, event, scope }) {
        const swipeSession = refs.get("swipeSession");
        return swipeSession.canStartDrag(
          event.point,
          event.target,
          dom.getContentEl(scope),
          prop("preventDragOnScroll"),
          computed("physicalSwipeDirection")
        );
      },
      shouldCloseOnSwipe({ prop, context, computed, refs }) {
        if (prop("snapToSequentialPoints")) return false;
        const swipeSession = refs.get("swipeSession");
        return swipeSession.shouldDismissOnRelease(
          context.get("contentSize"),
          computed("resolvedSnapPoints"),
          getActiveSnapOffset(context)
        );
      },
      hasSwipeIntent({ refs, computed, event }) {
        const swipeSession = refs.get("swipeSession");
        const start = swipeSession.getSwipeStart();
        if (!start || !event.point) return false;
        return (0, import_drawer_session.hasOpeningSwipeIntent)(start, event.point, computed("physicalSwipeDirection"));
      },
      shouldOpenOnSwipe({ context, refs, prop }) {
        const swipeSession = refs.get("swipeSession");
        return swipeSession.shouldOpenOnRelease(
          context.get("contentSize"),
          prop("swipeVelocityThreshold"),
          prop("closeThreshold")
        );
      }
    },
    actions: {
      setInitialFocus({ prop, scope }) {
        if (prop("trapFocus")) return;
        (0, import_dom_query.raf)(() => {
          const element = (0, import_dom_query.getInitialFocus)({
            root: dom.getContentEl(scope),
            getInitialEl: prop("initialFocusEl")
          });
          element?.focus({ preventScroll: true });
        });
      },
      checkRenderedElements({ context, scope }) {
        (0, import_dom_query.raf)(() => {
          context.set("rendered", {
            title: !!dom.getTitleEl(scope),
            description: !!dom.getDescriptionEl(scope)
          });
        });
      },
      deferClearDragOffset({ context, refs, scope }) {
        const dragOffset = context.get("dragOffset");
        if (dragOffset === null) return;
        const contentEl = dom.getContentEl(scope);
        const backdropEl = dom.getBackdropEl(scope);
        if (contentEl) contentEl.style.setProperty("animation", "none", "important");
        if (backdropEl) backdropEl.style.setProperty("animation", "none", "important");
        (0, import_dom_query.raf)(() => {
          refs.get("swipeSession").resetDragOffset();
          context.set("dragOffset", null);
        });
      },
      suppressBackdropAnimation({ scope }) {
        const backdropEl = dom.getBackdropEl(scope);
        if (!backdropEl) return;
        backdropEl.style.setProperty("animation", "none", "important");
      },
      clearSwipeOpenAnimation({ scope }) {
        const contentEl = dom.getContentEl(scope);
        const backdropEl = dom.getBackdropEl(scope);
        if (contentEl) contentEl.style.removeProperty("animation");
        if (backdropEl) backdropEl.style.removeProperty("animation");
      },
      setTriggerValue({ context, event }) {
        if (event.value === void 0) return;
        context.set("triggerValue", event.value);
      },
      invokeOnOpen({ prop }) {
        prop("onOpenChange")?.({ open: true });
      },
      invokeOnClose({ prop }) {
        prop("onOpenChange")?.({ open: false });
      },
      setSnapPoint({ context, event }) {
        context.set("snapPoint", event.snapPoint);
      },
      setPointerStart({ event, refs }) {
        refs.get("swipeSession").beginSwipe(event.point);
      },
      setDragOffset({ context, event, refs, computed }) {
        const swipeSession = refs.get("swipeSession");
        const physicalSwipeDirection = event.swipeDirection ?? computed("physicalSwipeDirection");
        swipeSession.setDragOffset(event.point, getActiveSnapOffset(context), physicalSwipeDirection);
        context.set("dragOffset", swipeSession.getDragOffset());
      },
      setSwipeOpenDragOffset({ context, event, refs, computed }) {
        const swipeSession = refs.get("swipeSession");
        const contentSize = context.get("contentSize");
        if (!contentSize) return;
        swipeSession.setSwipeOpenOffset(event.point, contentSize, computed("physicalSwipeDirection"));
        context.set("dragOffset", swipeSession.getDragOffset());
      },
      setClosestSnapPoint({ computed, context, refs, prop, send }) {
        const snapPoints = computed("resolvedSnapPoints");
        const contentSize = context.get("contentSize");
        const viewportSize = context.get("viewportSize");
        const rootFontSize = context.get("rootFontSize");
        if (!snapPoints.length || contentSize === null) return;
        const swipeSession = refs.get("swipeSession");
        const closestSnapPoint = swipeSession.resolveSnapPointOnRelease(
          snapPoints,
          context.get("resolvedActiveSnapPoint"),
          prop("snapToSequentialPoints"),
          contentSize
        );
        if (closestSnapPoint === null) {
          send({ type: "CLOSE" });
          return;
        }
        context.set("snapPoint", closestSnapPoint);
        const resolved = (0, import_snap_point.resolveSnapPoint)(closestSnapPoint, { contentSize, viewportSize, rootFontSize });
        context.set("resolvedActiveSnapPoint", resolved);
      },
      clearDragOffset({ context, refs }) {
        refs.get("swipeSession").resetDragOffset();
        context.set("dragOffset", null);
      },
      clearActiveSnapPoint({ context }) {
        context.set("snapPoint", context.initial("snapPoint"));
      },
      clearSizeMeasurements({ context }) {
        context.set("contentSize", null);
        context.set("viewportSize", 0);
        context.set("rootFontSize", 16);
      },
      clearResolvedActiveSnapPoint({ context }) {
        context.set("resolvedActiveSnapPoint", null);
      },
      clearPointerStart({ refs }) {
        refs.get("swipeSession").clearSwipeStart();
      },
      clearVelocityTracking({ refs }) {
        refs.get("swipeSession").resetVelocity();
      },
      setSnapSwipeStrength({ context, refs, computed, prop }) {
        const swipeSession = refs.get("swipeSession");
        const snapPoints = computed("resolvedSnapPoints");
        const contentSize = context.get("contentSize");
        const closestSnapPoint = swipeSession.resolveSnapPointOnRelease(
          snapPoints,
          context.get("resolvedActiveSnapPoint"),
          prop("snapToSequentialPoints"),
          contentSize ?? 0
        );
        if (closestSnapPoint === null) return;
        const viewportSize = context.get("viewportSize");
        const rootFontSize = context.get("rootFontSize");
        const resolved = (0, import_snap_point.resolveSnapPoint)(closestSnapPoint, {
          contentSize: contentSize ?? 0,
          viewportSize,
          rootFontSize
        });
        const restOffset = getActiveSnapOffset(context);
        context.set("swipeStrength", swipeSession.getSwipeStrength(resolved?.offset ?? 0, restOffset));
      },
      setDismissSwipeStrength({ context, refs }) {
        const swipeSession = refs.get("swipeSession");
        const contentSize = context.get("contentSize");
        const restOffset = getActiveSnapOffset(context);
        context.set("swipeStrength", swipeSession.getSwipeStrength(contentSize ?? 0, restOffset));
      },
      resetSwipeStrength({ context }) {
        context.set("swipeStrength", 1);
      },
      scheduleSnapBack({ refs, send, prop }) {
        if (prop("onOpenChange") != null) return;
        refs.get("snapBackFrame").request(() => {
          send({ type: "SNAP_BACK" });
        });
      },
      cancelSnapBack({ refs }) {
        refs.get("snapBackFrame").cancel();
      },
      setRegistrySwiping({ computed }) {
        import_drawer.drawerRegistry.setSwiping(computed("drawerId"), true);
      },
      clearRegistrySwiping({ computed }) {
        import_drawer.drawerRegistry.setSwiping(computed("drawerId"), false);
      },
      toggleVisibility({ event, send, prop }) {
        send({ type: prop("open") ? "CONTROLLED.OPEN" : "CONTROLLED.CLOSE", previousEvent: event });
      },
      syncDrawerStack({ context, prop, computed }) {
        const contentSize = context.get("contentSize");
        if (contentSize === null) return;
        const dragOffset = context.get("dragOffset");
        const snapPointOffset = getActiveSnapOffset(context);
        const progress = (0, import_drawer_session.resolveSwipeProgress)(contentSize, dragOffset, snapPointOffset);
        const id = computed("drawerId");
        if (dragOffset !== null) {
          import_drawer.drawerRegistry.setSwipeProgress(id, progress);
        }
        const stack = prop("stack");
        if (!stack) return;
        stack.setHeight(id, contentSize);
        stack.setSwipe(id, dragOffset !== null, progress);
      }
    },
    effects: {
      trackDrawerStack({ context, prop, computed }) {
        const stack = prop("stack");
        if (!stack) return;
        const id = computed("drawerId");
        stack.register(id);
        stack.setOpen(id, true);
        const sync = () => {
          const contentSize = context.get("contentSize");
          const dragOffset = context.get("dragOffset");
          const snapPointOffset = getActiveSnapOffset(context);
          stack.setHeight(id, contentSize ?? 0);
          stack.setSwipe(id, dragOffset !== null, (0, import_drawer_session.resolveSwipeProgress)(contentSize, dragOffset, snapPointOffset));
        };
        sync();
        return () => {
          stack.setSwipe(id, false, 0);
          stack.setOpen(id, false);
          stack.unregister(id);
        };
      },
      trackDismissableElement({ scope, prop, send }) {
        const getContentEl2 = () => dom.getContentEl(scope);
        return (0, import_dismissable.trackDismissableElement)(getContentEl2, {
          type: "drawer",
          defer: true,
          pointerBlocking: prop("modal"),
          layerStyleTargets: [() => dom.getBackdropEl(scope), () => dom.getPositionerEl(scope)],
          exclude: [dom.getTriggerEl(scope), ...dom.getTriggerEls(scope)].filter(Boolean),
          onInteractOutside(event) {
            prop("onInteractOutside")?.(event);
            if (!prop("closeOnInteractOutside")) {
              event.preventDefault();
            }
          },
          onFocusOutside: prop("onFocusOutside"),
          onEscapeKeyDown(event) {
            prop("onEscapeKeyDown")?.(event);
            if (!prop("closeOnEscape")) {
              event.preventDefault();
            }
          },
          onPointerDownOutside: prop("onPointerDownOutside"),
          onRequestDismiss: prop("onRequestDismiss"),
          onDismiss() {
            send({ type: "CLOSE", src: "interact-outside" });
          }
        });
      },
      preventScroll({ scope, prop }) {
        if (!prop("preventScroll")) return;
        return (0, import_remove_scroll.preventBodyScroll)(scope.getDoc());
      },
      trapFocus({ scope, prop, context }) {
        if (!prop("trapFocus")) return;
        const contentEl = () => dom.getContentEl(scope);
        return (0, import_focus_trap.trapFocus)(contentEl, {
          preventScroll: true,
          returnFocusOnDeactivate: !!prop("restoreFocus"),
          initialFocus: prop("initialFocusEl"),
          setReturnFocus: (el) => {
            const finalFocusEl = prop("finalFocusEl")?.();
            if (finalFocusEl) return finalFocusEl;
            const triggerValue = context.get("triggerValue");
            if (triggerValue) {
              const activeTriggerEl = dom.getActiveTriggerEl(scope, triggerValue);
              if (activeTriggerEl) return activeTriggerEl;
            }
            const fallbackTrigger = dom.getTriggerEls(scope)[0];
            if (fallbackTrigger) return fallbackTrigger;
            return el;
          },
          getShadowRoot: true
        });
      },
      hideContentBelow({ scope, prop }) {
        if (!prop("modal")) return;
        const getElements = () => [dom.getContentEl(scope)];
        return (0, import_aria_hidden.ariaHidden)(getElements, { defer: true });
      },
      trackPointerMove({ scope, send, refs, computed }) {
        return refs.get("swipeSession").bindDragTracking({
          getDoc: () => scope.getDoc(),
          getContentEl: () => dom.getContentEl(scope),
          getSwipeAreaEl: () => dom.getSwipeAreaEl(scope),
          swipeDirection: computed("physicalSwipeDirection"),
          onMove(details) {
            send({ type: "POINTER_MOVE", ...details });
          },
          onEnd(details) {
            send({ type: "POINTER_UP", ...details });
          },
          onCancel() {
            send({ type: "POINTER_CANCEL" });
          }
        });
      },
      trackSizeMeasurements({ context, scope, computed, prop }) {
        const doc = scope.getDoc();
        const html = doc.documentElement;
        const shouldMeasureRootFontSize = hasRemSnapPoints(prop("snapPoints"));
        return waitForContentEl(scope, (contentEl) => {
          const updateSize = () => {
            const direction = computed("physicalSwipeDirection");
            const rect = contentEl.getBoundingClientRect();
            const viewportSize = (0, import_session.isVerticalSwipeDirection)(direction) ? html.clientHeight : html.clientWidth;
            context.set("contentSize", (0, import_drawer_session.getSwipeDirectionSize)(rect, direction));
            context.set("viewportSize", viewportSize);
            if (shouldMeasureRootFontSize) {
              const rootFontSize = Number.parseFloat((0, import_dom_query.getComputedStyle)(html).fontSize);
              if (Number.isFinite(rootFontSize)) {
                context.set("rootFontSize", rootFontSize);
              }
            }
          };
          updateSize();
          const cleanups = [
            import_dom_query.resizeObserverBorderBox.observe(contentEl, updateSize),
            (0, import_dom_query.addDomEvent)(scope.getWin(), "resize", updateSize)
          ];
          return () => cleanups.forEach((cleanup) => cleanup?.());
        });
      },
      trackNestedDrawerMetrics({ scope, computed, context }) {
        return waitForContentEl(scope, (contentEl) => {
          const id = computed("drawerId");
          import_drawer.drawerRegistry.register(id, contentEl);
          const sync = () => {
            const entries = [...import_drawer.drawerRegistry.getEntries().entries()];
            const myIndex = entries.findIndex(([entryId]) => entryId === id);
            if (myIndex === -1) return;
            const count = entries.length - 1 - myIndex;
            const frontmostEl = entries[entries.length - 1]?.[1];
            const frontmostHeight = frontmostEl?.getBoundingClientRect().height ?? 0;
            const height = contentEl.getBoundingClientRect().height;
            context.set("nestedMetrics", {
              count,
              height,
              frontmostHeight,
              open: count > 0 && frontmostHeight > 0,
              swiping: import_drawer.drawerRegistry.hasSwipingAfter(id)
            });
          };
          sync();
          const cleanups = [
            import_drawer.drawerRegistry.subscribe(sync),
            import_dom_query.resizeObserverBorderBox.observe(contentEl, () => import_drawer.drawerRegistry.notify()),
            (0, import_dom_query.addDomEvent)(scope.getWin(), "resize", () => import_drawer.drawerRegistry.notify())
          ];
          return () => {
            cleanups.forEach((cleanup) => cleanup?.());
            import_drawer.drawerRegistry.unregister(id);
          };
        });
      },
      trackSwipeOpenPointerMove({ scope, send, refs, computed }) {
        return refs.get("swipeSession").bindSwipeOpenTracking({
          getDoc: () => scope.getDoc(),
          getContentEl: () => dom.getContentEl(scope),
          getSwipeAreaEl: () => dom.getSwipeAreaEl(scope),
          swipeDirection: computed("physicalSwipeDirection"),
          onMove(details) {
            send({ type: "POINTER_MOVE", ...details });
          },
          onEnd(details) {
            send({ type: "POINTER_UP", ...details });
          },
          onCancel() {
            send({ type: "POINTER_CANCEL" });
          }
        });
      },
      trackExitAnimation({ send, scope }) {
        const contentEl = dom.getContentEl(scope);
        if (!contentEl) return;
        return (0, import_dom_query.addDomEvent)(contentEl, "exitcomplete", () => {
          send({ type: "ANIMATION_END" });
        });
      }
    }
  }
});
function waitForContentEl(scope, setup) {
  const contentEl = dom.getContentEl(scope);
  let cleanup = contentEl ? setup(contentEl) : void 0;
  let abort;
  if (!cleanup) {
    const [promise, cancel] = (0, import_dom_query.waitForElement)(() => dom.getContentEl(scope), {
      timeout: 1e3,
      rootNode: scope.getDoc()
    });
    abort = cancel;
    promise.then((el) => cleanup = setup(el)).catch(import_utils.noop);
  }
  return () => {
    abort?.();
    cleanup?.();
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  machine
});
