// @zag-js/date-input/dist/date-input.props.mjs
import "./chunk-QZ7TP4HQ.mjs";

// src/date-input.props.ts
import { createProps } from "@zag-js/types";
import { createSplitProps } from "@zag-js/utils";
var props = createProps()([
  "createCalendar",
  "dir",
  "disabled",
  "format",
  "getRootNode",
  "id",
  "ids",
  "invalid",
  "isDateUnavailable",
  "locale",
  "max",
  "min",
  "name",
  "form",
  "onFocusChange",
  "onPlaceholderChange",
  "onValueChange",
  "readOnly",
  "required",
  "selectionMode",
  "timeZone",
  "translations",
  "value",
  "defaultValue",
  "hourCycle",
  "hideTimeZone",
  "granularity",
  "shouldForceLeadingZeros",
  "allSegments",
  "formatter",
  "placeholderValue",
  "defaultPlaceholderValue"
]);
var splitProps = createSplitProps(props);
var segmentProps = createProps()(["segment", "index"]);
var splitSegmentProps = createSplitProps(segmentProps);
var segmentGroupProps = createProps()(["index"]);
var splitSegmentGroupProps = createSplitProps(segmentGroupProps);
var segmentsProps = createProps()(["index"]);
var splitSegmentsProps = createSplitProps(segmentsProps);
var labelProps = createProps()(["index"]);
var splitLabelProps = createSplitProps(labelProps);
var hiddenInputProps = createProps()(["index", "name"]);
var splitHiddenInputProps = createSplitProps(hiddenInputProps);
export {
  hiddenInputProps,
  labelProps,
  props,
  segmentGroupProps,
  segmentProps,
  segmentsProps,
  splitHiddenInputProps,
  splitLabelProps,
  splitProps,
  splitSegmentGroupProps,
  splitSegmentProps,
  splitSegmentsProps
};
