// @zag-js/date-input/dist/date-input.dom.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/date-input.dom.ts
var date_input_dom_exports = {};
__export(date_input_dom_exports, {
  getControlEl: () => getControlEl,
  getControlId: () => getControlId,
  getHiddenInputId: () => getHiddenInputId,
  getLabelId: () => getLabelId,
  getRootId: () => getRootId,
  getSegmentEls: () => getSegmentEls,
  getSegmentGroupId: () => getSegmentGroupId
});
module.exports = __toCommonJS(date_input_dom_exports);
var import_dom_query = require("@zag-js/dom-query");
var getRootId = (ctx) => ctx.ids?.root ?? `date-input:${ctx.id}`;
var getLabelId = (ctx, index) => ctx.ids?.label?.(index) ?? `date-input:${ctx.id}:label:${index}`;
var getControlId = (ctx) => ctx.ids?.control ?? `date-input:${ctx.id}:control`;
var getSegmentGroupId = (ctx, index) => ctx.ids?.segmentGroup?.(index) ?? `date-input:${ctx.id}:segment-group:${index}`;
var getHiddenInputId = (ctx, index) => ctx.ids?.hiddenInput?.(index) ?? `date-input:${ctx.id}:hidden-input:${index}`;
var getControlEl = (ctx) => ctx.getById(getControlId(ctx));
var getSegmentEls = (ctx) => (0, import_dom_query.queryAll)(getControlEl(ctx), `[data-part=segment]`);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getControlEl,
  getControlId,
  getHiddenInputId,
  getLabelId,
  getRootId,
  getSegmentEls,
  getSegmentGroupId
});
