// @zag-js/date-input/dist/date-input.connect.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/date-input.connect.ts
var date_input_connect_exports = {};
__export(date_input_connect_exports, {
  connect: () => connect
});
module.exports = __toCommonJS(date_input_connect_exports);
var import_dom_query = require("@zag-js/dom-query");
var import_date_input = require("./date-input.anatomy.js");
var dom = __toESM(require("./date-input.dom.js"));
var import_date_utils = require("@zag-js/date-utils");
var import_segments = require("./utils/segments.js");
var import_validity = require("./utils/validity.js");
function connect(service, normalize) {
  const { state, context, prop, send, computed, scope } = service;
  const disabled = Boolean(prop("disabled"));
  const readOnly = Boolean(prop("readOnly"));
  const isDateUnavailableFn = prop("isDateUnavailable");
  const value = context.get("value");
  const valueAsDate = value.filter((date) => date != null).map((date) => date.toDate(prop("timeZone")));
  const isDateUnavailable = isDateUnavailableFn ? value.some((date) => date != null && isDateUnavailableFn(date, prop("locale"))) : false;
  const invalid = prop("invalid") ?? isDateUnavailable;
  const focused = state.matches("focused");
  const locale = prop("locale");
  const separator = (0, import_date_utils.getLocaleSeparator)(locale);
  function getSegmentState(props) {
    const { segment, index = 0 } = props;
    const isEditable = !disabled && !readOnly && segment.isEditable;
    const activeIndex = context.get("activeIndex");
    const isFocused = focused && activeIndex === index;
    return {
      editable: isEditable,
      focused: isFocused,
      readonly: !segment.isEditable || readOnly
    };
  }
  return {
    focused,
    disabled,
    invalid,
    value,
    valueAsDate,
    valueAsString: computed("valueAsString"),
    placeholderValue: context.get("placeholderValue"),
    displayValues: context.get("displayValues"),
    focus() {
      dom.getSegmentEls(scope)[0]?.focus();
    },
    setValue(values) {
      send({ type: "VALUE.SET", value: values });
    },
    clearValue() {
      send({ type: "VALUE.CLEAR" });
    },
    getSegments(props = {}) {
      const { index = 0 } = props;
      const allSegments = computed("segments");
      const segments = allSegments[index] ?? [];
      const enteredKeys = context.get("enteredKeys");
      const activeIndex = context.get("activeIndex");
      const activeSegmentIndex = context.get("activeSegmentIndex");
      if (focused && enteredKeys && index === activeIndex && activeSegmentIndex >= 0) {
        const localActiveSegmentIndex = activeSegmentIndex - (0, import_validity.getGroupOffset)(allSegments, index);
        return segments.map((segment, index2) => {
          if (index2 !== localActiveSegmentIndex) return segment;
          return { ...segment, text: enteredKeys, isPlaceholder: false };
        });
      }
      return segments;
    },
    getSegmentState,
    getRootProps() {
      return normalize.element({
        ...import_date_input.parts.root.attrs,
        dir: prop("dir"),
        id: dom.getRootId(scope),
        "data-disabled": (0, import_dom_query.dataAttr)(disabled),
        "data-readonly": (0, import_dom_query.dataAttr)(readOnly),
        "data-invalid": (0, import_dom_query.dataAttr)(invalid)
      });
    },
    getLabelProps(props = {}) {
      const { index = 0 } = props;
      return normalize.label({
        ...import_date_input.parts.label.attrs,
        id: dom.getLabelId(scope, index),
        dir: prop("dir"),
        htmlFor: dom.getHiddenInputId(scope, index),
        "data-disabled": (0, import_dom_query.dataAttr)(disabled),
        "data-readonly": (0, import_dom_query.dataAttr)(readOnly),
        "data-invalid": (0, import_dom_query.dataAttr)(invalid),
        onClick() {
          if (disabled) return;
          dom.getSegmentEls(scope)[0]?.focus();
        }
      });
    },
    getControlProps() {
      return normalize.element({
        ...import_date_input.parts.control.attrs,
        dir: prop("dir"),
        id: dom.getControlId(scope),
        "data-disabled": (0, import_dom_query.dataAttr)(disabled),
        "data-readonly": (0, import_dom_query.dataAttr)(readOnly),
        "data-invalid": (0, import_dom_query.dataAttr)(invalid),
        "data-focus": (0, import_dom_query.dataAttr)(focused)
      });
    },
    getSegmentGroupProps(props = {}) {
      const { index = 0 } = props;
      const activeIndex = context.get("activeIndex");
      return normalize.element({
        ...import_date_input.parts.segmentGroup.attrs,
        id: dom.getSegmentGroupId(scope, index),
        dir: prop("dir"),
        role: "group",
        "aria-labelledby": dom.getLabelId(scope, index),
        "data-disabled": (0, import_dom_query.dataAttr)(disabled),
        "data-readonly": (0, import_dom_query.dataAttr)(readOnly),
        "data-invalid": (0, import_dom_query.dataAttr)(invalid),
        "data-focus": (0, import_dom_query.dataAttr)(focused && activeIndex === index),
        style: {
          unicodeBidi: "isolate"
        }
      });
    },
    getSegmentProps(props) {
      const { segment, index = 0 } = props;
      const segmentState = getSegmentState(props);
      if (segment.type === "literal") {
        return normalize.element({
          ...import_date_input.parts.segment.attrs,
          dir: prop("dir"),
          "aria-hidden": true,
          "data-type": segment.type,
          "data-readonly": "",
          "data-disabled": ""
        });
      }
      return normalize.element({
        ...import_date_input.parts.segment.attrs,
        dir: prop("dir"),
        role: "spinbutton",
        tabIndex: disabled ? void 0 : 0,
        autoComplete: "off",
        spellCheck: segmentState.editable ? "false" : void 0,
        autoCorrect: segmentState.editable ? "off" : void 0,
        contentEditable: segmentState.editable,
        suppressContentEditableWarning: segmentState.editable,
        inputMode: disabled || segment.type === "dayPeriod" || segment.type === "era" || !segmentState.editable ? void 0 : "numeric",
        enterKeyHint: "next",
        "aria-label": (0, import_segments.getSegmentLabel)(segment.type),
        "aria-valuenow": segment.isPlaceholder ? void 0 : segment.value,
        "aria-valuetext": segment.isPlaceholder ? segment.placeholder : segment.text,
        "aria-valuemin": segment.minValue,
        "aria-valuemax": segment.maxValue,
        "aria-invalid": (0, import_dom_query.ariaAttr)(invalid),
        "aria-readonly": (0, import_dom_query.ariaAttr)(segmentState.readonly),
        "aria-disabled": (0, import_dom_query.ariaAttr)(disabled),
        "data-value": segment.value,
        "data-type": segment.type,
        "data-readonly": (0, import_dom_query.dataAttr)(segmentState.readonly),
        "data-disabled": (0, import_dom_query.dataAttr)(disabled),
        "data-editable": (0, import_dom_query.dataAttr)(segment.isEditable && !readOnly && !disabled),
        "data-placeholder-shown": (0, import_dom_query.dataAttr)(segment.isPlaceholder),
        style: {
          caretColor: "transparent"
        },
        onFocus(event) {
          const segmentEls = dom.getSegmentEls(scope);
          const target = event.currentTarget;
          const segmentIndex = segmentEls.indexOf(target);
          send({ type: "SEGMENT.FOCUS", dateIndex: index, segmentIndex });
          const selection = target.ownerDocument?.getSelection?.();
          if (selection && target) {
            selection.collapse(target);
          }
        },
        onBlur(event) {
          const next = event.relatedTarget;
          const control = dom.getControlEl(scope);
          if ((0, import_dom_query.contains)(control, next)) return;
          send({ type: "SEGMENT.BLUR", index: -1 });
        },
        onKeyDown(event) {
          if (event.defaultPrevented || event.ctrlKey || event.metaKey || event.shiftKey || event.altKey || readOnly || (0, import_dom_query.isComposingEvent)(event)) {
            return;
          }
          const keyMap = {
            ArrowLeft() {
              send({ type: "SEGMENT.ARROW_LEFT" });
            },
            ArrowRight() {
              send({ type: "SEGMENT.ARROW_RIGHT" });
            },
            ArrowUp() {
              send({ type: "SEGMENT.ADJUST", segment, amount: 1 });
            },
            ArrowDown() {
              send({ type: "SEGMENT.ADJUST", segment, amount: -1 });
            },
            PageUp() {
              send({
                type: "SEGMENT.ADJUST",
                segment,
                amount: import_segments.PAGE_STEP[segment.type] ?? 1
              });
            },
            PageDown() {
              send({
                type: "SEGMENT.ADJUST",
                segment,
                amount: -(import_segments.PAGE_STEP[segment.type] ?? 1)
              });
            },
            Backspace() {
              send({ type: "SEGMENT.BACKSPACE", segment });
            },
            Delete() {
              send({ type: "SEGMENT.BACKSPACE", segment });
            },
            Home() {
              send({ type: "SEGMENT.HOME", segment });
            },
            End() {
              send({ type: "SEGMENT.END", segment });
            }
          };
          const exec = keyMap[(0, import_dom_query.getEventKey)(event, { dir: prop("dir") })];
          if (exec) {
            exec(event);
            event.preventDefault();
            event.stopPropagation();
          }
        },
        onPointerDown(event) {
          event.stopPropagation();
        },
        onMouseDown(event) {
          event.stopPropagation();
        },
        onBeforeInput(event) {
          const { data, inputType } = (0, import_dom_query.getNativeEvent)(event);
          if (ALLOWED_INPUT_TYPES.includes(inputType)) {
            event.preventDefault();
            return;
          }
          if (inputType === "insertFromPaste") {
            event.preventDefault();
            return;
          }
          if (inputType === "insertCompositionText") {
            const target = event.currentTarget || event.target;
            if (target) {
              event.preventDefault();
              if (data != null) {
                send({ type: "SEGMENT.INPUT", segment, input: data });
              }
            }
            return;
          }
          const isTextSegment = segment.type === "dayPeriod" || segment.type === "era";
          if (data && (isTextSegment || (0, import_date_utils.isValidCharacter)(data, separator))) {
            event.preventDefault();
            send({ type: "SEGMENT.INPUT", segment, input: data });
          } else {
            event.preventDefault();
          }
        },
        onPaste(event) {
          event.preventDefault();
          const text = event.clipboardData?.getData("text/plain")?.trim();
          if (text) {
            send({ type: "SEGMENT.PASTE", value: text });
          }
        }
      });
    },
    getHiddenInputProps(props = {}) {
      const { index = 0, name } = props;
      const value2 = context.get("value");
      const inputName = name || prop("name");
      return normalize.input({
        ...import_date_input.parts.hiddenInput.attrs,
        type: "hidden",
        id: dom.getHiddenInputId(scope, index),
        name: inputName ? value2.length > 1 ? `${inputName}[${index}]` : inputName : void 0,
        form: prop("form"),
        required: prop("required"),
        disabled,
        readOnly,
        value: computed("valueAsString")[index] ?? "",
        style: import_dom_query.visuallyHiddenStyle
      });
    }
  };
}
var ALLOWED_INPUT_TYPES = ["deleteContentBackward", "deleteContentForward", "deleteByCut", "deleteByDrag"];
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  connect
});
