// @zag-js/date-input/dist/date-input.props.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/date-input.props.ts
var date_input_props_exports = {};
__export(date_input_props_exports, {
  hiddenInputProps: () => hiddenInputProps,
  labelProps: () => labelProps,
  props: () => props,
  segmentGroupProps: () => segmentGroupProps,
  segmentProps: () => segmentProps,
  segmentsProps: () => segmentsProps,
  splitHiddenInputProps: () => splitHiddenInputProps,
  splitLabelProps: () => splitLabelProps,
  splitProps: () => splitProps,
  splitSegmentGroupProps: () => splitSegmentGroupProps,
  splitSegmentProps: () => splitSegmentProps,
  splitSegmentsProps: () => splitSegmentsProps
});
module.exports = __toCommonJS(date_input_props_exports);
var import_types = require("@zag-js/types");
var import_utils = require("@zag-js/utils");
var props = (0, import_types.createProps)()([
  "createCalendar",
  "dir",
  "disabled",
  "format",
  "getRootNode",
  "id",
  "ids",
  "invalid",
  "isDateUnavailable",
  "locale",
  "max",
  "min",
  "name",
  "form",
  "onFocusChange",
  "onPlaceholderChange",
  "onValueChange",
  "readOnly",
  "required",
  "selectionMode",
  "timeZone",
  "translations",
  "value",
  "defaultValue",
  "hourCycle",
  "hideTimeZone",
  "granularity",
  "shouldForceLeadingZeros",
  "allSegments",
  "formatter",
  "placeholderValue",
  "defaultPlaceholderValue"
]);
var splitProps = (0, import_utils.createSplitProps)(props);
var segmentProps = (0, import_types.createProps)()(["segment", "index"]);
var splitSegmentProps = (0, import_utils.createSplitProps)(segmentProps);
var segmentGroupProps = (0, import_types.createProps)()(["index"]);
var splitSegmentGroupProps = (0, import_utils.createSplitProps)(segmentGroupProps);
var segmentsProps = (0, import_types.createProps)()(["index"]);
var splitSegmentsProps = (0, import_utils.createSplitProps)(segmentsProps);
var labelProps = (0, import_types.createProps)()(["index"]);
var splitLabelProps = (0, import_utils.createSplitProps)(labelProps);
var hiddenInputProps = (0, import_types.createProps)()(["index", "name"]);
var splitHiddenInputProps = (0, import_utils.createSplitProps)(hiddenInputProps);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  hiddenInputProps,
  labelProps,
  props,
  segmentGroupProps,
  segmentProps,
  segmentsProps,
  splitHiddenInputProps,
  splitLabelProps,
  splitProps,
  splitSegmentGroupProps,
  splitSegmentProps,
  splitSegmentsProps
});
