// @zag-js/date-input/dist/date-input.machine.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/date-input.machine.ts
var date_input_machine_exports = {};
__export(date_input_machine_exports, {
  machine: () => machine
});
module.exports = __toCommonJS(date_input_machine_exports);
var import_date = require("@internationalized/date");
var import_core = require("@zag-js/core");
var import_date_utils = require("@zag-js/date-utils");
var import_dom_query = require("@zag-js/dom-query");
var import_live_region = require("@zag-js/live-region");
var dom = __toESM(require("./date-input.dom.js"));
var import_formatting = require("./utils/formatting.js");
var import_incomplete_date = require("./utils/incomplete-date.js");
var import_input = require("./utils/input.js");
var import_placeholders = require("./utils/placeholders.js");
var import_date_input = require("./date-input.parse.js");
var import_segments = require("./utils/segments.js");
var import_validity = require("./utils/validity.js");
var machine = (0, import_core.createMachine)({
  props({ props }) {
    const locale = props.locale || "en-US";
    const timeZone = props.timeZone || "UTC";
    const selectionMode = props.selectionMode || "single";
    const granularity = props.granularity || "day";
    const translations = { ...import_placeholders.defaultTranslations, ...props.translations };
    const calendar = resolveCalendar(locale, props.createCalendar);
    const defaultValue = props.defaultValue ? props.defaultValue.map((date) => (0, import_date_utils.constrainSegments)(toTargetCalendar(date, calendar), props.min, props.max)) : [];
    const value = props.value ? props.value.map((date) => (0, import_date_utils.constrainSegments)(toTargetCalendar(date, calendar), props.min, props.max)) : void 0;
    const placeholderValue = (0, import_formatting.resolvePlaceholderValue)(props, timeZone, granularity, value, defaultValue, calendar);
    const hourCycle = (0, import_formatting.resolveHourCycleProp)(props.hourCycle);
    const shouldForceLeadingZeros = props.shouldForceLeadingZeros ?? false;
    const digitStyle = shouldForceLeadingZeros ? "2-digit" : "numeric";
    const firstValue = value?.[0] ?? defaultValue?.[0] ?? placeholderValue;
    const hasTimeZone = firstValue != null && "timeZone" in firstValue;
    const formatter = props.formatter ?? new import_date.DateFormatter(
      locale,
      (0, import_segments.getFormatterOptions)({
        granularity,
        digitStyle,
        hourCycle,
        timeZone,
        hasTimeZone,
        hideTimeZone: props.hideTimeZone
      })
    );
    const allSegments = props.allSegments ?? (0, import_segments.resolveAllSegments)(formatter);
    return {
      locale,
      timeZone,
      selectionMode,
      format: (0, import_formatting.createFormatFn)(formatter),
      ...props,
      translations,
      value,
      defaultValue,
      granularity,
      shouldForceLeadingZeros,
      formatter,
      placeholderValue: typeof props.placeholderValue === "undefined" ? void 0 : placeholderValue,
      defaultPlaceholderValue: placeholderValue,
      allSegments
    };
  },
  initialState() {
    return "idle";
  },
  refs() {
    return {
      announcer: null,
      segmentToAnnounceIndex: null
    };
  },
  effects: ["setupLiveRegion"],
  context({ prop, bindable }) {
    const formatter = prop("formatter");
    const hourCycle = (0, import_incomplete_date.resolvedHourCycle)(formatter);
    const placeholderValue = prop("defaultPlaceholderValue");
    const initialValue = prop("value") || prop("defaultValue");
    const groupCount = (0, import_validity.getGroupCount)(prop("selectionMode"));
    return {
      value: bindable(() => ({
        defaultValue: prop("defaultValue"),
        value: prop("value"),
        isEqual(a, b) {
          if (a?.length !== b?.length) return false;
          const len = Math.max(a.length, b.length);
          for (let i = 0; i < len; i++) {
            if (!(0, import_date_utils.isDateEqual)(a[i], b[i])) return false;
          }
          return true;
        },
        hash: (v) => v.map((date) => date?.toString() ?? "").join(","),
        onChange(value) {
          const valueAsString = (0, import_formatting.getValueAsString)(value, prop);
          prop("onValueChange")?.({ value, valueAsString });
        }
      })),
      activeIndex: bindable(() => ({
        defaultValue: 0,
        sync: true
      })),
      activeSegmentIndex: bindable(() => ({
        defaultValue: -1,
        sync: true
      })),
      placeholderValue: bindable(() => ({
        defaultValue: prop("defaultPlaceholderValue"),
        isEqual: import_date_utils.isDateEqual,
        hash: (v) => v.toString(),
        onChange(placeholderValue2) {
          prop("onPlaceholderChange")?.({ value: prop("value") ?? [], valueAsString: [], placeholderValue: placeholderValue2 });
        }
      })),
      displayValues: bindable(() => ({
        defaultValue: (0, import_incomplete_date.initDisplayValues)(
          initialValue,
          placeholderValue ?? (0, import_date_utils.getTodayDate)(prop("timeZone")),
          hourCycle,
          groupCount
        ),
        isEqual: (a, b) => b != null && a.length === b.length && a.every((d, i) => (0, import_incomplete_date.incompleteDateEqual)(d, b[i])),
        hash: (v) => v.map(import_incomplete_date.incompleteDateHash).join("||")
      })),
      enteredKeys: bindable(() => ({
        defaultValue: "",
        sync: true
      }))
    };
  },
  computed: {
    isInteractive: ({ prop }) => !prop("disabled") && !prop("readOnly"),
    groupCount: ({ prop }) => (0, import_validity.getGroupCount)(prop("selectionMode")),
    valueAsString: ({ context, prop }) => (0, import_formatting.getValueAsString)(context.get("value"), prop),
    segments: (0, import_core.memo)(
      ({ context, prop }) => [
        context.hash("value"),
        prop("selectionMode"),
        context.hash("placeholderValue"),
        context.hash("displayValues"),
        prop("allSegments"),
        prop("timeZone"),
        prop("translations"),
        prop("granularity"),
        JSON.stringify(prop("formatter").resolvedOptions()),
        prop("locale")
      ],
      (_deps, { context, prop, computed }) => {
        const value = context.get("value");
        const placeholderValue = context.get("placeholderValue");
        const displayValues = context.get("displayValues");
        const allSegments = prop("allSegments");
        const timeZone = prop("timeZone");
        const translations = prop("translations") || import_placeholders.defaultTranslations;
        const granularity = prop("granularity");
        const formatter = prop("formatter");
        const locale = prop("locale");
        const allSegmentTypes = Object.keys(allSegments);
        return Array.from({ length: computed("groupCount") }, (_, i) => {
          const displayValue = displayValues[i] ?? new import_incomplete_date.IncompleteDate(placeholderValue.calendar, (0, import_incomplete_date.resolvedHourCycle)(formatter));
          const committedValue = value?.[i];
          const isFullyCommitted = committedValue && displayValue.isComplete(allSegmentTypes);
          const displayDate = isFullyCommitted ? committedValue : displayValue.toValue(placeholderValue);
          const showEra = displayValue.era === "BC" && displayValue.calendar.identifier === "gregory";
          const segmentFormatter = showEra ? new import_date.DateFormatter(locale, {
            ...formatter.resolvedOptions(),
            era: "short"
          }) : formatter;
          return (0, import_segments.processSegments)({
            dateValue: displayDate.toDate(timeZone),
            displayValue,
            formatter: segmentFormatter,
            locale,
            translations,
            granularity
          });
        });
      }
    )
  },
  watch({ track, context, prop, action }) {
    track([() => context.hash("value")], () => {
      action(["syncDisplayValues"]);
    });
    track([() => (0, import_incomplete_date.resolvedHourCycle)(prop("formatter"))], () => {
      action(["syncDisplayValueHourCycle"]);
    });
    track([() => context.get("activeSegmentIndex")], () => {
      action(["focusActiveSegment"]);
    });
    track([() => prop("placeholderValue")?.toString()], () => {
      action(["syncPlaceholderProp"]);
    });
    track([() => prop("defaultPlaceholderValue")?.toString()], () => {
      action(["syncDefaultPlaceholderValue"]);
    });
  },
  on: {
    "VALUE.SET": {
      actions: ["setDateValue"]
    },
    "VALUE.CLEAR": {
      actions: ["clearDateValue", "clearDisplayValues", "clearEnteredKeys"]
    }
  },
  states: {
    idle: {
      on: {
        "SEGMENT.FOCUS": {
          target: "focused",
          actions: ["setActiveSegmentIndex", "invokeOnFocus"]
        }
      }
    },
    focused: {
      on: {
        "SEGMENT.FOCUS": {
          actions: ["setActiveSegmentIndex", "clearEnteredKeys"]
        },
        "SEGMENT.BLUR": {
          target: "idle",
          actions: ["confirmPlaceholder", "clearEnteredKeys", "invokeOnBlur"]
        },
        "SEGMENT.INPUT": {
          actions: ["setSegmentValue", "announceSegmentValue"]
        },
        "SEGMENT.ADJUST": {
          actions: ["invokeOnSegmentAdjust", "clearEnteredKeys", "announceSegmentValue"]
        },
        "SEGMENT.ARROW_LEFT": {
          actions: ["setPreviousActiveSegmentIndex", "clearEnteredKeys"]
        },
        "SEGMENT.ARROW_RIGHT": {
          actions: ["setNextActiveSegmentIndex", "clearEnteredKeys"]
        },
        "SEGMENT.BACKSPACE": [
          {
            guard: "isActiveSegmentPlaceholder",
            actions: ["setPreviousActiveSegmentIndex", "clearEnteredKeys"]
          },
          {
            actions: ["clearSegmentValue", "announceSegmentValue"]
          }
        ],
        "SEGMENT.HOME": {
          actions: ["setSegmentToLowestValue", "clearEnteredKeys", "announceSegmentValue"]
        },
        "SEGMENT.END": {
          actions: ["setSegmentToHighestValue", "clearEnteredKeys", "announceSegmentValue"]
        },
        "SEGMENT.PASTE": {
          actions: ["setPastedValue", "clearEnteredKeys"]
        }
      }
    }
  },
  implementations: {
    effects: {
      setupLiveRegion({ scope, refs }) {
        const liveRegion = (0, import_live_region.createLiveRegion)({
          level: "assertive",
          document: scope.getDoc()
        });
        refs.set("announcer", liveRegion);
        return () => liveRegion.destroy();
      }
    },
    guards: {
      isActiveSegmentPlaceholder: (ctx) => {
        const hasEnteredKeys = ctx.context.get("enteredKeys") !== "";
        if (hasEnteredKeys) return false;
        return (0, import_validity.getActiveSegment)(ctx)?.isPlaceholder === true;
      }
    },
    actions: {
      invokeOnFocus({ prop }) {
        prop("onFocusChange")?.({ focused: true });
      },
      invokeOnBlur({ prop }) {
        prop("onFocusChange")?.({ focused: false });
      },
      setActiveSegmentIndex({ context, event }) {
        if (event.dateIndex != null) {
          context.set("activeIndex", event.dateIndex);
        }
        context.set("activeSegmentIndex", event.segmentIndex);
      },
      clearDisplayValues({ context, prop, computed }) {
        const hourCycle = (0, import_incomplete_date.resolvedHourCycle)(prop("formatter"));
        const placeholderValue = context.get("placeholderValue");
        context.set(
          "displayValues",
          Array.from(
            { length: computed("groupCount") },
            () => new import_incomplete_date.IncompleteDate(placeholderValue.calendar, hourCycle)
          )
        );
      },
      clearEnteredKeys({ context }) {
        context.set("enteredKeys", "");
      },
      setPreviousActiveSegmentIndex(ctx) {
        (0, import_validity.goToPreviousSegment)(ctx);
      },
      setNextActiveSegmentIndex(ctx) {
        (0, import_validity.goToNextSegment)(ctx);
      },
      focusActiveSegment({ scope, context }) {
        (0, import_dom_query.raf)(() => {
          const segmentEls = dom.getSegmentEls(scope);
          const activeSegmentEl = segmentEls[context.get("activeSegmentIndex")];
          activeSegmentEl?.focus({ preventScroll: true });
        });
      },
      clearSegmentValue(params) {
        const { context, prop, event } = params;
        const index = context.get("activeIndex");
        const allSegments = prop("allSegments");
        const allSegmentTypes = Object.keys(allSegments);
        const placeholderValue = context.get("placeholderValue");
        const segment = (0, import_validity.getActiveSegment)(params) ?? event.segment;
        const type = segment.type;
        let displayValue = (0, import_validity.getActiveDisplayValue)(params);
        if (type === "dayPeriod") {
          const cleared = displayValue.clear(type);
          (0, import_validity.setDisplayValue)(params, index, cleared);
          if (cleared.isCleared(allSegmentTypes)) commitClear(params, index);
          return;
        }
        if (type === "hour") {
          const cleared = displayValue.clear(type);
          (0, import_validity.setDisplayValue)(params, index, cleared);
          if (cleared.isCleared(allSegmentTypes)) commitClear(params, index);
          return;
        }
        const enteredKeys = context.get("enteredKeys");
        const textToUse = enteredKeys !== "" ? enteredKeys : segment.text;
        const newValue = textToUse.slice(0, -1);
        if (newValue === "" || newValue === "0") {
          context.set("enteredKeys", "");
          const cleared = displayValue.clear(type);
          (0, import_validity.setDisplayValue)(params, index, cleared);
          if (cleared.isCleared(allSegmentTypes)) commitClear(params, index);
        } else {
          context.set("enteredKeys", newValue);
          displayValue = displayValue.set(type, Number(newValue), placeholderValue);
          (0, import_validity.setDisplayValue)(params, index, displayValue);
        }
      },
      invokeOnSegmentAdjust(params) {
        const { context, prop, event } = params;
        const { segment, amount } = event;
        const type = segment.type;
        const index = context.get("activeIndex");
        const allSegments = prop("allSegments");
        const allSegmentTypes = Object.keys(allSegments);
        const placeholderValue = context.get("placeholderValue");
        const displaySegmentTypes = allSegmentTypes;
        const displayValue = (0, import_validity.getActiveDisplayValue)(params);
        const next = displayValue.cycle(type, amount, placeholderValue, displaySegmentTypes);
        (0, import_validity.setDisplayValue)(params, index, next);
        if (next.isComplete(allSegmentTypes)) {
          commitValue(params, index, next);
        }
      },
      setSegmentValue(params) {
        const { event, context, refs } = params;
        const { segment, input } = event;
        refs.set("segmentToAnnounceIndex", context.get("activeSegmentIndex"));
        const index = context.get("activeIndex");
        const displayValue = (0, import_input.updateSegmentValue)(params, segment, input);
        const allSegmentTypes = Object.keys(params.prop("allSegments"));
        if (displayValue && displayValue.isComplete(allSegmentTypes) && context.get("enteredKeys") === "") {
          commitValue(params, index, displayValue);
        }
      },
      setSegmentToLowestValue(params) {
        const { event, context, prop } = params;
        const { segment } = event;
        const index = context.get("activeIndex");
        const allSegmentTypes = Object.keys(prop("allSegments"));
        const placeholderValue = context.get("placeholderValue");
        if (segment.minValue == null) return;
        const displayValue = (0, import_validity.getActiveDisplayValue)(params).set(
          segment.type,
          segment.minValue,
          placeholderValue
        );
        (0, import_validity.setDisplayValue)(params, index, displayValue);
        if (displayValue.isComplete(allSegmentTypes)) commitValue(params, index, displayValue);
      },
      setSegmentToHighestValue(params) {
        const { event, context, prop } = params;
        const { segment } = event;
        const index = context.get("activeIndex");
        const allSegmentTypes = Object.keys(prop("allSegments"));
        const placeholderValue = context.get("placeholderValue");
        if (segment.maxValue == null) return;
        const displayValue = (0, import_validity.getActiveDisplayValue)(params).set(
          segment.type,
          segment.maxValue,
          placeholderValue
        );
        (0, import_validity.setDisplayValue)(params, index, displayValue);
        if (displayValue.isComplete(allSegmentTypes)) commitValue(params, index, displayValue);
      },
      setDateValue({ context, event, prop }) {
        if (!Array.isArray(event.value)) return;
        const value = event.value.map((date) => (0, import_date_utils.constrainSegments)(date, prop("min"), prop("max")));
        context.set("value", value);
      },
      clearDateValue({ context }) {
        context.set("value", []);
      },
      setPastedValue({ context, event, prop }) {
        try {
          const parsed = (0, import_date_input.parse)(event.value);
          const constrained = (0, import_date_utils.constrainSegments)(parsed, prop("min"), prop("max"));
          const index = context.get("activeIndex");
          const values = Array.from(context.get("value"));
          values[index] = constrained;
          context.set("value", values);
        } catch {
        }
      },
      syncDisplayValues({ context, prop, computed }) {
        const value = context.get("value");
        const hourCycle = (0, import_incomplete_date.resolvedHourCycle)(prop("formatter"));
        const placeholderValue = context.get("placeholderValue");
        context.set(
          "displayValues",
          (0, import_incomplete_date.initDisplayValues)(value?.length ? value : void 0, placeholderValue, hourCycle, computed("groupCount"))
        );
      },
      syncDisplayValueHourCycle({ context, prop }) {
        const hourCycle = (0, import_incomplete_date.resolvedHourCycle)(prop("formatter"));
        const dvs = context.get("displayValues");
        context.set(
          "displayValues",
          dvs.map((dv) => dv.withHourCycle(hourCycle))
        );
      },
      syncDefaultPlaceholderValue({ prop, context }) {
        if (prop("placeholderValue") != null) return;
        const defaultPlaceholder = prop("defaultPlaceholderValue");
        if (!defaultPlaceholder) return;
        context.set("placeholderValue", defaultPlaceholder);
      },
      syncPlaceholderProp({ prop, context, computed }) {
        const propValue = prop("placeholderValue");
        if (propValue) {
          context.set("placeholderValue", propValue);
          const value = context.get("value");
          if (value?.length) {
            const hourCycle = (0, import_incomplete_date.resolvedHourCycle)(prop("formatter"));
            context.set("displayValues", (0, import_incomplete_date.initDisplayValues)(value, propValue, hourCycle, computed("groupCount")));
          }
        }
      },
      announceSegmentValue({ refs, computed, context }) {
        const announcer = refs.get("announcer");
        if (!announcer) return;
        const index = context.get("activeIndex");
        const activeSegmentIndex = context.get("activeSegmentIndex");
        const segmentIndexToUse = refs.get("segmentToAnnounceIndex") ?? activeSegmentIndex;
        refs.set("segmentToAnnounceIndex", null);
        const allSegments = computed("segments");
        const segment = allSegments[index]?.[segmentIndexToUse - (0, import_validity.getGroupOffset)(allSegments, index)];
        if (!segment || segment.type === "literal") return;
        const valueText = segment.isPlaceholder ? "Empty" : segment.text;
        announcer.announce(`${(0, import_segments.getSegmentLabel)(segment.type)}, ${valueText}`);
      },
      // On blur, if only dayPeriod is unfilled, auto-fill it and commit the value.
      // Also commit any complete display values that were deferred during partial typing.
      confirmPlaceholder(params) {
        const { context, prop, computed } = params;
        const allSegments = prop("allSegments");
        const allSegmentTypes = Object.keys(allSegments);
        const dateCount = computed("groupCount");
        const placeholderValue = context.get("placeholderValue");
        let values = Array.from(context.get("value"));
        let shouldUpdateValue = false;
        for (let i = 0; i < dateCount; i++) {
          const displayValue = context.get("displayValues")[i];
          if (!displayValue) continue;
          const allExceptDayPeriod = allSegmentTypes.filter((s) => s !== "dayPeriod");
          if (allSegments.dayPeriod && displayValue.isComplete(allExceptDayPeriod) && displayValue.dayPeriod == null) {
            const filled = displayValue.set(
              "dayPeriod",
              placeholderValue && "hour" in placeholderValue ? placeholderValue.hour >= 12 ? 1 : 0 : 0,
              placeholderValue
            );
            (0, import_validity.setDisplayValue)(params, i, filled);
            values[i] = filled.toValue(placeholderValue);
            shouldUpdateValue = true;
          } else if (displayValue.isComplete(allSegmentTypes)) {
            values[i] = displayValue.toValue(placeholderValue);
            shouldUpdateValue = true;
          }
        }
        const min = prop("min");
        const max = prop("max");
        if ((min || max) && values.length > 0) {
          values = values.map((d) => (0, import_date_utils.constrainSegments)(d, min, max));
          shouldUpdateValue = true;
        }
        if (shouldUpdateValue) {
          context.set("value", values);
        }
      }
    }
  }
});
function commitValue(params, index, displayValue) {
  const { context } = params;
  const placeholderValue = context.get("placeholderValue");
  const date = displayValue.toValue(placeholderValue);
  const values = Array.from(context.get("value"));
  values[index] = date;
  context.set("value", values);
}
function commitClear(params, index) {
  const { context } = params;
  const values = context.get("value");
  if (index < values.length) {
    context.set("value", values.slice(0, index));
  }
}
function toTargetCalendar(date, calendar) {
  if (!calendar) return date;
  if (date.calendar.identifier === calendar.identifier) return date;
  return (0, import_date.toCalendar)(date, calendar);
}
function resolveCalendar(locale, createCalendar) {
  if (!createCalendar) return void 0;
  const calendarId = new Intl.DateTimeFormat(locale).resolvedOptions().calendar;
  if (calendarId === "gregory" || calendarId === "iso8601") return void 0;
  return createCalendar(calendarId);
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  machine
});
