// @zag-js/drawer/dist/utils/drawer-session.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var __publicField = (obj, key, value) => __defNormalProp(obj, typeof key !== "symbol" ? key + "" : key, value);

// src/utils/drawer-session.ts
var drawer_session_exports = {};
__export(drawer_session_exports, {
  DrawerSwipeSession: () => DrawerSwipeSession,
  getSwipeDirectionSize: () => getSwipeDirectionSize,
  hasOpeningSwipeIntent: () => hasOpeningSwipeIntent,
  isTextSelectionInDrawer: () => isTextSelectionInDrawer,
  oppositeSwipeDirection: () => oppositeSwipeDirection,
  resolveSwipeDirection: () => resolveSwipeDirection,
  resolveSwipeProgress: () => resolveSwipeProgress,
  shouldIgnorePointerDownForDrag: () => shouldIgnorePointerDownForDrag
});
module.exports = __toCommonJS(drawer_session_exports);
var import_dom_query = require("@zag-js/dom-query");
var import_utils = require("@zag-js/utils");
var import_snap_point = require("./snap-point.js");
var import_session = require("./session.js");
var RELEASE_DISPLACEMENT_TRUST_PX = 24;
var OPEN_SWIPE_HIDDEN_VISIBLE_RATIO = 0.22;
var OPEN_SWIPE_HIDDEN_VELOCITY_MULTIPLIER = 1.25;
var OPEN_SWIPE_REVEALED_VISIBLE_RATIO = 0.5;
var OPEN_SWIPE_REVEALED_OPPOSING_MAX_ABS_VELOCITY = 650;
var DRAG_START_THRESHOLD = 0.3;
var CROSS_AXIS_BIAS = 0.58;
var SCROLL_SLACK_GATE = 0.5;
var SCROLL_SLACK_EPSILON = 1;
var SEQUENTIAL_THRESHOLD = 24;
var SNAP_VELOCITY_THRESHOLD = 400;
var SNAP_VELOCITY_MULTIPLIER = 0.4;
var MAX_SNAP_VELOCITY = 4e3;
var SWIPE_STRENGTH_MAX_DURATION_MS = 360;
var SWIPE_STRENGTH_MIN_SCALAR = 0.1;
var SWIPE_STRENGTH_MAX_SCALAR = 1;
var SWIPE_AREA_OPEN_INTENT_MIN_PX = 5;
var NO_DRAG_DATA_ATTR = "data-no-drag";
var NO_DRAG_SELECTOR = `[${NO_DRAG_DATA_ATTR}]`;
var DrawerSwipeSession = class {
  constructor(options) {
    __publicField(this, "session", new import_session.SwipeSession());
    __publicField(this, "dragOffset", null);
    __publicField(this, "preventDragOnScroll");
    this.preventDragOnScroll = options.preventDragOnScroll;
  }
  contentPointerDown(options) {
    const { event, getDoc, getContentEl, getWin, swipeDirection, canCommit, onCommit } = options;
    if (shouldIgnorePointerDownForDrag(event)) return;
    if (isTextSelectionInDrawer(getDoc(), getContentEl())) return;
    if (!canCommit()) return;
    const point = (0, import_dom_query.getEventPoint)(event);
    const shouldDefer = event.pointerType === "mouse" || event.pointerType === "pen";
    if (!shouldDefer) {
      onCommit(point);
      return;
    }
    this.session.startDeferredSwipe({
      getWin,
      pointerId: event.pointerId,
      startPoint: point,
      swipeDirection,
      onCommit,
      canCommit
    });
  }
  grabberPointerDown(options) {
    const { event, point, canCommit, onCommit } = options;
    if (shouldIgnorePointerDownForDrag(event)) return;
    this.session.cancelDeferredSwipe();
    if (!canCommit()) return;
    onCommit(point);
  }
  adjustReleaseVelocityAgainstDisplacement(velocity, displacementFromSnap) {
    const displacementSign = Math.sign(displacementFromSnap);
    const velocitySign = Math.sign(velocity);
    if (displacementSign !== 0 && Math.abs(displacementFromSnap) >= RELEASE_DISPLACEMENT_TRUST_PX && velocitySign !== 0 && velocitySign !== displacementSign) {
      return 0;
    }
    return velocity;
  }
  adjustReleaseVelocityForOpenSwipe(velocity, visibleRatio, swipeVelocityThreshold) {
    if (visibleRatio < OPEN_SWIPE_HIDDEN_VISIBLE_RATIO && velocity < 0 && Math.abs(velocity) < swipeVelocityThreshold * OPEN_SWIPE_HIDDEN_VELOCITY_MULTIPLIER) {
      return 0;
    }
    if (visibleRatio > OPEN_SWIPE_REVEALED_VISIBLE_RATIO && velocity > 0 && Math.abs(velocity) < OPEN_SWIPE_REVEALED_OPPOSING_MAX_ABS_VELOCITY) {
      return 0;
    }
    return velocity;
  }
  beginSwipe(point) {
    this.session.setStartPoint(point);
  }
  clearSwipeStart() {
    this.session.clearStartPoint();
  }
  getSwipeStart() {
    return this.session.getStartPoint();
  }
  getDragOffset() {
    return this.dragOffset;
  }
  resetDragOffset() {
    this.dragOffset = null;
  }
  resetVelocity() {
    this.session.clearVelocityTracking();
  }
  reset() {
    this.dragOffset = null;
    this.session.clear();
  }
  setDragOffset(point, resolvedActiveSnapPointOffset, direction) {
    if (!this.session.getStartPoint()) {
      this.dragOffset = null;
      return;
    }
    const axis = this.session.getGestureAxis(direction);
    const sign = this.session.getGestureSign(direction);
    let delta = this.session.getMainAxisDisplacement(point, axis, sign) - resolvedActiveSnapPointOffset;
    if (delta > 0) {
      delta = Math.sqrt(delta);
    }
    this.dragOffset = -delta;
  }
  setSwipeOpenOffset(point, contentSize, direction) {
    if (!this.session.getStartPoint()) {
      this.dragOffset = null;
      return;
    }
    const axis = this.session.getGestureAxis(direction);
    const sign = this.session.getGestureSign(direction);
    const openDisplacement = this.session.getMainAxisDisplacement(point, axis, sign);
    let dragOffset = contentSize - Math.max(0, openDisplacement);
    if (dragOffset < 0) {
      dragOffset = -Math.sqrt(Math.abs(dragOffset));
    }
    this.dragOffset = dragOffset;
  }
  canStartDrag(point, target, container, preventDragOnScroll, direction) {
    if (!(0, import_dom_query.isHTMLElement)(target)) return false;
    if (isDragExemptElement(target)) return false;
    if (!this.session.getStartPoint() || !container) return false;
    if (!preventDragOnScroll) return true;
    const axis = this.session.getGestureAxis(direction);
    const sign = this.session.getGestureSign(direction);
    const delta = this.session.getMainAxisDisplacement(point, axis, sign);
    if (Math.abs(delta) < DRAG_START_THRESHOLD) return false;
    const crossDelta = Math.abs(this.session.getCrossAxisDisplacement(point, axis));
    if (crossDelta > Math.abs(delta) * CROSS_AXIS_BIAS) {
      const crossDirection = (0, import_session.isVerticalSwipeDirection)(direction) ? "right" : "down";
      const crossScroll = getScrollInfo(target, container, crossDirection);
      if (crossScroll.availableForwardScroll > SCROLL_SLACK_GATE || crossScroll.availableBackwardScroll > SCROLL_SLACK_GATE) {
        return false;
      }
    }
    const { availableForwardScroll, availableBackwardScroll } = getScrollInfo(target, container, direction);
    if (delta > 0 && availableForwardScroll > SCROLL_SLACK_GATE || delta < 0 && availableBackwardScroll > SCROLL_SLACK_GATE) {
      return false;
    }
    return true;
  }
  resolveSnapPointOnRelease(snapPoints, snapPoint, snapToSequentialPoints, contentSize) {
    const dragOffset = this.dragOffset;
    if (dragOffset === null) return snapPoints[0]?.value ?? 1;
    const releaseVelocity = this.session.getReleaseVelocity();
    if (snapToSequentialPoints && snapPoint) {
      const ordered = [...snapPoints].sort((a, b) => a.offset - b.offset);
      let currentIndex = 0;
      let closestDist = Math.abs(snapPoint.offset - ordered[0].offset);
      for (let i = 1; i < ordered.length; i++) {
        const dist = Math.abs(snapPoint.offset - ordered[i].offset);
        if (dist < closestDist) {
          closestDist = dist;
          currentIndex = i;
        }
      }
      const currentPoint = ordered[currentIndex];
      const delta = dragOffset - currentPoint.offset;
      const dragDirection = Math.sign(delta);
      const velocityAdjusted = this.adjustReleaseVelocityAgainstDisplacement(releaseVelocity, delta);
      const velocityDirection = Math.sign(velocityAdjusted);
      let targetSnapPoint = currentPoint;
      let effectiveTargetOffset = dragOffset;
      const shouldAdvance = dragDirection !== 0 && velocityDirection === dragDirection && Math.abs(velocityAdjusted) >= SNAP_VELOCITY_THRESHOLD;
      if (shouldAdvance) {
        const adjacentIndex = Math.min(Math.max(currentIndex + dragDirection, 0), ordered.length - 1);
        if (adjacentIndex !== currentIndex) {
          targetSnapPoint = ordered[adjacentIndex];
          effectiveTargetOffset = targetSnapPoint.offset;
        } else if (dragDirection > 0) {
          return null;
        }
      } else if (delta > SEQUENTIAL_THRESHOLD) {
        const nextPoint = ordered[Math.min(currentIndex + 1, ordered.length - 1)];
        if (nextPoint) {
          targetSnapPoint = nextPoint;
          effectiveTargetOffset = nextPoint.offset;
        }
      } else if (delta < -SEQUENTIAL_THRESHOLD) {
        const prevPoint = ordered[Math.max(currentIndex - 1, 0)];
        if (prevPoint) {
          targetSnapPoint = prevPoint;
          effectiveTargetOffset = prevPoint.offset;
        }
      }
      const closeDistance = Math.abs(effectiveTargetOffset - contentSize);
      const snapDistance = Math.abs(effectiveTargetOffset - targetSnapPoint.offset);
      if (closeDistance < snapDistance) return null;
      return targetSnapPoint.value;
    }
    const snapRestOffset = snapPoint?.offset ?? 0;
    const velocity = this.adjustReleaseVelocityAgainstDisplacement(releaseVelocity, dragOffset - snapRestOffset);
    let targetOffset = dragOffset;
    if (Math.abs(velocity) >= SNAP_VELOCITY_THRESHOLD) {
      const clamped = (0, import_utils.clampValue)(velocity, -MAX_SNAP_VELOCITY, MAX_SNAP_VELOCITY);
      targetOffset += clamped * SNAP_VELOCITY_MULTIPLIER;
      targetOffset = Math.max(0, targetOffset);
    }
    return (0, import_snap_point.findClosestSnapPoint)(targetOffset, snapPoints)?.value ?? null;
  }
  shouldOpenOnRelease(contentSize, swipeVelocityThreshold, openThreshold) {
    const dragOffset = this.dragOffset;
    if (dragOffset === null || contentSize === null) return false;
    const visibleSize = contentSize - dragOffset;
    const visibleRatio = visibleSize / contentSize;
    const velocity = this.adjustReleaseVelocityForOpenSwipe(
      this.session.getReleaseVelocity(),
      visibleRatio,
      swipeVelocityThreshold
    );
    return velocity < 0 && Math.abs(velocity) >= swipeVelocityThreshold || visibleSize >= contentSize * openThreshold;
  }
  shouldDismissOnRelease(contentSize, snapPoints, resolvedSnapOffset) {
    const dragOffset = this.dragOffset;
    if (dragOffset === null || contentSize === null) return false;
    const velocity = this.adjustReleaseVelocityAgainstDisplacement(
      this.session.getReleaseVelocity(),
      dragOffset - resolvedSnapOffset
    );
    const visibleSize = contentSize - dragOffset;
    if (visibleSize <= 0) return true;
    let targetOffset = dragOffset;
    if (Math.abs(velocity) >= SNAP_VELOCITY_THRESHOLD) {
      const clamped = (0, import_utils.clampValue)(velocity, -MAX_SNAP_VELOCITY, MAX_SNAP_VELOCITY);
      targetOffset += clamped * SNAP_VELOCITY_MULTIPLIER;
      targetOffset = Math.max(0, targetOffset);
    }
    const closest = (0, import_snap_point.findClosestSnapPoint)(targetOffset, snapPoints);
    if (!closest) return false;
    const closeDistance = Math.abs(targetOffset - contentSize);
    const snapDistance = Math.abs(targetOffset - closest.offset);
    return closeDistance < snapDistance;
  }
  getSwipeStrength(targetOffset, resolvedSnapOffset = null) {
    const dragOffset = this.dragOffset;
    if (dragOffset === null) return SWIPE_STRENGTH_MAX_SCALAR;
    let velocity = this.session.getReleaseVelocity();
    if (resolvedSnapOffset != null) {
      velocity = this.adjustReleaseVelocityAgainstDisplacement(velocity, dragOffset - resolvedSnapOffset);
    }
    const distance = Math.abs(dragOffset - targetOffset);
    const absVelocity = Math.abs(velocity);
    if (absVelocity <= 0 || distance <= 0) return SWIPE_STRENGTH_MAX_SCALAR;
    const estimatedTimeMs = distance / absVelocity * 1e3;
    const normalized = (0, import_utils.clampValue)(estimatedTimeMs / SWIPE_STRENGTH_MAX_DURATION_MS, 0, 1);
    return SWIPE_STRENGTH_MIN_SCALAR + normalized * (SWIPE_STRENGTH_MAX_SCALAR - SWIPE_STRENGTH_MIN_SCALAR);
  }
  bindDragTracking(options) {
    const { getDoc, getContentEl, getSwipeAreaEl, swipeDirection, onMove, onEnd, onCancel } = options;
    const preventDragOnScroll = this.preventDragOnScroll;
    const isVertical = (0, import_session.isVerticalSwipeDirection)(swipeDirection);
    let lastAxis = 0;
    return this.session.bind({
      getDoc,
      getSelectionTarget: getContentEl,
      swipeDirection,
      onMove,
      onEnd,
      onCancel,
      cancelOnInterrupt: ({ reason, target }) => {
        if (reason !== "lost-pointer-capture") return true;
        return isWithinDrawerInteractionSurface(target, getContentEl(), getSwipeAreaEl());
      },
      onStart({ pointerType, point }) {
        if (pointerType !== "touch") return;
        lastAxis = isVertical ? point.y : point.x;
      },
      preventDefault({ event, pointerType, point, target }) {
        if (pointerType !== "touch") return false;
        const contentEl = getContentEl();
        const resolvedTarget = target ?? event.target;
        if (!preventDragOnScroll()) return false;
        if (!contentEl || !resolvedTarget || isDragExemptElement(resolvedTarget)) return false;
        const scrollParent = findClosestScrollableAncestorOnSwipeAxis(resolvedTarget, contentEl, swipeDirection);
        if (scrollParent) {
          const currentAxis = isVertical ? point.y : point.x;
          const shouldPrevent = shouldPreventTouchScroll({
            scrollParent,
            swipeDirection,
            lastMainAxis: lastAxis,
            currentMainAxis: currentAxis
          });
          lastAxis = currentAxis;
          return shouldPrevent;
        }
        lastAxis = isVertical ? point.y : point.x;
        return false;
      }
    });
  }
  bindSwipeOpenTracking(options) {
    const { getDoc, getContentEl, getSwipeAreaEl, swipeDirection, onMove, onEnd, onCancel } = options;
    return this.session.bind({
      getDoc,
      getSelectionTarget: getSwipeAreaEl,
      swipeDirection,
      onMove({ point }) {
        onMove({ point });
      },
      onEnd,
      onCancel,
      cancelOnInterrupt: ({ reason, target }) => {
        if (reason !== "lost-pointer-capture") return true;
        return isWithinDrawerInteractionSurface(target, getContentEl(), getSwipeAreaEl());
      }
    });
  }
};
function isWithinDrawerInteractionSurface(target, contentEl, swipeAreaEl) {
  if (!target) return false;
  return (0, import_dom_query.contains)(contentEl, target) || (0, import_dom_query.contains)(swipeAreaEl, target);
}
var oppositeSwipeDirection = {
  up: "down",
  down: "up",
  start: "end",
  end: "start"
};
function resolveSwipeDirection(direction, dir) {
  if (direction === "start") return dir === "rtl" ? "right" : "left";
  if (direction === "end") return dir === "rtl" ? "left" : "right";
  return direction;
}
function getSwipeDirectionSize(rect, direction) {
  return (0, import_session.isVerticalSwipeDirection)(direction) ? rect.height : rect.width;
}
function resolveSwipeProgress(contentSize, dragOffset, snapPointOffset) {
  if (!contentSize || contentSize <= 0) return 0;
  const currentOffset = dragOffset ?? snapPointOffset;
  return (0, import_utils.clampValue)(currentOffset / contentSize, 0, 1);
}
function hasOpeningSwipeIntent(start, current, direction) {
  const axis = (0, import_session.isVerticalSwipeDirection)(direction) ? "y" : "x";
  const sign = (0, import_session.isNegativeSwipeDirection)(direction) ? -1 : 1;
  const displacement = (start[axis] - current[axis]) * sign;
  return displacement > SWIPE_AREA_OPEN_INTENT_MIN_PX;
}
function overflowAllowsScroll(overflow) {
  return overflow === "auto" || overflow === "scroll" || overflow === "overlay";
}
function canScrollAlongY(el) {
  const style = (0, import_dom_query.getComputedStyle)(el);
  if (!overflowAllowsScroll(style.overflowY)) return false;
  return el.scrollHeight > el.clientHeight + SCROLL_SLACK_EPSILON;
}
function canScrollAlongX(el) {
  const style = (0, import_dom_query.getComputedStyle)(el);
  if (!overflowAllowsScroll(style.overflowX)) return false;
  return el.scrollWidth > el.clientWidth + SCROLL_SLACK_EPSILON;
}
function canScrollOnSwipeAxis(el, direction) {
  return (0, import_session.isVerticalSwipeDirection)(direction) ? canScrollAlongY(el) : canScrollAlongX(el);
}
function findClosestScrollableAncestorOnSwipeAxis(target, container, direction) {
  if (!container) return null;
  let el = target;
  while (el && el !== container) {
    if (canScrollOnSwipeAxis(el, direction)) return el;
    el = el.parentElement;
  }
  return null;
}
function getScrollInfo(target, container, direction) {
  let availableForwardScroll = 0;
  let availableBackwardScroll = 0;
  if (!container) return { availableForwardScroll, availableBackwardScroll };
  const vertical = (0, import_session.isVerticalSwipeDirection)(direction);
  let element = target;
  while (element) {
    if (vertical ? canScrollAlongY(element) : canScrollAlongX(element)) {
      const clientSize = vertical ? element.clientHeight : element.clientWidth;
      const scrollPos = vertical ? element.scrollTop : element.scrollLeft;
      const scrollSize = vertical ? element.scrollHeight : element.scrollWidth;
      const scrolled = scrollSize - scrollPos - clientSize;
      availableForwardScroll += scrolled;
      availableBackwardScroll += scrollPos;
    }
    if (element === container || element === element.ownerDocument.documentElement) break;
    element = element.parentElement;
  }
  return { availableForwardScroll, availableBackwardScroll };
}
function shouldPreventTouchScroll(options) {
  const { scrollParent, swipeDirection, lastMainAxis, currentMainAxis } = options;
  const vertical = (0, import_session.isVerticalSwipeDirection)(swipeDirection);
  const movingPositive = currentMainAxis > lastMainAxis;
  if (vertical) {
    const scrollPos = scrollParent.scrollTop;
    const maxScroll = Math.max(0, scrollParent.scrollHeight - scrollParent.clientHeight);
    if (swipeDirection === "down") return scrollPos <= SCROLL_SLACK_EPSILON && movingPositive;
    if (swipeDirection === "up") return scrollPos >= maxScroll - SCROLL_SLACK_EPSILON && !movingPositive;
  } else {
    const scrollPos = scrollParent.scrollLeft;
    const maxScroll = Math.max(0, scrollParent.scrollWidth - scrollParent.clientWidth);
    if (swipeDirection === "right") return scrollPos <= SCROLL_SLACK_EPSILON && movingPositive;
    if (swipeDirection === "left") return scrollPos >= maxScroll - SCROLL_SLACK_EPSILON && !movingPositive;
  }
  return false;
}
function isDragExemptElement(el) {
  if (!(0, import_dom_query.isHTMLElement)(el)) return false;
  if (el.closest(NO_DRAG_SELECTOR)) return true;
  let node = el;
  while (node) {
    if ((0, import_dom_query.isEditableElement)(node)) return true;
    node = node.parentElement;
  }
  const input = el.closest("input");
  if ((0, import_dom_query.isInputElement)(input)) {
    const type = input.type;
    if (type === "range" || type === "file") return true;
  }
  return false;
}
function isTextSelectionInDrawer(doc, contentEl) {
  if (!contentEl) return false;
  const selection = doc.getSelection();
  if (!selection || selection.rangeCount === 0 || selection.isCollapsed) return false;
  try {
    const range = selection.getRangeAt(0);
    if ((0, import_dom_query.contains)(contentEl, range.commonAncestorContainer)) return true;
    if ((0, import_dom_query.contains)(contentEl, selection.anchorNode)) return true;
    if ((0, import_dom_query.contains)(contentEl, selection.focusNode)) return true;
    if (typeof range.intersectsNode === "function" && range.intersectsNode(contentEl)) return true;
  } catch {
    return false;
  }
  return false;
}
function isDragExemptFromComposedPath(event) {
  const path = typeof event.composedPath === "function" ? event.composedPath() : [];
  for (const node of path) {
    if (isDragExemptElement(node)) return true;
  }
  return isDragExemptElement(event.target);
}
function shouldIgnorePointerDownForDrag(event) {
  if (!(0, import_dom_query.isLeftClick)(event)) return true;
  const target = (0, import_dom_query.getEventTarget)(event);
  if (target?.hasAttribute(NO_DRAG_DATA_ATTR) || target?.closest(NO_DRAG_SELECTOR)) return true;
  return isDragExemptFromComposedPath(event);
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  DrawerSwipeSession,
  getSwipeDirectionSize,
  hasOpeningSwipeIntent,
  isTextSelectionInDrawer,
  oppositeSwipeDirection,
  resolveSwipeDirection,
  resolveSwipeProgress,
  shouldIgnorePointerDownForDrag
});
