// @zag-js/drawer/dist/utils/session.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var __publicField = (obj, key, value) => __defNormalProp(obj, typeof key !== "symbol" ? key + "" : key, value);

// src/utils/session.ts
var session_exports = {};
__export(session_exports, {
  SwipeSession: () => SwipeSession,
  isNegativeSwipeDirection: () => isNegativeSwipeDirection,
  isVerticalSwipeDirection: () => isVerticalSwipeDirection
});
module.exports = __toCommonJS(session_exports);
var import_dom_query = require("@zag-js/dom-query");
var VELOCITY_WINDOW_MS = 100;
var MAX_RELEASE_VELOCITY_AGE_MS = 80;
var MIN_GESTURE_DURATION_MS = 50;
var MIN_VELOCITY_SAMPLES = 2;
var SAMPLE_BUFFER_COMPACT_THRESHOLD = 8;
var DEFERRED_DRAG_MIN_MAIN_AXIS_PX = 6;
var DEFERRED_DRAG_MAIN_OVER_CROSS_RATIO = 1.35;
function isVerticalSwipeDirection(direction) {
  return direction === "down" || direction === "up";
}
function isNegativeSwipeDirection(direction) {
  return direction === "up" || direction === "left";
}
var SwipeSession = class {
  constructor() {
    __publicField(this, "startPoint", null);
    __publicField(this, "velocity", null);
    __publicField(this, "samples", []);
    __publicField(this, "sampleStartIndex", 0);
    __publicField(this, "gestureStartAxis", null);
    __publicField(this, "gestureStartTime", null);
    __publicField(this, "gestureSign", 1);
    __publicField(this, "pendingSwipe", null);
  }
  setStartPoint(point) {
    this.startPoint = point;
  }
  clearStartPoint() {
    this.startPoint = null;
  }
  getStartPoint() {
    return this.startPoint;
  }
  getGestureAxis(direction) {
    return direction === "left" || direction === "right" ? "x" : "y";
  }
  getGestureSign(direction) {
    return isNegativeSwipeDirection(direction) ? -1 : 1;
  }
  getAxisValue(point, axis) {
    return point[axis];
  }
  getMainAxisDisplacement(point, axis, sign) {
    if (!this.startPoint) return 0;
    const startAxis = this.getAxisValue(this.startPoint, axis);
    const currentAxis = this.getAxisValue(point, axis);
    return (startAxis - currentAxis) * sign;
  }
  getCrossAxisDisplacement(point, axis) {
    if (!this.startPoint) return 0;
    const crossAxis = axis === "x" ? "y" : "x";
    const startAxis = this.getAxisValue(this.startPoint, crossAxis);
    const currentAxis = this.getAxisValue(point, crossAxis);
    return currentAxis - startAxis;
  }
  track(point, axis, sign) {
    const axisValue = this.getAxisValue(point, axis);
    const now = performance.now();
    if (this.gestureStartAxis === null) {
      this.gestureStartAxis = axisValue;
      this.gestureStartTime = now;
      this.gestureSign = sign;
    }
    this.samples.push({ axis: axisValue, time: now });
    const cutoff = now - VELOCITY_WINDOW_MS;
    while (this.sampleStartIndex < this.samples.length && this.samples[this.sampleStartIndex].time < cutoff) {
      this.sampleStartIndex += 1;
    }
    if (this.sampleStartIndex >= SAMPLE_BUFFER_COMPACT_THRESHOLD) {
      this.samples = this.samples.slice(this.sampleStartIndex);
      this.sampleStartIndex = 0;
    }
    const sampleCount = this.samples.length - this.sampleStartIndex;
    if (sampleCount < MIN_VELOCITY_SAMPLES) {
      this.velocity = 0;
      return;
    }
    const oldest = this.samples[this.sampleStartIndex];
    const newest = this.samples[this.samples.length - 1];
    const dt = newest.time - oldest.time;
    if (dt <= 0) {
      this.velocity = 0;
      return;
    }
    const delta = (newest.axis - oldest.axis) * sign;
    const velocity = delta / dt * 1e3;
    this.velocity = Number.isFinite(velocity) ? velocity : 0;
  }
  getReleaseVelocity() {
    const now = performance.now();
    const sampleCount = this.samples.length - this.sampleStartIndex;
    if (sampleCount >= MIN_VELOCITY_SAMPLES) {
      const newest = this.samples[this.samples.length - 1];
      if (now - newest.time <= MAX_RELEASE_VELOCITY_AGE_MS) {
        return this.velocity ?? 0;
      }
    }
    if (this.gestureStartAxis !== null && this.gestureStartTime !== null) {
      const lastSample = this.samples[this.samples.length - 1];
      if (lastSample) {
        const dt = Math.max(lastSample.time - this.gestureStartTime, MIN_GESTURE_DURATION_MS);
        const delta = (lastSample.axis - this.gestureStartAxis) * this.gestureSign;
        const velocity = delta / dt * 1e3;
        return Number.isFinite(velocity) ? velocity : 0;
      }
    }
    return this.velocity ?? 0;
  }
  clearVelocityTracking() {
    this.samples = [];
    this.sampleStartIndex = 0;
    this.velocity = null;
    this.gestureStartAxis = null;
    this.gestureStartTime = null;
    this.gestureSign = 1;
  }
  clear() {
    this.cancelDeferredSwipe();
    this.clearStartPoint();
    this.clearVelocityTracking();
  }
  startDeferredSwipe(options) {
    const { getWin, pointerId, startPoint, swipeDirection, onCommit, canCommit, onCancel } = options;
    this.cancelDeferredSwipe();
    const win = getWin();
    const vertical = isVerticalSwipeDirection(swipeDirection);
    const onMove = (event) => {
      if (event.pointerId !== pointerId) return;
      const dx = event.clientX - startPoint.x;
      const dy = event.clientY - startPoint.y;
      const mainDelta = vertical ? dy : dx;
      const crossDelta = vertical ? dx : dy;
      const absMain = Math.abs(mainDelta);
      const absCross = Math.abs(crossDelta);
      if (absMain >= DEFERRED_DRAG_MIN_MAIN_AXIS_PX && absMain >= absCross * DEFERRED_DRAG_MAIN_OVER_CROSS_RATIO) {
        if (!canCommit || canCommit()) {
          onCommit(startPoint);
        }
        this.cancelDeferredSwipe();
      }
    };
    const onEnd = (event) => {
      if (event.pointerId !== pointerId) return;
      onCancel?.();
      this.cancelDeferredSwipe();
    };
    const cleanups = [
      (0, import_dom_query.addDomEvent)(win, "pointermove", onMove, { capture: true }),
      (0, import_dom_query.addDomEvent)(win, "pointerup", onEnd, { capture: true }),
      (0, import_dom_query.addDomEvent)(win, "pointercancel", onEnd, { capture: true }),
      (0, import_dom_query.addDomEvent)(win, "lostpointercapture", onEnd, { capture: true })
    ];
    this.pendingSwipe = { pointerId, startPoint, cleanups };
  }
  cancelDeferredSwipe() {
    if (!this.pendingSwipe) return;
    this.pendingSwipe.cleanups.forEach((cleanup) => cleanup());
    this.pendingSwipe = null;
  }
  bind(options) {
    const {
      getDoc,
      getSelectionTarget,
      swipeDirection,
      onStart,
      onMove,
      onEnd,
      onCancel,
      preventDefault,
      cancelOnInterrupt
    } = options;
    const doc = getDoc();
    let usingTouchEvents = false;
    let restoreSelection;
    const axis = this.getGestureAxis(swipeDirection);
    const sign = this.getGestureSign(swipeDirection);
    const trackPoint = (point) => {
      this.track(point, axis, sign);
    };
    const startSelectionGuard = () => {
      restoreSelection ?? (restoreSelection = (0, import_dom_query.disableTextSelection)({
        doc,
        target: getSelectionTarget?.()
      }));
    };
    const stopSelectionGuard = () => {
      restoreSelection?.();
      restoreSelection = void 0;
    };
    function onPointerMove(event) {
      if (event.pointerType === "touch" && usingTouchEvents) return;
      const point = (0, import_dom_query.getEventPoint)(event);
      const target = (0, import_dom_query.getEventTarget)(event);
      startSelectionGuard();
      trackPoint(point);
      onMove({
        point,
        target,
        event,
        pointerType: event.pointerType,
        axis,
        swipeDirection
      });
    }
    function onPointerUp(event) {
      if (event.pointerType === "touch" && usingTouchEvents) {
        usingTouchEvents = false;
        return;
      }
      stopSelectionGuard();
      onEnd({ point: (0, import_dom_query.getEventPoint)(event), swipeDirection });
    }
    function onPointerCancel(event) {
      if (event.pointerType === "touch" && usingTouchEvents) {
        usingTouchEvents = false;
        return;
      }
      stopSelectionGuard();
      onCancel();
    }
    function onTouchStartEvent(event) {
      if (!event.touches[0]) return;
      usingTouchEvents = true;
      const point = (0, import_dom_query.getEventPoint)(event);
      const target = (0, import_dom_query.getEventTarget)(event);
      onStart?.({ point, target, event, pointerType: "touch", axis, swipeDirection });
    }
    function onTouchMoveEvent(event) {
      if (!event.touches[0]) return;
      usingTouchEvents = true;
      const point = (0, import_dom_query.getEventPoint)(event);
      const target = (0, import_dom_query.getEventTarget)(event);
      const details = { point, target, event, pointerType: "touch", axis, swipeDirection };
      if (preventDefault?.(details) && event.cancelable) {
        event.preventDefault();
      }
      startSelectionGuard();
      trackPoint(point);
      onMove(details);
    }
    function onTouchEnd(event) {
      if (event.touches.length !== 0) return;
      stopSelectionGuard();
      onEnd({ point: (0, import_dom_query.getEventPoint)(event), swipeDirection });
    }
    function onTouchCancel() {
      stopSelectionGuard();
      onCancel();
    }
    function onVisibilityChange() {
      if (doc.visibilityState !== "hidden") return;
      const shouldCancel = cancelOnInterrupt?.({
        reason: "visibility-hidden",
        event: doc,
        target: null,
        pointerType: null
      });
      if (shouldCancel === false) return;
      stopSelectionGuard();
      onCancel();
    }
    function onLostPointerCapture(event) {
      if (event.pointerType === "touch") return;
      const target = (0, import_dom_query.getEventTarget)(event);
      const shouldCancel = cancelOnInterrupt?.({
        reason: "lost-pointer-capture",
        event,
        target,
        pointerType: event.pointerType
      });
      if (shouldCancel === false) return;
      onCancel();
    }
    const cleanups = [
      (0, import_dom_query.addDomEvent)(doc, "pointermove", onPointerMove),
      (0, import_dom_query.addDomEvent)(doc, "pointerup", onPointerUp),
      (0, import_dom_query.addDomEvent)(doc, "pointercancel", onPointerCancel),
      (0, import_dom_query.addDomEvent)(doc, "touchstart", onTouchStartEvent, { capture: true, passive: false }),
      (0, import_dom_query.addDomEvent)(doc, "touchmove", onTouchMoveEvent, { capture: true, passive: false }),
      (0, import_dom_query.addDomEvent)(doc, "touchend", onTouchEnd, { capture: true }),
      (0, import_dom_query.addDomEvent)(doc, "touchcancel", onTouchCancel, { capture: true }),
      (0, import_dom_query.addDomEvent)(doc, "visibilitychange", onVisibilityChange),
      (0, import_dom_query.addDomEvent)(doc, "lostpointercapture", onLostPointerCapture, true)
    ];
    return () => {
      stopSelectionGuard();
      cleanups.forEach((cleanup) => cleanup());
    };
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SwipeSession,
  isNegativeSwipeDirection,
  isVerticalSwipeDirection
});
