// @ark-ui/react/dist/components/collection/use-list-selection.cjs
"use client";
const require_use_event = require("../../utils/use-event.cjs");
let react = require("react");
let _zag_js_collection = require("@zag-js/collection");
//#region src/components/collection/use-list-selection.ts
function useListSelection(props) {
	const { collection, selectionMode = "single", deselectable = true, initialSelectedValues = [], resetOnCollectionChange = false } = props;
	const createSelection = (0, react.useCallback)((values = []) => {
		const selection = new _zag_js_collection.Selection(values);
		selection.selectionMode = selectionMode;
		selection.deselectable = deselectable;
		return selection;
	}, [selectionMode, deselectable]);
	const [selection, setSelectionState] = (0, react.useState)(() => createSelection(initialSelectedValues));
	(0, react.useEffect)(() => {
		if (resetOnCollectionChange) setSelectionState(createSelection());
	}, [
		collection.toString(),
		resetOnCollectionChange,
		createSelection
	]);
	return {
		selectedValues: (0, react.useMemo)(() => Array.from(selection), [selection]),
		isEmpty: (0, react.useMemo)(() => selection.isEmpty(), [selection]),
		firstSelectedValue: (0, react.useMemo)(() => selection.firstSelectedValue(collection), [selection, collection]),
		lastSelectedValue: (0, react.useMemo)(() => selection.lastSelectedValue(collection), [selection, collection]),
		isSelected: require_use_event.useEvent((value) => {
			return selection.isSelected(value);
		}),
		isAllSelected: require_use_event.useEvent(() => {
			const allValues = collection.getValues();
			return allValues.length > 0 && allValues.every((value) => selection.isSelected(value));
		}),
		isSomeSelected: require_use_event.useEvent(() => {
			return collection.getValues().some((value) => selection.isSelected(value));
		}),
		canSelect: require_use_event.useEvent((value) => {
			return selection.canSelect(collection, value);
		}),
		select: require_use_event.useEvent((value, forceToggle) => {
			setSelectionState(selection.select(collection, value, forceToggle));
		}),
		deselect: require_use_event.useEvent((value) => {
			setSelectionState(selection.deselect(value));
		}),
		toggle: require_use_event.useEvent((value) => {
			setSelectionState(selection.toggleSelection(collection, value));
		}),
		replace: require_use_event.useEvent((value) => {
			setSelectionState(selection.replaceSelection(collection, value));
		}),
		extend: require_use_event.useEvent((anchorValue, targetValue) => {
			setSelectionState(selection.extendSelection(collection, anchorValue, targetValue));
		}),
		setSelectedValues: require_use_event.useEvent((values) => {
			setSelectionState(selection.setSelection(values));
		}),
		clear: require_use_event.useEvent(() => {
			setSelectionState(selection.clearSelection());
		}),
		resetSelection: require_use_event.useEvent(() => {
			setSelectionState(createSelection());
		})
	};
}
//#endregion
exports.useListSelection = useListSelection;
