// @ark-ui/react/dist/components/collection/use-list-selection.js
"use client";
import { useEvent } from "../../utils/use-event.js";
import { useCallback, useEffect, useMemo, useState } from "react";
import { Selection } from "@zag-js/collection";
//#region src/components/collection/use-list-selection.ts
function useListSelection(props) {
	const { collection, selectionMode = "single", deselectable = true, initialSelectedValues = [], resetOnCollectionChange = false } = props;
	const createSelection = useCallback((values = []) => {
		const selection = new Selection(values);
		selection.selectionMode = selectionMode;
		selection.deselectable = deselectable;
		return selection;
	}, [selectionMode, deselectable]);
	const [selection, setSelectionState] = useState(() => createSelection(initialSelectedValues));
	useEffect(() => {
		if (resetOnCollectionChange) setSelectionState(createSelection());
	}, [
		collection.toString(),
		resetOnCollectionChange,
		createSelection
	]);
	return {
		selectedValues: useMemo(() => Array.from(selection), [selection]),
		isEmpty: useMemo(() => selection.isEmpty(), [selection]),
		firstSelectedValue: useMemo(() => selection.firstSelectedValue(collection), [selection, collection]),
		lastSelectedValue: useMemo(() => selection.lastSelectedValue(collection), [selection, collection]),
		isSelected: useEvent((value) => {
			return selection.isSelected(value);
		}),
		isAllSelected: useEvent(() => {
			const allValues = collection.getValues();
			return allValues.length > 0 && allValues.every((value) => selection.isSelected(value));
		}),
		isSomeSelected: useEvent(() => {
			return collection.getValues().some((value) => selection.isSelected(value));
		}),
		canSelect: useEvent((value) => {
			return selection.canSelect(collection, value);
		}),
		select: useEvent((value, forceToggle) => {
			setSelectionState(selection.select(collection, value, forceToggle));
		}),
		deselect: useEvent((value) => {
			setSelectionState(selection.deselect(value));
		}),
		toggle: useEvent((value) => {
			setSelectionState(selection.toggleSelection(collection, value));
		}),
		replace: useEvent((value) => {
			setSelectionState(selection.replaceSelection(collection, value));
		}),
		extend: useEvent((anchorValue, targetValue) => {
			setSelectionState(selection.extendSelection(collection, anchorValue, targetValue));
		}),
		setSelectedValues: useEvent((values) => {
			setSelectionState(selection.setSelection(values));
		}),
		clear: useEvent(() => {
			setSelectionState(selection.clearSelection());
		}),
		resetSelection: useEvent(() => {
			setSelectionState(createSelection());
		})
	};
}
//#endregion
export { useListSelection };
