// @ark-ui/react/dist/components/accordion/accordion.cjs
const require_runtime = require("../../_virtual/_rolldown/runtime.cjs");
const require_accordion_context = require("./accordion-context.cjs");
const require_accordion_item = require("./accordion-item.cjs");
const require_accordion_item_content = require("./accordion-item-content.cjs");
const require_accordion_item_context = require("./accordion-item-context.cjs");
const require_accordion_item_indicator = require("./accordion-item-indicator.cjs");
const require_accordion_item_trigger = require("./accordion-item-trigger.cjs");
const require_accordion_root = require("./accordion-root.cjs");
const require_accordion_root_provider = require("./accordion-root-provider.cjs");
//#region src/components/accordion/accordion.ts
var accordion_exports = /* @__PURE__ */ require_runtime.__exportAll({
	Context: () => require_accordion_context.AccordionContext,
	Item: () => require_accordion_item.AccordionItem,
	ItemContent: () => require_accordion_item_content.AccordionItemContent,
	ItemContext: () => require_accordion_item_context.AccordionItemContext,
	ItemIndicator: () => require_accordion_item_indicator.AccordionItemIndicator,
	ItemTrigger: () => require_accordion_item_trigger.AccordionItemTrigger,
	Root: () => require_accordion_root.AccordionRoot,
	RootProvider: () => require_accordion_root_provider.AccordionRootProvider
});
//#endregion
Object.defineProperty(exports, "accordion_exports", {
	enumerable: true,
	get: function() {
		return accordion_exports;
	}
});
