// @ark-ui/react/dist/components/accordion/accordion.js
import { __exportAll } from "../../_virtual/_rolldown/runtime.js";
import { AccordionContext } from "./accordion-context.js";
import { AccordionItem } from "./accordion-item.js";
import { AccordionItemContent } from "./accordion-item-content.js";
import { AccordionItemContext } from "./accordion-item-context.js";
import { AccordionItemIndicator } from "./accordion-item-indicator.js";
import { AccordionItemTrigger } from "./accordion-item-trigger.js";
import { AccordionRoot } from "./accordion-root.js";
import { AccordionRootProvider } from "./accordion-root-provider.js";
//#region src/components/accordion/accordion.ts
var accordion_exports = /* @__PURE__ */ __exportAll({
	Context: () => AccordionContext,
	Item: () => AccordionItem,
	ItemContent: () => AccordionItemContent,
	ItemContext: () => AccordionItemContext,
	ItemIndicator: () => AccordionItemIndicator,
	ItemTrigger: () => AccordionItemTrigger,
	Root: () => AccordionRoot,
	RootProvider: () => AccordionRootProvider
});
//#endregion
export { accordion_exports };
