// @ark-ui/react/dist/components/tabs/index.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const require_use_tabs_context = require("./use-tabs-context.cjs");
const require_tab_content = require("./tab-content.cjs");
const require_tab_indicator = require("./tab-indicator.cjs");
const require_tab_list = require("./tab-list.cjs");
const require_tab_trigger = require("./tab-trigger.cjs");
const require_tabs_context = require("./tabs-context.cjs");
const require_use_tabs = require("./use-tabs.cjs");
const require_tabs_root = require("./tabs-root.cjs");
const require_tabs_root_provider = require("./tabs-root-provider.cjs");
const require_tabs = require("./tabs.cjs");
let _zag_js_tabs = require("@zag-js/tabs");
exports.TabContent = require_tab_content.TabContent;
exports.TabIndicator = require_tab_indicator.TabIndicator;
exports.TabList = require_tab_list.TabList;
exports.TabTrigger = require_tab_trigger.TabTrigger;
Object.defineProperty(exports, "Tabs", {
	enumerable: true,
	get: function() {
		return require_tabs.tabs_exports;
	}
});
exports.TabsContext = require_tabs_context.TabsContext;
exports.TabsRoot = require_tabs_root.TabsRoot;
exports.TabsRootProvider = require_tabs_root_provider.TabsRootProvider;
Object.defineProperty(exports, "tabsAnatomy", {
	enumerable: true,
	get: function() {
		return _zag_js_tabs.anatomy;
	}
});
exports.useTabs = require_use_tabs.useTabs;
exports.useTabsContext = require_use_tabs_context.useTabsContext;
