// @ark-ui/react/dist/components/tabs/tabs.cjs
const require_runtime = require("../../_virtual/_rolldown/runtime.cjs");
const require_tab_content = require("./tab-content.cjs");
const require_tab_indicator = require("./tab-indicator.cjs");
const require_tab_list = require("./tab-list.cjs");
const require_tab_trigger = require("./tab-trigger.cjs");
const require_tabs_context = require("./tabs-context.cjs");
const require_tabs_root = require("./tabs-root.cjs");
const require_tabs_root_provider = require("./tabs-root-provider.cjs");
//#region src/components/tabs/tabs.ts
var tabs_exports = /* @__PURE__ */ require_runtime.__exportAll({
	Content: () => require_tab_content.TabContent,
	Context: () => require_tabs_context.TabsContext,
	Indicator: () => require_tab_indicator.TabIndicator,
	List: () => require_tab_list.TabList,
	Root: () => require_tabs_root.TabsRoot,
	RootProvider: () => require_tabs_root_provider.TabsRootProvider,
	Trigger: () => require_tab_trigger.TabTrigger
});
//#endregion
Object.defineProperty(exports, "tabs_exports", {
	enumerable: true,
	get: function() {
		return tabs_exports;
	}
});
