// @zag-js/pin-input/dist/pin-input.connect.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/pin-input.connect.ts
var pin_input_connect_exports = {};
__export(pin_input_connect_exports, {
  connect: () => connect
});
module.exports = __toCommonJS(pin_input_connect_exports);
var import_dom_query = require("@zag-js/dom-query");
var import_utils = require("@zag-js/utils");
var import_pin_input = require("./pin-input.anatomy.js");
var dom = __toESM(require("./pin-input.dom.js"));
var import_pin_input2 = require("./pin-input.utils.js");
function connect(service, normalize) {
  const { send, context, computed, prop, scope } = service;
  const complete = computed("isValueComplete");
  const disabled = !!prop("disabled");
  const readOnly = !!prop("readOnly");
  const invalid = !!prop("invalid");
  const required = !!prop("required");
  const translations = prop("translations");
  const focusedIndex = context.get("focusedIndex");
  function focus() {
    dom.getFirstInputEl(scope)?.focus();
  }
  return {
    focus,
    count: context.get("count"),
    items: Array.from({ length: context.get("count") }).map((_, i) => i),
    value: context.get("value"),
    valueAsString: computed("valueAsString"),
    complete,
    setValue(value) {
      if (!Array.isArray(value)) {
        (0, import_utils.invariant)("[pin-input/setValue] value must be an array");
      }
      send({ type: "VALUE.SET", value });
    },
    clearValue() {
      send({ type: "VALUE.CLEAR" });
    },
    setValueAtIndex(index, value) {
      send({ type: "VALUE.SET", value, index });
    },
    getRootProps() {
      return normalize.element({
        dir: prop("dir"),
        ...import_pin_input.parts.root.attrs,
        id: dom.getRootId(scope),
        "data-invalid": (0, import_dom_query.dataAttr)(invalid),
        "data-disabled": (0, import_dom_query.dataAttr)(disabled),
        "data-complete": (0, import_dom_query.dataAttr)(complete),
        "data-readonly": (0, import_dom_query.dataAttr)(readOnly)
      });
    },
    getLabelProps() {
      return normalize.label({
        ...import_pin_input.parts.label.attrs,
        dir: prop("dir"),
        htmlFor: dom.getHiddenInputId(scope),
        id: dom.getLabelId(scope),
        "data-invalid": (0, import_dom_query.dataAttr)(invalid),
        "data-disabled": (0, import_dom_query.dataAttr)(disabled),
        "data-complete": (0, import_dom_query.dataAttr)(complete),
        "data-required": (0, import_dom_query.dataAttr)(required),
        "data-readonly": (0, import_dom_query.dataAttr)(readOnly),
        onClick(event) {
          event.preventDefault();
          focus();
        }
      });
    },
    getHiddenInputProps() {
      return normalize.input({
        "aria-hidden": true,
        type: "text",
        tabIndex: -1,
        id: dom.getHiddenInputId(scope),
        readOnly,
        disabled,
        required,
        name: prop("name"),
        form: prop("form"),
        style: import_dom_query.visuallyHiddenStyle,
        maxLength: computed("valueLength"),
        defaultValue: computed("valueAsString")
      });
    },
    getControlProps() {
      return normalize.element({
        ...import_pin_input.parts.control.attrs,
        dir: prop("dir"),
        id: dom.getControlId(scope)
      });
    },
    getInputProps(props) {
      const { index } = props;
      const inputType = prop("type") === "numeric" ? "tel" : "text";
      const valueLength = computed("valueLength");
      const tabbableIndex = focusedIndex !== -1 ? focusedIndex : Math.min(computed("filledValueLength"), valueLength - 1);
      return normalize.input({
        ...import_pin_input.parts.input.attrs,
        dir: prop("dir"),
        disabled,
        tabIndex: index === tabbableIndex ? 0 : -1,
        "data-disabled": (0, import_dom_query.dataAttr)(disabled),
        "data-complete": (0, import_dom_query.dataAttr)(complete),
        "data-filled": (0, import_dom_query.dataAttr)(computed("_value")[index] !== ""),
        id: dom.getInputId(scope, index.toString()),
        "data-index": index,
        "data-ownedby": dom.getRootId(scope),
        "aria-label": translations?.inputLabel?.(index, computed("valueLength")),
        inputMode: prop("otp") || prop("type") === "numeric" ? "numeric" : "text",
        "aria-invalid": (0, import_dom_query.ariaAttr)(invalid),
        "data-invalid": (0, import_dom_query.dataAttr)(invalid),
        enterKeyHint: index === valueLength - 1 ? "done" : "next",
        type: prop("mask") ? "password" : inputType,
        defaultValue: computed("_value")[index] || "",
        readOnly,
        autoCapitalize: "none",
        autoComplete: prop("otp") ? "one-time-code" : "off",
        placeholder: focusedIndex === index ? "" : prop("placeholder"),
        onPaste(event) {
          let pastedValue = event.clipboardData?.getData("text/plain");
          if (!pastedValue) return;
          const transformer = prop("sanitizeValue");
          if (transformer) pastedValue = transformer(pastedValue);
          const isValid = (0, import_pin_input2.isValidValue)(pastedValue, prop("type"), prop("pattern"));
          if (!isValid) {
            send({ type: "VALUE.INVALID", value: pastedValue });
            event.preventDefault();
            return;
          }
          event.preventDefault();
          send({ type: "INPUT.PASTE", value: pastedValue });
        },
        onBeforeInput(event) {
          try {
            const value = (0, import_dom_query.getBeforeInputValue)(event);
            const isValid = (0, import_pin_input2.isValidValue)(value, prop("type"), prop("pattern"));
            if (!isValid) {
              send({ type: "VALUE.INVALID", value });
              event.preventDefault();
            }
            if (value.length > 1) {
              event.currentTarget.setSelectionRange(0, 1, "forward");
            }
          } catch {
          }
        },
        onChange(event) {
          const evt = (0, import_dom_query.getNativeEvent)(event);
          const { value } = event.currentTarget;
          if (evt.inputType === "insertFromPaste") {
            event.currentTarget.value = value[0] || "";
            return;
          }
          if (value.length > 2) {
            send({ type: "INPUT.PASTE", value });
            event.currentTarget.value = value[0];
            event.preventDefault();
            return;
          }
          if (evt.inputType === "deleteContentBackward") {
            send({ type: "INPUT.BACKSPACE" });
            return;
          }
          if (evt.inputType === "deleteByCut") {
            send({ type: "INPUT.DELETE" });
            return;
          }
          if (value === computed("focusedValue")) return;
          send({ type: "INPUT.CHANGE", value, index });
        },
        onKeyDown(event) {
          if (event.defaultPrevented) return;
          if ((0, import_dom_query.isComposingEvent)(event)) return;
          if ((0, import_dom_query.isModifierKey)(event)) return;
          if (event.key.length === 1 && computed("focusedValue") === event.key) {
            event.preventDefault();
            send({ type: "INPUT.ADVANCE" });
            return;
          }
          const keyMap = {
            Backspace() {
              send({ type: "INPUT.BACKSPACE" });
            },
            Delete() {
              send({ type: "INPUT.DELETE" });
            },
            ArrowLeft() {
              send({ type: "INPUT.ARROW_LEFT" });
            },
            ArrowRight() {
              send({ type: "INPUT.ARROW_RIGHT" });
            },
            Enter() {
              send({ type: "INPUT.ENTER" });
            },
            Home() {
              send({ type: "INPUT.HOME" });
            },
            End() {
              send({ type: "INPUT.END" });
            }
          };
          const exec = keyMap[(0, import_dom_query.getEventKey)(event, {
            dir: prop("dir"),
            orientation: "horizontal"
          })];
          if (exec) {
            exec(event);
            event.preventDefault();
          }
        },
        onFocus() {
          send({ type: "INPUT.FOCUS", index });
        },
        onBlur(event) {
          const target = event.relatedTarget;
          if ((0, import_dom_query.isHTMLElement)(target) && target.dataset.ownedby === dom.getRootId(scope)) return;
          send({ type: "INPUT.BLUR", index });
        }
      });
    }
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  connect
});
