// @chakra-ui/react/dist/cjs/components/floating-panel/floating-panel.cjs
"use strict";
"use client";
'use strict';

var jsxRuntime = require('react/jsx-runtime');
var floatingPanel = require('@ark-ui/react/floating-panel');
var createSlotRecipeContext = require('../../styled-system/create-slot-recipe-context.cjs');
var _for = require('../for/for.cjs');

const {
  withRootProvider,
  withContext,
  useStyles: useFloatingPanelStyles,
  PropsProvider
} = createSlotRecipeContext.createSlotRecipeContext({ key: "floatingPanel" });
const FloatingPanelRootProvider = withRootProvider(
  floatingPanel.FloatingPanel.RootProvider
);
const FloatingPanelRoot = withRootProvider(
  floatingPanel.FloatingPanel.Root,
  {
    defaultProps: {
      lazyMount: true,
      unmountOnExit: true,
      minSize: { width: 240, height: 100 }
    }
  }
);
const FloatingPanelPropsProvider = PropsProvider;
const FloatingPanelTrigger = withContext(floatingPanel.FloatingPanel.Trigger, "trigger", { forwardAsChild: true });
const FloatingPanelPositioner = withContext(floatingPanel.FloatingPanel.Positioner, "positioner", { forwardAsChild: true });
const FloatingPanelContent = withContext(floatingPanel.FloatingPanel.Content, "content", { forwardAsChild: true });
const FloatingPanelHeader = withContext(floatingPanel.FloatingPanel.Header, "header", { forwardAsChild: true });
const FloatingPanelBody = withContext(floatingPanel.FloatingPanel.Body, "body", { forwardAsChild: true });
const FloatingPanelTitle = withContext(floatingPanel.FloatingPanel.Title, "title", { forwardAsChild: true });
const FloatingPanelDragTrigger = withContext(floatingPanel.FloatingPanel.DragTrigger, "dragTrigger", { forwardAsChild: true });
const FloatingPanelResizeTrigger = withContext(floatingPanel.FloatingPanel.ResizeTrigger, "resizeTrigger", { forwardAsChild: true });
const RESIZE_AXES = ["n", "s", "e", "w", "ne", "nw", "se", "sw"];
const FloatingPanelResizeTriggers = (props) => /* @__PURE__ */ jsxRuntime.jsx(_for.For, { each: props.axes ?? RESIZE_AXES, children: (axis) => /* @__PURE__ */ jsxRuntime.jsx(FloatingPanelResizeTrigger, { ...props, axis }, axis) });
const FloatingPanelStageTrigger = withContext(floatingPanel.FloatingPanel.StageTrigger, "stageTrigger", { forwardAsChild: true });
const FloatingPanelCloseTrigger = withContext(floatingPanel.FloatingPanel.CloseTrigger, "closeTrigger", { forwardAsChild: true });
const FloatingPanelControl = withContext(floatingPanel.FloatingPanel.Control, "control", { forwardAsChild: true });

exports.FloatingPanelBody = FloatingPanelBody;
exports.FloatingPanelCloseTrigger = FloatingPanelCloseTrigger;
exports.FloatingPanelContent = FloatingPanelContent;
exports.FloatingPanelControl = FloatingPanelControl;
exports.FloatingPanelDragTrigger = FloatingPanelDragTrigger;
exports.FloatingPanelHeader = FloatingPanelHeader;
exports.FloatingPanelPositioner = FloatingPanelPositioner;
exports.FloatingPanelPropsProvider = FloatingPanelPropsProvider;
exports.FloatingPanelResizeTrigger = FloatingPanelResizeTrigger;
exports.FloatingPanelResizeTriggers = FloatingPanelResizeTriggers;
exports.FloatingPanelRoot = FloatingPanelRoot;
exports.FloatingPanelRootProvider = FloatingPanelRootProvider;
exports.FloatingPanelStageTrigger = FloatingPanelStageTrigger;
exports.FloatingPanelTitle = FloatingPanelTitle;
exports.FloatingPanelTrigger = FloatingPanelTrigger;
exports.useFloatingPanelStyles = useFloatingPanelStyles;
