// @chakra-ui/react/dist/cjs/components/floating-panel/index.cjs
"use strict";
'use strict';

var floatingPanel = require('./floating-panel.cjs');
var floatingPanel$1 = require('@ark-ui/react/floating-panel');
var namespace = require('./namespace.cjs');



exports.FloatingPanelBody = floatingPanel.FloatingPanelBody;
exports.FloatingPanelCloseTrigger = floatingPanel.FloatingPanelCloseTrigger;
exports.FloatingPanelContent = floatingPanel.FloatingPanelContent;
exports.FloatingPanelControl = floatingPanel.FloatingPanelControl;
exports.FloatingPanelDragTrigger = floatingPanel.FloatingPanelDragTrigger;
exports.FloatingPanelHeader = floatingPanel.FloatingPanelHeader;
exports.FloatingPanelPositioner = floatingPanel.FloatingPanelPositioner;
exports.FloatingPanelPropsProvider = floatingPanel.FloatingPanelPropsProvider;
exports.FloatingPanelResizeTrigger = floatingPanel.FloatingPanelResizeTrigger;
exports.FloatingPanelResizeTriggers = floatingPanel.FloatingPanelResizeTriggers;
exports.FloatingPanelRoot = floatingPanel.FloatingPanelRoot;
exports.FloatingPanelRootProvider = floatingPanel.FloatingPanelRootProvider;
exports.FloatingPanelStageTrigger = floatingPanel.FloatingPanelStageTrigger;
exports.FloatingPanelTitle = floatingPanel.FloatingPanelTitle;
exports.FloatingPanelTrigger = floatingPanel.FloatingPanelTrigger;
exports.useFloatingPanelStyles = floatingPanel.useFloatingPanelStyles;
Object.defineProperty(exports, "FloatingPanelContext", {
  enumerable: true,
  get: function () { return floatingPanel$1.FloatingPanelContext; }
});
Object.defineProperty(exports, "useFloatingPanel", {
  enumerable: true,
  get: function () { return floatingPanel$1.useFloatingPanel; }
});
Object.defineProperty(exports, "useFloatingPanelContext", {
  enumerable: true,
  get: function () { return floatingPanel$1.useFloatingPanelContext; }
});
exports.FloatingPanel = namespace;
