// @chakra-ui/react/dist/cjs/components/floating-panel/namespace.cjs
"use strict";
'use strict';

var floatingPanel = require('./floating-panel.cjs');
var floatingPanel$1 = require('@ark-ui/react/floating-panel');



exports.Body = floatingPanel.FloatingPanelBody;
exports.CloseTrigger = floatingPanel.FloatingPanelCloseTrigger;
exports.Content = floatingPanel.FloatingPanelContent;
exports.Control = floatingPanel.FloatingPanelControl;
exports.DragTrigger = floatingPanel.FloatingPanelDragTrigger;
exports.Header = floatingPanel.FloatingPanelHeader;
exports.Positioner = floatingPanel.FloatingPanelPositioner;
exports.PropsProvider = floatingPanel.FloatingPanelPropsProvider;
exports.ResizeTrigger = floatingPanel.FloatingPanelResizeTrigger;
exports.ResizeTriggers = floatingPanel.FloatingPanelResizeTriggers;
exports.Root = floatingPanel.FloatingPanelRoot;
exports.RootProvider = floatingPanel.FloatingPanelRootProvider;
exports.StageTrigger = floatingPanel.FloatingPanelStageTrigger;
exports.Title = floatingPanel.FloatingPanelTitle;
exports.Trigger = floatingPanel.FloatingPanelTrigger;
Object.defineProperty(exports, "Context", {
  enumerable: true,
  get: function () { return floatingPanel$1.FloatingPanelContext; }
});
