// @zag-js/splitter/dist/splitter.props.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/splitter.props.ts
var splitter_props_exports = {};
__export(splitter_props_exports, {
  panelProps: () => panelProps,
  props: () => props,
  resizeTriggerProps: () => resizeTriggerProps,
  splitPanelProps: () => splitPanelProps,
  splitProps: () => splitProps,
  splitResizeTriggerProps: () => splitResizeTriggerProps
});
module.exports = __toCommonJS(splitter_props_exports);
var import_types = require("@zag-js/types");
var import_utils = require("@zag-js/utils");
var props = (0, import_types.createProps)()([
  "dir",
  "getRootNode",
  "id",
  "ids",
  "onResize",
  "onResizeStart",
  "onResizeEnd",
  "onCollapse",
  "onExpand",
  "orientation",
  "size",
  "defaultSize",
  "panels",
  "keyboardResizeBy",
  "nonce",
  "registry"
]);
var splitProps = (0, import_utils.createSplitProps)(props);
var panelProps = (0, import_types.createProps)()(["id"]);
var splitPanelProps = (0, import_utils.createSplitProps)(panelProps);
var resizeTriggerProps = (0, import_types.createProps)()(["disabled", "id"]);
var splitResizeTriggerProps = (0, import_utils.createSplitProps)(resizeTriggerProps);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  panelProps,
  props,
  resizeTriggerProps,
  splitPanelProps,
  splitProps,
  splitResizeTriggerProps
});
