// @ark-ui/react/dist/components/field/field-item.cjs
"use client";
const require_field_anatomy = require("./field.anatomy.cjs");
const require_use_field_context = require("./use-field-context.cjs");
let react = require("react");
let react_jsx_runtime = require("react/jsx-runtime");
//#region src/components/field/field-item.tsx
var FieldItem = (props) => {
	const { value, children } = props;
	const parentField = require_use_field_context.useFieldContext();
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_use_field_context.FieldProvider, {
		value: (0, react.useMemo)(() => {
			if (!parentField) throw new Error("Field.Item must be used within Field.Root");
			const controlId = `field::${parentField.ids.control}::item::${value}`;
			const labelId = `${controlId}::label`;
			const getControlProps = () => ({
				...parentField.getInputProps(),
				id: controlId
			});
			return {
				...parentField,
				ids: {
					...parentField.ids,
					control: controlId,
					label: labelId
				},
				getLabelProps: () => ({
					...parentField.getLabelProps(),
					id: labelId,
					htmlFor: controlId
				}),
				getInputProps: () => ({
					...getControlProps(),
					...require_field_anatomy.parts.input.attrs
				}),
				getSelectProps: () => ({
					...getControlProps(),
					...require_field_anatomy.parts.select.attrs
				}),
				getTextareaProps: () => ({
					...getControlProps(),
					...require_field_anatomy.parts.textarea.attrs
				})
			};
		}, [parentField, value]),
		children
	});
};
FieldItem.displayName = "FieldItem";
//#endregion
exports.FieldItem = FieldItem;
