// @ark-ui/react/dist/components/pagination/index.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const require_use_pagination_context = require("./use-pagination-context.cjs");
const require_pagination_context = require("./pagination-context.cjs");
const require_pagination_ellipsis = require("./pagination-ellipsis.cjs");
const require_pagination_first_trigger = require("./pagination-first-trigger.cjs");
const require_pagination_item = require("./pagination-item.cjs");
const require_pagination_last_trigger = require("./pagination-last-trigger.cjs");
const require_pagination_next_trigger = require("./pagination-next-trigger.cjs");
const require_pagination_prev_trigger = require("./pagination-prev-trigger.cjs");
const require_use_pagination = require("./use-pagination.cjs");
const require_pagination_root = require("./pagination-root.cjs");
const require_pagination_root_provider = require("./pagination-root-provider.cjs");
const require_pagination = require("./pagination.cjs");
let _zag_js_pagination = require("@zag-js/pagination");
Object.defineProperty(exports, "Pagination", {
	enumerable: true,
	get: function() {
		return require_pagination.pagination_exports;
	}
});
exports.PaginationContext = require_pagination_context.PaginationContext;
exports.PaginationEllipsis = require_pagination_ellipsis.PaginationEllipsis;
exports.PaginationFirstTrigger = require_pagination_first_trigger.PaginationFirstTrigger;
exports.PaginationItem = require_pagination_item.PaginationItem;
exports.PaginationLastTrigger = require_pagination_last_trigger.PaginationLastTrigger;
exports.PaginationNextTrigger = require_pagination_next_trigger.PaginationNextTrigger;
exports.PaginationPrevTrigger = require_pagination_prev_trigger.PaginationPrevTrigger;
exports.PaginationRoot = require_pagination_root.PaginationRoot;
exports.PaginationRootProvider = require_pagination_root_provider.PaginationRootProvider;
Object.defineProperty(exports, "paginationAnatomy", {
	enumerable: true,
	get: function() {
		return _zag_js_pagination.anatomy;
	}
});
exports.usePagination = require_use_pagination.usePagination;
exports.usePaginationContext = require_use_pagination_context.usePaginationContext;
