// @ark-ui/react/dist/components/pagination/pagination.cjs
const require_runtime = require("../../_virtual/_rolldown/runtime.cjs");
const require_pagination_context = require("./pagination-context.cjs");
const require_pagination_ellipsis = require("./pagination-ellipsis.cjs");
const require_pagination_first_trigger = require("./pagination-first-trigger.cjs");
const require_pagination_item = require("./pagination-item.cjs");
const require_pagination_last_trigger = require("./pagination-last-trigger.cjs");
const require_pagination_next_trigger = require("./pagination-next-trigger.cjs");
const require_pagination_prev_trigger = require("./pagination-prev-trigger.cjs");
const require_pagination_root = require("./pagination-root.cjs");
const require_pagination_root_provider = require("./pagination-root-provider.cjs");
//#region src/components/pagination/pagination.ts
var pagination_exports = /* @__PURE__ */ require_runtime.__exportAll({
	Context: () => require_pagination_context.PaginationContext,
	Ellipsis: () => require_pagination_ellipsis.PaginationEllipsis,
	FirstTrigger: () => require_pagination_first_trigger.PaginationFirstTrigger,
	Item: () => require_pagination_item.PaginationItem,
	LastTrigger: () => require_pagination_last_trigger.PaginationLastTrigger,
	NextTrigger: () => require_pagination_next_trigger.PaginationNextTrigger,
	PrevTrigger: () => require_pagination_prev_trigger.PaginationPrevTrigger,
	Root: () => require_pagination_root.PaginationRoot,
	RootProvider: () => require_pagination_root_provider.PaginationRootProvider
});
//#endregion
Object.defineProperty(exports, "pagination_exports", {
	enumerable: true,
	get: function() {
		return pagination_exports;
	}
});
