// @ark-ui/react/dist/components/pagination/pagination.js
import { __exportAll } from "../../_virtual/_rolldown/runtime.js";
import { PaginationContext } from "./pagination-context.js";
import { PaginationEllipsis } from "./pagination-ellipsis.js";
import { PaginationFirstTrigger } from "./pagination-first-trigger.js";
import { PaginationItem } from "./pagination-item.js";
import { PaginationLastTrigger } from "./pagination-last-trigger.js";
import { PaginationNextTrigger } from "./pagination-next-trigger.js";
import { PaginationPrevTrigger } from "./pagination-prev-trigger.js";
import { PaginationRoot } from "./pagination-root.js";
import { PaginationRootProvider } from "./pagination-root-provider.js";
//#region src/components/pagination/pagination.ts
var pagination_exports = /* @__PURE__ */ __exportAll({
	Context: () => PaginationContext,
	Ellipsis: () => PaginationEllipsis,
	FirstTrigger: () => PaginationFirstTrigger,
	Item: () => PaginationItem,
	LastTrigger: () => PaginationLastTrigger,
	NextTrigger: () => PaginationNextTrigger,
	PrevTrigger: () => PaginationPrevTrigger,
	Root: () => PaginationRoot,
	RootProvider: () => PaginationRootProvider
});
//#endregion
export { pagination_exports };
