// @chakra-ui/react/dist/cjs/components/carousel/index.cjs
"use strict";
'use strict';

var carousel = require('./carousel.cjs');
var carousel$1 = require('@ark-ui/react/carousel');
var namespace = require('./namespace.cjs');



exports.CarouselAutoplayIndicator = carousel.CarouselAutoplayIndicator;
exports.CarouselAutoplayTrigger = carousel.CarouselAutoplayTrigger;
exports.CarouselContext = carousel.CarouselContext;
exports.CarouselControl = carousel.CarouselControl;
exports.CarouselIndicator = carousel.CarouselIndicator;
exports.CarouselIndicatorGroup = carousel.CarouselIndicatorGroup;
exports.CarouselIndicators = carousel.CarouselIndicators;
exports.CarouselItem = carousel.CarouselItem;
exports.CarouselItemGroup = carousel.CarouselItemGroup;
exports.CarouselNextTrigger = carousel.CarouselNextTrigger;
exports.CarouselPrevTrigger = carousel.CarouselPrevTrigger;
exports.CarouselProgressText = carousel.CarouselProgressText;
exports.CarouselPropsProvider = carousel.CarouselPropsProvider;
exports.CarouselRoot = carousel.CarouselRoot;
exports.CarouselRootProvider = carousel.CarouselRootProvider;
exports.useCarouselStyles = carousel.useCarouselStyles;
Object.defineProperty(exports, "useCarousel", {
  enumerable: true,
  get: function () { return carousel$1.useCarousel; }
});
Object.defineProperty(exports, "useCarouselContext", {
  enumerable: true,
  get: function () { return carousel$1.useCarouselContext; }
});
exports.Carousel = namespace;
