// @ark-ui/react/dist/components/color-picker/use-color-picker.js
"use client";
import { useEnvironmentContext } from "../../providers/environment/use-environment-context.js";
import { useLocaleContext } from "../../providers/locale/use-locale-context.js";
import { useFieldContext } from "../field/use-field-context.js";
import * as colorPicker from "@zag-js/color-picker";
import { useId } from "react";
import { normalizeProps, useMachine } from "@zag-js/react";
//#region src/components/color-picker/use-color-picker.ts
var useColorPicker = (props) => {
	const id = useId();
	const { getRootNode } = useEnvironmentContext();
	const { dir } = useLocaleContext();
	const field = useFieldContext();
	const machineProps = {
		id,
		ids: {
			label: field?.ids.label,
			input: field?.ids.control
		},
		dir,
		disabled: field?.disabled,
		invalid: field?.invalid,
		readOnly: field?.readOnly,
		required: field?.required,
		getRootNode,
		...props
	};
	const service = useMachine(colorPicker.machine, machineProps);
	return colorPicker.connect(service, normalizeProps);
};
//#endregion
export { useColorPicker };
