// @ark-ui/react/dist/components/select/use-select.js
"use client";
import { useEnvironmentContext } from "../../providers/environment/use-environment-context.js";
import { useLocaleContext } from "../../providers/locale/use-locale-context.js";
import { useFieldContext } from "../field/use-field-context.js";
import * as select from "@zag-js/select";
import { useId } from "react";
import { normalizeProps, useMachine } from "@zag-js/react";
//#region src/components/select/use-select.ts
var useSelect = (props) => {
	const id = useId();
	const { dir } = useLocaleContext();
	const { getRootNode } = useEnvironmentContext();
	const field = useFieldContext();
	const machineProps = {
		id,
		ids: {
			label: field?.ids.label,
			hiddenSelect: field?.ids.control
		},
		disabled: field?.disabled,
		readOnly: field?.readOnly,
		invalid: field?.invalid,
		required: field?.required,
		dir,
		getRootNode,
		...props
	};
	const service = useMachine(select.machine, machineProps);
	return select.connect(service, normalizeProps);
};
//#endregion
export { useSelect };
