// @ark-ui/react/dist/components/pin-input/use-pin-input.js
"use client";
import { useEnvironmentContext } from "../../providers/environment/use-environment-context.js";
import { useLocaleContext } from "../../providers/locale/use-locale-context.js";
import { useFieldContext } from "../field/use-field-context.js";
import * as pinInput from "@zag-js/pin-input";
import { useId } from "react";
import { normalizeProps, useMachine } from "@zag-js/react";
//#region src/components/pin-input/use-pin-input.ts
var usePinInput = (props) => {
	const id = useId();
	const { getRootNode } = useEnvironmentContext();
	const { dir } = useLocaleContext();
	const field = useFieldContext();
	const machineProps = {
		id,
		ids: {
			label: field?.ids.label,
			hiddenInput: field?.ids.control
		},
		disabled: field?.disabled,
		readOnly: field?.readOnly,
		required: field?.required,
		invalid: field?.invalid,
		dir,
		getRootNode,
		...props
	};
	const service = useMachine(pinInput.machine, machineProps);
	return pinInput.connect(service, normalizeProps);
};
//#endregion
export { usePinInput };
