// @ark-ui/react/dist/components/collapsible/collapsible.cjs
const require_runtime = require("../../_virtual/_rolldown/runtime.cjs");
const require_collapsible_content = require("./collapsible-content.cjs");
const require_collapsible_context = require("./collapsible-context.cjs");
const require_collapsible_indicator = require("./collapsible-indicator.cjs");
const require_collapsible_root = require("./collapsible-root.cjs");
const require_collapsible_root_provider = require("./collapsible-root-provider.cjs");
const require_collapsible_trigger = require("./collapsible-trigger.cjs");
//#region src/components/collapsible/collapsible.ts
var collapsible_exports = /* @__PURE__ */ require_runtime.__exportAll({
	Content: () => require_collapsible_content.CollapsibleContent,
	Context: () => require_collapsible_context.CollapsibleContext,
	Indicator: () => require_collapsible_indicator.CollapsibleIndicator,
	Root: () => require_collapsible_root.CollapsibleRoot,
	RootProvider: () => require_collapsible_root_provider.CollapsibleRootProvider,
	Trigger: () => require_collapsible_trigger.CollapsibleTrigger
});
//#endregion
Object.defineProperty(exports, "collapsible_exports", {
	enumerable: true,
	get: function() {
		return collapsible_exports;
	}
});
