// @ark-ui/react/dist/components/collapsible/index.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const require_use_collapsible_context = require("./use-collapsible-context.cjs");
const require_collapsible_content = require("./collapsible-content.cjs");
const require_collapsible_context = require("./collapsible-context.cjs");
const require_collapsible_indicator = require("./collapsible-indicator.cjs");
const require_use_collapsible = require("./use-collapsible.cjs");
const require_collapsible_root = require("./collapsible-root.cjs");
const require_collapsible_root_provider = require("./collapsible-root-provider.cjs");
const require_collapsible_trigger = require("./collapsible-trigger.cjs");
const require_collapsible = require("./collapsible.cjs");
let _zag_js_collapsible = require("@zag-js/collapsible");
Object.defineProperty(exports, "Collapsible", {
	enumerable: true,
	get: function() {
		return require_collapsible.collapsible_exports;
	}
});
exports.CollapsibleContent = require_collapsible_content.CollapsibleContent;
exports.CollapsibleContext = require_collapsible_context.CollapsibleContext;
exports.CollapsibleIndicator = require_collapsible_indicator.CollapsibleIndicator;
exports.CollapsibleRoot = require_collapsible_root.CollapsibleRoot;
exports.CollapsibleRootProvider = require_collapsible_root_provider.CollapsibleRootProvider;
exports.CollapsibleTrigger = require_collapsible_trigger.CollapsibleTrigger;
Object.defineProperty(exports, "collapsibleAnatomy", {
	enumerable: true,
	get: function() {
		return _zag_js_collapsible.anatomy;
	}
});
exports.useCollapsible = require_use_collapsible.useCollapsible;
exports.useCollapsibleContext = require_use_collapsible_context.useCollapsibleContext;
