// @ark-ui/react/dist/components/scroll-area/index.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const require_use_scroll_area_context = require("./use-scroll-area-context.cjs");
const require_scroll_area_context = require("./scroll-area-context.cjs");
const require_scroll_area_content = require("./scroll-area-content.cjs");
const require_scroll_area_corner = require("./scroll-area-corner.cjs");
const require_use_scroll_area = require("./use-scroll-area.cjs");
const require_scroll_area_root = require("./scroll-area-root.cjs");
const require_scroll_area_root_provider = require("./scroll-area-root-provider.cjs");
const require_scroll_area_scrollbar = require("./scroll-area-scrollbar.cjs");
const require_scroll_area_thumb = require("./scroll-area-thumb.cjs");
const require_scroll_area_viewport = require("./scroll-area-viewport.cjs");
const require_scroll_area = require("./scroll-area.cjs");
let _zag_js_scroll_area = require("@zag-js/scroll-area");
Object.defineProperty(exports, "ScrollArea", {
	enumerable: true,
	get: function() {
		return require_scroll_area.scroll_area_exports;
	}
});
exports.ScrollAreaContent = require_scroll_area_content.ScrollAreaContent;
exports.ScrollAreaContext = require_scroll_area_context.ScrollAreaContext;
exports.ScrollAreaCorner = require_scroll_area_corner.ScrollAreaCorner;
exports.ScrollAreaRoot = require_scroll_area_root.ScrollAreaRoot;
exports.ScrollAreaRootProvider = require_scroll_area_root_provider.ScrollAreaRootProvider;
exports.ScrollAreaScrollbar = require_scroll_area_scrollbar.ScrollAreaScrollbar;
exports.ScrollAreaThumb = require_scroll_area_thumb.ScrollAreaThumb;
exports.ScrollAreaViewport = require_scroll_area_viewport.ScrollAreaViewport;
Object.defineProperty(exports, "scrollAreaAnatomy", {
	enumerable: true,
	get: function() {
		return _zag_js_scroll_area.anatomy;
	}
});
exports.useScrollArea = require_use_scroll_area.useScrollArea;
exports.useScrollAreaContext = require_use_scroll_area_context.useScrollAreaContext;
