// @ark-ui/react/dist/components/scroll-area/scroll-area.cjs
const require_runtime = require("../../_virtual/_rolldown/runtime.cjs");
const require_scroll_area_context = require("./scroll-area-context.cjs");
const require_scroll_area_content = require("./scroll-area-content.cjs");
const require_scroll_area_corner = require("./scroll-area-corner.cjs");
const require_scroll_area_root = require("./scroll-area-root.cjs");
const require_scroll_area_root_provider = require("./scroll-area-root-provider.cjs");
const require_scroll_area_scrollbar = require("./scroll-area-scrollbar.cjs");
const require_scroll_area_thumb = require("./scroll-area-thumb.cjs");
const require_scroll_area_viewport = require("./scroll-area-viewport.cjs");
//#region src/components/scroll-area/scroll-area.ts
var scroll_area_exports = /* @__PURE__ */ require_runtime.__exportAll({
	Content: () => require_scroll_area_content.ScrollAreaContent,
	Context: () => require_scroll_area_context.ScrollAreaContext,
	Corner: () => require_scroll_area_corner.ScrollAreaCorner,
	Root: () => require_scroll_area_root.ScrollAreaRoot,
	RootProvider: () => require_scroll_area_root_provider.ScrollAreaRootProvider,
	Scrollbar: () => require_scroll_area_scrollbar.ScrollAreaScrollbar,
	Thumb: () => require_scroll_area_thumb.ScrollAreaThumb,
	Viewport: () => require_scroll_area_viewport.ScrollAreaViewport
});
//#endregion
Object.defineProperty(exports, "scroll_area_exports", {
	enumerable: true,
	get: function() {
		return scroll_area_exports;
	}
});
