// @ark-ui/react/dist/components/fieldset/use-fieldset.cjs
"use client";
const require_fieldset_anatomy = require("./fieldset.anatomy.cjs");
const require_use_environment_context = require("../../providers/environment/use-environment-context.cjs");
const require_use_safe_layout_effect = require("../../utils/use-safe-layout-effect.cjs");
let react = require("react");
let _zag_js_dom_query = require("@zag-js/dom-query");
//#region src/components/fieldset/use-fieldset.ts
var useFieldset = (props = {}) => {
	const { disabled = false, invalid = false } = props;
	const env = require_use_environment_context.useEnvironmentContext();
	const [textElements, setTextElements] = (0, react.useState)({
		hasErrorText: false,
		hasHelperText: false
	});
	const uid = (0, react.useId)();
	const id = props.id ?? uid;
	const rootRef = (0, react.useRef)(null);
	const legendId = `fieldset::${id}::legend`;
	const errorTextId = `fieldset::${id}::error-text`;
	const helperTextId = `fieldset::${id}::helper-text`;
	require_use_safe_layout_effect.useSafeLayoutEffect(() => {
		const rootNode = rootRef.current;
		if (!rootNode) return;
		const checkTextElements = () => {
			const docOrShadowRoot = env.getRootNode();
			setTextElements({
				hasErrorText: !!docOrShadowRoot.getElementById(errorTextId),
				hasHelperText: !!docOrShadowRoot.getElementById(helperTextId)
			});
		};
		checkTextElements();
		const observer = new (env.getWindow()).MutationObserver(checkTextElements);
		observer.observe(rootNode, {
			childList: true,
			subtree: true
		});
		return () => observer.disconnect();
	}, [
		env,
		errorTextId,
		helperTextId
	]);
	const ids = [];
	if (textElements.hasErrorText && invalid) ids.push(errorTextId);
	if (textElements.hasHelperText) ids.push(helperTextId);
	const labelIds = ids.length > 0 ? ids.join(" ") : void 0;
	const getRootProps = () => ({
		...require_fieldset_anatomy.parts.root.attrs,
		ref: rootRef,
		disabled,
		"data-disabled": (0, _zag_js_dom_query.dataAttr)(disabled),
		"data-invalid": (0, _zag_js_dom_query.dataAttr)(invalid),
		"aria-labelledby": legendId,
		"aria-describedby": labelIds
	});
	const getLegendProps = () => ({
		id: legendId,
		...require_fieldset_anatomy.parts.legend.attrs,
		"data-disabled": (0, _zag_js_dom_query.dataAttr)(disabled),
		"data-invalid": (0, _zag_js_dom_query.dataAttr)(invalid)
	});
	const getHelperTextProps = () => ({
		id: helperTextId,
		...require_fieldset_anatomy.parts.helperText.attrs
	});
	const getErrorTextProps = () => ({
		id: errorTextId,
		...require_fieldset_anatomy.parts.errorText.attrs,
		"aria-live": "polite"
	});
	return {
		refs: { rootRef },
		ids: {
			legend: legendId,
			errorText: errorTextId,
			helperText: helperTextId
		},
		disabled,
		invalid,
		getRootProps,
		getLegendProps,
		getHelperTextProps,
		getErrorTextProps
	};
};
//#endregion
exports.useFieldset = useFieldset;
