// @zag-js/hover-card/dist/hover-card.dom.mjs
// src/hover-card.dom.ts
import { queryAll } from "@zag-js/dom-query";
import { isFunction } from "@zag-js/utils";
var getTriggerId = (scope, value) => {
  const customId = scope.ids?.trigger;
  if (customId != null) return isFunction(customId) ? customId(value) : customId;
  return value ? `hover-card:${scope.id}:trigger:${value}` : `hover-card:${scope.id}:trigger`;
};
var getContentId = (scope) => scope.ids?.content ?? `hover-card:${scope.id}:content`;
var getPositionerId = (scope) => scope.ids?.positioner ?? `hover-card:${scope.id}:popper`;
var getArrowId = (scope) => scope.ids?.arrow ?? `hover-card:${scope.id}:arrow`;
var getTriggerEl = (scope) => scope.getById(getTriggerId(scope));
var getContentEl = (scope) => scope.getById(getContentId(scope));
var getPositionerEl = (scope) => scope.getById(getPositionerId(scope));
var getTriggerEls = (scope) => queryAll(
  scope.getRootNode(),
  `[data-scope="hover-card"][data-part="trigger"][data-ownedby="${scope.id}"]`
);
var getActiveTriggerEl = (scope, value) => {
  if (value == null) {
    return getTriggerEl(scope) ?? getTriggerEls(scope)[0];
  }
  return scope.getById(getTriggerId(scope, value));
};
export {
  getActiveTriggerEl,
  getArrowId,
  getContentEl,
  getContentId,
  getPositionerEl,
  getPositionerId,
  getTriggerEl,
  getTriggerEls,
  getTriggerId
};
