// @zag-js/toggle-group/dist/toggle-group.dom.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/toggle-group.dom.ts
var toggle_group_dom_exports = {};
__export(toggle_group_dom_exports, {
  getElements: () => getElements,
  getFirstEl: () => getFirstEl,
  getItemId: () => getItemId,
  getLastEl: () => getLastEl,
  getNextEl: () => getNextEl,
  getPrevEl: () => getPrevEl,
  getRootEl: () => getRootEl,
  getRootId: () => getRootId
});
module.exports = __toCommonJS(toggle_group_dom_exports);
var import_dom_query = require("@zag-js/dom-query");
var import_utils = require("@zag-js/utils");
var getRootId = (ctx) => ctx.ids?.root ?? `toggle-group:${ctx.id}`;
var getItemId = (ctx, value) => ctx.ids?.item?.(value) ?? `toggle-group:${ctx.id}:${value}`;
var getRootEl = (ctx) => ctx.getById(getRootId(ctx));
var getElements = (ctx) => {
  const ownerId = CSS.escape(getRootId(ctx));
  const selector = `[data-ownedby='${ownerId}']:not([data-disabled])`;
  return (0, import_dom_query.queryAll)(getRootEl(ctx), selector);
};
var getFirstEl = (ctx) => (0, import_utils.first)(getElements(ctx));
var getLastEl = (ctx) => (0, import_utils.last)(getElements(ctx));
var getNextEl = (ctx, id, loopFocus) => (0, import_dom_query.nextById)(getElements(ctx), id, loopFocus);
var getPrevEl = (ctx, id, loopFocus) => (0, import_dom_query.prevById)(getElements(ctx), id, loopFocus);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getElements,
  getFirstEl,
  getItemId,
  getLastEl,
  getNextEl,
  getPrevEl,
  getRootEl,
  getRootId
});
