// @zag-js/timer/dist/timer.connect.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/timer.connect.ts
var timer_connect_exports = {};
__export(timer_connect_exports, {
  connect: () => connect
});
module.exports = __toCommonJS(timer_connect_exports);
var import_utils = require("@zag-js/utils");
var import_timer = require("./timer.anatomy.js");
var dom = __toESM(require("./timer.dom.js"));
var validActions = /* @__PURE__ */ new Set(["start", "pause", "resume", "reset", "restart"]);
function connect(service, normalize) {
  const { state, send, computed, scope, prop } = service;
  const translations = prop("translations");
  const running = state.matches("running");
  const paused = state.matches("paused");
  const time = computed("time");
  const formattedTime = computed("formattedTime");
  const progressPercent = computed("progressPercent");
  return {
    running,
    paused,
    time,
    formattedTime,
    progressPercent,
    start() {
      send({ type: "START" });
    },
    pause() {
      send({ type: "PAUSE" });
    },
    resume() {
      send({ type: "RESUME" });
    },
    reset() {
      send({ type: "RESET" });
    },
    restart() {
      send({ type: "RESTART" });
    },
    getRootProps() {
      return normalize.element({
        id: dom.getRootId(scope),
        ...import_timer.parts.root.attrs
      });
    },
    getAreaProps() {
      return normalize.element({
        role: "timer",
        id: dom.getAreaId(scope),
        "aria-label": translations.areaLabel?.(time, formattedTime),
        "aria-atomic": true,
        ...import_timer.parts.area.attrs
      });
    },
    getControlProps() {
      return normalize.element({
        ...import_timer.parts.control.attrs
      });
    },
    getItemProps(props) {
      const value = time[props.type];
      return normalize.element({
        ...import_timer.parts.item.attrs,
        "data-type": props.type,
        style: {
          "--value": value
        }
      });
    },
    getItemLabelProps(props) {
      return normalize.element({
        ...import_timer.parts.itemLabel.attrs,
        "data-type": props.type
      });
    },
    getItemValueProps(props) {
      return normalize.element({
        ...import_timer.parts.itemValue.attrs,
        "data-type": props.type
      });
    },
    getSeparatorProps() {
      return normalize.element({
        "aria-hidden": true,
        ...import_timer.parts.separator.attrs
      });
    },
    getActionTriggerProps(props) {
      if (!validActions.has(props.action)) {
        throw new Error(
          `[zag-js] Invalid action: ${props.action}. Must be one of: ${Array.from(validActions).join(", ")}`
        );
      }
      return normalize.button({
        ...import_timer.parts.actionTrigger.attrs,
        hidden: (0, import_utils.match)(props.action, {
          start: () => running || paused,
          pause: () => !running,
          reset: () => !running && !paused,
          resume: () => !paused,
          restart: () => false
        }),
        type: "button",
        onClick(event) {
          if (event.defaultPrevented) return;
          send({ type: props.action.toUpperCase() });
        }
      });
    }
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  connect
});
