// @zag-js/steps/dist/steps.connect.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/steps.connect.ts
var steps_connect_exports = {};
__export(steps_connect_exports, {
  connect: () => connect
});
module.exports = __toCommonJS(steps_connect_exports);
var import_dom_query = require("@zag-js/dom-query");
var import_utils = require("@zag-js/utils");
var import_steps = require("./steps.anatomy.js");
var dom = __toESM(require("./steps.dom.js"));
function connect(service, normalize) {
  const { context, send, computed, prop, scope } = service;
  const step = context.get("step");
  const count = prop("count");
  const percent = computed("percent");
  const hasNextStep = computed("hasNextStep");
  const hasPrevStep = computed("hasPrevStep");
  const isStepValid = (index) => {
    return prop("isStepValid")?.(index) ?? true;
  };
  const isStepSkippable = (index) => {
    return prop("isStepSkippable")?.(index) ?? false;
  };
  const getItemState = (props) => ({
    triggerId: dom.getTriggerId(scope, props.index),
    contentId: dom.getContentId(scope, props.index),
    current: props.index === step,
    completed: props.index < step,
    incomplete: props.index > step,
    index: props.index,
    first: props.index === 0,
    last: props.index === count - 1,
    skippable: isStepSkippable(props.index),
    isValid: () => isStepValid(props.index)
  });
  const goToNextStep = () => {
    send({ type: "STEP.NEXT", src: "next.trigger.click" });
  };
  const goToPrevStep = () => {
    send({ type: "STEP.PREV", src: "prev.trigger.click" });
  };
  const resetStep = () => {
    send({ type: "STEP.RESET", src: "reset.trigger.click" });
  };
  const setStep = (value) => {
    send({ type: "STEP.SET", value, src: "api.setValue" });
  };
  return {
    value: step,
    count,
    percent,
    hasNextStep,
    hasPrevStep,
    isCompleted: computed("completed"),
    isStepValid,
    isStepSkippable,
    goToNextStep,
    goToPrevStep,
    resetStep,
    getItemState,
    setStep,
    getRootProps() {
      return normalize.element({
        ...import_steps.parts.root.attrs,
        id: dom.getRootId(scope),
        dir: prop("dir"),
        "data-orientation": prop("orientation"),
        style: {
          "--percent": `${percent}%`
        }
      });
    },
    getListProps() {
      const arr = (0, import_utils.fromLength)(count);
      const triggerIds = arr.map((_, index) => dom.getTriggerId(scope, index));
      return normalize.element({
        ...import_steps.parts.list.attrs,
        dir: prop("dir"),
        id: dom.getListId(scope),
        role: "tablist",
        "aria-owns": triggerIds.join(" "),
        "aria-orientation": prop("orientation"),
        "data-orientation": prop("orientation")
      });
    },
    getItemProps(props) {
      const itemState = getItemState(props);
      return normalize.element({
        ...import_steps.parts.item.attrs,
        dir: prop("dir"),
        "aria-current": itemState.current ? "step" : void 0,
        "data-orientation": prop("orientation"),
        "data-skippable": (0, import_dom_query.dataAttr)(itemState.skippable)
      });
    },
    getTriggerProps(props) {
      const itemState = getItemState(props);
      return normalize.button({
        ...import_steps.parts.trigger.attrs,
        id: itemState.triggerId,
        role: "tab",
        dir: prop("dir"),
        tabIndex: !prop("linear") || itemState.current ? 0 : -1,
        "aria-selected": itemState.current,
        "aria-controls": itemState.contentId,
        "data-state": itemState.current ? "open" : "closed",
        "data-orientation": prop("orientation"),
        "data-complete": (0, import_dom_query.dataAttr)(itemState.completed),
        "data-current": (0, import_dom_query.dataAttr)(itemState.current),
        "data-incomplete": (0, import_dom_query.dataAttr)(itemState.incomplete),
        onClick(event) {
          if (event.defaultPrevented) return;
          if (prop("linear")) return;
          send({ type: "STEP.SET", value: props.index, src: "trigger.click" });
        }
      });
    },
    getContentProps(props) {
      const itemState = getItemState(props);
      return normalize.element({
        ...import_steps.parts.content.attrs,
        dir: prop("dir"),
        id: itemState.contentId,
        role: "tabpanel",
        tabIndex: 0,
        hidden: !itemState.current,
        "data-state": itemState.current ? "open" : "closed",
        "data-orientation": prop("orientation"),
        "aria-labelledby": itemState.triggerId
      });
    },
    getIndicatorProps(props) {
      const itemState = getItemState(props);
      return normalize.element({
        ...import_steps.parts.indicator.attrs,
        dir: prop("dir"),
        "aria-hidden": true,
        "data-complete": (0, import_dom_query.dataAttr)(itemState.completed),
        "data-current": (0, import_dom_query.dataAttr)(itemState.current),
        "data-incomplete": (0, import_dom_query.dataAttr)(itemState.incomplete)
      });
    },
    getSeparatorProps(props) {
      const itemState = getItemState(props);
      return normalize.element({
        ...import_steps.parts.separator.attrs,
        dir: prop("dir"),
        "data-orientation": prop("orientation"),
        "data-complete": (0, import_dom_query.dataAttr)(itemState.completed),
        "data-current": (0, import_dom_query.dataAttr)(itemState.current),
        "data-incomplete": (0, import_dom_query.dataAttr)(itemState.incomplete)
      });
    },
    getNextTriggerProps() {
      return normalize.button({
        ...import_steps.parts.nextTrigger.attrs,
        dir: prop("dir"),
        type: "button",
        disabled: !hasNextStep,
        onClick(event) {
          if (event.defaultPrevented) return;
          goToNextStep();
        }
      });
    },
    getPrevTriggerProps() {
      return normalize.button({
        dir: prop("dir"),
        ...import_steps.parts.prevTrigger.attrs,
        type: "button",
        disabled: !hasPrevStep,
        onClick(event) {
          if (event.defaultPrevented) return;
          goToPrevStep();
        }
      });
    },
    getProgressProps() {
      return normalize.element({
        dir: prop("dir"),
        ...import_steps.parts.progress.attrs,
        role: "progressbar",
        "aria-valuenow": percent,
        "aria-valuemin": 0,
        "aria-valuemax": 100,
        "aria-valuetext": `${percent}% complete`,
        "data-complete": (0, import_dom_query.dataAttr)(percent === 100)
      });
    }
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  connect
});
