// @ark-ui/react/dist/components/marquee/index.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const require_use_marquee_context = require("./use-marquee-context.cjs");
const require_marquee_content = require("./marquee-content.cjs");
const require_marquee_context = require("./marquee-context.cjs");
const require_marquee_edge = require("./marquee-edge.cjs");
const require_marquee_item = require("./marquee-item.cjs");
const require_use_marquee = require("./use-marquee.cjs");
const require_marquee_root = require("./marquee-root.cjs");
const require_marquee_root_provider = require("./marquee-root-provider.cjs");
const require_marquee_viewport = require("./marquee-viewport.cjs");
const require_marquee = require("./marquee.cjs");
let _zag_js_marquee = require("@zag-js/marquee");
Object.defineProperty(exports, "Marquee", {
	enumerable: true,
	get: function() {
		return require_marquee.marquee_exports;
	}
});
exports.MarqueeContent = require_marquee_content.MarqueeContent;
exports.MarqueeContext = require_marquee_context.MarqueeContext;
exports.MarqueeEdge = require_marquee_edge.MarqueeEdge;
exports.MarqueeItem = require_marquee_item.MarqueeItem;
exports.MarqueeRoot = require_marquee_root.MarqueeRoot;
exports.MarqueeRootProvider = require_marquee_root_provider.MarqueeRootProvider;
exports.MarqueeViewport = require_marquee_viewport.MarqueeViewport;
Object.defineProperty(exports, "marqueeAnatomy", {
	enumerable: true,
	get: function() {
		return _zag_js_marquee.anatomy;
	}
});
exports.useMarquee = require_use_marquee.useMarquee;
exports.useMarqueeContext = require_use_marquee_context.useMarqueeContext;
