// @ark-ui/react/dist/components/marquee/marquee.cjs
const require_runtime = require("../../_virtual/_rolldown/runtime.cjs");
const require_marquee_content = require("./marquee-content.cjs");
const require_marquee_context = require("./marquee-context.cjs");
const require_marquee_edge = require("./marquee-edge.cjs");
const require_marquee_item = require("./marquee-item.cjs");
const require_marquee_root = require("./marquee-root.cjs");
const require_marquee_root_provider = require("./marquee-root-provider.cjs");
const require_marquee_viewport = require("./marquee-viewport.cjs");
//#region src/components/marquee/marquee.ts
var marquee_exports = /* @__PURE__ */ require_runtime.__exportAll({
	Content: () => require_marquee_content.MarqueeContent,
	Context: () => require_marquee_context.MarqueeContext,
	Edge: () => require_marquee_edge.MarqueeEdge,
	Item: () => require_marquee_item.MarqueeItem,
	Root: () => require_marquee_root.MarqueeRoot,
	RootProvider: () => require_marquee_root_provider.MarqueeRootProvider,
	Viewport: () => require_marquee_viewport.MarqueeViewport
});
//#endregion
Object.defineProperty(exports, "marquee_exports", {
	enumerable: true,
	get: function() {
		return marquee_exports;
	}
});
