// @zag-js/navigation-menu/dist/navigation-menu.dom.mjs
// src/navigation-menu.dom.ts
import { getTabbables, getWindow, queryAll } from "@zag-js/dom-query";
var getRootId = (ctx) => ctx.ids?.root ?? `nav-menu:${ctx.id}`;
var getTriggerId = (ctx, value) => ctx.ids?.trigger?.(value) ?? `nav-menu:${ctx.id}:trigger:${value}`;
var getTriggerProxyId = (ctx, value) => ctx.ids?.triggerProxy?.(value) ?? `nav-menu:${ctx.id}:trigger-proxy:${value}`;
var getContentId = (ctx, value) => ctx.ids?.content?.(value) ?? `nav-menu:${ctx.id}:content:${value}`;
var getViewportId = (ctx) => ctx.ids?.viewport ?? `nav-menu:${ctx.id}:viewport`;
var getListId = (ctx) => ctx.ids?.list ?? `nav-menu:${ctx.id}:list`;
var getItemId = (ctx, value) => ctx.ids?.item?.(value) ?? `nav-menu:${ctx.id}:item:${value}`;
var getRootEl = (ctx) => ctx.getById(getRootId(ctx));
var getViewportEl = (ctx) => ctx.getById(getViewportId(ctx));
var getTriggerEl = (ctx, value) => {
  if (!value) return null;
  return ctx.getById(getTriggerId(ctx, value));
};
var getTriggerProxyEl = (ctx, value) => {
  if (!value) return null;
  return ctx.getById(getTriggerProxyId(ctx, value));
};
var getListEl = (ctx) => ctx.getById(getListId(ctx));
var getContentEl = (ctx, value) => {
  if (!value) return null;
  return ctx.getById(getContentId(ctx, value));
};
var getContentEls = (ctx) => queryAll(ctx.getDoc(), `[data-scope=navigation-menu][data-part=content][data-uid='${ctx.id}']`);
var getTabbableEls = (ctx, value) => {
  return getTabbables(getContentEl(ctx, value));
};
var getTriggerEls = (ctx) => queryAll(getListEl(ctx), `[data-part=trigger][data-uid='${ctx.id}']`);
var getLinkEls = (ctx, value) => {
  const contentEl = getContentEl(ctx, value);
  return queryAll(contentEl, `[data-part=link][data-ownedby="${getContentId(ctx, value)}"]`);
};
var getElements = (ctx) => {
  const topLevelTriggerSelector = `[data-part=trigger][data-uid='${ctx.id}']:not([data-disabled])`;
  const topLevelLinkSelector = `[data-part=item] > [data-part=link]`;
  return queryAll(getListEl(ctx), `${topLevelTriggerSelector}, ${topLevelLinkSelector}`);
};
function trackResizeObserver(element, onResize) {
  if (!element.length) return;
  let frame = 0;
  const win = getWindow(element[0]);
  const obs = new win.ResizeObserver(() => {
    cancelAnimationFrame(frame);
    frame = requestAnimationFrame(onResize);
  });
  element.forEach((el) => {
    if (el) obs.observe(el);
  });
  return () => {
    cancelAnimationFrame(frame);
    element.forEach((el) => {
      if (el) obs.unobserve(el);
    });
  };
}
function setMotionAttr(scope, value, previousValue) {
  const triggers = getTriggerEls(scope);
  const dir = triggers[0].dir;
  let values = triggers.map((trigger) => trigger.getAttribute("data-value"));
  if (dir === "rtl") values.reverse();
  const index = values.indexOf(value);
  const prevIndex = values.indexOf(previousValue);
  const contentEls = getContentEls(scope);
  contentEls.forEach((contentEl) => {
    const itemValue = contentEl.dataset.value;
    const selected = value === itemValue;
    const prevSelected = prevIndex === values.indexOf(itemValue);
    if (!selected && !prevSelected) {
      delete contentEl.dataset.motion;
      return;
    }
    const attribute = (() => {
      if (index !== prevIndex) {
        if (selected && prevIndex !== -1) return index > prevIndex ? "from-end" : "from-start";
        if (prevSelected && index !== -1) return index > prevIndex ? "to-start" : "to-end";
      }
      return null;
    })();
    if (attribute) {
      contentEl.dataset.motion = attribute;
    } else {
      delete contentEl.dataset.motion;
    }
  });
}
function focusFirst(scope, candidates) {
  const previouslyFocusedElement = scope.getActiveElement();
  return candidates.some((candidate) => {
    if (candidate === previouslyFocusedElement) return true;
    candidate.focus();
    return scope.getActiveElement() !== previouslyFocusedElement;
  });
}
function removeFromTabOrder(candidates) {
  candidates.forEach((candidate) => {
    candidate.dataset.tabindex = candidate.getAttribute("tabindex") || "";
    candidate.setAttribute("tabindex", "-1");
  });
  return () => {
    candidates.forEach((candidate) => {
      const prevTabIndex = candidate.dataset.tabindex;
      candidate.setAttribute("tabindex", prevTabIndex);
    });
  };
}
export {
  focusFirst,
  getContentEl,
  getContentEls,
  getContentId,
  getElements,
  getItemId,
  getLinkEls,
  getListEl,
  getListId,
  getRootEl,
  getRootId,
  getTabbableEls,
  getTriggerEl,
  getTriggerEls,
  getTriggerId,
  getTriggerProxyEl,
  getTriggerProxyId,
  getViewportEl,
  getViewportId,
  removeFromTabOrder,
  setMotionAttr,
  trackResizeObserver
};
