// @zag-js/navigation-menu/dist/navigation-menu.connect.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/navigation-menu.connect.ts
var navigation_menu_connect_exports = {};
__export(navigation_menu_connect_exports, {
  connect: () => connect
});
module.exports = __toCommonJS(navigation_menu_connect_exports);
var import_dom_query = require("@zag-js/dom-query");
var import_utils = require("@zag-js/utils");
var import_navigation_menu = require("./navigation-menu.anatomy.js");
var dom = __toESM(require("./navigation-menu.dom.js"));
function connect(service, normalize) {
  const { context, send, prop, scope } = service;
  const translations = prop("translations");
  const triggerRect = context.get("triggerRect");
  const viewportSize = context.get("viewportSize");
  const viewportPosition = context.get("viewportPosition");
  const value = context.get("value");
  const previousValue = context.get("previousValue");
  const open = Boolean(value);
  const isViewportRendered = context.get("isViewportRendered");
  const preventTransition = value && !previousValue;
  function getItemState(props) {
    const selected = value === props.value;
    const wasSelected = !value && previousValue === props.value;
    return {
      itemId: dom.getItemId(scope, props.value),
      triggerId: dom.getTriggerId(scope, props.value),
      triggerProxyId: dom.getTriggerProxyId(scope, props.value),
      contentId: dom.getContentId(scope, props.value),
      selected,
      wasSelected,
      open: selected || wasSelected,
      disabled: !!props.disabled
    };
  }
  return {
    open,
    value,
    orientation: prop("orientation"),
    isViewportRendered,
    getViewportNode() {
      return dom.getViewportEl(scope);
    },
    setValue(value2) {
      send({ type: "VALUE.SET", value: value2 });
    },
    reposition() {
      send({ type: "VIEWPORT.POSITION" });
    },
    getRootProps() {
      return normalize.element({
        ...import_navigation_menu.parts.root.attrs,
        id: dom.getRootId(scope),
        "aria-label": translations?.rootLabel,
        "data-orientation": prop("orientation"),
        dir: prop("dir"),
        style: {
          "--trigger-width": (0, import_utils.toPx)(triggerRect?.width),
          "--trigger-height": (0, import_utils.toPx)(triggerRect?.height),
          "--trigger-x": (0, import_utils.toPx)(triggerRect?.x),
          "--trigger-y": (0, import_utils.toPx)(triggerRect?.y),
          "--viewport-width": (0, import_utils.toPx)(viewportSize?.width),
          "--viewport-height": (0, import_utils.toPx)(viewportSize?.height),
          "--viewport-x": (0, import_utils.toPx)(viewportPosition?.x),
          "--viewport-y": (0, import_utils.toPx)(viewportPosition?.y)
        }
      });
    },
    getListProps() {
      return normalize.element({
        ...import_navigation_menu.parts.list.attrs,
        id: dom.getListId(scope),
        dir: prop("dir"),
        "data-orientation": prop("orientation"),
        style: { position: "relative" }
      });
    },
    getItemProps(props) {
      const itemState = getItemState(props);
      return normalize.element({
        ...import_navigation_menu.parts.item.attrs,
        id: itemState.itemId,
        dir: prop("dir"),
        "data-value": props.value,
        "data-state": itemState.open ? "open" : "closed",
        "data-orientation": prop("orientation"),
        "data-disabled": (0, import_dom_query.dataAttr)(itemState.disabled),
        onKeyDown(event) {
          switch (event.key) {
            case "ArrowDown":
            case "ArrowUp":
            case "ArrowLeft":
            case "ArrowRight":
            case "Home":
            case "End": {
              send({ type: "ITEM.NAVIGATE", value: props.value, key: event.key });
              event.preventDefault();
              event.stopPropagation();
              break;
            }
            default:
              break;
          }
        }
      });
    },
    getIndicatorProps() {
      return normalize.element({
        ...import_navigation_menu.parts.indicator.attrs,
        "aria-hidden": true,
        dir: prop("dir"),
        hidden: !open,
        "data-state": open ? "open" : "closed",
        "data-orientation": prop("orientation"),
        style: {
          position: "absolute",
          transition: preventTransition ? "none" : void 0
        }
      });
    },
    getArrowProps() {
      return normalize.element({
        ...import_navigation_menu.parts.arrow.attrs,
        "aria-hidden": true,
        dir: prop("dir"),
        "data-orientation": prop("orientation")
      });
    },
    getTriggerProps(props) {
      const itemState = getItemState(props);
      return normalize.button({
        ...import_navigation_menu.parts.trigger.attrs,
        id: itemState.triggerId,
        "data-uid": prop("id"),
        "data-trigger-proxy-id": dom.getTriggerProxyId(scope, props.value),
        dir: prop("dir"),
        disabled: itemState.disabled,
        "data-value": props.value,
        "data-state": itemState.selected ? "open" : "closed",
        "data-disabled": (0, import_dom_query.dataAttr)(itemState.disabled),
        "aria-controls": itemState.contentId,
        "aria-expanded": itemState.selected,
        onPointerEnter(event) {
          if (prop("disableHoverTrigger")) return;
          if (event.pointerType !== "mouse") return;
          if (itemState.disabled) return;
          send({ type: "TRIGGER.POINTERENTER", value: props.value });
        },
        onPointerLeave(event) {
          if (prop("disableHoverTrigger")) return;
          if (event.pointerType !== "mouse") return;
          if (itemState.disabled) return;
          send({ type: "TRIGGER.POINTERLEAVE", value: props.value });
        },
        onClick() {
          if (prop("disableClickTrigger")) return;
          send({ type: "TRIGGER.CLICK", value: props.value });
        },
        onKeyDown(event) {
          const verticalEntryKey = prop("dir") === "rtl" ? "ArrowLeft" : "ArrowRight";
          const entryKey = {
            horizontal: "ArrowDown",
            vertical: verticalEntryKey
          }[prop("orientation")];
          if (open && event.key === entryKey) {
            send({ type: "CONTENT.FOCUS", side: "start" });
            event.preventDefault();
            event.stopPropagation();
            return;
          }
          const elements = dom.getElements(scope);
          const currentElement = event.currentTarget;
          const nextElement = (0, import_dom_query.navigate)(elements, currentElement, {
            key: event.key,
            orientation: prop("orientation"),
            dir: prop("dir"),
            loop: false
          });
          if (nextElement) {
            nextElement.focus();
            event.preventDefault();
            event.stopPropagation();
          }
        }
      });
    },
    getTriggerProxyProps(props) {
      const itemState = getItemState(props);
      return normalize.element({
        "aria-hidden": true,
        tabIndex: 0,
        "data-trigger-proxy": "",
        id: itemState.triggerProxyId,
        "data-trigger-id": itemState.triggerId,
        hidden: !itemState.selected,
        style: import_dom_query.visuallyHiddenStyle,
        onFocus(event) {
          const contentEl = dom.getContentEl(scope, props.value);
          if (!contentEl) return;
          const prevFocusedEl = event.relatedTarget;
          const wasTriggerFocused = prevFocusedEl === dom.getTriggerEl(scope, props.value);
          const wasFocusFromContent = (0, import_dom_query.contains)(contentEl, prevFocusedEl);
          if (wasTriggerFocused || !wasFocusFromContent) {
            send({ type: "CONTENT.FOCUS", side: wasTriggerFocused ? "start" : "end" });
          }
        }
      });
    },
    getViewportProxyProps(props) {
      const itemState = getItemState(props);
      if (!itemState.selected || !isViewportRendered) {
        return { hidden: true };
      }
      return normalize.element({
        "aria-owns": dom.getContentId(scope, props.value)
      });
    },
    getLinkProps(props) {
      const { closeOnClick = true } = props;
      return normalize.element({
        ...import_navigation_menu.parts.link.attrs,
        dir: prop("dir"),
        "data-value": props.value,
        "data-current": (0, import_dom_query.dataAttr)(props.current),
        "aria-current": props.current ? "page" : void 0,
        "data-ownedby": dom.getContentId(scope, props.value),
        onClick(event) {
          const target = event.currentTarget;
          const win = (0, import_dom_query.getWindow)(target);
          const onSelect = props.onSelect;
          target.addEventListener("link.select", onSelect, { once: true });
          const linkSelectEvent = new win.CustomEvent("link.select", {
            bubbles: true,
            cancelable: true,
            detail: { originalEvent: event }
          });
          target.dispatchEvent(linkSelectEvent);
          if (closeOnClick && !linkSelectEvent.defaultPrevented && !event.metaKey) {
            send({ type: "CLOSE" });
          }
        },
        onKeyDown(event) {
          const currentElement = event.currentTarget;
          const contentEl = currentElement.closest('[data-scope="navigation-menu"][data-part="content"]');
          const elements = contentEl ? dom.getLinkEls(scope, props.value) : dom.getElements(scope);
          const nextElement = (0, import_dom_query.navigate)(elements, currentElement, {
            key: event.key,
            orientation: prop("orientation"),
            dir: prop("dir"),
            loop: false
          });
          if (nextElement) {
            nextElement.focus();
            event.preventDefault();
            event.stopPropagation();
          }
        }
      });
    },
    getContentProps(props) {
      const itemState = getItemState(props);
      const currentValue = context.get("value") || context.get("previousValue");
      const selected = isViewportRendered ? currentValue === props.value : itemState.selected;
      return normalize.element({
        ...import_navigation_menu.parts.content.attrs,
        id: itemState.contentId,
        dir: prop("dir"),
        hidden: !selected,
        "aria-labelledby": itemState.triggerId,
        "data-uid": prop("id"),
        "data-state": selected ? "open" : "closed",
        "data-orientation": prop("orientation"),
        "data-value": props.value,
        onPointerEnter(event) {
          if (event.pointerType !== "mouse") return;
          send({ type: "CONTENT.POINTERENTER", value: props.value });
        },
        onPointerLeave(event) {
          if (event.pointerType !== "mouse") return;
          send({ type: "CONTENT.POINTERLEAVE", value: props.value });
        },
        onKeyDown(event) {
          if (event.defaultPrevented) return;
          if (event.currentTarget.closest("[data-scope=navigation-menu][data-part=root]") !== dom.getRootEl(scope))
            return;
          const isMetaKey = event.altKey || event.ctrlKey || event.metaKey;
          const isTabKey = event.key === "Tab" && !isMetaKey;
          const candidates = (0, import_dom_query.getTabbables)(event.currentTarget);
          if (isTabKey) {
            const focusedElement = scope.getActiveElement();
            const index = candidates.findIndex((candidate) => candidate === focusedElement);
            const isMovingBackwards = event.shiftKey;
            const nextCandidates = isMovingBackwards ? candidates.slice(0, index).reverse() : candidates.slice(index + 1, candidates.length);
            if (dom.focusFirst(scope, nextCandidates)) {
              event.preventDefault();
            } else {
              dom.getTriggerProxyEl(scope, props.value)?.focus();
              return;
            }
          }
          if (!(0, import_dom_query.isSelfTarget)(event)) return;
          const el = (0, import_dom_query.navigate)(candidates, scope.getActiveElement(), {
            key: event.key,
            dir: prop("dir"),
            loop: false
          });
          if (el) {
            el.focus();
            event.preventDefault();
            event.stopPropagation();
          }
        }
      });
    },
    getViewportPositionerProps(props = {}) {
      const { align = "center" } = props;
      return normalize.element({
        ...import_navigation_menu.parts.viewportPositioner.attrs,
        dir: prop("dir"),
        "data-orientation": prop("orientation"),
        "data-align": align
      });
    },
    getViewportProps(props = {}) {
      const { align = "center" } = props;
      const open2 = Boolean(value);
      return normalize.element({
        ...import_navigation_menu.parts.viewport.attrs,
        id: dom.getViewportId(scope),
        dir: prop("dir"),
        hidden: !open2,
        "data-state": open2 ? "open" : "closed",
        "data-orientation": prop("orientation"),
        "data-align": align,
        style: {
          transition: preventTransition ? "none" : void 0,
          pointerEvents: !open2 ? "none" : void 0,
          "--viewport-width": (0, import_utils.toPx)(viewportSize?.width),
          "--viewport-height": (0, import_utils.toPx)(viewportSize?.height),
          "--viewport-x": (0, import_utils.toPx)(viewportPosition?.x),
          "--viewport-y": (0, import_utils.toPx)(viewportPosition?.y)
        },
        onPointerEnter() {
          send({ type: "CONTENT.POINTERENTER" });
        },
        onPointerLeave(event) {
          if (prop("disablePointerLeaveClose")) return;
          if (event.pointerType !== "mouse") return;
          send({ type: "CONTENT.POINTERLEAVE" });
        }
      });
    },
    getItemState,
    getItemIndicatorProps(props) {
      const itemState = getItemState(props);
      return normalize.element({
        ...import_navigation_menu.parts.itemIndicator.attrs,
        "aria-hidden": true,
        dir: prop("dir"),
        hidden: !itemState.selected,
        "data-state": itemState.selected ? "open" : "closed",
        "data-orientation": prop("orientation"),
        "data-value": props.value
      });
    }
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  connect
});
