// @zag-js/navigation-menu/dist/navigation-menu.dom.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/navigation-menu.dom.ts
var navigation_menu_dom_exports = {};
__export(navigation_menu_dom_exports, {
  focusFirst: () => focusFirst,
  getContentEl: () => getContentEl,
  getContentEls: () => getContentEls,
  getContentId: () => getContentId,
  getElements: () => getElements,
  getItemId: () => getItemId,
  getLinkEls: () => getLinkEls,
  getListEl: () => getListEl,
  getListId: () => getListId,
  getRootEl: () => getRootEl,
  getRootId: () => getRootId,
  getTabbableEls: () => getTabbableEls,
  getTriggerEl: () => getTriggerEl,
  getTriggerEls: () => getTriggerEls,
  getTriggerId: () => getTriggerId,
  getTriggerProxyEl: () => getTriggerProxyEl,
  getTriggerProxyId: () => getTriggerProxyId,
  getViewportEl: () => getViewportEl,
  getViewportId: () => getViewportId,
  removeFromTabOrder: () => removeFromTabOrder,
  setMotionAttr: () => setMotionAttr,
  trackResizeObserver: () => trackResizeObserver
});
module.exports = __toCommonJS(navigation_menu_dom_exports);
var import_dom_query = require("@zag-js/dom-query");
var getRootId = (ctx) => ctx.ids?.root ?? `nav-menu:${ctx.id}`;
var getTriggerId = (ctx, value) => ctx.ids?.trigger?.(value) ?? `nav-menu:${ctx.id}:trigger:${value}`;
var getTriggerProxyId = (ctx, value) => ctx.ids?.triggerProxy?.(value) ?? `nav-menu:${ctx.id}:trigger-proxy:${value}`;
var getContentId = (ctx, value) => ctx.ids?.content?.(value) ?? `nav-menu:${ctx.id}:content:${value}`;
var getViewportId = (ctx) => ctx.ids?.viewport ?? `nav-menu:${ctx.id}:viewport`;
var getListId = (ctx) => ctx.ids?.list ?? `nav-menu:${ctx.id}:list`;
var getItemId = (ctx, value) => ctx.ids?.item?.(value) ?? `nav-menu:${ctx.id}:item:${value}`;
var getRootEl = (ctx) => ctx.getById(getRootId(ctx));
var getViewportEl = (ctx) => ctx.getById(getViewportId(ctx));
var getTriggerEl = (ctx, value) => {
  if (!value) return null;
  return ctx.getById(getTriggerId(ctx, value));
};
var getTriggerProxyEl = (ctx, value) => {
  if (!value) return null;
  return ctx.getById(getTriggerProxyId(ctx, value));
};
var getListEl = (ctx) => ctx.getById(getListId(ctx));
var getContentEl = (ctx, value) => {
  if (!value) return null;
  return ctx.getById(getContentId(ctx, value));
};
var getContentEls = (ctx) => (0, import_dom_query.queryAll)(ctx.getDoc(), `[data-scope=navigation-menu][data-part=content][data-uid='${ctx.id}']`);
var getTabbableEls = (ctx, value) => {
  return (0, import_dom_query.getTabbables)(getContentEl(ctx, value));
};
var getTriggerEls = (ctx) => (0, import_dom_query.queryAll)(getListEl(ctx), `[data-part=trigger][data-uid='${ctx.id}']`);
var getLinkEls = (ctx, value) => {
  const contentEl = getContentEl(ctx, value);
  return (0, import_dom_query.queryAll)(contentEl, `[data-part=link][data-ownedby="${getContentId(ctx, value)}"]`);
};
var getElements = (ctx) => {
  const topLevelTriggerSelector = `[data-part=trigger][data-uid='${ctx.id}']:not([data-disabled])`;
  const topLevelLinkSelector = `[data-part=item] > [data-part=link]`;
  return (0, import_dom_query.queryAll)(getListEl(ctx), `${topLevelTriggerSelector}, ${topLevelLinkSelector}`);
};
function trackResizeObserver(element, onResize) {
  if (!element.length) return;
  let frame = 0;
  const win = (0, import_dom_query.getWindow)(element[0]);
  const obs = new win.ResizeObserver(() => {
    cancelAnimationFrame(frame);
    frame = requestAnimationFrame(onResize);
  });
  element.forEach((el) => {
    if (el) obs.observe(el);
  });
  return () => {
    cancelAnimationFrame(frame);
    element.forEach((el) => {
      if (el) obs.unobserve(el);
    });
  };
}
function setMotionAttr(scope, value, previousValue) {
  const triggers = getTriggerEls(scope);
  const dir = triggers[0].dir;
  let values = triggers.map((trigger) => trigger.getAttribute("data-value"));
  if (dir === "rtl") values.reverse();
  const index = values.indexOf(value);
  const prevIndex = values.indexOf(previousValue);
  const contentEls = getContentEls(scope);
  contentEls.forEach((contentEl) => {
    const itemValue = contentEl.dataset.value;
    const selected = value === itemValue;
    const prevSelected = prevIndex === values.indexOf(itemValue);
    if (!selected && !prevSelected) {
      delete contentEl.dataset.motion;
      return;
    }
    const attribute = (() => {
      if (index !== prevIndex) {
        if (selected && prevIndex !== -1) return index > prevIndex ? "from-end" : "from-start";
        if (prevSelected && index !== -1) return index > prevIndex ? "to-start" : "to-end";
      }
      return null;
    })();
    if (attribute) {
      contentEl.dataset.motion = attribute;
    } else {
      delete contentEl.dataset.motion;
    }
  });
}
function focusFirst(scope, candidates) {
  const previouslyFocusedElement = scope.getActiveElement();
  return candidates.some((candidate) => {
    if (candidate === previouslyFocusedElement) return true;
    candidate.focus();
    return scope.getActiveElement() !== previouslyFocusedElement;
  });
}
function removeFromTabOrder(candidates) {
  candidates.forEach((candidate) => {
    candidate.dataset.tabindex = candidate.getAttribute("tabindex") || "";
    candidate.setAttribute("tabindex", "-1");
  });
  return () => {
    candidates.forEach((candidate) => {
      const prevTabIndex = candidate.dataset.tabindex;
      candidate.setAttribute("tabindex", prevTabIndex);
    });
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  focusFirst,
  getContentEl,
  getContentEls,
  getContentId,
  getElements,
  getItemId,
  getLinkEls,
  getListEl,
  getListId,
  getRootEl,
  getRootId,
  getTabbableEls,
  getTriggerEl,
  getTriggerEls,
  getTriggerId,
  getTriggerProxyEl,
  getTriggerProxyId,
  getViewportEl,
  getViewportId,
  removeFromTabOrder,
  setMotionAttr,
  trackResizeObserver
});
