// @ark-ui/react/dist/components/timer/index.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const require_use_timer_context = require("./use-timer-context.cjs");
const require_timer_action_trigger = require("./timer-action-trigger.cjs");
const require_timer_area = require("./timer-area.cjs");
const require_timer_context = require("./timer-context.cjs");
const require_timer_control = require("./timer-control.cjs");
const require_timer_item = require("./timer-item.cjs");
const require_use_timer = require("./use-timer.cjs");
const require_timer_root = require("./timer-root.cjs");
const require_timer_root_provider = require("./timer-root-provider.cjs");
const require_timer_separator = require("./timer-separator.cjs");
const require_timer = require("./timer.cjs");
let _zag_js_timer = require("@zag-js/timer");
Object.defineProperty(exports, "Timer", {
	enumerable: true,
	get: function() {
		return require_timer.timer_exports;
	}
});
exports.TimerActionTrigger = require_timer_action_trigger.TimerActionTrigger;
exports.TimerArea = require_timer_area.TimerArea;
exports.TimerContext = require_timer_context.TimerContext;
exports.TimerControl = require_timer_control.TimerControl;
exports.TimerItem = require_timer_item.TimerItem;
exports.TimerRoot = require_timer_root.TimerRoot;
exports.TimerRootProvider = require_timer_root_provider.TimerRootProvider;
exports.TimerSeparator = require_timer_separator.TimerSeparator;
Object.defineProperty(exports, "timerAnatomy", {
	enumerable: true,
	get: function() {
		return _zag_js_timer.anatomy;
	}
});
exports.useTimer = require_use_timer.useTimer;
exports.useTimerContext = require_use_timer_context.useTimerContext;
