// @ark-ui/react/dist/components/timer/timer.cjs
const require_runtime = require("../../_virtual/_rolldown/runtime.cjs");
const require_timer_action_trigger = require("./timer-action-trigger.cjs");
const require_timer_area = require("./timer-area.cjs");
const require_timer_context = require("./timer-context.cjs");
const require_timer_control = require("./timer-control.cjs");
const require_timer_item = require("./timer-item.cjs");
const require_timer_root = require("./timer-root.cjs");
const require_timer_root_provider = require("./timer-root-provider.cjs");
const require_timer_separator = require("./timer-separator.cjs");
//#region src/components/timer/timer.ts
var timer_exports = /* @__PURE__ */ require_runtime.__exportAll({
	ActionTrigger: () => require_timer_action_trigger.TimerActionTrigger,
	Area: () => require_timer_area.TimerArea,
	Context: () => require_timer_context.TimerContext,
	Control: () => require_timer_control.TimerControl,
	Item: () => require_timer_item.TimerItem,
	Root: () => require_timer_root.TimerRoot,
	RootProvider: () => require_timer_root_provider.TimerRootProvider,
	Separator: () => require_timer_separator.TimerSeparator
});
//#endregion
Object.defineProperty(exports, "timer_exports", {
	enumerable: true,
	get: function() {
		return timer_exports;
	}
});
