// @zag-js/clipboard/dist/clipboard.connect.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/clipboard.connect.ts
var clipboard_connect_exports = {};
__export(clipboard_connect_exports, {
  connect: () => connect
});
module.exports = __toCommonJS(clipboard_connect_exports);
var import_dom_query = require("@zag-js/dom-query");
var import_clipboard = require("./clipboard.anatomy.js");
var dom = __toESM(require("./clipboard.dom.js"));
function connect(service, normalize) {
  const { state, send, context, scope, prop } = service;
  const copied = state.matches("copied");
  const translations = prop("translations");
  return {
    copied,
    value: context.get("value"),
    setValue(value) {
      send({ type: "VALUE.SET", value });
    },
    copy() {
      send({ type: "COPY" });
    },
    getRootProps() {
      return normalize.element({
        ...import_clipboard.parts.root.attrs,
        "data-copied": (0, import_dom_query.dataAttr)(copied),
        id: dom.getRootId(scope)
      });
    },
    getLabelProps() {
      return normalize.label({
        ...import_clipboard.parts.label.attrs,
        htmlFor: dom.getInputId(scope),
        "data-copied": (0, import_dom_query.dataAttr)(copied),
        id: dom.getLabelId(scope)
      });
    },
    getControlProps() {
      return normalize.element({
        ...import_clipboard.parts.control.attrs,
        "data-copied": (0, import_dom_query.dataAttr)(copied)
      });
    },
    getInputProps() {
      return normalize.input({
        ...import_clipboard.parts.input.attrs,
        defaultValue: context.get("value"),
        "data-copied": (0, import_dom_query.dataAttr)(copied),
        readOnly: true,
        "data-readonly": "true",
        id: dom.getInputId(scope),
        onFocus(event) {
          event.currentTarget.select();
        },
        onCopy() {
          send({ type: "INPUT.COPY" });
        }
      });
    },
    getTriggerProps() {
      return normalize.button({
        ...import_clipboard.parts.trigger.attrs,
        type: "button",
        "aria-label": translations.triggerLabel?.(copied),
        "data-copied": (0, import_dom_query.dataAttr)(copied),
        onClick() {
          send({ type: "COPY" });
        }
      });
    },
    getIndicatorProps(props) {
      return normalize.element({
        ...import_clipboard.parts.indicator.attrs,
        hidden: props.copied !== copied
      });
    }
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  connect
});
