// @zag-js/date-input/dist/utils/incomplete-date.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var __publicField = (obj, key, value) => __defNormalProp(obj, typeof key !== "symbol" ? key + "" : key, value);

// src/utils/incomplete-date.ts
var incomplete_date_exports = {};
__export(incomplete_date_exports, {
  IncompleteDate: () => IncompleteDate,
  incompleteDateEqual: () => incompleteDateEqual,
  incompleteDateHash: () => incompleteDateHash,
  initDisplayValues: () => initDisplayValues,
  resolvedHourCycle: () => resolvedHourCycle,
  toHourCycle: () => toHourCycle
});
module.exports = __toCommonJS(incomplete_date_exports);
var import_date = require("@internationalized/date");
function resolvedHourCycle(formatter) {
  const hc = formatter.resolvedOptions().hourCycle;
  if (hc === "h11" || hc === "h12" || hc === "h23" || hc === "h24") return hc;
  return "h23";
}
function incompleteDateHash(dv) {
  return `${dv.year}|${dv.month}|${dv.day}|${dv.hour}|${dv.dayPeriod}|${dv.minute}|${dv.second}|${dv.era}`;
}
function incompleteDateEqual(a, b) {
  return incompleteDateHash(a) === incompleteDateHash(b);
}
function initDisplayValues(value, placeholderValue, hourCycle, count) {
  const calendar = placeholderValue.calendar;
  if (value?.length) {
    return Array.from(
      { length: count },
      (_, i) => value[i] ? new IncompleteDate(calendar, hourCycle, value[i]) : new IncompleteDate(calendar, hourCycle)
    );
  }
  return Array.from({ length: count }, () => new IncompleteDate(calendar, hourCycle));
}
var IncompleteDate = class _IncompleteDate {
  constructor(calendar, hourCycle, dateValue) {
    __publicField(this, "calendar");
    __publicField(this, "era");
    __publicField(this, "year");
    __publicField(this, "month");
    __publicField(this, "day");
    __publicField(this, "hour");
    __publicField(this, "hourCycle");
    __publicField(this, "dayPeriod");
    __publicField(this, "minute");
    __publicField(this, "second");
    __publicField(this, "millisecond");
    __publicField(this, "offset");
    this.era = dateValue?.era ?? null;
    this.calendar = calendar;
    this.year = dateValue?.year ?? null;
    this.month = dateValue?.month ?? null;
    this.day = dateValue?.day ?? null;
    this.hour = dateValue?.hour ?? null;
    this.hourCycle = hourCycle;
    this.dayPeriod = null;
    this.minute = dateValue?.minute ?? null;
    this.second = dateValue?.second ?? null;
    this.millisecond = dateValue?.millisecond ?? null;
    this.offset = "offset" in (dateValue ?? {}) ? dateValue.offset : null;
    if (this.hour != null) {
      const [dayPeriod, hour] = toHourCycle(this.hour, hourCycle);
      this.dayPeriod = dayPeriod;
      this.hour = hour;
    }
  }
  copy() {
    const res = new _IncompleteDate(this.calendar, this.hourCycle);
    res.era = this.era;
    res.year = this.year;
    res.month = this.month;
    res.day = this.day;
    res.hour = this.hour;
    res.dayPeriod = this.dayPeriod;
    res.minute = this.minute;
    res.second = this.second;
    res.millisecond = this.millisecond;
    res.offset = this.offset;
    return res;
  }
  /** Rebuild this date under a new hour cycle, preserving all field values. */
  withHourCycle(nextHourCycle) {
    if (nextHourCycle === this.hourCycle) return this;
    const hour24 = this.hour != null ? fromHourCycle(this.hour, this.dayPeriod ?? 0, this.hourCycle) : null;
    const res = new _IncompleteDate(this.calendar, nextHourCycle, {
      era: this.era,
      year: this.year,
      month: this.month,
      day: this.day,
      hour: hour24,
      minute: this.minute,
      second: this.second,
      millisecond: this.millisecond
    });
    res.offset = this.offset;
    if (this.hour == null && this.dayPeriod != null && (nextHourCycle === "h12" || nextHourCycle === "h11")) {
      res.dayPeriod = this.dayPeriod;
    }
    return res;
  }
  /** Checks whether all the specified segments have a value. */
  isComplete(segments) {
    return segments.every((segment) => this[segment] != null);
  }
  /** Checks if the date is empty (i.e. all specified segments are null). */
  isCleared(segments) {
    return segments.every((segment) => this[segment] === null);
  }
  /** Sets the given field. */
  set(field, value, placeholder) {
    const result = this.copy();
    result[field] = value;
    if (field === "hour" && result.dayPeriod == null && "hour" in placeholder) {
      result.dayPeriod = toHourCycle(placeholder.hour, this.hourCycle)[0];
    }
    if (field === "year" && result.era == null) {
      result.era = placeholder.era;
    }
    if (field !== "second" && field !== "literal" && field !== "timeZoneName") {
      result.offset = null;
    }
    return result;
  }
  /** Sets the given field to null. */
  clear(field) {
    const result = this.copy();
    result[field] = null;
    if (field === "year") {
      result.era = null;
    }
    result.offset = null;
    return result;
  }
  /** Increments or decrements the given field. If it is null, then it is set to the placeholder value (no increment on first press). */
  cycle(field, amount, placeholder, displaySegments) {
    const res = this.copy();
    if (res[field] == null && field !== "dayPeriod" && field !== "era") {
      if (field === "hour" && "hour" in placeholder) {
        const [dayPeriod, hour] = toHourCycle(placeholder.hour, this.hourCycle);
        res.dayPeriod = dayPeriod;
        res.hour = hour;
      } else {
        ;
        res[field] = placeholder[field];
      }
      if (field === "year" && res.era == null) {
        res.era = placeholder.era;
      }
      return res;
    }
    switch (field) {
      case "era": {
        const eras = this.calendar.getEras();
        let index = eras.indexOf(res.era);
        index = cycleValue(index, amount, 0, eras.length - 1);
        res.era = eras[index];
        break;
      }
      case "year": {
        let date = new import_date.CalendarDate(
          this.calendar,
          this.era ?? placeholder.era,
          this.year ?? placeholder.year,
          this.month ?? 1,
          this.day ?? 1
        );
        date = date.cycle(field, amount, { round: field === "year" });
        res.era = date.era;
        res.year = date.year;
        break;
      }
      case "month":
        res.month = cycleValue(res.month ?? 1, amount, 1, this.calendar.getMaximumMonthsInYear());
        break;
      case "day":
        res.day = cycleValue(res.day ?? 1, amount, 1, this.calendar.getMaximumDaysInMonth());
        break;
      case "hour": {
        const hasDateSegments = displaySegments.some((s) => ["year", "month", "day"].includes(s));
        if ("timeZone" in placeholder && (!hasDateSegments || res.year != null && res.month != null && res.day != null)) {
          let date = this.toValue(placeholder);
          date = date.cycle("hour", amount, { hourCycle: this.hourCycle === "h12" ? 12 : 24, round: false });
          const [dayPeriod, adjustedHour] = toHourCycle(date.hour, this.hourCycle);
          res.hour = adjustedHour;
          res.dayPeriod = dayPeriod;
          res.offset = date.offset;
        } else {
          const hours = res.hour ?? 0;
          const limits = this.getSegmentLimits("hour");
          res.hour = cycleValue(hours, amount, limits.minValue, limits.maxValue);
          if (res.dayPeriod == null && "hour" in placeholder) {
            res.dayPeriod = toHourCycle(placeholder.hour, this.hourCycle)[0];
          }
        }
        break;
      }
      case "dayPeriod":
        res.dayPeriod = cycleValue(res.dayPeriod ?? 0, amount, 0, 1);
        break;
      case "minute":
        res.minute = cycleValue(res.minute ?? 0, amount, 0, 59, true);
        break;
      case "second":
        res.second = cycleValue(res.second ?? 0, amount, 0, 59, true);
        break;
    }
    return res;
  }
  /** Converts the incomplete date to a full date value, using the provided value for any unset fields. */
  toValue(value) {
    if ("hour" in value) {
      let hour = this.hour;
      if (hour != null) {
        hour = fromHourCycle(hour, this.dayPeriod ?? 0, this.hourCycle);
      } else if (this.hourCycle === "h12" || this.hourCycle === "h11") {
        hour = this.dayPeriod === 1 ? 12 : 0;
      }
      const res = value.set({
        era: this.era ?? value.era,
        year: this.year ?? value.year,
        month: this.month ?? value.month,
        day: this.day ?? value.day,
        hour: hour ?? value.hour,
        minute: this.minute ?? value.minute,
        second: this.second ?? value.second,
        millisecond: this.millisecond ?? value.millisecond
      });
      if ("offset" in res && this.offset != null && res.offset !== this.offset) {
        return res.add({ milliseconds: res.offset - this.offset });
      }
      return res;
    } else {
      return value.set({
        era: this.era ?? value.era,
        year: this.year ?? value.year,
        month: this.month ?? value.month,
        day: this.day ?? value.day
      });
    }
  }
  getSegmentLimits(type) {
    switch (type) {
      case "era": {
        const eras = this.calendar.getEras();
        return {
          value: this.era != null ? eras.indexOf(this.era) : eras.length - 1,
          minValue: 0,
          maxValue: eras.length - 1
        };
      }
      case "year":
        return {
          value: this.year,
          minValue: 1,
          maxValue: 9999
        };
      case "month":
        return {
          value: this.month,
          minValue: 1,
          maxValue: this.calendar.getMaximumMonthsInYear()
        };
      case "day":
        return {
          value: this.day,
          minValue: 1,
          maxValue: this.calendar.getMaximumDaysInMonth()
        };
      case "dayPeriod":
        return {
          value: this.dayPeriod,
          minValue: 0,
          maxValue: 1
        };
      case "hour": {
        let minValue = 0;
        let maxValue = 23;
        if (this.hourCycle === "h12") {
          minValue = 1;
          maxValue = 12;
        } else if (this.hourCycle === "h11") {
          minValue = 0;
          maxValue = 11;
        }
        return {
          value: this.hour,
          minValue,
          maxValue
        };
      }
      case "minute":
        return {
          value: this.minute,
          minValue: 0,
          maxValue: 59
        };
      case "second":
        return {
          value: this.second,
          minValue: 0,
          maxValue: 59
        };
    }
  }
  /** Returns a debug string of all non-null fields. Returns "-" when all fields are null. */
  toString() {
    const fields = {
      era: this.era,
      year: this.year,
      month: this.month,
      day: this.day,
      hour: this.hour,
      dayPeriod: this.dayPeriod,
      minute: this.minute,
      second: this.second
    };
    const parts = Object.entries(fields).filter(([, v]) => v != null).map(([k, v]) => `${k}=${v}`);
    return parts.length > 0 ? parts.join(",") : "-";
  }
};
function cycleValue(value, amount, min, max, round = false) {
  if (round) {
    value += Math.sign(amount);
    if (value < min) {
      value = max;
    }
    const div = Math.abs(amount);
    if (amount > 0) {
      value = Math.ceil(value / div) * div;
    } else {
      value = Math.floor(value / div) * div;
    }
    if (value > max) {
      value = min;
    }
  } else {
    value += amount;
    if (value < min) {
      value = max - (min - value - 1);
    } else if (value > max) {
      value = min + (value - max - 1);
    }
  }
  return value;
}
function toHourCycle(hour, hourCycle) {
  let dayPeriod = hour >= 12 ? 1 : 0;
  switch (hourCycle) {
    case "h11":
      if (hour >= 12) {
        hour -= 12;
      }
      break;
    case "h12":
      if (hour === 0) {
        hour = 12;
      } else if (hour > 12) {
        hour -= 12;
      }
      break;
    case "h23":
      dayPeriod = null;
      break;
    case "h24":
      hour += 1;
      dayPeriod = null;
  }
  return [dayPeriod, hour];
}
function fromHourCycle(hour, dayPeriod, hourCycle) {
  switch (hourCycle) {
    case "h11":
      if (dayPeriod === 1) {
        hour += 12;
      }
      break;
    case "h12":
      if (hour === 12) {
        hour = 0;
      }
      if (dayPeriod === 1) {
        hour += 12;
      }
      break;
    case "h24":
      hour -= 1;
      break;
  }
  return hour;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  IncompleteDate,
  incompleteDateEqual,
  incompleteDateHash,
  initDisplayValues,
  resolvedHourCycle,
  toHourCycle
});
