// @zag-js/date-input/dist/utils/placeholders.mjs
import "../chunk-QZ7TP4HQ.mjs";

// src/utils/placeholders.ts
var LOCALE_PLACEHOLDERS = {
  ach: ["mwaka", "dwe", "nino"],
  af: ["jjjj", "mm", "dd"],
  am: ["\u12D3\u12D3\u12D3\u12D3", "\u121A\u121C", "\u1240\u1240"],
  an: ["aaaa", "mm", "dd"],
  ar: ["\u0633\u0646\u0629", "\u0634\u0647\u0631", "\u064A\u0648\u0645"],
  ast: ["aaaa", "mm", "dd"],
  az: ["iiii", "aa", "gg"],
  be: ["\u0433\u0433\u0433\u0433", "\u043C\u043C", "\u0434\u0434"],
  bg: ["\u0433\u0433\u0433\u0433", "\u043C\u043C", "\u0434\u0434"],
  bn: ["yyyy", "\u09AE\u09BF\u09AE\u09BF", "dd"],
  br: ["bbbb", "mm", "dd"],
  bs: ["gggg", "mm", "dd"],
  ca: ["aaaa", "mm", "dd"],
  cak: ["jjjj", "ii", "q'q'"],
  ckb: ["\u0633\u0627\u06B5", "\u0645\u0627\u0646\u06AF", "\u0695\u06C6\u0698"],
  cs: ["rrrr", "mm", "dd"],
  cy: ["bbbb", "mm", "dd"],
  da: ["\xE5\xE5\xE5\xE5", "mm", "dd"],
  de: ["jjjj", "mm", "tt"],
  dsb: ["llll", "mm", "\u017A\u017A"],
  el: ["\u03B5\u03B5\u03B5\u03B5", "\u03BC\u03BC", "\u03B7\u03B7"],
  en: ["yyyy", "mm", "dd"],
  eo: ["jjjj", "mm", "tt"],
  es: ["aaaa", "mm", "dd"],
  et: ["aaaa", "kk", "pp"],
  eu: ["uuuu", "hh", "ee"],
  fa: ["\u0633\u0627\u0644", "\u0645\u0627\u0647", "\u0631\u0648\u0632"],
  ff: ["hhhh", "ll", "\xF1\xF1"],
  fi: ["vvvv", "kk", "pp"],
  fr: ["aaaa", "mm", "jj"],
  fy: ["jjjj", "mm", "dd"],
  ga: ["bbbb", "mm", "ll"],
  gd: ["bbbb", "mm", "ll"],
  gl: ["aaaa", "mm", "dd"],
  he: ["\u05E9\u05E0\u05D4", "\u05D7\u05D5\u05D3\u05E9", "\u05D9\u05D5\u05DD"],
  hr: ["gggg", "mm", "dd"],
  hsb: ["llll", "mm", "dd"],
  hu: ["\xE9\xE9\xE9\xE9", "hh", "nn"],
  ia: ["aaaa", "mm", "dd"],
  id: ["tttt", "bb", "hh"],
  it: ["aaaa", "mm", "gg"],
  ja: ["\u5E74", "\u6708", "\u65E5"],
  ka: ["\u10EC\u10EC\u10EC\u10EC", "\u10D7\u10D7", "\u10E0\u10E0"],
  kk: ["\u0436\u0436\u0436\u0436", "\u0430\u0430", "\u043A\u043A"],
  kn: ["\u0CB5\u0CB5\u0CB5\u0CB5", "\u0CAE\u0CBF\u0CAE\u0CC0", "\u0CA6\u0CBF\u0CA6\u0CBF"],
  ko: ["\uC5F0\uB3C4", "\uC6D4", "\uC77C"],
  lb: ["jjjj", "mm", "dd"],
  lo: ["\u0E9B\u0E9B\u0E9B\u0E9B", "\u0E94\u0E94", "\u0EA7\u0EA7"],
  lt: ["mmmm", "mm", "dd"],
  lv: ["gggg", "mm", "dd"],
  meh: ["aaaa", "mm", "dd"],
  ml: ["\u0D35\u0D7C\u0D37\u0D02", "\u0D2E\u0D3E\u0D38\u0D02", "\u0D24\u0D40\u0D2F\u0D24\u0D3F"],
  ms: ["tttt", "mm", "hh"],
  nl: ["jjjj", "mm", "dd"],
  nn: ["\xE5\xE5\xE5\xE5", "mm", "dd"],
  no: ["\xE5\xE5\xE5\xE5", "mm", "dd"],
  oc: ["aaaa", "mm", "jj"],
  pl: ["rrrr", "mm", "dd"],
  pt: ["aaaa", "mm", "dd"],
  rm: ["oooo", "mm", "dd"],
  ro: ["aaaa", "ll", "zz"],
  ru: ["\u0433\u0433\u0433\u0433", "\u043C\u043C", "\u0434\u0434"],
  sc: ["aaaa", "mm", "dd"],
  scn: ["aaaa", "mm", "jj"],
  sk: ["rrrr", "mm", "dd"],
  sl: ["llll", "mm", "dd"],
  sr: ["\u0433\u0433\u0433\u0433", "\u043C\u043C", "\u0434\u0434"],
  sv: ["\xE5\xE5\xE5\xE5", "mm", "dd"],
  szl: ["rrrr", "mm", "dd"],
  tg: ["\u0441\u0441\u0441\u0441", "\u043C\u043C", "\u0440\u0440"],
  th: ["\u0E1B\u0E1B\u0E1B\u0E1B", "\u0E14\u0E14", "\u0E27\u0E27"],
  tr: ["yyyy", "aa", "gg"],
  uk: ["\u0440\u0440\u0440\u0440", "\u043C\u043C", "\u0434\u0434"],
  "sr-Latn": ["gggg", "mm", "dd"],
  "zh-CN": ["\u5E74", "\u6708", "\u65E5"],
  "zh-TW": ["\u5E74", "\u6708", "\u65E5"]
};
function getLocaleLanguage(locale) {
  if (typeof Intl !== "undefined" && Intl.Locale) {
    return new Intl.Locale(locale).language;
  }
  return locale.split("-")[0];
}
function getLocaleScript(locale) {
  if (typeof Intl !== "undefined" && Intl.Locale) {
    return new Intl.Locale(locale).script || void 0;
  }
  const script = locale.split("-").find((part) => part.length === 4);
  if (!script) return void 0;
  return script.charAt(0).toUpperCase() + script.slice(1).toLowerCase();
}
function getLocalePlaceholders(locale) {
  const exact = LOCALE_PLACEHOLDERS[locale];
  if (exact) return exact;
  const lang = getLocaleLanguage(locale);
  const script = getLocaleScript(locale);
  if (script) {
    const langScript = `${lang}-${script}`;
    const langScriptPlaceholder = LOCALE_PLACEHOLDERS[langScript];
    if (langScriptPlaceholder) return langScriptPlaceholder;
  }
  return LOCALE_PLACEHOLDERS[lang] || LOCALE_PLACEHOLDERS.en;
}
var defaultTranslations = {
  placeholder(locale) {
    const [year, month, day] = getLocalePlaceholders(locale);
    return {
      day,
      month,
      year,
      hour: "\u2013\u2013",
      minute: "\u2013\u2013",
      second: "\u2013\u2013",
      dayPeriod: "AM/PM",
      era: "era",
      timeZoneName: "timeZone",
      weekday: "weekday",
      unknown: "unknown",
      fractionalSecond: "ff"
    };
  }
};
export {
  defaultTranslations
};
