// @zag-js/date-input/dist/utils/segments.mjs
import "../chunk-QZ7TP4HQ.mjs";

// src/utils/segments.ts
import "@internationalized/date";
function needsTimeGranularity(granularity) {
  return granularity === "hour" || granularity === "minute" || granularity === "second";
}
function getFormatterOptions(opts) {
  const { granularity, digitStyle, hourCycle, timeZone, hasTimeZone, hideTimeZone } = opts;
  const options = {
    timeZone,
    day: digitStyle,
    month: digitStyle,
    year: "numeric",
    hourCycle
  };
  if (needsTimeGranularity(granularity)) options.hour = digitStyle;
  if (granularity === "minute" || granularity === "second") options.minute = "2-digit";
  if (granularity === "second") options.second = "2-digit";
  if (hasTimeZone && !hideTimeZone) options.timeZoneName = "short";
  return options;
}
function resolveAllSegments(formatter) {
  const segs = formatter.formatToParts(/* @__PURE__ */ new Date()).filter((seg) => EDITABLE_SEGMENTS[seg.type]).reduce((p, seg) => {
    const key = TYPE_MAPPING[seg.type] || seg.type;
    p[key] = true;
    return p;
  }, {});
  if (segs.year) segs.era = true;
  return segs;
}
var EDITABLE_SEGMENTS = {
  year: true,
  month: true,
  day: true,
  hour: true,
  minute: true,
  second: true,
  dayPeriod: true,
  era: true,
  literal: false,
  timeZoneName: false,
  weekday: false,
  unknown: false,
  fractionalSecond: false
};
var PAGE_STEP = {
  year: 5,
  month: 2,
  day: 7,
  hour: 2,
  minute: 15,
  second: 15,
  dayPeriod: void 0,
  era: void 0,
  literal: void 0,
  timeZoneName: void 0,
  weekday: void 0,
  unknown: void 0,
  fractionalSecond: void 0
};
var SEGMENT_LABELS = {
  era: "Era",
  year: "Year",
  month: "Month",
  day: "Day",
  hour: "Hour",
  minute: "Minute",
  second: "Second",
  dayPeriod: "AM/PM",
  timeZoneName: "Time zone"
};
function getSegmentLabel(type) {
  return SEGMENT_LABELS[type] ?? type;
}
var TYPE_MAPPING = {
  // Node seems to convert everything to lowercase...
  dayperiod: "dayPeriod",
  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/formatToParts#named_years
  relatedYear: "year",
  yearName: "literal",
  // not editable
  unknown: "literal"
};
function getSafeType(type) {
  return TYPE_MAPPING[type] ?? type;
}
function getPlaceholder(type, translations, locale) {
  return translations.placeholder(locale)[type];
}
function isEditableSegment(type) {
  return EDITABLE_SEGMENTS[type] === true;
}
function getSafeFormatParts(formatter, dateValue, displayValue) {
  try {
    return formatter.formatToParts(dateValue);
  } catch {
    return formatter.formatToParts(/* @__PURE__ */ new Date()).map((part) => {
      const type = part.type;
      switch (type) {
        case "year":
        case "relatedYear":
          return { ...part, value: String(displayValue.year ?? 0) };
        case "month":
          return { ...part, value: String(displayValue.month ?? 0) };
        case "day":
          return { ...part, value: String(displayValue.day ?? 0) };
      }
      return part;
    });
  }
}
function processSegments({
  dateValue,
  displayValue,
  formatter,
  locale,
  translations,
  granularity
}) {
  const timeValue = ["hour", "minute", "second"];
  const segments = getSafeFormatParts(formatter, dateValue, displayValue);
  const processedSegments = [];
  for (const segment of segments) {
    const type = getSafeType(segment.type);
    let isEditable = isEditableSegment(type);
    if (type === "era" && displayValue.calendar.getEras().length === 1) {
      isEditable = false;
    }
    const isPlaceholder = isEditable && displayValue[type] == null;
    let placeholder = isEditableSegment(type) ? getPlaceholder(type, translations, locale) : null;
    if ((type === "dayPeriod" || type === "era") && segment.value) {
      placeholder = segment.value;
    }
    const dateSegment = {
      type,
      text: isPlaceholder ? placeholder : segment.value,
      ...displayValue.getSegmentLimits(type) ?? {},
      isPlaceholder,
      placeholder,
      isEditable
    };
    if (type === "hour") {
      processedSegments.push({
        type: "literal",
        text: "\u2066",
        isPlaceholder: false,
        placeholder: "",
        isEditable: false
      });
      processedSegments.push(dateSegment);
      if (type === granularity) {
        processedSegments.push({
          type: "literal",
          text: "\u2069",
          isPlaceholder: false,
          placeholder: "",
          isEditable: false
        });
      }
    } else if (timeValue.includes(type) && type === granularity) {
      processedSegments.push(dateSegment);
      processedSegments.push({
        type: "literal",
        text: "\u2069",
        isPlaceholder: false,
        placeholder: "",
        isEditable: false
      });
    } else {
      processedSegments.push(dateSegment);
    }
  }
  return processedSegments;
}
export {
  EDITABLE_SEGMENTS,
  PAGE_STEP,
  TYPE_MAPPING,
  getFormatterOptions,
  getSegmentLabel,
  needsTimeGranularity,
  processSegments,
  resolveAllSegments
};
