// @zag-js/date-input/dist/utils/validity.mjs
import "../chunk-QZ7TP4HQ.mjs";

// src/utils/validity.ts
var isFocusableSegment = (segment) => segment.type !== "literal";
function getGroupCount(selectionMode) {
  return selectionMode === "range" ? 2 : 1;
}
function getGroupOffset(allSegments, index) {
  return allSegments.slice(0, index).reduce((acc, segs) => acc + segs.length, 0);
}
function resolveActiveSegment(ctx) {
  const { context, computed } = ctx;
  const index = context.get("activeIndex");
  const activeSegmentIndex = context.get("activeSegmentIndex");
  const allSegments = computed("segments");
  const offset = getGroupOffset(allSegments, index);
  return { allSegments, segments: allSegments[index], offset, localIndex: activeSegmentIndex - offset };
}
function getActiveSegment(ctx) {
  const { allSegments, localIndex } = resolveActiveSegment(ctx);
  const index = ctx.context.get("activeIndex");
  return allSegments[index]?.[localIndex];
}
function goToNextSegment(ctx, predicate = isFocusableSegment) {
  const { context } = ctx;
  const index = context.get("activeIndex");
  const { allSegments, segments, offset, localIndex } = resolveActiveSegment(ctx);
  const nextLocalIndex = segments.findIndex((s, i) => i > localIndex && predicate(s));
  if (nextLocalIndex !== -1) {
    context.set("activeSegmentIndex", offset + nextLocalIndex);
    return;
  }
  const nextGroupSegments = allSegments[index + 1];
  if (!nextGroupSegments) return;
  const firstNextGroupLocalIndex = nextGroupSegments.findIndex(predicate);
  if (firstNextGroupLocalIndex === -1) return;
  context.set("activeIndex", index + 1);
  context.set("activeSegmentIndex", offset + segments.length + firstNextGroupLocalIndex);
}
function goToPreviousSegment(ctx, predicate = isFocusableSegment) {
  const { context } = ctx;
  const index = context.get("activeIndex");
  const { allSegments, segments, offset, localIndex } = resolveActiveSegment(ctx);
  const prevLocalIndex = segments.findLastIndex((s, i) => i < localIndex && predicate(s));
  if (prevLocalIndex !== -1) {
    context.set("activeSegmentIndex", offset + prevLocalIndex);
    return;
  }
  const prevGroupIndex = index - 1;
  if (prevGroupIndex < 0) return;
  const prevGroupSegments = allSegments[prevGroupIndex];
  if (!prevGroupSegments) return;
  const lastPrevGroupLocalIndex = prevGroupSegments.findLastIndex(predicate);
  if (lastPrevGroupLocalIndex === -1) return;
  context.set("activeIndex", prevGroupIndex);
  context.set("activeSegmentIndex", getGroupOffset(allSegments, prevGroupIndex) + lastPrevGroupLocalIndex);
}
function getActiveDisplayValue(ctx) {
  const index = ctx.context.get("activeIndex");
  return ctx.context.get("displayValues")[index];
}
function setDisplayValue(ctx, index, dv) {
  const displayValues = [...ctx.context.get("displayValues")];
  displayValues[index] = dv;
  ctx.context.set("displayValues", displayValues);
}
export {
  getActiveDisplayValue,
  getActiveSegment,
  getGroupCount,
  getGroupOffset,
  goToNextSegment,
  goToPreviousSegment,
  isFocusableSegment,
  resolveActiveSegment,
  setDisplayValue
};
