// @zag-js/date-input/dist/utils/validity.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/utils/validity.ts
var validity_exports = {};
__export(validity_exports, {
  getActiveDisplayValue: () => getActiveDisplayValue,
  getActiveSegment: () => getActiveSegment,
  getGroupCount: () => getGroupCount,
  getGroupOffset: () => getGroupOffset,
  goToNextSegment: () => goToNextSegment,
  goToPreviousSegment: () => goToPreviousSegment,
  isFocusableSegment: () => isFocusableSegment,
  resolveActiveSegment: () => resolveActiveSegment,
  setDisplayValue: () => setDisplayValue
});
module.exports = __toCommonJS(validity_exports);
var isFocusableSegment = (segment) => segment.type !== "literal";
function getGroupCount(selectionMode) {
  return selectionMode === "range" ? 2 : 1;
}
function getGroupOffset(allSegments, index) {
  return allSegments.slice(0, index).reduce((acc, segs) => acc + segs.length, 0);
}
function resolveActiveSegment(ctx) {
  const { context, computed } = ctx;
  const index = context.get("activeIndex");
  const activeSegmentIndex = context.get("activeSegmentIndex");
  const allSegments = computed("segments");
  const offset = getGroupOffset(allSegments, index);
  return { allSegments, segments: allSegments[index], offset, localIndex: activeSegmentIndex - offset };
}
function getActiveSegment(ctx) {
  const { allSegments, localIndex } = resolveActiveSegment(ctx);
  const index = ctx.context.get("activeIndex");
  return allSegments[index]?.[localIndex];
}
function goToNextSegment(ctx, predicate = isFocusableSegment) {
  const { context } = ctx;
  const index = context.get("activeIndex");
  const { allSegments, segments, offset, localIndex } = resolveActiveSegment(ctx);
  const nextLocalIndex = segments.findIndex((s, i) => i > localIndex && predicate(s));
  if (nextLocalIndex !== -1) {
    context.set("activeSegmentIndex", offset + nextLocalIndex);
    return;
  }
  const nextGroupSegments = allSegments[index + 1];
  if (!nextGroupSegments) return;
  const firstNextGroupLocalIndex = nextGroupSegments.findIndex(predicate);
  if (firstNextGroupLocalIndex === -1) return;
  context.set("activeIndex", index + 1);
  context.set("activeSegmentIndex", offset + segments.length + firstNextGroupLocalIndex);
}
function goToPreviousSegment(ctx, predicate = isFocusableSegment) {
  const { context } = ctx;
  const index = context.get("activeIndex");
  const { allSegments, segments, offset, localIndex } = resolveActiveSegment(ctx);
  const prevLocalIndex = segments.findLastIndex((s, i) => i < localIndex && predicate(s));
  if (prevLocalIndex !== -1) {
    context.set("activeSegmentIndex", offset + prevLocalIndex);
    return;
  }
  const prevGroupIndex = index - 1;
  if (prevGroupIndex < 0) return;
  const prevGroupSegments = allSegments[prevGroupIndex];
  if (!prevGroupSegments) return;
  const lastPrevGroupLocalIndex = prevGroupSegments.findLastIndex(predicate);
  if (lastPrevGroupLocalIndex === -1) return;
  context.set("activeIndex", prevGroupIndex);
  context.set("activeSegmentIndex", getGroupOffset(allSegments, prevGroupIndex) + lastPrevGroupLocalIndex);
}
function getActiveDisplayValue(ctx) {
  const index = ctx.context.get("activeIndex");
  return ctx.context.get("displayValues")[index];
}
function setDisplayValue(ctx, index, dv) {
  const displayValues = [...ctx.context.get("displayValues")];
  displayValues[index] = dv;
  ctx.context.set("displayValues", displayValues);
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getActiveDisplayValue,
  getActiveSegment,
  getGroupCount,
  getGroupOffset,
  goToNextSegment,
  goToPreviousSegment,
  isFocusableSegment,
  resolveActiveSegment,
  setDisplayValue
});
