// @ark-ui/react/dist/components/floating-panel/floating-panel.js
import { __exportAll } from "../../_virtual/_rolldown/runtime.js";
import { FloatingPanelBody } from "./floating-panel-body.js";
import { FloatingPanelCloseTrigger } from "./floating-panel-close-trigger.js";
import { FloatingPanelContent } from "./floating-panel-content.js";
import { FloatingPanelContext } from "./floating-panel-context.js";
import { FloatingPanelDragTrigger } from "./floating-panel-drag-trigger.js";
import { FloatingPanelHeader } from "./floating-panel-header.js";
import { FloatingPanelPositioner } from "./floating-panel-positioner.js";
import { FloatingPanelResizeTrigger } from "./floating-panel-resize-trigger.js";
import { FloatingPanelStageTrigger } from "./floating-panel-stage-trigger.js";
import { FloatingPanelRoot } from "./floating-panel-root.js";
import { FloatingPanelRootProvider } from "./floating-panel-root-provider.js";
import { FloatingPanelTitle } from "./floating-panel-title.js";
import { FloatingPanelTrigger } from "./floating-panel-trigger.js";
import { FloatingPanelControl } from "./floating-panel-control.js";
import { resizeTriggerAxes } from "@zag-js/floating-panel";
//#region src/components/floating-panel/floating-panel.ts
var floating_panel_exports = /* @__PURE__ */ __exportAll({
	Body: () => FloatingPanelBody,
	CloseTrigger: () => FloatingPanelCloseTrigger,
	Content: () => FloatingPanelContent,
	Context: () => FloatingPanelContext,
	Control: () => FloatingPanelControl,
	DragTrigger: () => FloatingPanelDragTrigger,
	Header: () => FloatingPanelHeader,
	Positioner: () => FloatingPanelPositioner,
	ResizeTrigger: () => FloatingPanelResizeTrigger,
	Root: () => FloatingPanelRoot,
	RootProvider: () => FloatingPanelRootProvider,
	StageTrigger: () => FloatingPanelStageTrigger,
	Title: () => FloatingPanelTitle,
	Trigger: () => FloatingPanelTrigger,
	resizeTriggerAxes: () => resizeTriggerAxes
});
//#endregion
export { floating_panel_exports };
