// @ark-ui/react/dist/components/floating-panel/index.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const require_use_floating_panel_context = require("./use-floating-panel-context.cjs");
const require_floating_panel_body = require("./floating-panel-body.cjs");
const require_floating_panel_close_trigger = require("./floating-panel-close-trigger.cjs");
const require_floating_panel_content = require("./floating-panel-content.cjs");
const require_floating_panel_context = require("./floating-panel-context.cjs");
const require_floating_panel_drag_trigger = require("./floating-panel-drag-trigger.cjs");
const require_floating_panel_header = require("./floating-panel-header.cjs");
const require_floating_panel_positioner = require("./floating-panel-positioner.cjs");
const require_floating_panel_resize_trigger = require("./floating-panel-resize-trigger.cjs");
const require_floating_panel_stage_trigger = require("./floating-panel-stage-trigger.cjs");
const require_use_floating_panel = require("./use-floating-panel.cjs");
const require_floating_panel_root = require("./floating-panel-root.cjs");
const require_floating_panel_root_provider = require("./floating-panel-root-provider.cjs");
const require_floating_panel_title = require("./floating-panel-title.cjs");
const require_floating_panel_trigger = require("./floating-panel-trigger.cjs");
const require_floating_panel_control = require("./floating-panel-control.cjs");
const require_floating_panel = require("./floating-panel.cjs");
let _zag_js_floating_panel = require("@zag-js/floating-panel");
Object.defineProperty(exports, "FloatingPanel", {
	enumerable: true,
	get: function() {
		return require_floating_panel.floating_panel_exports;
	}
});
exports.FloatingPanelBody = require_floating_panel_body.FloatingPanelBody;
exports.FloatingPanelCloseTrigger = require_floating_panel_close_trigger.FloatingPanelCloseTrigger;
exports.FloatingPanelContent = require_floating_panel_content.FloatingPanelContent;
exports.FloatingPanelContext = require_floating_panel_context.FloatingPanelContext;
exports.FloatingPanelControl = require_floating_panel_control.FloatingPanelControl;
exports.FloatingPanelDragTrigger = require_floating_panel_drag_trigger.FloatingPanelDragTrigger;
exports.FloatingPanelHeader = require_floating_panel_header.FloatingPanelHeader;
exports.FloatingPanelPositioner = require_floating_panel_positioner.FloatingPanelPositioner;
exports.FloatingPanelResizeTrigger = require_floating_panel_resize_trigger.FloatingPanelResizeTrigger;
exports.FloatingPanelRoot = require_floating_panel_root.FloatingPanelRoot;
exports.FloatingPanelRootProvider = require_floating_panel_root_provider.FloatingPanelRootProvider;
exports.FloatingPanelStageTrigger = require_floating_panel_stage_trigger.FloatingPanelStageTrigger;
exports.FloatingPanelTitle = require_floating_panel_title.FloatingPanelTitle;
exports.FloatingPanelTrigger = require_floating_panel_trigger.FloatingPanelTrigger;
Object.defineProperty(exports, "floatingPanelAnatomy", {
	enumerable: true,
	get: function() {
		return _zag_js_floating_panel.anatomy;
	}
});
exports.useFloatingPanel = require_use_floating_panel.useFloatingPanel;
exports.useFloatingPanelContext = require_use_floating_panel_context.useFloatingPanelContext;
