// @chakra-ui/react/dist/cjs/styled-system/token-middleware.cjs
"use strict";
'use strict';

var is = require('../utils/is.cjs');
var calc = require('./calc.cjs');
var unitConversion = require('./unit-conversion.cjs');

const addNegativeTokens = {
  enforce: "pre",
  transform(dictionary) {
    const { prefix, allTokens, formatCssVar, formatTokenName, registerToken } = dictionary;
    const tokens = allTokens.filter(
      ({ extensions }) => extensions.category === "spacing"
    );
    tokens.forEach((token) => {
      const originalPath = token.path.slice();
      const originalVar = formatCssVar(originalPath, prefix);
      if (is.isString(token.value) && token.value === "0rem") {
        return;
      }
      const newPath = [...token.path];
      const lastPath = newPath[newPath.length - 1];
      if (lastPath != null) {
        newPath[newPath.length - 1] = `-${lastPath}`;
      }
      const nextToken = {
        ...token,
        value: calc.calc.negate(originalVar.ref),
        name: formatTokenName(newPath),
        path: newPath,
        extensions: {
          ...token.extensions,
          negative: true,
          prop: `-${token.extensions.prop}`,
          originalPath
        }
      };
      registerToken(nextToken);
    });
  }
};
const units = /* @__PURE__ */ new Set([
  "spacing",
  "sizes",
  "borderWidths",
  "fontSizes",
  "radii"
]);
const addPixelUnit = {
  enforce: "post",
  transform(dictionary) {
    const tokens = dictionary.allTokens.filter((token) => {
      return units.has(token.extensions.category) && !token.extensions.negative;
    });
    tokens.forEach((token) => {
      Object.assign(token.extensions, {
        pixelValue: unitConversion.toPx(token.value)
      });
    });
  }
};
const addVirtualPalette = {
  enforce: "post",
  transform(dictionary) {
    const { allTokens, registerToken, formatTokenName } = dictionary;
    const tokens = allTokens.filter(
      ({ extensions }) => extensions.category === "colors"
    );
    const keys = /* @__PURE__ */ new Map();
    const colorPalettes = /* @__PURE__ */ new Map();
    tokens.forEach((token) => {
      const { colorPalette } = token.extensions;
      if (!colorPalette) return;
      colorPalette.keys.forEach((keyPath) => {
        keys.set(formatTokenName(keyPath), keyPath);
      });
      colorPalette.roots.forEach((colorPaletteRoot) => {
        const name = formatTokenName(colorPaletteRoot);
        const colorPaletteList = colorPalettes.get(name) || [];
        colorPaletteList.push(token);
        colorPalettes.set(name, colorPaletteList);
        if (token.extensions.default && colorPaletteRoot.length === 1) {
          const keyPath = colorPalette.keys[0]?.filter(Boolean);
          if (!keyPath.length) return;
          const path = colorPaletteRoot.concat(keyPath);
          keys.set(formatTokenName(path), []);
        }
      });
    });
    keys.forEach((segments) => {
      const path = ["colors", "colorPalette", ...segments].filter(Boolean);
      const name = formatTokenName(path);
      const prop = formatTokenName(path.slice(1));
      const token = {
        name,
        value: name,
        originalValue: name,
        path,
        extensions: {
          condition: "base",
          originalPath: path,
          category: "colors",
          prop,
          virtual: true
        }
      };
      registerToken(token, "pre");
    });
  }
};
const removeEmptyTokens = {
  enforce: "post",
  transform(dictionary) {
    const removed = [];
    const next = [];
    dictionary.allTokens.forEach((token) => {
      if (token.value === "") {
        removed.push(token);
      } else {
        next.push(token);
      }
    });
    dictionary.allTokens.splice(0, dictionary.allTokens.length, ...next);
    removed.forEach((token) => {
      if (dictionary.tokenMap.get(token.name) !== token) return;
      const replacement = dictionary.allTokens.find(
        (t) => t.name === token.name
      );
      if (!replacement) {
        dictionary.tokenMap.delete(token.name);
        return;
      }
      if (token.extensions.conditions && !replacement.extensions.conditions) {
        replacement.extensions.conditions = token.extensions.conditions;
      }
      dictionary.tokenMap.set(token.name, replacement);
    });
  }
};
const tokenMiddlewares = [
  addNegativeTokens,
  addVirtualPalette,
  addPixelUnit,
  removeEmptyTokens
];

exports.addNegativeTokens = addNegativeTokens;
exports.addPixelUnit = addPixelUnit;
exports.addVirtualPalette = addVirtualPalette;
exports.removeEmptyTokens = removeEmptyTokens;
exports.tokenMiddlewares = tokenMiddlewares;
