// @ark-ui/react/dist/components/checkbox/use-checkbox.cjs
"use client";
const require_runtime = require("../../_virtual/_rolldown/runtime.cjs");
const require_use_environment_context = require("../../providers/environment/use-environment-context.cjs");
const require_use_locale_context = require("../../providers/locale/use-locale-context.cjs");
const require_use_checkbox_group_context = require("./use-checkbox-group-context.cjs");
const require_use_field_context = require("../field/use-field-context.cjs");
let _zag_js_checkbox = require("@zag-js/checkbox");
_zag_js_checkbox = require_runtime.__toESM(_zag_js_checkbox, 1);
let react = require("react");
let _zag_js_react = require("@zag-js/react");
//#region src/components/checkbox/use-checkbox.ts
var useCheckbox = (ownProps = {}) => {
	const checkboxGroup = require_use_checkbox_group_context.useCheckboxGroupContext();
	const field = require_use_field_context.useFieldContext();
	const props = (0, react.useMemo)(() => {
		return (0, _zag_js_react.mergeProps)(checkboxGroup?.getItemProps({ value: ownProps.value }) ?? {}, ownProps);
	}, [ownProps, checkboxGroup]);
	const id = (0, react.useId)();
	const { getRootNode } = require_use_environment_context.useEnvironmentContext();
	const { dir } = require_use_locale_context.useLocaleContext();
	const machineProps = {
		id,
		ids: {
			label: field?.ids.label,
			hiddenInput: field?.ids.control
		},
		dir,
		disabled: field?.disabled,
		readOnly: field?.readOnly,
		invalid: field?.invalid,
		required: field?.required,
		getRootNode,
		...props
	};
	const service = (0, _zag_js_react.useMachine)(_zag_js_checkbox.machine, machineProps);
	return _zag_js_checkbox.connect(service, _zag_js_react.normalizeProps);
};
//#endregion
exports.useCheckbox = useCheckbox;
