// @zag-js/floating-panel/dist/floating-panel.connect.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/floating-panel.connect.ts
var floating_panel_connect_exports = {};
__export(floating_panel_connect_exports, {
  connect: () => connect
});
module.exports = __toCommonJS(floating_panel_connect_exports);
var import_dom_query = require("@zag-js/dom-query");
var import_utils = require("@zag-js/utils");
var import_floating_panel = require("./floating-panel.anatomy.js");
var dom = __toESM(require("./floating-panel.dom.js"));
var import_get_resize_axis_style = require("./get-resize-axis-style.js");
var validStages = /* @__PURE__ */ new Set(["minimized", "maximized", "default"]);
function connect(service, normalize) {
  const { state, send, scope, prop, computed, context } = service;
  const open = state.hasTag("open");
  const dragging = state.matches("open.dragging");
  const resizing = state.matches("open.resizing");
  const isTopmost = context.get("isTopmost");
  const size = context.get("size");
  const position = context.get("position");
  const isMaximized = computed("isMaximized");
  const isMinimized = computed("isMinimized");
  const isStaged = computed("isStaged");
  const canResize = computed("canResize");
  const canDrag = computed("canDrag");
  return {
    open,
    resizable: prop("resizable"),
    draggable: prop("draggable"),
    setOpen(nextOpen) {
      const open2 = state.hasTag("open");
      if (open2 === nextOpen) return;
      send({ type: nextOpen ? "OPEN" : "CLOSE" });
    },
    dragging,
    resizing,
    position,
    size,
    setPosition(position2) {
      send({ type: "SET_POSITION", position: position2 });
    },
    setSize(size2) {
      send({ type: "SET_SIZE", size: size2 });
    },
    minimize() {
      send({ type: "MINIMIZE" });
    },
    maximize() {
      send({ type: "MAXIMIZE" });
    },
    restore() {
      send({ type: "RESTORE" });
    },
    getTriggerProps() {
      return normalize.button({
        ...import_floating_panel.parts.trigger.attrs,
        dir: prop("dir"),
        type: "button",
        disabled: prop("disabled"),
        id: dom.getTriggerId(scope),
        "data-state": open ? "open" : "closed",
        "data-dragging": (0, import_dom_query.dataAttr)(dragging),
        "aria-controls": dom.getContentId(scope),
        onClick(event) {
          if (event.defaultPrevented) return;
          if (prop("disabled")) return;
          const open2 = state.hasTag("open");
          send({ type: open2 ? "CLOSE" : "OPEN", src: "trigger" });
        }
      });
    },
    getPositionerProps() {
      return normalize.element({
        ...import_floating_panel.parts.positioner.attrs,
        dir: prop("dir"),
        id: dom.getPositionerId(scope),
        style: {
          "--width": (0, import_utils.toPx)(size?.width),
          "--height": (0, import_utils.toPx)(size?.height),
          "--x": (0, import_utils.toPx)(position?.x),
          "--y": (0, import_utils.toPx)(position?.y),
          position: prop("strategy"),
          top: "var(--y)",
          left: "var(--x)"
        }
      });
    },
    getContentProps() {
      return normalize.element({
        ...import_floating_panel.parts.content.attrs,
        dir: prop("dir"),
        role: "dialog",
        tabIndex: 0,
        hidden: !open,
        id: dom.getContentId(scope),
        "aria-labelledby": dom.getTitleId(scope),
        "data-state": open ? "open" : "closed",
        "data-dragging": (0, import_dom_query.dataAttr)(dragging),
        "data-topmost": (0, import_dom_query.dataAttr)(isTopmost),
        "data-behind": (0, import_dom_query.dataAttr)(!isTopmost),
        "data-minimized": (0, import_dom_query.dataAttr)(isMinimized),
        "data-maximized": (0, import_dom_query.dataAttr)(isMaximized),
        "data-staged": (0, import_dom_query.dataAttr)(isStaged),
        style: {
          width: "var(--width)",
          height: "var(--height)",
          overflow: isMinimized ? "hidden" : void 0
        },
        onFocus() {
          send({ type: "CONTENT_FOCUS" });
        },
        onKeyDown(event) {
          if (event.defaultPrevented) return;
          if (event.key === "Escape" && isTopmost) {
            send({ type: "ESCAPE" });
            return;
          }
          if (event.currentTarget !== (0, import_dom_query.getEventTarget)(event)) return;
          const step = (0, import_dom_query.getEventStep)(event) * prop("gridSize");
          const keyMap = {
            ArrowLeft() {
              send({ type: "MOVE", direction: "left", step });
            },
            ArrowRight() {
              send({ type: "MOVE", direction: "right", step });
            },
            ArrowUp() {
              send({ type: "MOVE", direction: "up", step });
            },
            ArrowDown() {
              send({ type: "MOVE", direction: "down", step });
            }
          };
          const handler = keyMap[(0, import_dom_query.getEventKey)(event, { dir: prop("dir") })];
          if (handler) {
            event.preventDefault();
            handler(event);
          }
        }
      });
    },
    getCloseTriggerProps() {
      return normalize.button({
        ...import_floating_panel.parts.closeTrigger.attrs,
        dir: prop("dir"),
        disabled: prop("disabled"),
        "aria-label": "Close Window",
        type: "button",
        onClick(event) {
          if (event.defaultPrevented) return;
          send({ type: "CLOSE" });
        }
      });
    },
    getStageTriggerProps(props) {
      if (!validStages.has(props.stage)) {
        throw new Error(`[zag-js] Invalid stage: ${props.stage}. Must be one of: ${Array.from(validStages).join(", ")}`);
      }
      const translations = prop("translations");
      const actionProps = (0, import_utils.match)(props.stage, {
        minimized: () => ({
          "aria-label": translations.minimize,
          hidden: isStaged
        }),
        maximized: () => ({
          "aria-label": translations.maximize,
          hidden: isStaged
        }),
        default: () => ({
          "aria-label": translations.restore,
          hidden: !isStaged
        })
      });
      return normalize.button({
        ...import_floating_panel.parts.stageTrigger.attrs,
        dir: prop("dir"),
        disabled: prop("disabled"),
        "data-stage": props.stage,
        ...actionProps,
        type: "button",
        onClick(event) {
          if (event.defaultPrevented) return;
          if (!prop("resizable")) return;
          const type = (0, import_utils.match)(props.stage, {
            minimized: () => "MINIMIZE",
            maximized: () => "MAXIMIZE",
            default: () => "RESTORE"
          });
          send({ type: type.toUpperCase() });
        }
      });
    },
    getResizeTriggerProps(props) {
      return normalize.element({
        ...import_floating_panel.parts.resizeTrigger.attrs,
        dir: prop("dir"),
        "data-disabled": (0, import_dom_query.dataAttr)(!canResize),
        "data-axis": props.axis,
        onPointerDown(event) {
          if (!canResize) return;
          if (!(0, import_dom_query.isLeftClick)(event)) return;
          event.currentTarget.setPointerCapture(event.pointerId);
          event.stopPropagation();
          send({
            type: "RESIZE_START",
            axis: props.axis,
            position: { x: event.clientX, y: event.clientY }
          });
        },
        onPointerUp(event) {
          if (!canResize) return;
          const node = event.currentTarget;
          if (node.hasPointerCapture(event.pointerId)) {
            node.releasePointerCapture(event.pointerId);
          }
        },
        style: {
          position: "absolute",
          touchAction: "none",
          ...(0, import_get_resize_axis_style.getResizeAxisStyle)(props.axis)
        }
      });
    },
    getDragTriggerProps() {
      return normalize.element({
        ...import_floating_panel.parts.dragTrigger.attrs,
        dir: prop("dir"),
        "data-disabled": (0, import_dom_query.dataAttr)(!canDrag),
        onPointerDown(event) {
          if (!canDrag) return;
          if (!(0, import_dom_query.isLeftClick)(event)) return;
          const target = (0, import_dom_query.getEventTarget)(event);
          if (target?.closest("button") || target?.closest("[data-no-drag]")) {
            return;
          }
          event.currentTarget.setPointerCapture(event.pointerId);
          event.stopPropagation();
          send({
            type: "DRAG_START",
            pointerId: event.pointerId,
            position: { x: event.clientX, y: event.clientY }
          });
        },
        onPointerUp(event) {
          if (!canDrag) return;
          const node = event.currentTarget;
          if (node.hasPointerCapture(event.pointerId)) {
            node.releasePointerCapture(event.pointerId);
          }
        },
        onDoubleClick(event) {
          if (event.defaultPrevented) return;
          if (!prop("resizable")) return;
          send({ type: isStaged ? "RESTORE" : "MAXIMIZE" });
        },
        style: {
          WebkitUserSelect: "none",
          userSelect: "none",
          touchAction: "none",
          cursor: "move"
        }
      });
    },
    getControlProps() {
      return normalize.element({
        ...import_floating_panel.parts.control.attrs,
        dir: prop("dir"),
        "data-disabled": (0, import_dom_query.dataAttr)(prop("disabled")),
        "data-stage": context.get("stage"),
        "data-minimized": (0, import_dom_query.dataAttr)(isMinimized),
        "data-maximized": (0, import_dom_query.dataAttr)(isMaximized),
        "data-staged": (0, import_dom_query.dataAttr)(isStaged)
      });
    },
    getTitleProps() {
      return normalize.element({
        ...import_floating_panel.parts.title.attrs,
        dir: prop("dir"),
        id: dom.getTitleId(scope)
      });
    },
    getHeaderProps() {
      return normalize.element({
        ...import_floating_panel.parts.header.attrs,
        dir: prop("dir"),
        id: dom.getHeaderId(scope),
        "data-dragging": (0, import_dom_query.dataAttr)(dragging),
        "data-topmost": (0, import_dom_query.dataAttr)(isTopmost),
        "data-behind": (0, import_dom_query.dataAttr)(!isTopmost),
        "data-minimized": (0, import_dom_query.dataAttr)(isMinimized),
        "data-maximized": (0, import_dom_query.dataAttr)(isMaximized),
        "data-staged": (0, import_dom_query.dataAttr)(isStaged)
      });
    },
    getBodyProps() {
      return normalize.element({
        ...import_floating_panel.parts.body.attrs,
        dir: prop("dir"),
        "data-dragging": (0, import_dom_query.dataAttr)(dragging),
        "data-minimized": (0, import_dom_query.dataAttr)(isMinimized),
        "data-maximized": (0, import_dom_query.dataAttr)(isMaximized),
        "data-staged": (0, import_dom_query.dataAttr)(isStaged),
        hidden: isMinimized
      });
    }
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  connect
});
