// @chakra-ui/react/dist/cjs/components/tags-input/index.cjs
"use strict";
'use strict';

var tagsInput = require('./tags-input.cjs');
var tagsInput$1 = require('@ark-ui/react/tags-input');
var namespace = require('./namespace.cjs');



exports.TagsInputClearTrigger = tagsInput.TagsInputClearTrigger;
exports.TagsInputContext = tagsInput.TagsInputContext;
exports.TagsInputControl = tagsInput.TagsInputControl;
exports.TagsInputHiddenInput = tagsInput.TagsInputHiddenInput;
exports.TagsInputInput = tagsInput.TagsInputInput;
exports.TagsInputItem = tagsInput.TagsInputItem;
exports.TagsInputItemContext = tagsInput.TagsInputItemContext;
exports.TagsInputItemDeleteTrigger = tagsInput.TagsInputItemDeleteTrigger;
exports.TagsInputItemText = tagsInput.TagsInputItemText;
exports.TagsInputItems = tagsInput.TagsInputItems;
exports.TagsInputLabel = tagsInput.TagsInputLabel;
exports.TagsInputPropsProvider = tagsInput.TagsInputPropsProvider;
exports.TagsInputRoot = tagsInput.TagsInputRoot;
exports.TagsInputRootProvider = tagsInput.TagsInputRootProvider;
exports.useTagsInputStyles = tagsInput.useTagsInputStyles;
Object.defineProperty(exports, "useTagsInput", {
  enumerable: true,
  get: function () { return tagsInput$1.useTagsInput; }
});
Object.defineProperty(exports, "useTagsInputContext", {
  enumerable: true,
  get: function () { return tagsInput$1.useTagsInputContext; }
});
Object.defineProperty(exports, "useTagsInputItemContext", {
  enumerable: true,
  get: function () { return tagsInput$1.useTagsInputItemContext; }
});
exports.TagsInput = namespace;
