// @zag-js/tree-view/dist/tree-view.connect.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/tree-view.connect.ts
var tree_view_connect_exports = {};
__export(tree_view_connect_exports, {
  connect: () => connect
});
module.exports = __toCommonJS(tree_view_connect_exports);
var import_dom_query = require("@zag-js/dom-query");
var import_utils = require("@zag-js/utils");
var import_tree_view = require("./tree-view.anatomy.js");
var dom = __toESM(require("./tree-view.dom.js"));
var import_checked_state = require("./utils/checked-state.js");
function connect(service, normalize) {
  const { context, scope, computed, prop, send } = service;
  const collection = prop("collection");
  const translations = prop("translations");
  const expandedValue = Array.from(context.get("expandedValue"));
  const selectedValue = Array.from(context.get("selectedValue"));
  const checkedValue = Array.from(context.get("checkedValue"));
  const isTypingAhead = computed("isTypingAhead");
  const focusedValue = context.get("focusedValue");
  const loadingStatus = context.get("loadingStatus");
  const renamingValue = context.get("renamingValue");
  const skip = ({ indexPath }) => {
    const paths = collection.getValuePath(indexPath).slice(0, -1);
    return paths.some((value) => !expandedValue.includes(value));
  };
  const firstNode = collection.getFirstNode(void 0, { skip });
  const firstNodeValue = firstNode ? collection.getNodeValue(firstNode) : null;
  function getNodeState(props) {
    const { node, indexPath } = props;
    const value = collection.getNodeValue(node);
    return {
      id: dom.getNodeId(scope, value),
      value,
      indexPath,
      valuePath: collection.getValuePath(indexPath),
      disabled: Boolean(node.disabled),
      focused: focusedValue == null ? firstNodeValue === value : focusedValue === value,
      selected: selectedValue.includes(value),
      expanded: expandedValue.includes(value),
      loading: loadingStatus[value] === "loading",
      depth: indexPath.length,
      isBranch: collection.isBranchNode(node),
      renaming: renamingValue === value,
      get checked() {
        return (0, import_checked_state.getCheckedState)(collection, node, checkedValue);
      }
    };
  }
  return {
    collection,
    expandedValue,
    selectedValue,
    checkedValue,
    toggleChecked(value, isBranch) {
      send({ type: "CHECKED.TOGGLE", value, isBranch });
    },
    setChecked(value) {
      send({ type: "CHECKED.SET", value });
    },
    clearChecked() {
      send({ type: "CHECKED.CLEAR" });
    },
    getCheckedMap() {
      return (0, import_checked_state.getCheckedValueMap)(collection, checkedValue);
    },
    expand(value) {
      send({ type: value ? "BRANCH.EXPAND" : "EXPANDED.ALL", value });
    },
    collapse(value) {
      send({ type: value ? "BRANCH.COLLAPSE" : "EXPANDED.CLEAR", value });
    },
    deselect(value) {
      send({ type: value ? "NODE.DESELECT" : "SELECTED.CLEAR", value });
    },
    select(value) {
      send({ type: value ? "NODE.SELECT" : "SELECTED.ALL", value, isTrusted: false });
    },
    getVisibleNodes() {
      return computed("visibleNodes");
    },
    focus(value) {
      dom.focusNode(scope, value);
    },
    selectParent(value) {
      const parentNode = collection.getParentNode(value);
      if (!parentNode) return;
      const _selectedValue = (0, import_utils.add)(selectedValue, collection.getNodeValue(parentNode));
      send({ type: "SELECTED.SET", value: _selectedValue, src: "select.parent" });
    },
    expandParent(value) {
      const parentNode = collection.getParentNode(value);
      if (!parentNode) return;
      const _expandedValue = (0, import_utils.add)(expandedValue, collection.getNodeValue(parentNode));
      send({ type: "EXPANDED.SET", value: _expandedValue, src: "expand.parent" });
    },
    setExpandedValue(value) {
      const _expandedValue = (0, import_utils.uniq)(value);
      send({ type: "EXPANDED.SET", value: _expandedValue });
    },
    setSelectedValue(value) {
      const _selectedValue = (0, import_utils.uniq)(value);
      send({ type: "SELECTED.SET", value: _selectedValue });
    },
    startRenaming(value) {
      send({ type: "NODE.RENAME", value });
    },
    submitRenaming(value, label) {
      send({ type: "RENAME.SUBMIT", value, label });
    },
    cancelRenaming() {
      send({ type: "RENAME.CANCEL" });
    },
    getRootProps() {
      return normalize.element({
        ...import_tree_view.parts.root.attrs,
        id: dom.getRootId(scope),
        dir: prop("dir")
      });
    },
    getLabelProps() {
      return normalize.element({
        ...import_tree_view.parts.label.attrs,
        id: dom.getLabelId(scope),
        dir: prop("dir")
      });
    },
    getTreeProps() {
      return normalize.element({
        ...import_tree_view.parts.tree.attrs,
        id: dom.getTreeId(scope),
        dir: prop("dir"),
        role: "tree",
        "aria-label": translations.treeLabel,
        "aria-labelledby": dom.getLabelId(scope),
        "aria-multiselectable": prop("selectionMode") === "multiple" || void 0,
        tabIndex: -1,
        onKeyDown(event) {
          if (event.defaultPrevented) return;
          if ((0, import_dom_query.isComposingEvent)(event)) return;
          const target = (0, import_dom_query.getEventTarget)(event);
          if ((0, import_dom_query.isEditableElement)(target)) return;
          const node = target?.closest("[data-part=branch-control], [data-part=item]");
          if (!node) return;
          const nodeId = node.dataset.value;
          if (nodeId == null) {
            console.warn(`[zag-js/tree-view] Node id not found for node`, node);
            return;
          }
          const isBranchNode = node.matches("[data-part=branch-control]");
          const keyMap = {
            ArrowDown(event2) {
              if ((0, import_dom_query.isModifierKey)(event2)) return;
              event2.preventDefault();
              send({ type: "NODE.ARROW_DOWN", id: nodeId, shiftKey: event2.shiftKey });
            },
            ArrowUp(event2) {
              if ((0, import_dom_query.isModifierKey)(event2)) return;
              event2.preventDefault();
              send({ type: "NODE.ARROW_UP", id: nodeId, shiftKey: event2.shiftKey });
            },
            ArrowLeft(event2) {
              if ((0, import_dom_query.isModifierKey)(event2) || node.dataset.disabled) return;
              event2.preventDefault();
              send({ type: isBranchNode ? "BRANCH_NODE.ARROW_LEFT" : "NODE.ARROW_LEFT", id: nodeId });
            },
            ArrowRight(event2) {
              if (!isBranchNode || node.dataset.disabled) return;
              event2.preventDefault();
              send({ type: "BRANCH_NODE.ARROW_RIGHT", id: nodeId });
            },
            Home(event2) {
              if ((0, import_dom_query.isModifierKey)(event2)) return;
              event2.preventDefault();
              send({ type: "NODE.HOME", id: nodeId, shiftKey: event2.shiftKey });
            },
            End(event2) {
              if ((0, import_dom_query.isModifierKey)(event2)) return;
              event2.preventDefault();
              send({ type: "NODE.END", id: nodeId, shiftKey: event2.shiftKey });
            },
            Space(event2) {
              if (node.dataset.disabled) return;
              if (isTypingAhead) {
                send({ type: "TREE.TYPEAHEAD", key: event2.key });
              } else {
                keyMap.Enter?.(event2);
              }
            },
            Enter(event2) {
              if (node.dataset.disabled) return;
              if ((0, import_dom_query.isAnchorElement)(target) && (0, import_dom_query.isModifierKey)(event2)) return;
              send({ type: isBranchNode ? "BRANCH_NODE.CLICK" : "NODE.CLICK", id: nodeId, src: "keyboard" });
              if (!(0, import_dom_query.isAnchorElement)(target)) {
                event2.preventDefault();
              }
            },
            "*"(event2) {
              if (node.dataset.disabled) return;
              event2.preventDefault();
              send({ type: "SIBLINGS.EXPAND", id: nodeId });
            },
            a(event2) {
              if (!event2.metaKey || node.dataset.disabled) return;
              event2.preventDefault();
              send({ type: "SELECTED.ALL", moveFocus: true });
            },
            F2(event2) {
              if (node.dataset.disabled) return;
              const canRenameFn = prop("canRename");
              if (!canRenameFn) return;
              const indexPath = collection.getIndexPath(nodeId);
              if (indexPath) {
                const node2 = collection.at(indexPath);
                if (node2 && !canRenameFn(node2, indexPath)) {
                  return;
                }
              }
              event2.preventDefault();
              send({ type: "NODE.RENAME", value: nodeId });
            }
          };
          const key = (0, import_dom_query.getEventKey)(event, { dir: prop("dir") });
          const exec = keyMap[key];
          if (exec) {
            exec(event);
            return;
          }
          if (!import_dom_query.getByTypeahead.isValidEvent(event)) return;
          send({ type: "TREE.TYPEAHEAD", key: event.key, id: nodeId });
          event.preventDefault();
        }
      });
    },
    getNodeState,
    getItemProps(props) {
      const nodeState = getNodeState(props);
      return normalize.element({
        ...import_tree_view.parts.item.attrs,
        id: nodeState.id,
        dir: prop("dir"),
        "data-ownedby": dom.getTreeId(scope),
        "data-path": props.indexPath.join("/"),
        "data-value": nodeState.value,
        tabIndex: nodeState.focused ? 0 : -1,
        "data-focus": (0, import_dom_query.dataAttr)(nodeState.focused),
        role: "treeitem",
        "aria-current": nodeState.selected ? "true" : void 0,
        "aria-selected": nodeState.disabled ? void 0 : nodeState.selected,
        "data-selected": (0, import_dom_query.dataAttr)(nodeState.selected),
        "aria-disabled": (0, import_dom_query.ariaAttr)(nodeState.disabled),
        "data-disabled": (0, import_dom_query.dataAttr)(nodeState.disabled),
        "data-renaming": (0, import_dom_query.dataAttr)(nodeState.renaming),
        "data-checked": (0, import_dom_query.dataAttr)(nodeState.checked === true),
        "data-indeterminate": (0, import_dom_query.dataAttr)(nodeState.checked === "indeterminate"),
        "aria-level": nodeState.depth,
        "data-depth": nodeState.depth,
        style: {
          "--depth": nodeState.depth
        },
        onFocus(event) {
          event.stopPropagation();
          send({ type: "NODE.FOCUS", id: nodeState.value });
        },
        onClick(event) {
          if (nodeState.disabled) return;
          if (!(0, import_dom_query.isLeftClick)(event)) return;
          if ((0, import_dom_query.isAnchorElement)(event.currentTarget) && (0, import_dom_query.isModifierKey)(event)) return;
          const isMetaKey = event.metaKey || event.ctrlKey;
          send({ type: "NODE.CLICK", id: nodeState.value, shiftKey: event.shiftKey, ctrlKey: isMetaKey });
          event.stopPropagation();
          if (!(0, import_dom_query.isAnchorElement)(event.currentTarget)) {
            event.preventDefault();
          }
        }
      });
    },
    getItemTextProps(props) {
      const itemState = getNodeState(props);
      return normalize.element({
        ...import_tree_view.parts.itemText.attrs,
        "data-disabled": (0, import_dom_query.dataAttr)(itemState.disabled),
        "data-selected": (0, import_dom_query.dataAttr)(itemState.selected),
        "data-focus": (0, import_dom_query.dataAttr)(itemState.focused)
      });
    },
    getItemIndicatorProps(props) {
      const itemState = getNodeState(props);
      return normalize.element({
        ...import_tree_view.parts.itemIndicator.attrs,
        "aria-hidden": true,
        "data-disabled": (0, import_dom_query.dataAttr)(itemState.disabled),
        "data-selected": (0, import_dom_query.dataAttr)(itemState.selected),
        "data-focus": (0, import_dom_query.dataAttr)(itemState.focused),
        hidden: !itemState.selected
      });
    },
    getBranchProps(props) {
      const nodeState = getNodeState(props);
      return normalize.element({
        ...import_tree_view.parts.branch.attrs,
        "data-depth": nodeState.depth,
        dir: prop("dir"),
        "data-branch": nodeState.value,
        role: "treeitem",
        "data-ownedby": dom.getTreeId(scope),
        "data-value": nodeState.value,
        "aria-level": nodeState.depth,
        "aria-selected": nodeState.disabled ? void 0 : nodeState.selected,
        "data-path": props.indexPath.join("/"),
        "data-selected": (0, import_dom_query.dataAttr)(nodeState.selected),
        "aria-expanded": nodeState.expanded,
        "data-state": nodeState.expanded ? "open" : "closed",
        "aria-disabled": (0, import_dom_query.ariaAttr)(nodeState.disabled),
        "data-disabled": (0, import_dom_query.dataAttr)(nodeState.disabled),
        "data-loading": (0, import_dom_query.dataAttr)(nodeState.loading),
        "aria-busy": (0, import_dom_query.ariaAttr)(nodeState.loading),
        style: {
          "--depth": nodeState.depth
        }
      });
    },
    getBranchIndicatorProps(props) {
      const nodeState = getNodeState(props);
      return normalize.element({
        ...import_tree_view.parts.branchIndicator.attrs,
        "aria-hidden": true,
        "data-state": nodeState.expanded ? "open" : "closed",
        "data-disabled": (0, import_dom_query.dataAttr)(nodeState.disabled),
        "data-selected": (0, import_dom_query.dataAttr)(nodeState.selected),
        "data-focus": (0, import_dom_query.dataAttr)(nodeState.focused),
        "data-loading": (0, import_dom_query.dataAttr)(nodeState.loading)
      });
    },
    getBranchTriggerProps(props) {
      const nodeState = getNodeState(props);
      return normalize.element({
        ...import_tree_view.parts.branchTrigger.attrs,
        role: "button",
        dir: prop("dir"),
        "data-disabled": (0, import_dom_query.dataAttr)(nodeState.disabled),
        "data-state": nodeState.expanded ? "open" : "closed",
        "data-value": nodeState.value,
        "data-loading": (0, import_dom_query.dataAttr)(nodeState.loading),
        disabled: nodeState.loading,
        onClick(event) {
          if (nodeState.disabled || nodeState.loading) return;
          send({ type: "BRANCH_TOGGLE.CLICK", id: nodeState.value });
          event.stopPropagation();
        }
      });
    },
    getBranchControlProps(props) {
      const nodeState = getNodeState(props);
      return normalize.element({
        ...import_tree_view.parts.branchControl.attrs,
        role: "button",
        id: nodeState.id,
        dir: prop("dir"),
        tabIndex: nodeState.focused ? 0 : -1,
        "data-path": props.indexPath.join("/"),
        "data-state": nodeState.expanded ? "open" : "closed",
        "data-disabled": (0, import_dom_query.dataAttr)(nodeState.disabled),
        "data-selected": (0, import_dom_query.dataAttr)(nodeState.selected),
        "data-focus": (0, import_dom_query.dataAttr)(nodeState.focused),
        "data-renaming": (0, import_dom_query.dataAttr)(nodeState.renaming),
        "data-checked": (0, import_dom_query.dataAttr)(nodeState.checked === true),
        "data-indeterminate": (0, import_dom_query.dataAttr)(nodeState.checked === "indeterminate"),
        "data-value": nodeState.value,
        "data-depth": nodeState.depth,
        "data-loading": (0, import_dom_query.dataAttr)(nodeState.loading),
        "aria-busy": (0, import_dom_query.ariaAttr)(nodeState.loading),
        onFocus(event) {
          send({ type: "NODE.FOCUS", id: nodeState.value });
          event.stopPropagation();
        },
        onClick(event) {
          if (nodeState.disabled) return;
          if (nodeState.loading) return;
          if (!(0, import_dom_query.isLeftClick)(event)) return;
          if ((0, import_dom_query.isAnchorElement)(event.currentTarget) && (0, import_dom_query.isModifierKey)(event)) return;
          const isMetaKey = event.metaKey || event.ctrlKey;
          send({ type: "BRANCH_NODE.CLICK", id: nodeState.value, shiftKey: event.shiftKey, ctrlKey: isMetaKey });
          event.stopPropagation();
        }
      });
    },
    getBranchTextProps(props) {
      const nodeState = getNodeState(props);
      return normalize.element({
        ...import_tree_view.parts.branchText.attrs,
        dir: prop("dir"),
        "data-disabled": (0, import_dom_query.dataAttr)(nodeState.disabled),
        "data-state": nodeState.expanded ? "open" : "closed",
        "data-loading": (0, import_dom_query.dataAttr)(nodeState.loading)
      });
    },
    getBranchContentProps(props) {
      const nodeState = getNodeState(props);
      return normalize.element({
        ...import_tree_view.parts.branchContent.attrs,
        role: "group",
        dir: prop("dir"),
        "data-state": nodeState.expanded ? "open" : "closed",
        "data-depth": nodeState.depth,
        "data-path": props.indexPath.join("/"),
        "data-value": nodeState.value,
        hidden: !nodeState.expanded
      });
    },
    getBranchIndentGuideProps(props) {
      const nodeState = getNodeState(props);
      return normalize.element({
        ...import_tree_view.parts.branchIndentGuide.attrs,
        "data-depth": nodeState.depth
      });
    },
    getNodeCheckboxProps(props) {
      const nodeState = getNodeState(props);
      const checkedState = nodeState.checked;
      return normalize.element({
        ...import_tree_view.parts.nodeCheckbox.attrs,
        tabIndex: -1,
        role: "checkbox",
        "data-state": checkedState === true ? "checked" : checkedState === false ? "unchecked" : "indeterminate",
        "aria-checked": checkedState === true ? "true" : checkedState === false ? "false" : "mixed",
        "data-disabled": (0, import_dom_query.dataAttr)(nodeState.disabled),
        onClick(event) {
          if (event.defaultPrevented) return;
          if (nodeState.disabled) return;
          if (!(0, import_dom_query.isLeftClick)(event)) return;
          send({ type: "CHECKED.TOGGLE", value: nodeState.value, isBranch: nodeState.isBranch });
          event.stopPropagation();
          const node = event.currentTarget.closest("[role=treeitem]");
          node?.focus({ preventScroll: true });
        }
      });
    },
    getNodeRenameInputProps(props) {
      const nodeState = getNodeState(props);
      return normalize.input({
        ...import_tree_view.parts.nodeRenameInput.attrs,
        id: dom.getRenameInputId(scope, nodeState.value),
        type: "text",
        "aria-label": translations.renameInputLabel,
        hidden: !nodeState.renaming,
        onKeyDown(event) {
          if ((0, import_dom_query.isComposingEvent)(event)) return;
          if (event.key === "Escape") {
            send({ type: "RENAME.CANCEL" });
            event.preventDefault();
          }
          if (event.key === "Enter") {
            send({ type: "RENAME.SUBMIT", label: event.currentTarget.value });
            event.preventDefault();
          }
          event.stopPropagation();
        },
        onBlur(event) {
          send({ type: "RENAME.SUBMIT", label: event.currentTarget.value });
        }
      });
    }
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  connect
});
