// @zag-js/tabs/dist/tabs.dom.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/tabs.dom.ts
var tabs_dom_exports = {};
__export(tabs_dom_exports, {
  getContentEl: () => getContentEl,
  getContentId: () => getContentId,
  getElements: () => getElements,
  getFirstTriggerEl: () => getFirstTriggerEl,
  getIndicatorEl: () => getIndicatorEl,
  getIndicatorId: () => getIndicatorId,
  getLastTriggerEl: () => getLastTriggerEl,
  getListEl: () => getListEl,
  getListId: () => getListId,
  getNextTriggerEl: () => getNextTriggerEl,
  getOffsetRect: () => getOffsetRect,
  getPrevTriggerEl: () => getPrevTriggerEl,
  getRectByValue: () => getRectByValue,
  getRootId: () => getRootId,
  getTriggerEl: () => getTriggerEl,
  getTriggerId: () => getTriggerId
});
module.exports = __toCommonJS(tabs_dom_exports);
var import_dom_query = require("@zag-js/dom-query");
var import_utils = require("@zag-js/utils");
var getRootId = (ctx) => ctx.ids?.root ?? `tabs:${ctx.id}`;
var getListId = (ctx) => ctx.ids?.list ?? `tabs:${ctx.id}:list`;
var getContentId = (ctx, value) => ctx.ids?.content?.(value) ?? `tabs:${ctx.id}:content-${value}`;
var getTriggerId = (ctx, value) => ctx.ids?.trigger?.(value) ?? `tabs:${ctx.id}:trigger-${value}`;
var getIndicatorId = (ctx) => ctx.ids?.indicator ?? `tabs:${ctx.id}:indicator`;
var getListEl = (ctx) => ctx.getById(getListId(ctx));
var getContentEl = (ctx, value) => ctx.getById(getContentId(ctx, value));
var getTriggerEl = (ctx, value) => value != null ? ctx.getById(getTriggerId(ctx, value)) : null;
var getIndicatorEl = (ctx) => ctx.getById(getIndicatorId(ctx));
var getElements = (ctx) => {
  const ownerId = CSS.escape(getListId(ctx));
  const selector = `[role=tab][data-ownedby='${ownerId}']:not([disabled])`;
  return (0, import_dom_query.queryAll)(getListEl(ctx), selector);
};
var getFirstTriggerEl = (ctx) => (0, import_utils.first)(getElements(ctx));
var getLastTriggerEl = (ctx) => (0, import_utils.last)(getElements(ctx));
var getNextTriggerEl = (ctx, opts) => (0, import_dom_query.nextById)(getElements(ctx), getTriggerId(ctx, opts.value), opts.loopFocus);
var getPrevTriggerEl = (ctx, opts) => (0, import_dom_query.prevById)(getElements(ctx), getTriggerId(ctx, opts.value), opts.loopFocus);
var getOffsetRect = (el) => ({
  x: el?.offsetLeft ?? 0,
  y: el?.offsetTop ?? 0,
  width: el?.offsetWidth ?? 0,
  height: el?.offsetHeight ?? 0
});
var getRectByValue = (ctx, value) => {
  const tab = (0, import_dom_query.itemById)(getElements(ctx), getTriggerId(ctx, value));
  return getOffsetRect(tab);
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getContentEl,
  getContentId,
  getElements,
  getFirstTriggerEl,
  getIndicatorEl,
  getIndicatorId,
  getLastTriggerEl,
  getListEl,
  getListId,
  getNextTriggerEl,
  getOffsetRect,
  getPrevTriggerEl,
  getRectByValue,
  getRootId,
  getTriggerEl,
  getTriggerId
});
