// @ark-ui/react/dist/components/switch/index.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const require_use_switch_context = require("./use-switch-context.cjs");
const require_switch_context = require("./switch-context.cjs");
const require_switch_control = require("./switch-control.cjs");
const require_switch_hidden_input = require("./switch-hidden-input.cjs");
const require_switch_label = require("./switch-label.cjs");
const require_use_switch = require("./use-switch.cjs");
const require_switch_root = require("./switch-root.cjs");
const require_switch_root_provider = require("./switch-root-provider.cjs");
const require_switch_thumb = require("./switch-thumb.cjs");
const require_switch = require("./switch.cjs");
let _zag_js_switch = require("@zag-js/switch");
Object.defineProperty(exports, "Switch", {
	enumerable: true,
	get: function() {
		return require_switch.switch_exports;
	}
});
exports.SwitchContext = require_switch_context.SwitchContext;
exports.SwitchControl = require_switch_control.SwitchControl;
exports.SwitchHiddenInput = require_switch_hidden_input.SwitchHiddenInput;
exports.SwitchLabel = require_switch_label.SwitchLabel;
exports.SwitchRoot = require_switch_root.SwitchRoot;
exports.SwitchRootProvider = require_switch_root_provider.SwitchRootProvider;
exports.SwitchThumb = require_switch_thumb.SwitchThumb;
Object.defineProperty(exports, "switchAnatomy", {
	enumerable: true,
	get: function() {
		return _zag_js_switch.anatomy;
	}
});
exports.useSwitch = require_use_switch.useSwitch;
exports.useSwitchContext = require_use_switch_context.useSwitchContext;
