// @ark-ui/react/dist/components/switch/switch.cjs
const require_runtime = require("../../_virtual/_rolldown/runtime.cjs");
const require_switch_context = require("./switch-context.cjs");
const require_switch_control = require("./switch-control.cjs");
const require_switch_hidden_input = require("./switch-hidden-input.cjs");
const require_switch_label = require("./switch-label.cjs");
const require_switch_root = require("./switch-root.cjs");
const require_switch_root_provider = require("./switch-root-provider.cjs");
const require_switch_thumb = require("./switch-thumb.cjs");
//#region src/components/switch/switch.ts
var switch_exports = /* @__PURE__ */ require_runtime.__exportAll({
	Context: () => require_switch_context.SwitchContext,
	Control: () => require_switch_control.SwitchControl,
	HiddenInput: () => require_switch_hidden_input.SwitchHiddenInput,
	Label: () => require_switch_label.SwitchLabel,
	Root: () => require_switch_root.SwitchRoot,
	RootProvider: () => require_switch_root_provider.SwitchRootProvider,
	Thumb: () => require_switch_thumb.SwitchThumb
});
//#endregion
Object.defineProperty(exports, "switch_exports", {
	enumerable: true,
	get: function() {
		return switch_exports;
	}
});
