// @ark-ui/react/dist/components/switch/use-switch.js
"use client";
import { useEnvironmentContext } from "../../providers/environment/use-environment-context.js";
import { useLocaleContext } from "../../providers/locale/use-locale-context.js";
import { useFieldContext } from "../field/use-field-context.js";
import * as zagSwitch from "@zag-js/switch";
import { useId } from "react";
import { normalizeProps, useMachine } from "@zag-js/react";
//#region src/components/switch/use-switch.ts
var useSwitch = (props) => {
	const id = useId();
	const { getRootNode } = useEnvironmentContext();
	const { dir } = useLocaleContext();
	const field = useFieldContext();
	const machineProps = {
		id,
		ids: {
			label: field?.ids.label,
			hiddenInput: field?.ids.control
		},
		dir,
		disabled: field?.disabled,
		readOnly: field?.readOnly,
		invalid: field?.invalid,
		required: field?.required,
		getRootNode,
		...props
	};
	const service = useMachine(zagSwitch.machine, machineProps);
	return zagSwitch.connect(service, normalizeProps);
};
//#endregion
export { useSwitch };
