// @ark-ui/react/dist/components/clipboard/clipboard.cjs
const require_runtime = require("../../_virtual/_rolldown/runtime.cjs");
const require_clipboard_context = require("./clipboard-context.cjs");
const require_clipboard_control = require("./clipboard-control.cjs");
const require_clipboard_indicator = require("./clipboard-indicator.cjs");
const require_clipboard_input = require("./clipboard-input.cjs");
const require_clipboard_label = require("./clipboard-label.cjs");
const require_clipboard_root = require("./clipboard-root.cjs");
const require_clipboard_root_provider = require("./clipboard-root-provider.cjs");
const require_clipboard_trigger = require("./clipboard-trigger.cjs");
const require_clipboard_value_text = require("./clipboard-value-text.cjs");
//#region src/components/clipboard/clipboard.ts
var clipboard_exports = /* @__PURE__ */ require_runtime.__exportAll({
	Context: () => require_clipboard_context.ClipboardContext,
	Control: () => require_clipboard_control.ClipboardControl,
	Indicator: () => require_clipboard_indicator.ClipboardIndicator,
	Input: () => require_clipboard_input.ClipboardInput,
	Label: () => require_clipboard_label.ClipboardLabel,
	Root: () => require_clipboard_root.ClipboardRoot,
	RootProvider: () => require_clipboard_root_provider.ClipboardRootProvider,
	Trigger: () => require_clipboard_trigger.ClipboardTrigger,
	ValueText: () => require_clipboard_value_text.ClipboardValueText
});
//#endregion
Object.defineProperty(exports, "clipboard_exports", {
	enumerable: true,
	get: function() {
		return clipboard_exports;
	}
});
