// @ark-ui/react/dist/components/clipboard/clipboard.js
import { __exportAll } from "../../_virtual/_rolldown/runtime.js";
import { ClipboardContext } from "./clipboard-context.js";
import { ClipboardControl } from "./clipboard-control.js";
import { ClipboardIndicator } from "./clipboard-indicator.js";
import { ClipboardInput } from "./clipboard-input.js";
import { ClipboardLabel } from "./clipboard-label.js";
import { ClipboardRoot } from "./clipboard-root.js";
import { ClipboardRootProvider } from "./clipboard-root-provider.js";
import { ClipboardTrigger } from "./clipboard-trigger.js";
import { ClipboardValueText } from "./clipboard-value-text.js";
//#region src/components/clipboard/clipboard.ts
var clipboard_exports = /* @__PURE__ */ __exportAll({
	Context: () => ClipboardContext,
	Control: () => ClipboardControl,
	Indicator: () => ClipboardIndicator,
	Input: () => ClipboardInput,
	Label: () => ClipboardLabel,
	Root: () => ClipboardRoot,
	RootProvider: () => ClipboardRootProvider,
	Trigger: () => ClipboardTrigger,
	ValueText: () => ClipboardValueText
});
//#endregion
export { clipboard_exports };
