// @ark-ui/react/dist/components/clipboard/index.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const require_use_clipboard_context = require("./use-clipboard-context.cjs");
const require_clipboard_context = require("./clipboard-context.cjs");
const require_clipboard_control = require("./clipboard-control.cjs");
const require_clipboard_indicator = require("./clipboard-indicator.cjs");
const require_clipboard_input = require("./clipboard-input.cjs");
const require_clipboard_label = require("./clipboard-label.cjs");
const require_use_clipboard = require("./use-clipboard.cjs");
const require_clipboard_root = require("./clipboard-root.cjs");
const require_clipboard_root_provider = require("./clipboard-root-provider.cjs");
const require_clipboard_trigger = require("./clipboard-trigger.cjs");
const require_clipboard_value_text = require("./clipboard-value-text.cjs");
const require_clipboard = require("./clipboard.cjs");
let _zag_js_clipboard = require("@zag-js/clipboard");
Object.defineProperty(exports, "Clipboard", {
	enumerable: true,
	get: function() {
		return require_clipboard.clipboard_exports;
	}
});
exports.ClipboardContext = require_clipboard_context.ClipboardContext;
exports.ClipboardControl = require_clipboard_control.ClipboardControl;
exports.ClipboardIndicator = require_clipboard_indicator.ClipboardIndicator;
exports.ClipboardInput = require_clipboard_input.ClipboardInput;
exports.ClipboardLabel = require_clipboard_label.ClipboardLabel;
exports.ClipboardRoot = require_clipboard_root.ClipboardRoot;
exports.ClipboardRootProvider = require_clipboard_root_provider.ClipboardRootProvider;
exports.ClipboardTrigger = require_clipboard_trigger.ClipboardTrigger;
exports.ClipboardValueText = require_clipboard_value_text.ClipboardValueText;
Object.defineProperty(exports, "clipboardAnatomy", {
	enumerable: true,
	get: function() {
		return _zag_js_clipboard.anatomy;
	}
});
exports.useClipboard = require_use_clipboard.useClipboard;
exports.useClipboardContext = require_use_clipboard_context.useClipboardContext;
