// @ark-ui/react/dist/components/avatar/index.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const require_use_avatar_context = require("./use-avatar-context.cjs");
const require_avatar_context = require("./avatar-context.cjs");
const require_avatar_fallback = require("./avatar-fallback.cjs");
const require_avatar_image = require("./avatar-image.cjs");
const require_use_avatar = require("./use-avatar.cjs");
const require_avatar_root = require("./avatar-root.cjs");
const require_avatar_root_provider = require("./avatar-root-provider.cjs");
const require_avatar = require("./avatar.cjs");
let _zag_js_avatar = require("@zag-js/avatar");
Object.defineProperty(exports, "Avatar", {
	enumerable: true,
	get: function() {
		return require_avatar.avatar_exports;
	}
});
exports.AvatarContext = require_avatar_context.AvatarContext;
exports.AvatarFallback = require_avatar_fallback.AvatarFallback;
exports.AvatarImage = require_avatar_image.AvatarImage;
exports.AvatarRoot = require_avatar_root.AvatarRoot;
exports.AvatarRootProvider = require_avatar_root_provider.AvatarRootProvider;
Object.defineProperty(exports, "avatarAnatomy", {
	enumerable: true,
	get: function() {
		return _zag_js_avatar.anatomy;
	}
});
exports.useAvatar = require_use_avatar.useAvatar;
exports.useAvatarContext = require_use_avatar_context.useAvatarContext;
