// @ark-ui/react/dist/components/toast/toaster.cjs
"use client";
const require_runtime = require("../../_virtual/_rolldown/runtime.cjs");
const require_factory = require("../factory.cjs");
const require_use_environment_context = require("../../providers/environment/use-environment-context.cjs");
const require_use_locale_context = require("../../providers/locale/use-locale-context.cjs");
const require_use_toast_context = require("./use-toast-context.cjs");
let _zag_js_toast = require("@zag-js/toast");
_zag_js_toast = require_runtime.__toESM(_zag_js_toast, 1);
let react = require("react");
let _zag_js_react = require("@zag-js/react");
let react_jsx_runtime = require("react/jsx-runtime");
//#region src/components/toast/toaster.tsx
var Toaster = (0, react.forwardRef)((props, ref) => {
	const { toaster, children, ...localProps } = props;
	const locale = require_use_locale_context.useLocaleContext();
	const env = require_use_environment_context.useEnvironmentContext();
	const service = (0, _zag_js_react.useMachine)(_zag_js_toast.group.machine, {
		store: toaster,
		id: (0, react.useId)(),
		dir: locale?.dir,
		getRootNode: env?.getRootNode
	});
	const api = _zag_js_toast.group.connect(service, _zag_js_react.normalizeProps);
	const mergedProps = (0, _zag_js_react.mergeProps)(api.getGroupProps(), localProps);
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_factory.ark.div, {
		...mergedProps,
		ref,
		children: api.getToasts().map((toast, index) => /* @__PURE__ */ (0, react_jsx_runtime.jsx)(ToastActor, {
			value: toast,
			parent: service,
			index,
			children: (ctx) => children(ctx)
		}, toast.id))
	});
});
Toaster.displayName = "Toaster";
var ToastActor = (props) => {
	const env = require_use_environment_context.useEnvironmentContext();
	const localProps = {
		...props.value,
		parent: props.parent,
		index: props.index,
		getRootNode: env.getRootNode
	};
	const service = (0, _zag_js_react.useMachine)(_zag_js_toast.machine, { ...localProps });
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_use_toast_context.ToastProvider, {
		value: _zag_js_toast.connect(service, _zag_js_react.normalizeProps),
		children: props.children(props.value)
	});
};
ToastActor.displayName = "ToastActor";
//#endregion
exports.Toaster = Toaster;
