// @zag-js/slider/dist/slider.props.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/slider.props.ts
var slider_props_exports = {};
__export(slider_props_exports, {
  markerProps: () => markerProps,
  props: () => props,
  splitMarkerProps: () => splitMarkerProps,
  splitProps: () => splitProps,
  splitThumbProps: () => splitThumbProps,
  thumbProps: () => thumbProps
});
module.exports = __toCommonJS(slider_props_exports);
var import_types = require("@zag-js/types");
var import_utils = require("@zag-js/utils");
var props = (0, import_types.createProps)()([
  "aria-label",
  "aria-labelledby",
  "dir",
  "disabled",
  "form",
  "getAriaValueText",
  "getRootNode",
  "id",
  "ids",
  "invalid",
  "max",
  "min",
  "minStepsBetweenThumbs",
  "name",
  "onFocusChange",
  "onValueChange",
  "onValueChangeEnd",
  "orientation",
  "origin",
  "readOnly",
  "step",
  "thumbAlignment",
  "thumbCollisionBehavior",
  "thumbSize",
  "value",
  "defaultValue"
]);
var splitProps = (0, import_utils.createSplitProps)(props);
var thumbProps = (0, import_types.createProps)()(["index", "name"]);
var splitThumbProps = (0, import_utils.createSplitProps)(thumbProps);
var markerProps = (0, import_types.createProps)()(["value"]);
var splitMarkerProps = (0, import_utils.createSplitProps)(markerProps);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  markerProps,
  props,
  splitMarkerProps,
  splitProps,
  splitThumbProps,
  thumbProps
});
