// @zag-js/marquee/dist/marquee.utils.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/marquee.utils.ts
var marquee_utils_exports = {};
__export(marquee_utils_exports, {
  getEdgePositionStyles: () => getEdgePositionStyles,
  getMarqueeTranslate: () => getMarqueeTranslate
});
module.exports = __toCommonJS(marquee_utils_exports);
var getEdgePositionStyles = (options) => {
  const { side } = options;
  switch (side) {
    case "start":
      return {
        top: 0,
        insetInlineStart: 0,
        height: "100%"
      };
    case "end":
      return {
        top: 0,
        insetInlineEnd: 0,
        height: "100%"
      };
    case "top":
      return {
        top: 0,
        insetInline: 0,
        width: "100%"
      };
    case "bottom":
      return {
        bottom: 0,
        insetInline: 0,
        width: "100%"
      };
  }
};
var getMarqueeTranslate = (options) => {
  const { side, dir } = options;
  if (side === "top") {
    return "-100%";
  }
  if (side === "bottom") {
    return "100%";
  }
  const shouldBeNegative = side === "start" && dir === "ltr" || side === "end" && dir === "rtl";
  return shouldBeNegative ? "-100%" : "100%";
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getEdgePositionStyles,
  getMarqueeTranslate
});
