// @zag-js/marquee/dist/marquee.connect.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/marquee.connect.ts
var marquee_connect_exports = {};
__export(marquee_connect_exports, {
  connect: () => connect
});
module.exports = __toCommonJS(marquee_connect_exports);
var import_dom_query = require("@zag-js/dom-query");
var import_marquee = require("./marquee.anatomy.js");
var import_marquee2 = require("./marquee.dom.js");
var import_marquee3 = require("./marquee.utils.js");
function connect(service, normalize) {
  const { scope, send, context, computed, prop } = service;
  const side = prop("side");
  const paused = context.get("paused");
  const duration = context.get("duration");
  const orientation = computed("orientation");
  const multiplier = computed("multiplier");
  const isVertical = computed("isVertical");
  return {
    paused,
    orientation,
    side,
    multiplier,
    contentCount: multiplier + 1,
    pause() {
      send({ type: "PAUSE" });
    },
    resume() {
      send({ type: "RESUME" });
    },
    togglePause() {
      send({ type: "TOGGLE_PAUSE" });
    },
    restart() {
      send({ type: "RESTART" });
    },
    getRootProps() {
      const dir = prop("dir");
      return normalize.element({
        ...import_marquee.parts.root.attrs,
        id: import_marquee2.dom.getRootId(scope),
        dir,
        role: "region",
        "aria-roledescription": "marquee",
        "aria-live": "off",
        "aria-label": prop("translations").root,
        "data-state": paused ? "paused" : "idle",
        "data-orientation": orientation,
        "data-paused": (0, import_dom_query.dataAttr)(paused),
        onMouseEnter: prop("pauseOnInteraction") ? () => send({ type: "PAUSE" }) : void 0,
        onMouseLeave: prop("pauseOnInteraction") ? () => send({ type: "RESUME" }) : void 0,
        onFocusCapture: prop("pauseOnInteraction") ? (event) => {
          if (event.target !== event.currentTarget) {
            send({ type: "PAUSE" });
          }
        } : void 0,
        onBlurCapture: prop("pauseOnInteraction") ? (event) => {
          if (!event.currentTarget.contains(event.relatedTarget)) {
            send({ type: "RESUME" });
          }
        } : void 0,
        style: {
          display: "flex",
          flexDirection: orientation === "vertical" ? "column" : "row",
          position: "relative",
          // Essential for clipping scrolling content
          overflow: "hidden",
          // CSS containment for performance - prevents layout recalculation of parent elements
          contain: "layout style paint",
          "--marquee-duration": `${duration}s`,
          "--marquee-spacing": prop("spacing"),
          "--marquee-delay": `${prop("delay")}s`,
          "--marquee-loop-count": prop("loopCount") === 0 ? "infinite" : prop("loopCount").toString(),
          "--marquee-translate": (0, import_marquee3.getMarqueeTranslate)({ side, dir })
        }
      });
    },
    getViewportProps() {
      return normalize.element({
        ...import_marquee.parts.viewport.attrs,
        id: import_marquee2.dom.getViewportId(scope),
        "data-part": "viewport",
        "data-orientation": orientation,
        "data-side": side,
        onAnimationIteration(event) {
          if (event.target === import_marquee2.dom.getContentEl(scope, 0)) {
            prop("onLoopComplete")?.();
          }
        },
        onAnimationEnd(event) {
          if (event.target === import_marquee2.dom.getContentEl(scope, 0)) {
            prop("onComplete")?.();
          }
        },
        style: {
          display: "flex",
          [isVertical ? "height" : "width"]: "100%",
          // For bottom/end sides, reverse flex direction so clones appear on the correct side
          flexDirection: orientation === "vertical" ? side === "bottom" ? "column-reverse" : "column" : side === "end" ? "row-reverse" : "row"
        }
      });
    },
    getContentProps(props) {
      const { index } = props;
      const clone = index > 0;
      return normalize.element({
        ...import_marquee.parts.content.attrs,
        id: import_marquee2.dom.getContentId(scope, index),
        dir: prop("dir"),
        "data-part": "content",
        "data-index": index,
        "data-orientation": orientation,
        "data-side": side,
        "data-reverse": prop("reverse") ? "" : void 0,
        "data-clone": (0, import_dom_query.dataAttr)(clone),
        role: clone ? "presentation" : void 0,
        "aria-hidden": clone ? true : void 0,
        style: {
          // Essential layout for marquee content
          display: "flex",
          flexDirection: orientation === "vertical" ? "column" : "row",
          flexShrink: 0,
          // Force compositor layer to prevent flicker during animation reset
          backfaceVisibility: "hidden",
          WebkitBackfaceVisibility: "hidden",
          willChange: paused ? "auto" : "transform",
          transform: "translateZ(0)",
          [isVertical ? "minWidth" : "minHeight"]: "auto",
          // Prevent subpixel rendering issues that cause flicker in Firefox
          contain: "paint"
        }
      });
    },
    getEdgeProps(props) {
      const { side: side2 } = props;
      const dir = prop("dir");
      return normalize.element({
        ...import_marquee.parts.edge.attrs,
        dir,
        "data-part": "edge",
        "data-side": side2,
        "data-orientation": orientation,
        style: {
          pointerEvents: "none",
          position: "absolute",
          ...(0, import_marquee3.getEdgePositionStyles)({ side: side2, dir })
        }
      });
    },
    getItemProps() {
      return normalize.element({
        ...import_marquee.parts.item.attrs,
        dir: prop("dir"),
        style: {
          [isVertical ? "marginBlock" : "marginInline"]: "calc(var(--marquee-spacing) / 2)"
        }
      });
    }
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  connect
});
