// @zag-js/marquee/dist/marquee.machine.mjs
// src/marquee.machine.ts
import { createMachine } from "@zag-js/core";
import { dom } from "./marquee.dom.mjs";
var machine = createMachine({
  props({ props }) {
    return {
      dir: "ltr",
      side: "start",
      speed: 50,
      delay: 0,
      loopCount: 0,
      spacing: "1rem",
      autoFill: false,
      pauseOnInteraction: false,
      reverse: false,
      defaultPaused: false,
      translations: {
        root: "Marquee content"
      },
      ...props
    };
  },
  refs() {
    return {
      dimensions: void 0,
      initialDurationSet: false
    };
  },
  context({ prop, bindable }) {
    return {
      paused: bindable(() => ({
        value: prop("paused"),
        defaultValue: prop("defaultPaused"),
        onChange(value) {
          prop("onPauseChange")?.({ paused: value });
        }
      })),
      duration: bindable(() => ({
        defaultValue: 2e3 / Math.max(1e-3, prop("speed"))
      }))
    };
  },
  initialState() {
    return "idle";
  },
  computed: {
    orientation: ({ prop }) => {
      const side = prop("side");
      return side === "top" || side === "bottom" ? "vertical" : "horizontal";
    },
    isVertical: ({ prop }) => {
      const side = prop("side");
      return side === "top" || side === "bottom";
    },
    multiplier: ({ refs, prop }) => {
      if (!prop("autoFill")) return 1;
      const dimensions = refs.get("dimensions");
      if (!dimensions) return 1;
      const { rootSize, contentSize } = dimensions;
      if (contentSize === 0) return 1;
      return contentSize < rootSize ? Math.ceil(rootSize / contentSize) : 1;
    }
  },
  watch({ track, action, prop }) {
    track([() => prop("speed")], () => {
      action(["recalculateDuration", "restartAnimation"]);
    });
    track([() => prop("spacing"), () => prop("side")], () => {
      action(["recalculateDuration"]);
    });
  },
  on: {
    PAUSE: {
      actions: ["setPaused"]
    },
    RESUME: {
      actions: ["setResumed"]
    },
    TOGGLE_PAUSE: {
      actions: ["togglePaused"]
    },
    RESTART: {
      actions: ["restartAnimation"]
    }
  },
  effects: ["trackDimensions"],
  states: {
    idle: {}
  },
  implementations: {
    actions: {
      setPaused({ context }) {
        context.set("paused", true);
      },
      setResumed({ context }) {
        context.set("paused", false);
      },
      togglePaused({ context }) {
        context.set("paused", (prev) => !prev);
      },
      restartAnimation({ scope }) {
        const viewportEl = dom.getViewportEl(scope);
        if (!viewportEl) return;
        const contentElements = viewportEl.querySelectorAll('[data-part="content"]');
        contentElements.forEach((el) => {
          el.style.animation = "none";
          el.offsetHeight;
          el.style.animation = "";
        });
      },
      recalculateDuration({ refs, computed, context, prop }) {
        const dimensions = refs.get("dimensions");
        if (!dimensions) return;
        const { rootSize, contentSize } = dimensions;
        const duration = calculateDuration({
          rootSize,
          contentSize,
          speed: Math.max(1e-3, prop("speed")),
          multiplier: computed("multiplier"),
          autoFill: prop("autoFill")
        });
        context.set("duration", duration);
      }
    },
    effects: {
      trackDimensions({ scope, refs, computed, context, prop }) {
        const rootEl = dom.getRootEl(scope);
        const contentEl = dom.getContentEl(scope, 0);
        if (!rootEl || !contentEl) return;
        const win = scope.getWin();
        const measureDimensions = () => {
          const rootSize = computed("isVertical") ? rootEl.clientHeight : rootEl.clientWidth;
          const contentSize = computed("isVertical") ? contentEl.clientHeight : contentEl.clientWidth;
          return { rootSize, contentSize };
        };
        const exec = () => {
          const { rootSize, contentSize } = measureDimensions();
          if (rootSize > 0 && contentSize > 0) {
            refs.set("dimensions", { rootSize, contentSize });
            if (!refs.get("initialDurationSet")) {
              const duration = calculateDuration({
                rootSize,
                contentSize,
                speed: Math.max(1e-3, prop("speed")),
                multiplier: computed("multiplier"),
                autoFill: prop("autoFill")
              });
              context.set("duration", duration);
              refs.set("initialDurationSet", true);
            }
          }
        };
        let rafId = null;
        const observer = new win.ResizeObserver(() => {
          if (rafId !== null) return;
          rafId = win.requestAnimationFrame(() => {
            const { rootSize, contentSize } = measureDimensions();
            refs.set("dimensions", { rootSize, contentSize });
            rafId = null;
          });
        });
        observer.observe(rootEl);
        observer.observe(contentEl);
        exec();
        return () => {
          observer.disconnect();
          if (rafId !== null) win.cancelAnimationFrame(rafId);
        };
      }
    }
  }
});
function calculateDuration(options) {
  const { rootSize, contentSize, speed, multiplier, autoFill } = options;
  if (autoFill) {
    return contentSize * multiplier / speed;
  }
  return contentSize < rootSize ? rootSize / speed : contentSize / speed;
}
export {
  machine
};
