// @ark-ui/react/dist/components/popover/popover.cjs
const require_runtime = require("../../_virtual/_rolldown/runtime.cjs");
const require_popover_anchor = require("./popover-anchor.cjs");
const require_popover_arrow = require("./popover-arrow.cjs");
const require_popover_arrow_tip = require("./popover-arrow-tip.cjs");
const require_popover_close_trigger = require("./popover-close-trigger.cjs");
const require_popover_content = require("./popover-content.cjs");
const require_popover_context = require("./popover-context.cjs");
const require_popover_description = require("./popover-description.cjs");
const require_popover_indicator = require("./popover-indicator.cjs");
const require_popover_positioner = require("./popover-positioner.cjs");
const require_popover_root = require("./popover-root.cjs");
const require_popover_root_provider = require("./popover-root-provider.cjs");
const require_popover_title = require("./popover-title.cjs");
const require_popover_trigger = require("./popover-trigger.cjs");
//#region src/components/popover/popover.ts
var popover_exports = /* @__PURE__ */ require_runtime.__exportAll({
	Anchor: () => require_popover_anchor.PopoverAnchor,
	Arrow: () => require_popover_arrow.PopoverArrow,
	ArrowTip: () => require_popover_arrow_tip.PopoverArrowTip,
	CloseTrigger: () => require_popover_close_trigger.PopoverCloseTrigger,
	Content: () => require_popover_content.PopoverContent,
	Context: () => require_popover_context.PopoverContext,
	Description: () => require_popover_description.PopoverDescription,
	Indicator: () => require_popover_indicator.PopoverIndicator,
	Positioner: () => require_popover_positioner.PopoverPositioner,
	Root: () => require_popover_root.PopoverRoot,
	RootProvider: () => require_popover_root_provider.PopoverRootProvider,
	Title: () => require_popover_title.PopoverTitle,
	Trigger: () => require_popover_trigger.PopoverTrigger
});
//#endregion
Object.defineProperty(exports, "popover_exports", {
	enumerable: true,
	get: function() {
		return popover_exports;
	}
});
