// @zag-js/combobox/dist/combobox.connect.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/combobox.connect.ts
var combobox_connect_exports = {};
__export(combobox_connect_exports, {
  connect: () => connect
});
module.exports = __toCommonJS(combobox_connect_exports);
var import_dom_query = require("@zag-js/dom-query");
var import_popper = require("@zag-js/popper");
var import_utils = require("@zag-js/utils");
var import_combobox = require("./combobox.anatomy.js");
var dom = __toESM(require("./combobox.dom.js"));
function connect(service, normalize) {
  const { context, prop, state, send, scope, computed, event } = service;
  const translations = prop("translations");
  const collection = prop("collection");
  const disabled = !!prop("disabled");
  const interactive = computed("isInteractive");
  const invalid = !!prop("invalid");
  const required = !!prop("required");
  const readOnly = !!prop("readOnly");
  const open = state.hasTag("open");
  const focused = state.hasTag("focused");
  const composite = prop("composite");
  const highlightedValue = context.get("highlightedValue");
  const currentPlacement = context.get("currentPlacement");
  const currentPlacementSide = currentPlacement ? (0, import_popper.getPlacementSide)(currentPlacement) : void 0;
  const popperStyles = (0, import_popper.getPlacementStyles)({
    ...prop("positioning"),
    placement: currentPlacement
  });
  function getItemState(props) {
    const itemDisabled = collection.getItemDisabled(props.item);
    const value = collection.getItemValue(props.item);
    (0, import_utils.ensure)(value, () => `[zag-js] No value found for item ${JSON.stringify(props.item)}`);
    return {
      value,
      disabled: Boolean(disabled || itemDisabled),
      highlighted: highlightedValue === value,
      selected: context.get("value").includes(value)
    };
  }
  return {
    focused,
    open,
    inputValue: context.get("inputValue"),
    highlightedValue,
    highlightedItem: context.get("highlightedItem"),
    value: context.get("value"),
    valueAsString: computed("valueAsString"),
    hasSelectedItems: computed("hasSelectedItems"),
    selectedItems: computed("selectedItems"),
    collection: prop("collection"),
    multiple: !!prop("multiple"),
    disabled: !!disabled,
    syncSelectedItems() {
      send({ type: "SELECTED_ITEMS.SYNC" });
    },
    reposition(options = {}) {
      send({ type: "POSITIONING.SET", options });
    },
    setHighlightValue(value) {
      send({ type: "HIGHLIGHTED_VALUE.SET", value });
    },
    clearHighlightValue() {
      send({ type: "HIGHLIGHTED_VALUE.CLEAR" });
    },
    selectValue(value) {
      send({ type: "ITEM.SELECT", value });
    },
    setValue(value) {
      send({ type: "VALUE.SET", value });
    },
    setInputValue(value, reason = "script") {
      send({ type: "INPUT_VALUE.SET", value, src: reason });
    },
    clearValue(value) {
      if (value != null) {
        send({ type: "ITEM.CLEAR", value });
      } else {
        send({ type: "VALUE.CLEAR" });
      }
    },
    focus() {
      dom.getInputEl(scope)?.focus();
    },
    setOpen(nextOpen, reason = "script") {
      const open2 = state.hasTag("open");
      if (open2 === nextOpen) return;
      send({ type: nextOpen ? "OPEN" : "CLOSE", src: reason });
    },
    getRootProps() {
      return normalize.element({
        ...import_combobox.parts.root.attrs,
        dir: prop("dir"),
        id: dom.getRootId(scope),
        "data-invalid": (0, import_dom_query.dataAttr)(invalid),
        "data-readonly": (0, import_dom_query.dataAttr)(readOnly)
      });
    },
    getLabelProps() {
      return normalize.label({
        ...import_combobox.parts.label.attrs,
        dir: prop("dir"),
        htmlFor: dom.getInputId(scope),
        id: dom.getLabelId(scope),
        "data-readonly": (0, import_dom_query.dataAttr)(readOnly),
        "data-disabled": (0, import_dom_query.dataAttr)(disabled),
        "data-invalid": (0, import_dom_query.dataAttr)(invalid),
        "data-required": (0, import_dom_query.dataAttr)(required),
        "data-focus": (0, import_dom_query.dataAttr)(focused),
        onClick(event2) {
          if (composite) return;
          event2.preventDefault();
          dom.getTriggerEl(scope)?.focus({ preventScroll: true });
        }
      });
    },
    getControlProps() {
      return normalize.element({
        ...import_combobox.parts.control.attrs,
        dir: prop("dir"),
        id: dom.getControlId(scope),
        "data-state": open ? "open" : "closed",
        "data-focus": (0, import_dom_query.dataAttr)(focused),
        "data-disabled": (0, import_dom_query.dataAttr)(disabled),
        "data-invalid": (0, import_dom_query.dataAttr)(invalid)
      });
    },
    getPositionerProps() {
      return normalize.element({
        ...import_combobox.parts.positioner.attrs,
        dir: prop("dir"),
        id: dom.getPositionerId(scope),
        style: popperStyles.floating
      });
    },
    getInputProps() {
      return normalize.input({
        ...import_combobox.parts.input.attrs,
        dir: prop("dir"),
        "aria-invalid": (0, import_dom_query.ariaAttr)(invalid),
        "data-invalid": (0, import_dom_query.dataAttr)(invalid),
        "data-autofocus": (0, import_dom_query.dataAttr)(prop("autoFocus")),
        name: prop("name"),
        form: prop("form"),
        disabled,
        required: prop("required"),
        autoComplete: "off",
        autoCorrect: "off",
        autoCapitalize: "none",
        spellCheck: "false",
        readOnly,
        placeholder: prop("placeholder"),
        id: dom.getInputId(scope),
        type: "text",
        role: "combobox",
        defaultValue: context.get("inputValue"),
        "aria-autocomplete": computed("autoComplete") ? "both" : "list",
        "aria-controls": dom.getContentId(scope),
        "aria-expanded": open,
        "data-state": open ? "open" : "closed",
        "aria-activedescendant": highlightedValue ? dom.getItemId(scope, highlightedValue) : void 0,
        onClick(event2) {
          if (event2.defaultPrevented) return;
          if (!prop("openOnClick")) return;
          if (!interactive) return;
          send({ type: "INPUT.CLICK", src: "input-click" });
        },
        onFocus() {
          if (disabled) return;
          send({ type: "INPUT.FOCUS" });
        },
        onBlur() {
          if (disabled) return;
          send({ type: "INPUT.BLUR" });
        },
        onChange(event2) {
          send({ type: "INPUT.CHANGE", value: event2.currentTarget.value, src: "input-change" });
        },
        onKeyDown(event2) {
          if (event2.defaultPrevented) return;
          if (!interactive) return;
          if (event2.ctrlKey || event2.shiftKey || (0, import_dom_query.isComposingEvent)(event2)) return;
          const openOnKeyPress = prop("openOnKeyPress");
          const isModifierKey = event2.ctrlKey || event2.metaKey || event2.shiftKey;
          const keypress = true;
          const keymap = {
            ArrowDown(event3) {
              if (!openOnKeyPress && !open) return;
              send({ type: event3.altKey ? "OPEN" : "INPUT.ARROW_DOWN", keypress, src: "arrow-key" });
              event3.preventDefault();
            },
            ArrowUp() {
              if (!openOnKeyPress && !open) return;
              send({ type: event2.altKey ? "CLOSE" : "INPUT.ARROW_UP", keypress, src: "arrow-key" });
              event2.preventDefault();
            },
            Home(event3) {
              if (isModifierKey) return;
              send({ type: "INPUT.HOME", keypress });
              if (open) {
                event3.preventDefault();
              }
            },
            End(event3) {
              if (isModifierKey) return;
              send({ type: "INPUT.END", keypress });
              if (open) {
                event3.preventDefault();
              }
            },
            Enter(event3) {
              send({ type: "INPUT.ENTER", keypress, src: "item-select" });
              const hasHighlight = highlightedValue != null;
              const alwaysSubmit = prop("alwaysSubmitOnEnter");
              const willBeRejected = computed("isCustomValue") && !prop("allowCustomValue");
              if (open && !alwaysSubmit && (hasHighlight || willBeRejected)) {
                event3.preventDefault();
              }
              if (highlightedValue == null) return;
              const itemEl = dom.getItemEl(scope, highlightedValue);
              if ((0, import_dom_query.isAnchorElement)(itemEl)) {
                prop("navigate")?.({ value: highlightedValue, node: itemEl, href: itemEl.href });
              }
            },
            Escape() {
              send({ type: "INPUT.ESCAPE", keypress, src: "escape-key" });
              event2.preventDefault();
            }
          };
          const key = (0, import_dom_query.getEventKey)(event2, { dir: prop("dir") });
          const exec = keymap[key];
          exec?.(event2);
        }
      });
    },
    getTriggerProps(props = {}) {
      return normalize.button({
        ...import_combobox.parts.trigger.attrs,
        dir: prop("dir"),
        id: dom.getTriggerId(scope),
        "aria-haspopup": composite ? "listbox" : "dialog",
        type: "button",
        tabIndex: props.focusable ? void 0 : -1,
        "aria-label": translations.triggerLabel,
        "aria-expanded": open,
        "data-state": open ? "open" : "closed",
        "aria-controls": open ? dom.getContentId(scope) : void 0,
        disabled,
        "data-invalid": (0, import_dom_query.dataAttr)(invalid),
        "data-focusable": (0, import_dom_query.dataAttr)(props.focusable),
        "data-readonly": (0, import_dom_query.dataAttr)(readOnly),
        "data-disabled": (0, import_dom_query.dataAttr)(disabled),
        onFocus() {
          if (!props.focusable) return;
          send({ type: "INPUT.FOCUS", src: "trigger" });
        },
        onClick(event2) {
          if (event2.defaultPrevented) return;
          if (!interactive) return;
          if (!(0, import_dom_query.isLeftClick)(event2)) return;
          send({ type: "TRIGGER.CLICK", src: "trigger-click" });
        },
        onPointerDown(event2) {
          if (!interactive) return;
          if (event2.pointerType === "touch") return;
          if (!(0, import_dom_query.isLeftClick)(event2)) return;
          event2.preventDefault();
          queueMicrotask(() => {
            dom.focusInputEl(scope);
          });
        },
        onKeyDown(event2) {
          if (event2.defaultPrevented) return;
          if (composite) return;
          const keyMap = {
            ArrowDown() {
              send({ type: "INPUT.ARROW_DOWN", src: "arrow-key" });
            },
            ArrowUp() {
              send({ type: "INPUT.ARROW_UP", src: "arrow-key" });
            }
          };
          const key = (0, import_dom_query.getEventKey)(event2, { dir: prop("dir") });
          const exec = keyMap[key];
          if (exec) {
            exec(event2);
            event2.preventDefault();
          }
        }
      });
    },
    getContentProps() {
      return normalize.element({
        ...import_combobox.parts.content.attrs,
        dir: prop("dir"),
        id: dom.getContentId(scope),
        role: !composite ? "dialog" : "listbox",
        tabIndex: -1,
        hidden: !open,
        "data-state": open ? "open" : "closed",
        "data-placement": currentPlacement,
        "data-side": currentPlacementSide,
        "aria-labelledby": dom.getLabelId(scope),
        "aria-multiselectable": prop("multiple") && composite ? true : void 0,
        "data-empty": (0, import_dom_query.dataAttr)(collection.size === 0),
        onPointerDown(event2) {
          if (!(0, import_dom_query.isLeftClick)(event2)) return;
          event2.preventDefault();
        }
      });
    },
    getListProps() {
      return normalize.element({
        ...import_combobox.parts.list.attrs,
        role: !composite ? "listbox" : void 0,
        "data-empty": (0, import_dom_query.dataAttr)(collection.size === 0),
        "aria-labelledby": dom.getLabelId(scope),
        "aria-multiselectable": prop("multiple") && !composite ? true : void 0
      });
    },
    getClearTriggerProps() {
      return normalize.button({
        ...import_combobox.parts.clearTrigger.attrs,
        dir: prop("dir"),
        id: dom.getClearTriggerId(scope),
        type: "button",
        tabIndex: -1,
        disabled,
        "data-invalid": (0, import_dom_query.dataAttr)(invalid),
        "aria-label": translations.clearTriggerLabel,
        "aria-controls": dom.getInputId(scope),
        hidden: !context.get("value").length,
        onPointerDown(event2) {
          if (!(0, import_dom_query.isLeftClick)(event2)) return;
          event2.preventDefault();
        },
        onClick(event2) {
          if (event2.defaultPrevented) return;
          if (!interactive) return;
          send({ type: "VALUE.CLEAR", src: "clear-trigger" });
        }
      });
    },
    getItemState,
    getItemProps(props) {
      const itemState = getItemState(props);
      const value = itemState.value;
      return normalize.element({
        ...import_combobox.parts.item.attrs,
        dir: prop("dir"),
        id: dom.getItemId(scope, value),
        role: "option",
        tabIndex: -1,
        "data-highlighted": (0, import_dom_query.dataAttr)(itemState.highlighted),
        "data-state": itemState.selected ? "checked" : "unchecked",
        "aria-selected": (0, import_dom_query.ariaAttr)(itemState.selected),
        "aria-disabled": (0, import_dom_query.ariaAttr)(itemState.disabled),
        "data-disabled": (0, import_dom_query.dataAttr)(itemState.disabled),
        "data-value": itemState.value,
        onPointerMove() {
          if (itemState.disabled) return;
          if (itemState.highlighted) return;
          send({ type: "ITEM.POINTER_MOVE", value });
        },
        onPointerLeave() {
          if (props.persistFocus) return;
          if (itemState.disabled) return;
          const prev = event.previous();
          const mouseMoved = prev?.type.includes("POINTER");
          if (!mouseMoved) return;
          send({ type: "ITEM.POINTER_LEAVE", value });
        },
        onClick(event2) {
          if ((0, import_dom_query.isDownloadingEvent)(event2)) return;
          if ((0, import_dom_query.isOpeningInNewTab)(event2)) return;
          if ((0, import_dom_query.isContextMenuEvent)(event2)) return;
          if (itemState.disabled) return;
          send({ type: "ITEM.CLICK", src: "item-select", value });
        }
      });
    },
    getItemTextProps(props) {
      const itemState = getItemState(props);
      return normalize.element({
        ...import_combobox.parts.itemText.attrs,
        dir: prop("dir"),
        "data-state": itemState.selected ? "checked" : "unchecked",
        "data-disabled": (0, import_dom_query.dataAttr)(itemState.disabled),
        "data-highlighted": (0, import_dom_query.dataAttr)(itemState.highlighted)
      });
    },
    getItemIndicatorProps(props) {
      const itemState = getItemState(props);
      return normalize.element({
        "aria-hidden": true,
        ...import_combobox.parts.itemIndicator.attrs,
        dir: prop("dir"),
        "data-state": itemState.selected ? "checked" : "unchecked",
        hidden: !itemState.selected
      });
    },
    getItemGroupProps(props) {
      const { id } = props;
      return normalize.element({
        ...import_combobox.parts.itemGroup.attrs,
        dir: prop("dir"),
        id: dom.getItemGroupId(scope, id),
        "aria-labelledby": dom.getItemGroupLabelId(scope, id),
        "data-empty": (0, import_dom_query.dataAttr)(collection.size === 0),
        role: "group"
      });
    },
    getItemGroupLabelProps(props) {
      const { htmlFor } = props;
      return normalize.element({
        ...import_combobox.parts.itemGroupLabel.attrs,
        dir: prop("dir"),
        id: dom.getItemGroupLabelId(scope, htmlFor),
        role: "presentation"
      });
    }
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  connect
});
