// @zag-js/accordion/dist/accordion.dom.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/accordion.dom.ts
var accordion_dom_exports = {};
__export(accordion_dom_exports, {
  getFirstTriggerEl: () => getFirstTriggerEl,
  getItemContentId: () => getItemContentId,
  getItemId: () => getItemId,
  getItemTriggerId: () => getItemTriggerId,
  getLastTriggerEl: () => getLastTriggerEl,
  getNextTriggerEl: () => getNextTriggerEl,
  getPrevTriggerEl: () => getPrevTriggerEl,
  getRootEl: () => getRootEl,
  getRootId: () => getRootId,
  getTriggerEls: () => getTriggerEls
});
module.exports = __toCommonJS(accordion_dom_exports);
var import_dom_query = require("@zag-js/dom-query");
var import_utils = require("@zag-js/utils");
var getRootId = (ctx) => ctx.ids?.root ?? `accordion:${ctx.id}`;
var getItemId = (ctx, value) => ctx.ids?.item?.(value) ?? `accordion:${ctx.id}:item:${value}`;
var getItemContentId = (ctx, value) => ctx.ids?.itemContent?.(value) ?? `accordion:${ctx.id}:content:${value}`;
var getItemTriggerId = (ctx, value) => ctx.ids?.itemTrigger?.(value) ?? `accordion:${ctx.id}:trigger:${value}`;
var getRootEl = (ctx) => ctx.getById(getRootId(ctx));
var getTriggerEls = (ctx) => {
  const ownerId = CSS.escape(getRootId(ctx));
  const selector = `[data-controls][data-ownedby='${ownerId}']:not([disabled])`;
  return (0, import_dom_query.queryAll)(getRootEl(ctx), selector);
};
var getFirstTriggerEl = (ctx) => (0, import_utils.first)(getTriggerEls(ctx));
var getLastTriggerEl = (ctx) => (0, import_utils.last)(getTriggerEls(ctx));
var getNextTriggerEl = (ctx, id) => (0, import_dom_query.nextById)(getTriggerEls(ctx), getItemTriggerId(ctx, id));
var getPrevTriggerEl = (ctx, id) => (0, import_dom_query.prevById)(getTriggerEls(ctx), getItemTriggerId(ctx, id));
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getFirstTriggerEl,
  getItemContentId,
  getItemId,
  getItemTriggerId,
  getLastTriggerEl,
  getNextTriggerEl,
  getPrevTriggerEl,
  getRootEl,
  getRootId,
  getTriggerEls
});
