// @ark-ui/react/dist/components/dialog/dialog.cjs
const require_runtime = require("../../_virtual/_rolldown/runtime.cjs");
const require_dialog_backdrop = require("./dialog-backdrop.cjs");
const require_dialog_close_trigger = require("./dialog-close-trigger.cjs");
const require_dialog_content = require("./dialog-content.cjs");
const require_dialog_context = require("./dialog-context.cjs");
const require_dialog_description = require("./dialog-description.cjs");
const require_dialog_positioner = require("./dialog-positioner.cjs");
const require_dialog_root = require("./dialog-root.cjs");
const require_dialog_root_provider = require("./dialog-root-provider.cjs");
const require_dialog_title = require("./dialog-title.cjs");
const require_dialog_trigger = require("./dialog-trigger.cjs");
//#region src/components/dialog/dialog.ts
var dialog_exports = /* @__PURE__ */ require_runtime.__exportAll({
	Backdrop: () => require_dialog_backdrop.DialogBackdrop,
	CloseTrigger: () => require_dialog_close_trigger.DialogCloseTrigger,
	Content: () => require_dialog_content.DialogContent,
	Context: () => require_dialog_context.DialogContext,
	Description: () => require_dialog_description.DialogDescription,
	Positioner: () => require_dialog_positioner.DialogPositioner,
	Root: () => require_dialog_root.DialogRoot,
	RootProvider: () => require_dialog_root_provider.DialogRootProvider,
	Title: () => require_dialog_title.DialogTitle,
	Trigger: () => require_dialog_trigger.DialogTrigger
});
//#endregion
Object.defineProperty(exports, "dialog_exports", {
	enumerable: true,
	get: function() {
		return dialog_exports;
	}
});
