// @ark-ui/react/dist/components/dialog/dialog.js
import { __exportAll } from "../../_virtual/_rolldown/runtime.js";
import { DialogBackdrop } from "./dialog-backdrop.js";
import { DialogCloseTrigger } from "./dialog-close-trigger.js";
import { DialogContent } from "./dialog-content.js";
import { DialogContext } from "./dialog-context.js";
import { DialogDescription } from "./dialog-description.js";
import { DialogPositioner } from "./dialog-positioner.js";
import { DialogRoot } from "./dialog-root.js";
import { DialogRootProvider } from "./dialog-root-provider.js";
import { DialogTitle } from "./dialog-title.js";
import { DialogTrigger } from "./dialog-trigger.js";
//#region src/components/dialog/dialog.ts
var dialog_exports = /* @__PURE__ */ __exportAll({
	Backdrop: () => DialogBackdrop,
	CloseTrigger: () => DialogCloseTrigger,
	Content: () => DialogContent,
	Context: () => DialogContext,
	Description: () => DialogDescription,
	Positioner: () => DialogPositioner,
	Root: () => DialogRoot,
	RootProvider: () => DialogRootProvider,
	Title: () => DialogTitle,
	Trigger: () => DialogTrigger
});
//#endregion
export { dialog_exports };
