// @ark-ui/react/dist/components/dialog/index.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const require_use_dialog_context = require("./use-dialog-context.cjs");
const require_dialog_backdrop = require("./dialog-backdrop.cjs");
const require_dialog_close_trigger = require("./dialog-close-trigger.cjs");
const require_dialog_content = require("./dialog-content.cjs");
const require_dialog_context = require("./dialog-context.cjs");
const require_dialog_description = require("./dialog-description.cjs");
const require_dialog_positioner = require("./dialog-positioner.cjs");
const require_use_dialog = require("./use-dialog.cjs");
const require_dialog_root = require("./dialog-root.cjs");
const require_dialog_root_provider = require("./dialog-root-provider.cjs");
const require_dialog_title = require("./dialog-title.cjs");
const require_dialog_trigger = require("./dialog-trigger.cjs");
const require_dialog = require("./dialog.cjs");
let _zag_js_dialog = require("@zag-js/dialog");
Object.defineProperty(exports, "Dialog", {
	enumerable: true,
	get: function() {
		return require_dialog.dialog_exports;
	}
});
exports.DialogBackdrop = require_dialog_backdrop.DialogBackdrop;
exports.DialogCloseTrigger = require_dialog_close_trigger.DialogCloseTrigger;
exports.DialogContent = require_dialog_content.DialogContent;
exports.DialogContext = require_dialog_context.DialogContext;
exports.DialogDescription = require_dialog_description.DialogDescription;
exports.DialogPositioner = require_dialog_positioner.DialogPositioner;
exports.DialogRoot = require_dialog_root.DialogRoot;
exports.DialogRootProvider = require_dialog_root_provider.DialogRootProvider;
exports.DialogTitle = require_dialog_title.DialogTitle;
exports.DialogTrigger = require_dialog_trigger.DialogTrigger;
Object.defineProperty(exports, "dialogAnatomy", {
	enumerable: true,
	get: function() {
		return _zag_js_dialog.anatomy;
	}
});
exports.useDialog = require_use_dialog.useDialog;
exports.useDialogContext = require_use_dialog_context.useDialogContext;
