// @zag-js/live-region/dist/index.mjs
// src/index.ts
var ID = "__live-region__";
var DEBUG_ID = "__live-region-debug__";
var DEBUG_STYLES = "position:fixed;inset-inline:0;bottom:0;z-index:2147483647;padding:12px 16px;background:black;color:white;font-size:14px;line-height:20px;text-align:center;pointer-events:none;";
function createLiveRegion(opts = {}) {
  const { level = "polite", document: doc = document, root, delay: _delay = 0, debug = false } = opts;
  const win = doc.defaultView ?? window;
  const parent = root ?? doc.body;
  function getDebugRegion() {
    if (!debug) return;
    let region = doc.getElementById(DEBUG_ID);
    if (region) return region;
    region = doc.createElement("div");
    region.id = DEBUG_ID;
    region.dataset.liveAnnouncerDebug = "true";
    region.setAttribute("aria-hidden", "true");
    region.style.cssText = DEBUG_STYLES;
    parent.appendChild(region);
    return region;
  }
  function announce(message, delay) {
    const oldRegion = doc.getElementById(ID);
    oldRegion?.remove();
    delay = delay ?? _delay;
    const region = doc.createElement("span");
    region.id = ID;
    region.dataset.liveAnnouncer = "true";
    const role = level !== "assertive" ? "status" : "alert";
    region.setAttribute("aria-live", level);
    region.setAttribute("role", role);
    Object.assign(region.style, {
      border: "0",
      clip: "rect(0 0 0 0)",
      height: "1px",
      margin: "-1px",
      overflow: "hidden",
      padding: "0",
      position: "absolute",
      width: "1px",
      whiteSpace: "nowrap",
      wordWrap: "normal"
    });
    parent.appendChild(region);
    win.setTimeout(() => {
      if (!region.isConnected) return;
      region.textContent = message;
      const debugRegion = getDebugRegion();
      if (debugRegion) debugRegion.textContent = message;
    }, delay);
  }
  function destroy() {
    const oldRegion = doc.getElementById(ID);
    oldRegion?.remove();
    const debugRegion = doc.getElementById(DEBUG_ID);
    debugRegion?.remove();
  }
  return {
    announce,
    destroy,
    toJSON() {
      return ID;
    }
  };
}
export {
  createLiveRegion
};
