// @zag-js/image-cropper/dist/image-cropper.connect.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/image-cropper.connect.ts
var image_cropper_connect_exports = {};
__export(image_cropper_connect_exports, {
  connect: () => connect
});
module.exports = __toCommonJS(image_cropper_connect_exports);
var import_dom_query = require("@zag-js/dom-query");
var import_utils = require("@zag-js/utils");
var import_get_resize_axis_style = require("./get-resize-axis-style.js");
var import_image_cropper = require("./image-cropper.anatomy.js");
var dom = __toESM(require("./image-cropper.dom.js"));
var import_image_cropper2 = require("./image-cropper.utils.js");
function connect(service, normalize) {
  const { scope, send, context, prop, state, computed } = service;
  const dragging = state.matches("dragging");
  const panning = state.matches("panning");
  const translations = prop("translations");
  const fixedCropArea = prop("fixedCropArea");
  const cropShape = prop("cropShape");
  const zoom = context.get("zoom");
  const rotation = context.get("rotation");
  const flip = context.get("flip");
  const crop = context.get("crop");
  const offset = context.get("offset");
  const naturalSize = context.get("naturalSize");
  const viewportRect = context.get("viewportRect");
  const isImageReady = computed("isImageReady");
  const isMeasured = computed("isMeasured");
  const roundedCrop = (0, import_image_cropper2.roundRect)(crop);
  const shouldIgnoreTouchPointer = (event) => {
    if (event.pointerType !== "touch") return false;
    const isSecondaryTouch = event.isPrimary === false;
    const pinchActive = context.get("pinchDistance") != null;
    return isSecondaryTouch || pinchActive;
  };
  return {
    zoom,
    rotation,
    flip,
    crop,
    offset,
    naturalSize,
    viewportRect,
    dragging,
    panning,
    setZoom(value) {
      send({ type: "SET_ZOOM", zoom: value });
    },
    zoomBy(delta) {
      send({ type: "SET_ZOOM", zoom: zoom + delta });
    },
    setRotation(value) {
      send({ type: "SET_ROTATION", rotation: value });
    },
    rotateBy(degrees) {
      send({ type: "SET_ROTATION", rotation: rotation + degrees });
    },
    setFlip(nextFlip) {
      if (!nextFlip) return;
      const normalized = (0, import_image_cropper2.normalizeFlipState)(nextFlip, flip);
      if ((0, import_image_cropper2.isEqualFlip)(normalized, flip)) return;
      send({ type: "SET_FLIP", flip: normalized });
    },
    flipHorizontally(value) {
      const nextValue = typeof value === "boolean" ? value : !flip.horizontal;
      if (nextValue === flip.horizontal) return;
      send({ type: "SET_FLIP", flip: { horizontal: nextValue } });
    },
    flipVertically(value) {
      const nextValue = typeof value === "boolean" ? value : !flip.vertical;
      if (nextValue === flip.vertical) return;
      send({ type: "SET_FLIP", flip: { vertical: nextValue } });
    },
    resize(handlePosition, delta) {
      if (!handlePosition) return;
      if (fixedCropArea) return;
      let deltaX = 0;
      let deltaY = 0;
      if ((0, import_image_cropper2.isLeftHandle)(handlePosition)) {
        deltaX = -delta;
      } else if ((0, import_image_cropper2.isRightHandle)(handlePosition)) {
        deltaX = delta;
      }
      if ((0, import_image_cropper2.isTopHandle)(handlePosition)) {
        deltaY = -delta;
      } else if ((0, import_image_cropper2.isBottomHandle)(handlePosition)) {
        deltaY = delta;
      }
      send({ type: "RESIZE_CROP", handlePosition, delta: { x: deltaX, y: deltaY } });
    },
    reset() {
      send({ type: "RESET" });
    },
    getCropData() {
      const sourceRect = (0, import_image_cropper2.getCropSourceRect)({
        crop,
        zoom,
        offset,
        viewportSize: viewportRect,
        naturalSize
      });
      return {
        x: Math.round(sourceRect.x),
        y: Math.round(sourceRect.y),
        width: Math.round(sourceRect.width),
        height: Math.round(sourceRect.height),
        rotate: rotation,
        flipX: flip.horizontal,
        flipY: flip.vertical
      };
    },
    async getCroppedImage(options = {}) {
      const { type = "image/png", quality = 1, output = "blob" } = options;
      if (!(0, import_image_cropper2.isVisibleRect)(naturalSize)) return null;
      const canvas = dom.drawCroppedImageToCanvas(service);
      if (!canvas) return null;
      if (output === "dataUrl") {
        return canvas.toDataURL(type, quality);
      }
      return new Promise((resolve) => {
        canvas.toBlob(
          (blob) => {
            resolve(blob);
          },
          type,
          quality
        );
      });
    },
    getRootProps() {
      return normalize.element({
        ...import_image_cropper.parts.root.attrs,
        id: dom.getRootId(scope),
        dir: prop("dir"),
        role: "group",
        "aria-roledescription": translations.rootRoleDescription,
        "aria-label": translations.rootLabel,
        "aria-description": isImageReady ? translations.previewDescription({
          crop: roundedCrop,
          zoom: Number.isFinite(zoom) ? zoom : null,
          rotation: Number.isFinite(rotation) ? rotation : null
        }) : translations.previewLoading,
        "aria-live": "polite",
        "aria-controls": `${dom.getViewportId(scope)} ${dom.getSelectionId(scope)}`,
        "aria-busy": isImageReady ? void 0 : "true",
        "data-fixed": (0, import_dom_query.dataAttr)(fixedCropArea),
        "data-shape": cropShape,
        "data-pinch": (0, import_dom_query.dataAttr)(context.get("pinchDistance") != null),
        "data-dragging": (0, import_dom_query.dataAttr)(dragging),
        "data-panning": (0, import_dom_query.dataAttr)(panning),
        style: {
          "--crop-width": (0, import_utils.toPx)(crop.width),
          "--crop-height": (0, import_utils.toPx)(crop.height),
          "--crop-x": (0, import_utils.toPx)(crop.x),
          "--crop-y": (0, import_utils.toPx)(crop.y),
          "--image-zoom": zoom,
          "--image-rotation": rotation,
          "--image-offset-x": (0, import_utils.toPx)(offset.x),
          "--image-offset-y": (0, import_utils.toPx)(offset.y)
        }
      });
    },
    getViewportProps() {
      const viewportId = dom.getViewportId(scope);
      return normalize.element({
        ...import_image_cropper.parts.viewport.attrs,
        id: viewportId,
        role: "presentation",
        "data-ownedby": dom.getRootId(scope),
        "data-disabled": (0, import_dom_query.dataAttr)(!!fixedCropArea),
        style: {
          position: "relative",
          overflow: "hidden",
          touchAction: "none",
          userSelect: "none"
        },
        onPointerDown(event) {
          if (event.pointerType === "mouse" && event.button !== 0) return;
          if (shouldIgnoreTouchPointer(event)) return;
          const target = (0, import_dom_query.getEventTarget)(event);
          const rootEl = dom.getRootEl(scope);
          if (!target || !rootEl || !(0, import_dom_query.contains)(rootEl, target)) return;
          const selectionEl = dom.getSelectionEl(scope);
          if (!fixedCropArea && (0, import_dom_query.contains)(selectionEl, target)) return;
          const handleEl = target.closest('[data-scope="image-cropper"][data-part="handle"]');
          if (handleEl && (0, import_dom_query.contains)(rootEl, handleEl)) return;
          const point = (0, import_dom_query.getEventPoint)(event);
          send({ type: "PAN_POINTER_DOWN", point });
        }
      });
    },
    getImageProps() {
      const flipHorizontal = flip.horizontal;
      const flipVertical = flip.vertical;
      const translate = `translate(${(0, import_utils.toPx)(offset.x)}, ${(0, import_utils.toPx)(offset.y)})`;
      const rotate = `rotate(${rotation}deg)`;
      const scaleX = zoom * (flipHorizontal ? -1 : 1);
      const scaleY = zoom * (flipVertical ? -1 : 1);
      const scale = `scale(${scaleX}, ${scaleY})`;
      return normalize.element({
        ...import_image_cropper.parts.image.attrs,
        id: dom.getImageId(scope),
        draggable: false,
        role: "presentation",
        alt: "",
        "aria-hidden": true,
        "data-ownedby": dom.getViewportId(scope),
        "data-ready": (0, import_dom_query.dataAttr)(isImageReady),
        "data-flip-horizontal": (0, import_dom_query.dataAttr)(flipHorizontal),
        "data-flip-vertical": (0, import_dom_query.dataAttr)(flipVertical),
        onLoad(event) {
          const imageEl = event.currentTarget;
          if (!imageEl?.complete) return;
          const { naturalWidth: width, naturalHeight: height } = imageEl;
          send({ type: "SET_NATURAL_SIZE", src: "element", size: { width, height } });
        },
        style: {
          pointerEvents: "none",
          userSelect: "none",
          transform: `${translate} ${rotate} ${scale}`,
          willChange: "transform"
        }
      });
    },
    getSelectionProps() {
      const disabled = !!fixedCropArea;
      return normalize.element({
        ...import_image_cropper.parts.selection.attrs,
        id: dom.getSelectionId(scope),
        tabIndex: disabled ? void 0 : 0,
        role: "slider",
        "aria-label": translations.selectionLabel({ shape: cropShape }),
        "aria-roledescription": translations.selectionRoleDescription,
        "aria-disabled": disabled ? "true" : void 0,
        "aria-valuemin": 0,
        "aria-valuemax": (0, import_image_cropper2.isVisibleRect)(viewportRect) ? Math.max(0, Math.round(viewportRect.width - crop.width)) : Math.max(roundedCrop.x, 0),
        "aria-valuenow": roundedCrop.x,
        "aria-valuetext": translations.selectionValueText({ shape: cropShape, ...roundedCrop }),
        "aria-description": translations.selectionInstructions,
        "data-disabled": (0, import_dom_query.dataAttr)(disabled),
        "data-shape": cropShape,
        "data-measured": (0, import_dom_query.dataAttr)(isMeasured),
        "data-dragging": (0, import_dom_query.dataAttr)(dragging),
        "data-panning": (0, import_dom_query.dataAttr)(panning),
        style: {
          position: "absolute",
          top: "var(--crop-y)",
          left: "var(--crop-x)",
          width: "var(--crop-width)",
          height: "var(--crop-height)",
          touchAction: "none",
          visibility: isMeasured ? void 0 : "hidden"
        },
        onPointerDown(event) {
          if (disabled) {
            event.preventDefault();
            return;
          }
          if (shouldIgnoreTouchPointer(event)) return;
          const point = (0, import_dom_query.getEventPoint)(event);
          send({ type: "POINTER_DOWN", point });
        },
        onKeyDown(event) {
          if (disabled) {
            event.preventDefault();
            return;
          }
          if (event.defaultPrevented) return;
          const src = "selection";
          const { shiftKey, ctrlKey, metaKey, altKey } = event;
          const key = (0, import_dom_query.getEventKey)(event, { dir: prop("dir") });
          const isZoomInKey = key === "+" || key === "=";
          const isZoomOutKey = key === "-" || key === "_";
          if (isZoomInKey || isZoomOutKey) {
            const delta = isZoomInKey ? -1 : 1;
            send({ type: "ZOOM", trigger: "keyboard", delta });
            event.preventDefault();
            return;
          }
          if (altKey && (key === "ArrowUp" || key === "ArrowDown" || key === "ArrowLeft" || key === "ArrowRight")) {
            const handlePosition = key === "ArrowUp" || key === "ArrowDown" ? "s" : "e";
            send({
              type: "NUDGE_RESIZE_CROP",
              handlePosition,
              key,
              src,
              shiftKey,
              ctrlKey,
              metaKey
            });
            event.preventDefault();
            return;
          }
          const keyMap = {
            ArrowUp() {
              send({ type: "NUDGE_MOVE_CROP", key: "ArrowUp", src, shiftKey, ctrlKey, metaKey });
            },
            ArrowDown() {
              send({ type: "NUDGE_MOVE_CROP", key: "ArrowDown", src, shiftKey, ctrlKey, metaKey });
            },
            ArrowLeft() {
              send({ type: "NUDGE_MOVE_CROP", key: "ArrowLeft", src, shiftKey, ctrlKey, metaKey });
            },
            ArrowRight() {
              send({ type: "NUDGE_MOVE_CROP", key: "ArrowRight", src, shiftKey, ctrlKey, metaKey });
            }
          };
          const exec = keyMap[key];
          if (exec) {
            exec(event);
            event.preventDefault();
          }
        }
      });
    },
    getHandleProps(props) {
      const handlePosition = props.position;
      const disabled = !!fixedCropArea;
      return normalize.element({
        ...import_image_cropper.parts.handle.attrs,
        id: dom.getHandleId(scope, handlePosition),
        "data-position": handlePosition,
        "aria-hidden": "true",
        role: "presentation",
        "data-disabled": (0, import_dom_query.dataAttr)(disabled),
        style: (0, import_get_resize_axis_style.getHandlePositionStyles)(handlePosition),
        onPointerDown(event) {
          if (disabled) {
            event.preventDefault();
            return;
          }
          if (shouldIgnoreTouchPointer(event)) return;
          const point = (0, import_dom_query.getEventPoint)(event);
          send({ type: "POINTER_DOWN", point, handlePosition });
        }
      });
    },
    getGridProps(props) {
      const axis = props.axis;
      const isMeasured2 = computed("isMeasured");
      return normalize.element({
        ...import_image_cropper.parts.grid.attrs,
        "aria-hidden": "true",
        "data-axis": axis,
        "data-dragging": (0, import_dom_query.dataAttr)(dragging),
        "data-panning": (0, import_dom_query.dataAttr)(panning),
        style: {
          position: "absolute",
          inset: axis === "horizontal" ? "33.33% 0" : "0 33.33%",
          pointerEvents: "none",
          visibility: isMeasured2 ? void 0 : "hidden"
        }
      });
    }
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  connect
});
