// @zag-js/image-cropper/dist/image-cropper.utils.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/image-cropper.utils.ts
var image_cropper_utils_exports = {};
__export(image_cropper_utils_exports, {
  MIN_PINCH_DISTANCE: () => MIN_PINCH_DISTANCE,
  ZERO_POINT: () => ZERO_POINT,
  addPoints: () => addPoints,
  centerCropOnPoint: () => centerCropOnPoint,
  centerRect: () => centerRect,
  clampOffset: () => clampOffset,
  clampPoint: () => clampPoint,
  computeAABB: () => computeAABB,
  computeDefaultCropDimensions: () => computeDefaultCropDimensions,
  computeKeyboardCrop: () => computeKeyboardCrop,
  computeMoveCrop: () => computeMoveCrop,
  computeResizeCrop: () => computeResizeCrop,
  getCenterPoint: () => getCenterPoint,
  getCropSizeLimits: () => getCropSizeLimits,
  getCropSourceRect: () => getCropSourceRect,
  getKeyboardMoveDelta: () => getKeyboardMoveDelta,
  getMaxBounds: () => getMaxBounds,
  getMidpoint: () => getMidpoint,
  getNudgeStep: () => getNudgeStep,
  getRotationTransform: () => getRotationTransform,
  getTouchDistance: () => getTouchDistance,
  getViewportCenter: () => getViewportCenter,
  isAspectRatioEqual: () => isAspectRatioEqual,
  isBottomHandle: () => isBottomHandle,
  isCornerHandle: () => isCornerHandle,
  isEqualFlip: () => isEqualFlip,
  isHorizontalEdgeHandle: () => isHorizontalEdgeHandle,
  isLeftHandle: () => isLeftHandle,
  isRightHandle: () => isRightHandle,
  isSameSize: () => isSameSize,
  isTopHandle: () => isTopHandle,
  isVerticalEdgeHandle: () => isVerticalEdgeHandle,
  isVisibleRect: () => isVisibleRect,
  normalizeFlipState: () => normalizeFlipState,
  resolveCropAspectRatio: () => resolveCropAspectRatio,
  roundRect: () => roundRect,
  scaleRect: () => scaleRect,
  scaleSize: () => scaleSize,
  subtractPoints: () => subtractPoints
});
module.exports = __toCommonJS(image_cropper_utils_exports);
var import_utils = require("@zag-js/utils");
var { min, max, abs, round, hypot, PI, cos, sin } = Math;
var ASPECT_RATIO_TOLERANCE = 1e-3;
var MIN_PINCH_DISTANCE = 1;
var isAspectRatioEqual = (a, b) => {
  return abs(a - b) < ASPECT_RATIO_TOLERANCE;
};
var isLeftHandle = (v) => v === "w" || v === "nw" || v === "sw";
var isRightHandle = (v) => v === "e" || v === "ne" || v === "se";
var isTopHandle = (v) => v === "n" || v === "nw" || v === "ne";
var isBottomHandle = (v) => v === "s" || v === "sw" || v === "se";
var isCornerHandle = (v) => (isLeftHandle(v) || isRightHandle(v)) && (isTopHandle(v) || isBottomHandle(v));
var isHorizontalEdgeHandle = (v) => (isLeftHandle(v) || isRightHandle(v)) && !(isTopHandle(v) || isBottomHandle(v));
var isVerticalEdgeHandle = (v) => (isTopHandle(v) || isBottomHandle(v)) && !(isLeftHandle(v) || isRightHandle(v));
var hasAspectRatio = (value) => typeof value === "number" && value > 0;
var resolveSizeLimits = (options) => {
  const { minSize, maxSize, viewportSize, aspectRatio } = options;
  let minWidth = min(minSize.width, viewportSize.width);
  let minHeight = min(minSize.height, viewportSize.height);
  let maxWidth = maxSize?.width ?? viewportSize.width;
  if (!Number.isFinite(maxWidth)) maxWidth = viewportSize.width;
  maxWidth = min(maxWidth, viewportSize.width);
  let maxHeight = maxSize?.height ?? viewportSize.height;
  if (!Number.isFinite(maxHeight)) maxHeight = viewportSize.height;
  maxHeight = min(maxHeight, viewportSize.height);
  maxWidth = max(minWidth, maxWidth);
  maxHeight = max(minHeight, maxHeight);
  const hasAspect = hasAspectRatio(aspectRatio);
  if (hasAspect) {
    const minWidthWithAspect = max(minWidth, minHeight * aspectRatio);
    const minHeightWithAspect = minWidthWithAspect / aspectRatio;
    minWidth = min(minWidthWithAspect, viewportSize.width);
    minHeight = min(minHeightWithAspect, viewportSize.height);
    let constrainedMaxWidth = min(maxWidth, maxHeight * aspectRatio, viewportSize.width);
    let constrainedMaxHeight = constrainedMaxWidth / aspectRatio;
    if (constrainedMaxHeight > maxHeight || constrainedMaxHeight > viewportSize.height) {
      constrainedMaxHeight = min(maxHeight, viewportSize.height);
      constrainedMaxWidth = constrainedMaxHeight * aspectRatio;
    }
    maxWidth = max(minWidth, min(constrainedMaxWidth, viewportSize.width));
    maxHeight = max(minHeight, min(constrainedMaxHeight, viewportSize.height));
  } else {
    maxWidth = max(minWidth, min(maxWidth, viewportSize.width));
    maxHeight = max(minHeight, min(maxHeight, viewportSize.height));
  }
  return { minWidth, minHeight, maxWidth, maxHeight, hasAspect };
};
var clampAspectSize = (params) => {
  const { widthValue, heightValue, limits, viewportRect, aspectRatio } = params;
  const { minWidth, minHeight, maxWidth, maxHeight } = limits;
  const constrainWidthFromHeight = (height) => {
    let width = (0, import_utils.clampValue)(height * aspectRatio, minWidth, maxWidth);
    width = min(width, viewportRect.width);
    return { width, height: width / aspectRatio };
  };
  const clampByWidth = (value) => {
    let width = (0, import_utils.clampValue)(value, minWidth, maxWidth);
    width = min(width, viewportRect.width);
    let height = width / aspectRatio;
    if (height < minHeight) {
      const constrained = constrainWidthFromHeight(minHeight);
      width = constrained.width;
      height = constrained.height;
    }
    if (height > maxHeight) {
      const clampedHeight = min(maxHeight, viewportRect.height);
      const constrained = constrainWidthFromHeight(clampedHeight);
      width = constrained.width;
      height = constrained.height;
    }
    if (height > viewportRect.height) {
      const constrained = constrainWidthFromHeight(viewportRect.height);
      width = constrained.width;
      height = constrained.height;
      if (height < minHeight) {
        const reconstrainted = constrainWidthFromHeight(minHeight);
        width = reconstrainted.width;
        height = reconstrainted.height;
      }
    }
    return { width, height };
  };
  const clampByHeight = (value) => {
    let height = (0, import_utils.clampValue)(value, minHeight, maxHeight);
    height = min(height, viewportRect.height);
    let width = height * aspectRatio;
    width = (0, import_utils.clampValue)(width, minWidth, maxWidth);
    width = min(width, viewportRect.width);
    let adjustedHeight = width / aspectRatio;
    if (adjustedHeight < minHeight) {
      const constrained = constrainWidthFromHeight(minHeight);
      width = constrained.width;
      adjustedHeight = constrained.height;
    }
    if (adjustedHeight > maxHeight) {
      const clampedHeight = min(maxHeight, viewportRect.height);
      const constrained = constrainWidthFromHeight(clampedHeight);
      width = constrained.width;
      adjustedHeight = constrained.height;
    }
    if (width > viewportRect.width) {
      width = viewportRect.width;
      adjustedHeight = width / aspectRatio;
      if (adjustedHeight > maxHeight) {
        const clampedHeight = min(maxHeight, viewportRect.height);
        const constrained = constrainWidthFromHeight(clampedHeight);
        width = constrained.width;
        adjustedHeight = constrained.height;
      }
      if (adjustedHeight < minHeight) {
        const constrained = constrainWidthFromHeight(minHeight);
        width = constrained.width;
        adjustedHeight = constrained.height;
      }
    }
    return { width, height: adjustedHeight };
  };
  const byWidth = clampByWidth(widthValue);
  const byHeight = clampByHeight(heightValue);
  const deltaWidth = abs(byWidth.width - widthValue) + abs(byWidth.height - heightValue);
  const deltaHeight = abs(byHeight.width - widthValue) + abs(byHeight.height - heightValue);
  return deltaHeight < deltaWidth ? byHeight : byWidth;
};
var applyDeltaToEdges = (params) => {
  const { bounds, delta, handlePosition, viewportRect, minSize, maxSize } = params;
  let { left, top, right, bottom } = bounds;
  if (isLeftHandle(handlePosition)) {
    const minLeft = max(0, right - maxSize.width);
    const maxLeft = right - minSize.width;
    left = (0, import_utils.clampValue)(left + delta.x, minLeft, maxLeft);
  }
  if (isRightHandle(handlePosition)) {
    const minRight = left + minSize.width;
    const maxRight = min(viewportRect.width, left + maxSize.width);
    right = (0, import_utils.clampValue)(right + delta.x, minRight, maxRight);
  }
  if (isTopHandle(handlePosition)) {
    const minTop = max(0, bottom - maxSize.height);
    const maxTop = bottom - minSize.height;
    top = (0, import_utils.clampValue)(top + delta.y, minTop, maxTop);
  }
  if (isBottomHandle(handlePosition)) {
    const minBottom = top + minSize.height;
    const maxBottom = min(viewportRect.height, top + maxSize.height);
    bottom = (0, import_utils.clampValue)(bottom + delta.y, minBottom, maxBottom);
  }
  return { left, top, right, bottom };
};
var applyAspectToHorizontalResize = (params) => {
  const { bounds, limits, viewportRect, aspectRatio, handlePosition } = params;
  const { left, top, right, bottom } = bounds;
  const centerY = (top + bottom) / 2;
  let nextWidth = right - left;
  let nextHeight = nextWidth / aspectRatio;
  const constrained = clampAspectSize({
    widthValue: nextWidth,
    heightValue: nextHeight,
    limits,
    viewportRect,
    aspectRatio
  });
  nextWidth = constrained.width;
  nextHeight = constrained.height;
  const halfH = nextHeight / 2;
  let newTop = centerY - halfH;
  let newBottom = centerY + halfH;
  if (newTop < 0) {
    newTop = 0;
    newBottom = nextHeight;
  }
  if (newBottom > viewportRect.height) {
    newBottom = viewportRect.height;
    newTop = newBottom - nextHeight;
  }
  return {
    left: isRightHandle(handlePosition) ? left : right - nextWidth,
    top: newTop,
    right: isRightHandle(handlePosition) ? left + nextWidth : right,
    bottom: newBottom
  };
};
var applyAspectToVerticalResize = (params) => {
  const { bounds, limits, viewportRect, aspectRatio, handlePosition } = params;
  const { left, top, right, bottom } = bounds;
  const centerX = (left + right) / 2;
  let nextHeight = bottom - top;
  let nextWidth = nextHeight * aspectRatio;
  const constrained = clampAspectSize({
    widthValue: nextWidth,
    heightValue: nextHeight,
    limits,
    viewportRect,
    aspectRatio
  });
  nextWidth = constrained.width;
  nextHeight = constrained.height;
  const halfW = nextWidth / 2;
  let newLeft = centerX - halfW;
  let newRight = centerX + halfW;
  if (newLeft < 0) {
    newLeft = 0;
    newRight = nextWidth;
  }
  if (newRight > viewportRect.width) {
    newRight = viewportRect.width;
    newLeft = newRight - nextWidth;
  }
  return {
    left: newLeft,
    top: isBottomHandle(handlePosition) ? top : bottom - nextHeight,
    right: newRight,
    bottom: isBottomHandle(handlePosition) ? top + nextHeight : bottom
  };
};
var applyCornerResize = (params) => {
  const { bounds, width, height, handlePosition } = params;
  const { left, top, right, bottom } = bounds;
  if (isRightHandle(handlePosition) && isBottomHandle(handlePosition)) {
    return { left, top, right: left + width, bottom: top + height };
  } else if (isRightHandle(handlePosition) && isTopHandle(handlePosition)) {
    return { left, top: bottom - height, right: left + width, bottom };
  } else if (isBottomHandle(handlePosition)) {
    return { left: right - width, top, right, bottom: top + height };
  } else {
    return { left: right - width, top: bottom - height, right, bottom };
  }
};
function computeResizeCrop(options) {
  const { cropStart, handlePosition, delta, viewportRect, minSize, maxSize, aspectRatio } = options;
  let { x, y, width, height } = cropStart;
  let left = x;
  let top = y;
  let right = x + width;
  let bottom = y + height;
  const { minWidth, minHeight, maxWidth, maxHeight, hasAspect } = resolveSizeLimits({
    minSize,
    maxSize,
    viewportSize: viewportRect,
    aspectRatio
  });
  const edgesAfterDelta = applyDeltaToEdges({
    bounds: { left, top, right, bottom },
    delta,
    handlePosition,
    viewportRect,
    minSize,
    maxSize
  });
  left = edgesAfterDelta.left;
  top = edgesAfterDelta.top;
  right = edgesAfterDelta.right;
  bottom = edgesAfterDelta.bottom;
  if (hasAspect) {
    const limits = { minWidth, minHeight, maxWidth, maxHeight, hasAspect };
    if (isCornerHandle(handlePosition)) {
      let tempW = right - left;
      let tempH = tempW / aspectRatio;
      if (tempH > bottom - top || top + tempH > viewportRect.height || left + tempW > viewportRect.width) {
        tempH = bottom - top;
        tempW = tempH * aspectRatio;
      }
      const constrained = clampAspectSize({
        widthValue: tempW,
        heightValue: tempH,
        limits,
        viewportRect,
        aspectRatio
      });
      const result = applyCornerResize({
        bounds: { left, top, right, bottom },
        width: constrained.width,
        height: constrained.height,
        handlePosition
      });
      left = result.left;
      top = result.top;
      right = result.right;
      bottom = result.bottom;
    } else if (isHorizontalEdgeHandle(handlePosition)) {
      const result = applyAspectToHorizontalResize({
        bounds: { left, top, right, bottom },
        limits,
        viewportRect,
        aspectRatio,
        handlePosition
      });
      left = result.left;
      top = result.top;
      right = result.right;
      bottom = result.bottom;
    } else if (isVerticalEdgeHandle(handlePosition)) {
      const result = applyAspectToVerticalResize({
        bounds: { left, top, right, bottom },
        limits,
        viewportRect,
        aspectRatio,
        handlePosition
      });
      left = result.left;
      top = result.top;
      right = result.right;
      bottom = result.bottom;
    }
  }
  const maxLeft = max(0, viewportRect.width - minWidth);
  const maxTop = max(0, viewportRect.height - minHeight);
  left = (0, import_utils.clampValue)(left, 0, maxLeft);
  top = (0, import_utils.clampValue)(top, 0, maxTop);
  const maxRight = min(viewportRect.width, left + maxWidth);
  const maxBottom = min(viewportRect.height, top + maxHeight);
  right = (0, import_utils.clampValue)(right, left + minWidth, maxRight);
  bottom = (0, import_utils.clampValue)(bottom, top + minHeight, maxBottom);
  return {
    x: left,
    y: top,
    width: right - left,
    height: bottom - top
  };
}
function getCropSourceRect(params) {
  const { crop, zoom, offset, viewportSize, naturalSize } = params;
  const scaleX = viewportSize.width > 0 ? naturalSize.width / viewportSize.width : 1;
  const scaleY = viewportSize.height > 0 ? naturalSize.height / viewportSize.height : 1;
  const safeZoom = zoom > 0 ? zoom : 1;
  const viewportCenterX = viewportSize.width / 2;
  const viewportCenterY = viewportSize.height / 2;
  const imageCenterX = naturalSize.width / 2;
  const imageCenterY = naturalSize.height / 2;
  const cropCenterX = crop.x + crop.width / 2;
  const cropCenterY = crop.y + crop.height / 2;
  const sourceCenterX = imageCenterX + (cropCenterX - viewportCenterX - offset.x) / safeZoom * scaleX;
  const sourceCenterY = imageCenterY + (cropCenterY - viewportCenterY - offset.y) / safeZoom * scaleY;
  const sourceWidth = crop.width / safeZoom * scaleX;
  const sourceHeight = crop.height / safeZoom * scaleY;
  return {
    x: sourceCenterX - sourceWidth / 2,
    y: sourceCenterY - sourceHeight / 2,
    width: sourceWidth,
    height: sourceHeight
  };
}
function computeMoveCrop(cropStart, delta, viewportRect) {
  return {
    x: (0, import_utils.clampValue)(cropStart.x + delta.x, 0, viewportRect.width - cropStart.width),
    y: (0, import_utils.clampValue)(cropStart.y + delta.y, 0, viewportRect.height - cropStart.height),
    width: cropStart.width,
    height: cropStart.height
  };
}
function clampOffset(params) {
  const { zoom, rotation, viewportSize, offset, fixedCropArea, crop } = params;
  const { cos: cos2, sin: sin2 } = getRotationTransform(rotation);
  if (fixedCropArea && crop) {
    const aabb2 = computeAABB(viewportSize, zoom, cos2, sin2);
    const center = getViewportCenter(viewportSize);
    const cropRight = crop.x + crop.width;
    const cropBottom = crop.y + crop.height;
    const minPoint2 = {
      x: cropRight - center.x - aabb2.width / 2,
      y: cropBottom - center.y - aabb2.height / 2
    };
    const maxPoint2 = {
      x: crop.x - center.x + aabb2.width / 2,
      y: crop.y - center.y + aabb2.height / 2
    };
    return clampPoint(offset, minPoint2, maxPoint2);
  }
  const aabb = computeAABB(viewportSize, zoom, cos2, sin2);
  const extraWidth = max(0, aabb.width - viewportSize.width);
  const extraHeight = max(0, aabb.height - viewportSize.height);
  const minPoint = { x: -extraWidth / 2, y: -extraHeight / 2 };
  const maxPoint = { x: extraWidth / 2, y: extraHeight / 2 };
  return clampPoint(offset, minPoint, maxPoint);
}
var expandLeft = (crop, step, maxWidth) => {
  const newX = max(0, crop.x - step);
  const newWidth = crop.width + (crop.x - newX);
  if (newWidth <= maxWidth) {
    return { x: newX, width: newWidth };
  }
  return { x: crop.x + crop.width - maxWidth, width: maxWidth };
};
var expandTop = (crop, step, maxHeight) => {
  const newY = max(0, crop.y - step);
  const newHeight = crop.height + (crop.y - newY);
  if (newHeight <= maxHeight) {
    return { y: newY, height: newHeight };
  }
  return { y: crop.y + crop.height - maxHeight, height: maxHeight };
};
var shrinkFromLeft = (crop, step, minWidth) => {
  const newX = min(crop.x + step, crop.x + crop.width - minWidth);
  return { x: newX, width: crop.width - (newX - crop.x) };
};
var shrinkFromTop = (crop, step, minHeight) => {
  const newY = min(crop.y + step, crop.y + crop.height - minHeight);
  return { y: newY, height: crop.height - (newY - crop.y) };
};
function computeKeyboardCrop(key, handlePosition, step, crop, viewportRect, minSize, maxSize) {
  const nextCrop = { ...crop };
  const { minWidth, minHeight, maxWidth, maxHeight } = resolveSizeLimits({
    minSize,
    maxSize,
    viewportSize: viewportRect
  });
  const isCorner = isCornerHandle(handlePosition);
  if (key === "ArrowLeft") {
    if (isLeftHandle(handlePosition)) {
      const expanded = expandLeft(crop, step, maxWidth);
      nextCrop.x = expanded.x;
      nextCrop.width = expanded.width;
      if (isCorner && isTopHandle(handlePosition)) {
        const expandedY = expandTop(crop, step, maxHeight);
        nextCrop.y = expandedY.y;
        nextCrop.height = expandedY.height;
      } else if (isCorner && isBottomHandle(handlePosition)) {
        const newHeight = nextCrop.height + step;
        nextCrop.height = min(viewportRect.height - nextCrop.y, min(maxHeight, newHeight));
      }
    } else if (isRightHandle(handlePosition)) {
      nextCrop.width = max(minWidth, nextCrop.width - step);
      if (isCorner && isTopHandle(handlePosition)) {
        const shrunk = shrinkFromTop(crop, step, minHeight);
        nextCrop.y = shrunk.y;
        nextCrop.height = shrunk.height;
      } else if (isCorner && isBottomHandle(handlePosition)) {
        nextCrop.height = max(minHeight, nextCrop.height - step);
      }
    }
  } else if (key === "ArrowRight") {
    if (isLeftHandle(handlePosition)) {
      const shrunk = shrinkFromLeft(crop, step, minWidth);
      nextCrop.x = shrunk.x;
      nextCrop.width = shrunk.width;
      if (isCorner && isTopHandle(handlePosition)) {
        const shrunkY = shrinkFromTop(crop, step, minHeight);
        nextCrop.y = shrunkY.y;
        nextCrop.height = shrunkY.height;
      } else if (isCorner && isBottomHandle(handlePosition)) {
        nextCrop.height = max(minHeight, nextCrop.height - step);
      }
    } else if (isRightHandle(handlePosition)) {
      const newWidth = nextCrop.width + step;
      nextCrop.width = min(viewportRect.width - nextCrop.x, min(maxWidth, newWidth));
      if (isCorner && isTopHandle(handlePosition)) {
        const expanded = expandTop(crop, step, maxHeight);
        nextCrop.y = expanded.y;
        nextCrop.height = expanded.height;
      } else if (isCorner && isBottomHandle(handlePosition)) {
        const newHeight = nextCrop.height + step;
        nextCrop.height = min(viewportRect.height - nextCrop.y, min(maxHeight, newHeight));
      }
    }
  }
  if (key === "ArrowUp") {
    if (isTopHandle(handlePosition)) {
      const expanded = expandTop(crop, step, maxHeight);
      nextCrop.y = expanded.y;
      nextCrop.height = expanded.height;
      if (isCorner && isLeftHandle(handlePosition)) {
        const expandedX = expandLeft(crop, step, maxWidth);
        nextCrop.x = expandedX.x;
        nextCrop.width = expandedX.width;
      } else if (isCorner && isRightHandle(handlePosition)) {
        const newWidth = nextCrop.width + step;
        nextCrop.width = min(viewportRect.width - nextCrop.x, min(maxWidth, newWidth));
      }
    } else if (isBottomHandle(handlePosition)) {
      nextCrop.height = max(minHeight, nextCrop.height - step);
      if (isCorner && isLeftHandle(handlePosition)) {
        const shrunk = shrinkFromLeft(crop, step, minWidth);
        nextCrop.x = shrunk.x;
        nextCrop.width = shrunk.width;
      } else if (isCorner && isRightHandle(handlePosition)) {
        nextCrop.width = max(minWidth, nextCrop.width - step);
      }
    }
  } else if (key === "ArrowDown") {
    if (isTopHandle(handlePosition)) {
      const shrunk = shrinkFromTop(crop, step, minHeight);
      nextCrop.y = shrunk.y;
      nextCrop.height = shrunk.height;
      if (isCorner && isLeftHandle(handlePosition)) {
        const shrunkX = shrinkFromLeft(crop, step, minWidth);
        nextCrop.x = shrunkX.x;
        nextCrop.width = shrunkX.width;
      } else if (isCorner && isRightHandle(handlePosition)) {
        nextCrop.width = max(minWidth, nextCrop.width - step);
      }
    } else if (isBottomHandle(handlePosition)) {
      const newHeight = nextCrop.height + step;
      nextCrop.height = min(viewportRect.height - nextCrop.y, min(maxHeight, newHeight));
      if (isCorner && isLeftHandle(handlePosition)) {
        const expanded = expandLeft(crop, step, maxWidth);
        nextCrop.x = expanded.x;
        nextCrop.width = expanded.width;
      } else if (isCorner && isRightHandle(handlePosition)) {
        const newWidth = nextCrop.width + step;
        nextCrop.width = min(viewportRect.width - nextCrop.x, min(maxWidth, newWidth));
      }
    }
  }
  return nextCrop;
}
function getKeyboardMoveDelta(key, step) {
  switch (key) {
    case "ArrowLeft":
      return { x: -step, y: 0 };
    case "ArrowRight":
      return { x: step, y: 0 };
    case "ArrowUp":
      return { x: 0, y: -step };
    case "ArrowDown":
      return { x: 0, y: step };
    default:
      return ZERO_POINT;
  }
}
var resolveCropAspectRatio = (shape, aspectRatio) => shape === "circle" ? 1 : aspectRatio;
var getCropSizeLimits = (prop) => ({
  minSize: { width: prop("minWidth"), height: prop("minHeight") },
  maxSize: { width: prop("maxWidth"), height: prop("maxHeight") }
});
var getNudgeStep = (prop, modifiers) => {
  if (modifiers.ctrlKey || modifiers.metaKey) return prop("nudgeStepCtrl");
  if (modifiers.shiftKey) return prop("nudgeStepShift");
  return prop("nudgeStep");
};
var DEFAULT_VIEWPORT_FILL = 0.8;
var computeDefaultCropDimensions = (viewportRect, aspectRatio, fixedCropArea) => {
  const targetWidth = viewportRect.width * DEFAULT_VIEWPORT_FILL;
  const targetHeight = viewportRect.height * DEFAULT_VIEWPORT_FILL;
  if (typeof aspectRatio === "number" && aspectRatio > 0) {
    if (fixedCropArea) {
      let height2 = viewportRect.height;
      let width2 = height2 * aspectRatio;
      if (width2 > viewportRect.width) {
        width2 = viewportRect.width;
        height2 = width2 / aspectRatio;
      }
      return { width: width2, height: height2 };
    }
    const targetAspect = targetWidth / targetHeight;
    if (aspectRatio > targetAspect) {
      const width2 = targetWidth;
      const height2 = width2 / aspectRatio;
      return { width: width2, height: height2 };
    }
    const height = targetHeight;
    const width = height * aspectRatio;
    return { width, height };
  }
  if (fixedCropArea) {
    const size = min(viewportRect.width, viewportRect.height);
    return { width: size, height: size };
  }
  return { width: targetWidth, height: targetHeight };
};
var normalizeFlipState = (nextFlip, currentFlip) => {
  if (!nextFlip) return currentFlip;
  return {
    horizontal: (0, import_utils.isBoolean)(nextFlip.horizontal) ? nextFlip.horizontal : currentFlip.horizontal,
    vertical: (0, import_utils.isBoolean)(nextFlip.vertical) ? nextFlip.vertical : currentFlip.vertical
  };
};
var isEqualFlip = (a, b) => {
  return a.horizontal === b.horizontal && a.vertical === b.vertical;
};
var isVisibleRect = (rect) => rect.width > 0 && rect.height > 0;
var getCenterPoint = (rect) => ({
  x: rect.x + rect.width / 2,
  y: rect.y + rect.height / 2
});
var getViewportCenter = (size) => ({
  x: size.width / 2,
  y: size.height / 2
});
var centerRect = (size, viewport) => ({
  x: max(0, (viewport.width - size.width) / 2),
  y: max(0, (viewport.height - size.height) / 2)
});
var getMidpoint = (p1, p2, offset = ZERO_POINT) => ({
  x: (p1.x + p2.x) / 2 - offset.x,
  y: (p1.y + p2.y) / 2 - offset.y
});
var getMaxBounds = (cropSize, viewportSize) => ({
  x: max(0, viewportSize.width - cropSize.width),
  y: max(0, viewportSize.height - cropSize.height)
});
var centerCropOnPoint = (cropSize, center, viewportSize) => {
  const newPos = {
    x: center.x - cropSize.width / 2,
    y: center.y - cropSize.height / 2
  };
  const maxBounds = getMaxBounds(cropSize, viewportSize);
  return clampPoint(newPos, ZERO_POINT, maxBounds);
};
var isSameSize = (a, b) => {
  return a.width === b.width && a.height === b.height;
};
var ZERO_POINT = { x: 0, y: 0 };
var getTouchDistance = (p1, p2) => {
  const dx = p1.x - p2.x;
  const dy = p1.y - p2.y;
  return hypot(dx, dy);
};
var clampPoint = (point, min2, max2) => ({
  x: (0, import_utils.clampValue)(point.x, min2.x, max2.x),
  y: (0, import_utils.clampValue)(point.y, min2.y, max2.y)
});
var subtractPoints = (a, b) => ({
  x: a.x - b.x,
  y: a.y - b.y
});
var addPoints = (a, b) => ({
  x: a.x + b.x,
  y: a.y + b.y
});
var roundRect = (rect) => ({
  x: round(rect.x),
  y: round(rect.y),
  width: round(rect.width),
  height: round(rect.height)
});
var scaleRect = (rect, scale) => ({
  x: rect.x * scale.x,
  y: rect.y * scale.y,
  width: rect.width * scale.x,
  height: rect.height * scale.y
});
var getRotationTransform = (rotation) => {
  const theta = rotation % 360 * PI / 180;
  return {
    cos: abs(cos(theta)),
    sin: abs(sin(theta))
  };
};
var computeAABB = (size, zoom, cos2, sin2) => {
  const w = size.width * zoom;
  const h = size.height * zoom;
  return {
    width: w * cos2 + h * sin2,
    height: w * sin2 + h * cos2
  };
};
var scaleSize = (size, scale) => ({
  width: size.width * scale,
  height: size.height * scale
});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  MIN_PINCH_DISTANCE,
  ZERO_POINT,
  addPoints,
  centerCropOnPoint,
  centerRect,
  clampOffset,
  clampPoint,
  computeAABB,
  computeDefaultCropDimensions,
  computeKeyboardCrop,
  computeMoveCrop,
  computeResizeCrop,
  getCenterPoint,
  getCropSizeLimits,
  getCropSourceRect,
  getKeyboardMoveDelta,
  getMaxBounds,
  getMidpoint,
  getNudgeStep,
  getRotationTransform,
  getTouchDistance,
  getViewportCenter,
  isAspectRatioEqual,
  isBottomHandle,
  isCornerHandle,
  isEqualFlip,
  isHorizontalEdgeHandle,
  isLeftHandle,
  isRightHandle,
  isSameSize,
  isTopHandle,
  isVerticalEdgeHandle,
  isVisibleRect,
  normalizeFlipState,
  resolveCropAspectRatio,
  roundRect,
  scaleRect,
  scaleSize,
  subtractPoints
});
