// @zag-js/image-cropper/dist/image-cropper.dom.mjs
// src/image-cropper.dom.ts
import { getCropSourceRect } from "./image-cropper.utils.mjs";
var getRootId = (ctx) => ctx.ids?.root ?? `image-cropper:${ctx.id}`;
var getViewportId = (ctx) => ctx.ids?.viewport ?? `image-cropper:${ctx.id}:viewport`;
var getImageId = (ctx) => ctx.ids?.image ?? `image-cropper:${ctx.id}:image`;
var getSelectionId = (ctx) => ctx.ids?.selection ?? `image-cropper:${ctx.id}:selection`;
var getHandleId = (ctx, position) => ctx.ids?.handle?.(position) ?? `image-cropper:${ctx.id}:handle:${position}`;
var getRootEl = (ctx) => ctx.getById(getRootId(ctx));
var getViewportEl = (ctx) => ctx.getById(getViewportId(ctx));
var getImageEl = (ctx) => ctx.getById(getImageId(ctx));
var getSelectionEl = (ctx) => ctx.getById(getSelectionId(ctx));
var getHandleEl = (ctx, position) => ctx.getById(getHandleId(ctx, position));
function drawCroppedImageToCanvas(params) {
  const { context, scope } = params;
  const imageEl = getImageEl(scope);
  if (!imageEl || !imageEl.complete) return null;
  const doc = imageEl.ownerDocument;
  const crop = context.get("crop");
  const zoom = context.get("zoom");
  const rotation = context.get("rotation");
  const flip = context.get("flip");
  const viewportRect = context.get("viewportRect");
  const naturalSize = context.get("naturalSize");
  const offset = context.get("offset");
  const canvas = doc.createElement("canvas");
  canvas.width = crop.width;
  canvas.height = crop.height;
  const ctx = canvas.getContext("2d");
  if (!ctx) return null;
  ctx.save();
  ctx.translate(canvas.width / 2, canvas.height / 2);
  ctx.rotate(rotation * Math.PI / 180);
  const scaleX = flip.horizontal ? -1 : 1;
  const scaleY = flip.vertical ? -1 : 1;
  ctx.scale(scaleX, scaleY);
  const sourceRect = getCropSourceRect({
    crop,
    zoom,
    offset,
    viewportSize: viewportRect,
    naturalSize
  });
  ctx.drawImage(
    imageEl,
    sourceRect.x,
    sourceRect.y,
    sourceRect.width,
    sourceRect.height,
    -canvas.width / 2,
    -canvas.height / 2,
    canvas.width,
    canvas.height
  );
  ctx.restore();
  return canvas;
}
export {
  drawCroppedImageToCanvas,
  getHandleEl,
  getHandleId,
  getImageEl,
  getImageId,
  getRootEl,
  getRootId,
  getSelectionEl,
  getSelectionId,
  getViewportEl,
  getViewportId
};
