// @zag-js/image-cropper/dist/image-cropper.dom.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/image-cropper.dom.ts
var image_cropper_dom_exports = {};
__export(image_cropper_dom_exports, {
  drawCroppedImageToCanvas: () => drawCroppedImageToCanvas,
  getHandleEl: () => getHandleEl,
  getHandleId: () => getHandleId,
  getImageEl: () => getImageEl,
  getImageId: () => getImageId,
  getRootEl: () => getRootEl,
  getRootId: () => getRootId,
  getSelectionEl: () => getSelectionEl,
  getSelectionId: () => getSelectionId,
  getViewportEl: () => getViewportEl,
  getViewportId: () => getViewportId
});
module.exports = __toCommonJS(image_cropper_dom_exports);
var import_image_cropper = require("./image-cropper.utils.js");
var getRootId = (ctx) => ctx.ids?.root ?? `image-cropper:${ctx.id}`;
var getViewportId = (ctx) => ctx.ids?.viewport ?? `image-cropper:${ctx.id}:viewport`;
var getImageId = (ctx) => ctx.ids?.image ?? `image-cropper:${ctx.id}:image`;
var getSelectionId = (ctx) => ctx.ids?.selection ?? `image-cropper:${ctx.id}:selection`;
var getHandleId = (ctx, position) => ctx.ids?.handle?.(position) ?? `image-cropper:${ctx.id}:handle:${position}`;
var getRootEl = (ctx) => ctx.getById(getRootId(ctx));
var getViewportEl = (ctx) => ctx.getById(getViewportId(ctx));
var getImageEl = (ctx) => ctx.getById(getImageId(ctx));
var getSelectionEl = (ctx) => ctx.getById(getSelectionId(ctx));
var getHandleEl = (ctx, position) => ctx.getById(getHandleId(ctx, position));
function drawCroppedImageToCanvas(params) {
  const { context, scope } = params;
  const imageEl = getImageEl(scope);
  if (!imageEl || !imageEl.complete) return null;
  const doc = imageEl.ownerDocument;
  const crop = context.get("crop");
  const zoom = context.get("zoom");
  const rotation = context.get("rotation");
  const flip = context.get("flip");
  const viewportRect = context.get("viewportRect");
  const naturalSize = context.get("naturalSize");
  const offset = context.get("offset");
  const canvas = doc.createElement("canvas");
  canvas.width = crop.width;
  canvas.height = crop.height;
  const ctx = canvas.getContext("2d");
  if (!ctx) return null;
  ctx.save();
  ctx.translate(canvas.width / 2, canvas.height / 2);
  ctx.rotate(rotation * Math.PI / 180);
  const scaleX = flip.horizontal ? -1 : 1;
  const scaleY = flip.vertical ? -1 : 1;
  ctx.scale(scaleX, scaleY);
  const sourceRect = (0, import_image_cropper.getCropSourceRect)({
    crop,
    zoom,
    offset,
    viewportSize: viewportRect,
    naturalSize
  });
  ctx.drawImage(
    imageEl,
    sourceRect.x,
    sourceRect.y,
    sourceRect.width,
    sourceRect.height,
    -canvas.width / 2,
    -canvas.height / 2,
    canvas.width,
    canvas.height
  );
  ctx.restore();
  return canvas;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  drawCroppedImageToCanvas,
  getHandleEl,
  getHandleId,
  getImageEl,
  getImageId,
  getRootEl,
  getRootId,
  getSelectionEl,
  getSelectionId,
  getViewportEl,
  getViewportId
});
