// @zag-js/image-cropper/dist/image-cropper.machine.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/image-cropper.machine.ts
var image_cropper_machine_exports = {};
__export(image_cropper_machine_exports, {
  machine: () => machine
});
module.exports = __toCommonJS(image_cropper_machine_exports);
var import_core = require("@zag-js/core");
var import_dom_query = require("@zag-js/dom-query");
var import_utils = require("@zag-js/utils");
var dom = __toESM(require("./image-cropper.dom.js"));
var import_image_cropper = require("./image-cropper.utils.js");
var machine = (0, import_core.createMachine)({
  props({ props }) {
    return {
      minWidth: 40,
      minHeight: 40,
      maxWidth: Number.POSITIVE_INFINITY,
      maxHeight: Number.POSITIVE_INFINITY,
      defaultZoom: 1,
      zoomStep: 0.1,
      zoomSensitivity: 2,
      minZoom: 1,
      maxZoom: 5,
      defaultRotation: 0,
      defaultFlip: { horizontal: false, vertical: false },
      fixedCropArea: false,
      cropShape: "rectangle",
      nudgeStep: 1,
      nudgeStepShift: 10,
      nudgeStepCtrl: 50,
      ...props,
      translations: {
        rootLabel: "Image cropper",
        rootRoleDescription: "Image cropper",
        previewLoading: "Image cropper preview loading",
        previewDescription({ crop, zoom, rotation }) {
          const zoomText = zoom != null && Number.isFinite(zoom) ? `${zoom.toFixed(2)}x zoom` : "default zoom";
          const rotationText = rotation != null && Number.isFinite(rotation) ? `${Math.round(rotation)} degrees rotation` : "0 degrees rotation";
          return `Image cropper preview, ${zoomText}, ${rotationText}. Crop positioned at ${crop.x}px from the left and ${crop.y}px from the top with a size of ${crop.width}px by ${crop.height}px.`;
        },
        selectionLabel: ({ shape }) => `Crop selection area (${shape === "circle" ? "circle" : "rectangle"})`,
        selectionRoleDescription: "2d slider",
        selectionInstructions: "Use arrow keys to move the crop. Hold Alt with arrow keys to resize width or height. Press plus or minus to zoom.",
        selectionValueText({ shape, x, y, width, height }) {
          if (shape === "circle") {
            return `Position X ${x}px, Y ${y}px. Diameter ${width}px.`;
          }
          return `Position X ${x}px, Y ${y}px. Size ${width}px by ${height}px.`;
        },
        ...props.translations
      }
    };
  },
  context({ bindable, prop }) {
    return {
      naturalSize: bindable(() => ({
        defaultValue: { width: 0, height: 0 }
      })),
      crop: bindable(() => ({
        defaultValue: { x: 0, y: 0, width: 0, height: 0 },
        onChange(crop) {
          prop("onCropChange")?.({ crop });
        }
      })),
      pointerStart: bindable(() => ({
        defaultValue: null
      })),
      cropStart: bindable(() => ({
        defaultValue: null
      })),
      handlePosition: bindable(() => ({
        defaultValue: null
      })),
      shiftLockRatio: bindable(() => ({
        defaultValue: null
      })),
      pinchDistance: bindable(() => ({
        defaultValue: null
      })),
      pinchMidpoint: bindable(() => ({
        defaultValue: null
      })),
      zoom: bindable(() => ({
        defaultValue: prop("zoom") ?? prop("defaultZoom"),
        onChange(zoom) {
          prop("onZoomChange")?.({ zoom });
        }
      })),
      rotation: bindable(() => ({
        defaultValue: prop("defaultRotation"),
        value: prop("rotation"),
        onChange(rotation) {
          prop("onRotationChange")?.({ rotation });
        }
      })),
      flip: bindable(() => {
        const defaultFlip = prop("defaultFlip");
        return {
          defaultValue: { ...defaultFlip },
          value: prop("flip"),
          onChange(flip) {
            prop("onFlipChange")?.({ flip });
          }
        };
      }),
      offset: bindable(() => ({
        defaultValue: import_image_cropper.ZERO_POINT
      })),
      offsetStart: bindable(() => ({
        defaultValue: null
      })),
      viewportRect: bindable(() => ({
        defaultValue: { width: 0, height: 0, top: 0, left: 0, right: 0, bottom: 0 }
      }))
    };
  },
  initialState() {
    return "idle";
  },
  on: {
    PINCH_START: {
      actions: ["setPinchDistance"]
    },
    PINCH_MOVE: {
      actions: ["handlePinchMove"]
    },
    PINCH_END: {
      actions: ["clearPinchDistance"]
    },
    SET_ZOOM: {
      actions: ["updateZoom"]
    },
    SET_ROTATION: {
      actions: ["setRotation"]
    },
    SET_FLIP: {
      actions: ["setFlip"]
    },
    RESIZE_CROP: {
      guard: "canResizeCrop",
      actions: ["resizeCrop"]
    },
    VIEWPORT_RESIZE: {
      actions: ["resizeViewport"]
    },
    RESET: {
      actions: ["resetToInitialState"]
    },
    ADJUST_ASPECT_RATIO: {
      guard: "hasViewportRect",
      actions: ["adjustCropAspectRatio"]
    }
  },
  computed: {
    isMeasured: ({ context }) => (0, import_image_cropper.isVisibleRect)(context.get("viewportRect")) && (0, import_image_cropper.isVisibleRect)(context.get("crop")),
    isImageReady: ({ context }) => (0, import_image_cropper.isVisibleRect)(context.get("naturalSize"))
  },
  watch({ track, context, prop, send }) {
    track([() => prop("zoom")], () => {
      const propZoom = prop("zoom");
      if (propZoom === void 0) return;
      const currentZoom = context.get("zoom");
      if (propZoom === currentZoom) return;
      send({ type: "SET_ZOOM", zoom: propZoom, src: "prop" });
    });
    track([() => prop("aspectRatio"), () => prop("cropShape")], () => {
      send({ type: "ADJUST_ASPECT_RATIO", src: "prop" });
    });
  },
  states: {
    idle: {
      entry: ["checkImageStatus"],
      effects: ["trackViewportResize", "trackWheelEvent", "trackTouchEvents"],
      on: {
        SET_NATURAL_SIZE: {
          actions: ["setNaturalSize"]
        },
        SET_DEFAULT_CROP: {
          actions: ["setDefaultCrop"]
        },
        POINTER_DOWN: {
          guard: "canDragSelection",
          target: "dragging",
          actions: ["setPointerStart", "setCropStart", "setHandlePosition"]
        },
        PAN_POINTER_DOWN: {
          guard: "canPan",
          target: "panning",
          actions: ["setPointerStart", "setOffsetStart"]
        },
        ZOOM: {
          guard: "hasViewportRect",
          actions: ["updateZoom"]
        },
        NUDGE_RESIZE_CROP: {
          guard: "hasViewportRect",
          actions: ["nudgeResizeCrop"]
        },
        NUDGE_MOVE_CROP: {
          guard: "hasViewportRect",
          actions: ["nudgeMoveCrop"]
        }
      }
    },
    dragging: {
      effects: ["trackPointerMove"],
      on: {
        POINTER_MOVE: {
          actions: ["updateCrop"]
        },
        POINTER_UP: {
          target: "idle",
          actions: [
            "clearPointerStart",
            "clearCropStart",
            "clearHandlePosition",
            "clearOffsetStart",
            "clearShiftRatio"
          ]
        }
      }
    },
    panning: {
      effects: ["trackPointerMove"],
      on: {
        POINTER_MOVE: {
          actions: ["updatePanOffset"]
        },
        POINTER_UP: {
          target: "idle",
          actions: ["clearPointerStart", "clearOffsetStart"]
        }
      }
    }
  },
  implementations: {
    guards: {
      hasViewportRect({ context }) {
        return (0, import_image_cropper.isVisibleRect)(context.get("viewportRect"));
      },
      canResizeCrop({ context, prop }) {
        return !prop("fixedCropArea") && (0, import_image_cropper.isVisibleRect)(context.get("viewportRect"));
      },
      canPan({ context }) {
        return (0, import_image_cropper.isVisibleRect)(context.get("naturalSize")) && (0, import_image_cropper.isVisibleRect)(context.get("viewportRect"));
      },
      canDragSelection({ context, prop }) {
        return (0, import_image_cropper.isVisibleRect)(context.get("viewportRect")) && !prop("fixedCropArea");
      }
    },
    actions: {
      checkImageStatus({ send, scope, context }) {
        const naturalSize = context.get("naturalSize");
        const imageEl = dom.getImageEl(scope);
        if (!imageEl?.complete) return;
        const { naturalWidth: width, naturalHeight: height } = imageEl;
        if ((0, import_image_cropper.isVisibleRect)({ width, height }) && !(0, import_image_cropper.isVisibleRect)(naturalSize)) {
          send({ type: "SET_NATURAL_SIZE", src: "ssr", size: { width, height } });
        }
      },
      setNaturalSize({ event, context, send }) {
        context.set("naturalSize", event.size);
        send({ type: "SET_DEFAULT_CROP", src: "init" });
      },
      setDefaultCrop({ context, prop, scope }) {
        const viewportEl = dom.getViewportEl(scope);
        if (!viewportEl) return;
        const viewportRect = getBoundingRect(viewportEl);
        if (!(0, import_image_cropper.isVisibleRect)(viewportRect)) return;
        const cropShape = prop("cropShape");
        const aspectRatio = (0, import_image_cropper.resolveCropAspectRatio)(cropShape, prop("aspectRatio"));
        const { minSize, maxSize } = (0, import_image_cropper.getCropSizeLimits)(prop);
        const clampSize = (rect) => {
          const result = (0, import_image_cropper.computeResizeCrop)({
            cropStart: rect,
            handlePosition: "se",
            delta: import_image_cropper.ZERO_POINT,
            viewportRect,
            minSize,
            maxSize,
            aspectRatio
          });
          return { width: result.width, height: result.height };
        };
        const initialCrop = prop("initialCrop");
        if (initialCrop) {
          const constrainedSize2 = clampSize({
            x: 0,
            y: 0,
            width: initialCrop.width,
            height: initialCrop.height
          });
          const { width: width2, height: height2 } = constrainedSize2;
          const max = (0, import_image_cropper.getMaxBounds)({ width: width2, height: height2 }, viewportRect);
          const { x: x2, y: y2 } = (0, import_image_cropper.clampPoint)(initialCrop, import_image_cropper.ZERO_POINT, max);
          context.set("crop", { x: x2, y: y2, width: width2, height: height2 });
          return;
        }
        const fixedCropArea = prop("fixedCropArea");
        const defaultSize = (0, import_image_cropper.computeDefaultCropDimensions)(viewportRect, aspectRatio, fixedCropArea);
        const constrainedSize = clampSize({
          x: 0,
          y: 0,
          width: defaultSize.width,
          height: defaultSize.height
        });
        const width = constrainedSize.width;
        const height = constrainedSize.height;
        const { x, y } = (0, import_image_cropper.centerRect)({ width, height }, viewportRect);
        context.set("crop", { x, y, width, height });
        context.set("viewportRect", viewportRect);
      },
      setPointerStart({ event, context }) {
        const point = event.point;
        if (!point) return;
        context.set("pointerStart", point);
      },
      setOffsetStart({ context }) {
        const offset = context.get("offset");
        context.set("offsetStart", { ...offset });
      },
      setCropStart({ context }) {
        const crop = context.get("crop");
        context.set("cropStart", crop);
      },
      updateCrop({ context, event, prop }) {
        const handlePosition = context.get("handlePosition");
        const pointerStart = context.get("pointerStart");
        const cropStart = context.get("cropStart");
        const viewportRect = context.get("viewportRect");
        const cropShape = prop("cropShape");
        const aspectRatioProp = prop("aspectRatio");
        let aspectRatio = (0, import_image_cropper.resolveCropAspectRatio)(cropShape, aspectRatioProp);
        const { minSize, maxSize } = (0, import_image_cropper.getCropSizeLimits)(prop);
        if (!pointerStart || !cropStart) return;
        const currentPoint = event.point;
        let delta = (0, import_image_cropper.subtractPoints)(currentPoint, pointerStart);
        let nextCrop;
        if (handlePosition) {
          const allowShiftLock = typeof aspectRatioProp === "undefined" && cropShape !== "circle";
          if (allowShiftLock) {
            if (event.shiftKey) {
              const currentCrop = context.get("crop");
              const w = currentCrop.width;
              const h = currentCrop.height;
              if (w > 0 && h > 0) {
                const ratio = w / h;
                if (ratio > 0) context.set("shiftLockRatio", ratio);
              }
              const lockRatio = context.get("shiftLockRatio");
              if (lockRatio !== null && lockRatio > 0) aspectRatio = lockRatio;
            } else {
              context.set("shiftLockRatio", null);
            }
          } else {
            context.set("shiftLockRatio", null);
          }
          if (event.altKey) {
            delta = { x: delta.x * 2, y: delta.y * 2 };
          }
          nextCrop = (0, import_image_cropper.computeResizeCrop)({
            cropStart,
            handlePosition,
            delta,
            viewportRect,
            minSize,
            maxSize,
            aspectRatio
          });
          if (event.altKey) {
            const originalCenter = (0, import_image_cropper.getCenterPoint)(cropStart);
            const pos = (0, import_image_cropper.centerCropOnPoint)(nextCrop, originalCenter, viewportRect);
            nextCrop = { ...nextCrop, x: pos.x, y: pos.y };
          }
        } else {
          nextCrop = (0, import_image_cropper.computeMoveCrop)(cropStart, delta, viewportRect);
        }
        context.set("crop", nextCrop);
      },
      updatePanOffset({ context, event, prop }) {
        const point = event.point;
        const pointerStart = context.get("pointerStart");
        const offsetStart = context.get("offsetStart");
        if (!point || !pointerStart || !offsetStart) return;
        const zoom = context.get("zoom");
        const rotation = context.get("rotation");
        const viewportRect = context.get("viewportRect");
        const delta = (0, import_image_cropper.subtractPoints)(point, pointerStart);
        const nextOffset = (0, import_image_cropper.clampOffset)({
          zoom,
          rotation,
          viewportSize: viewportRect,
          offset: (0, import_image_cropper.addPoints)(offsetStart, delta),
          fixedCropArea: prop("fixedCropArea"),
          crop: context.get("crop")
        });
        context.set("offset", nextOffset);
      },
      setHandlePosition({ event, context }) {
        const position = event.handlePosition;
        if (!position) return;
        context.set("handlePosition", position);
      },
      setRotation({ context, event }) {
        const rotation = event.rotation;
        const nextRotation = (0, import_utils.clampValue)(rotation, 0, 360);
        context.set("rotation", nextRotation);
      },
      setFlip({ context, event }) {
        const nextFlip = event.flip;
        if (!nextFlip) return;
        const currentFlip = context.get("flip");
        const normalized = (0, import_image_cropper.normalizeFlipState)(nextFlip, currentFlip);
        if ((0, import_image_cropper.isEqualFlip)(normalized, currentFlip)) return;
        context.set("flip", normalized);
      },
      resizeCrop({ context, event, prop }) {
        const { handlePosition, delta } = event;
        if (!handlePosition) return;
        const viewportRect = context.get("viewportRect");
        if (!(0, import_image_cropper.isVisibleRect)(viewportRect)) return;
        const cropShape = prop("cropShape");
        const aspectRatio = (0, import_image_cropper.resolveCropAspectRatio)(cropShape, prop("aspectRatio"));
        const { minSize, maxSize } = (0, import_image_cropper.getCropSizeLimits)(prop);
        const crop = context.get("crop");
        const nextCrop = (0, import_image_cropper.computeResizeCrop)({
          cropStart: crop,
          handlePosition,
          delta,
          viewportRect,
          minSize,
          maxSize,
          aspectRatio
        });
        context.set("crop", nextCrop);
      },
      clearPointerStart({ context }) {
        context.set("pointerStart", null);
      },
      clearCropStart({ context }) {
        context.set("cropStart", null);
      },
      clearHandlePosition({ context }) {
        context.set("handlePosition", null);
      },
      clearOffsetStart({ context }) {
        context.set("offsetStart", null);
      },
      clearShiftRatio({ context }) {
        context.set("shiftLockRatio", null);
      },
      updateZoom({ context, event, prop }) {
        let { delta, point, zoom: targetZoom, scale, panDelta } = event;
        const crop = context.get("crop");
        const currentZoom = context.get("zoom");
        const currentOffset = context.get("offset");
        const rotation = context.get("rotation");
        const viewportRect = context.get("viewportRect");
        const fixedCropArea = prop("fixedCropArea");
        if (!point) {
          point = (0, import_image_cropper.getCenterPoint)(crop);
        }
        const step = Math.abs(prop("zoomStep"));
        const sensitivity = Math.max(0, prop("zoomSensitivity"));
        const [minZoom, maxZoom] = [prop("minZoom"), prop("maxZoom")];
        const calculateNextZoom = () => {
          if (typeof targetZoom === "number") {
            return (0, import_utils.clampValue)(targetZoom, minZoom, maxZoom);
          }
          if (event.trigger === "touch" && typeof scale === "number") {
            const minScale = 0.5;
            const maxScale = 2;
            const clampedScale = (0, import_utils.clampValue)(scale, minScale, maxScale);
            const smoothing = sensitivity > 0 ? Math.pow(clampedScale, sensitivity) : clampedScale;
            return (0, import_utils.clampValue)(currentZoom * smoothing, minZoom, maxZoom);
          }
          if (typeof delta === "number") {
            const direction = Math.sign(delta) < 0 ? 1 : -1;
            return (0, import_utils.clampValue)(currentZoom + step * direction, minZoom, maxZoom);
          }
          return null;
        };
        const applyClampedOffset = (zoom, offset) => {
          return (0, import_image_cropper.clampOffset)({
            zoom,
            rotation,
            viewportSize: viewportRect,
            offset,
            fixedCropArea,
            crop
          });
        };
        const nextZoom = calculateNextZoom();
        if (nextZoom === null) return;
        if (nextZoom === currentZoom && panDelta) {
          const nextOffset2 = applyClampedOffset(currentZoom, (0, import_image_cropper.addPoints)(currentOffset, panDelta));
          context.set("offset", nextOffset2);
          return;
        }
        if (nextZoom === currentZoom) return;
        const { width: viewportWidth, height: viewportHeight } = viewportRect;
        const { x: centerX, y: centerY } = (0, import_image_cropper.getViewportCenter)(viewportRect);
        const zoomRatio = nextZoom / currentZoom;
        let nextOffset = {
          x: (1 - zoomRatio) * (point.x - centerX) + zoomRatio * currentOffset.x,
          y: (1 - zoomRatio) * (point.y - centerY) + zoomRatio * currentOffset.y
        };
        if (panDelta) {
          nextOffset = applyClampedOffset(nextZoom, (0, import_image_cropper.addPoints)(nextOffset, panDelta));
        } else if (nextZoom < currentZoom) {
          if (fixedCropArea) {
            nextOffset = applyClampedOffset(nextZoom, nextOffset);
          } else {
            const { width: scaledImageWidth, height: scaledImageHeight } = (0, import_image_cropper.scaleSize)(viewportRect, nextZoom);
            if (scaledImageWidth <= viewportWidth) {
              nextOffset.x = 0;
            } else {
              const minX = viewportWidth - centerX - scaledImageWidth / 2;
              const maxX = scaledImageWidth / 2 - centerX;
              nextOffset.x = Math.max(minX, Math.min(maxX, nextOffset.x));
            }
            if (scaledImageHeight <= viewportHeight) {
              nextOffset.y = 0;
            } else {
              const minY = viewportHeight - centerY - scaledImageHeight / 2;
              const maxY = scaledImageHeight / 2 - centerY;
              nextOffset.y = Math.max(minY, Math.min(maxY, nextOffset.y));
            }
          }
        }
        context.set("zoom", nextZoom);
        context.set("offset", nextOffset);
      },
      setPinchDistance({ context, event, send }) {
        const touches = Array.isArray(event.touches) ? event.touches : [];
        if (touches.length < 2) return;
        if (context.get("pointerStart") !== null) {
          send({ type: "POINTER_UP", src: "pinch" });
        }
        const [first, second] = touches;
        const distance = (0, import_image_cropper.getTouchDistance)(first, second);
        const viewportRect = context.get("viewportRect");
        const midpoint = (0, import_image_cropper.getMidpoint)(first, second, { x: viewportRect.left, y: viewportRect.top });
        context.set("pinchDistance", distance);
        context.set("pinchMidpoint", midpoint);
      },
      handlePinchMove({ context, event, send }) {
        const touches = Array.isArray(event.touches) ? event.touches : [];
        if (touches.length < 2) return;
        const [first, second] = touches;
        const distance = (0, import_image_cropper.getTouchDistance)(first, second);
        const lastDistance = context.get("pinchDistance");
        const lastMidpoint = context.get("pinchMidpoint");
        const viewportRect = context.get("viewportRect");
        const midpoint = (0, import_image_cropper.getMidpoint)(first, second, { x: viewportRect.left, y: viewportRect.top });
        if (lastDistance != null && lastDistance > 0 && lastMidpoint != null) {
          const delta = lastDistance - distance;
          const scale = distance / lastDistance;
          const distanceChange = Math.abs(delta);
          const hasSignificantZoom = distanceChange > import_image_cropper.MIN_PINCH_DISTANCE;
          const panDelta = (0, import_image_cropper.subtractPoints)(midpoint, lastMidpoint);
          send({
            type: "ZOOM",
            trigger: "touch",
            delta,
            scale: hasSignificantZoom ? scale : 1,
            point: midpoint,
            panDelta
          });
        }
        context.set("pinchDistance", distance);
        context.set("pinchMidpoint", midpoint);
      },
      clearPinchDistance({ context }) {
        context.set("pinchDistance", null);
        context.set("pinchMidpoint", null);
      },
      nudgeResizeCrop({ context, event, prop }) {
        const { key, handlePosition, shiftKey, ctrlKey, metaKey } = event;
        const crop = context.get("crop");
        const viewportRect = context.get("viewportRect");
        const step = (0, import_image_cropper.getNudgeStep)(prop, { shiftKey, ctrlKey, metaKey });
        const { minSize, maxSize } = (0, import_image_cropper.getCropSizeLimits)(prop);
        const nextCrop = (0, import_image_cropper.computeKeyboardCrop)(key, handlePosition, step, crop, viewportRect, minSize, maxSize);
        context.set("crop", nextCrop);
      },
      nudgeMoveCrop({ context, event, prop }) {
        const { key, shiftKey, ctrlKey, metaKey } = event;
        const crop = context.get("crop");
        const viewportRect = context.get("viewportRect");
        const step = (0, import_image_cropper.getNudgeStep)(prop, { shiftKey, ctrlKey, metaKey });
        const delta = (0, import_image_cropper.getKeyboardMoveDelta)(key, step);
        const nextCrop = (0, import_image_cropper.computeMoveCrop)(crop, delta, viewportRect);
        context.set("crop", nextCrop);
      },
      resizeViewport({ context, prop, scope, send }) {
        const viewportEl = dom.getViewportEl(scope);
        if (!viewportEl) return;
        const newViewportRect = getBoundingRect(viewportEl);
        if (!(0, import_image_cropper.isVisibleRect)(newViewportRect)) return;
        const oldViewportRect = context.get("viewportRect");
        if ((0, import_image_cropper.isSameSize)(oldViewportRect, newViewportRect)) {
          return;
        }
        context.set("viewportRect", newViewportRect);
        const oldCrop = context.get("crop");
        if (!(0, import_image_cropper.isVisibleRect)(oldCrop)) {
          send({ type: "SET_DEFAULT_CROP", src: "viewport-resize" });
          return;
        }
        if (!(0, import_image_cropper.isVisibleRect)(oldViewportRect)) {
          return;
        }
        const cropShape = prop("cropShape");
        const aspectRatio = (0, import_image_cropper.resolveCropAspectRatio)(cropShape, prop("aspectRatio"));
        const { minSize, maxSize } = (0, import_image_cropper.getCropSizeLimits)(prop);
        const scale = {
          x: newViewportRect.width / oldViewportRect.width,
          y: newViewportRect.height / oldViewportRect.height
        };
        let newCrop = (0, import_image_cropper.scaleRect)(oldCrop, scale);
        const constrainedCrop = (0, import_image_cropper.computeResizeCrop)({
          cropStart: newCrop,
          handlePosition: "se",
          delta: import_image_cropper.ZERO_POINT,
          viewportRect: newViewportRect,
          minSize,
          maxSize,
          aspectRatio
        });
        const max = (0, import_image_cropper.getMaxBounds)(constrainedCrop, newViewportRect);
        const { x, y } = (0, import_image_cropper.clampPoint)(constrainedCrop, import_image_cropper.ZERO_POINT, max);
        context.set("crop", {
          x,
          y,
          width: constrainedCrop.width,
          height: constrainedCrop.height
        });
      },
      resetToInitialState({ context, send }) {
        context.set("zoom", context.initial("zoom"));
        context.set("rotation", context.initial("rotation"));
        context.set("flip", context.initial("flip"));
        context.set("offset", import_image_cropper.ZERO_POINT);
        send({ type: "SET_DEFAULT_CROP", src: "reset" });
      },
      adjustCropAspectRatio({ context, prop }) {
        const viewportRect = context.get("viewportRect");
        if (!(0, import_image_cropper.isVisibleRect)(viewportRect)) return;
        const crop = context.get("crop");
        if (!(0, import_image_cropper.isVisibleRect)(crop)) return;
        const cropShape = prop("cropShape");
        const aspectRatio = (0, import_image_cropper.resolveCropAspectRatio)(cropShape, prop("aspectRatio"));
        if (aspectRatio === void 0) return;
        const currentAspect = crop.width / crop.height;
        if ((0, import_image_cropper.isAspectRatioEqual)(currentAspect, aspectRatio)) return;
        const { minSize, maxSize } = (0, import_image_cropper.getCropSizeLimits)(prop);
        const constrainedCrop = (0, import_image_cropper.computeResizeCrop)({
          cropStart: crop,
          handlePosition: "se",
          delta: import_image_cropper.ZERO_POINT,
          viewportRect,
          minSize,
          maxSize,
          aspectRatio
        });
        if ((0, import_image_cropper.isSameSize)(crop, constrainedCrop)) return;
        const center = (0, import_image_cropper.getCenterPoint)(crop);
        const pos = (0, import_image_cropper.centerCropOnPoint)(constrainedCrop, center, viewportRect);
        context.set("crop", {
          x: pos.x,
          y: pos.y,
          width: constrainedCrop.width,
          height: constrainedCrop.height
        });
      }
    },
    effects: {
      trackPointerMove({ scope, send }) {
        function onPointerMove(event) {
          const point = (0, import_dom_query.getEventPoint)(event);
          const target = (0, import_dom_query.getEventTarget)(event);
          send({ type: "POINTER_MOVE", point, target, shiftKey: event.shiftKey, altKey: event.altKey });
        }
        function onPointerUp() {
          send({ type: "POINTER_UP" });
        }
        return (0, import_utils.callAll)(
          (0, import_dom_query.addDomEvent)(scope.getDoc(), "pointermove", onPointerMove),
          (0, import_dom_query.addDomEvent)(scope.getDoc(), "pointerup", onPointerUp)
        );
      },
      trackViewportResize({ scope, send }) {
        const viewportEl = dom.getViewportEl(scope);
        if (!viewportEl) return;
        return import_dom_query.resizeObserverBorderBox.observe(viewportEl, () => {
          send({ type: "VIEWPORT_RESIZE", src: "resize" });
        });
      },
      trackWheelEvent({ scope, send }) {
        const viewportEl = dom.getViewportEl(scope);
        if (!viewportEl) return;
        function onWheel(event) {
          event.preventDefault();
          if (!viewportEl) return;
          const rect = viewportEl.getBoundingClientRect();
          const point = {
            x: event.clientX - rect.left,
            y: event.clientY - rect.top
          };
          send({ type: "ZOOM", trigger: "wheel", delta: event.deltaY, point });
        }
        return (0, import_dom_query.addDomEvent)(viewportEl, "wheel", onWheel, { passive: false });
      },
      trackTouchEvents({ scope, send }) {
        const viewportEl = dom.getViewportEl(scope);
        if (!viewportEl) return;
        function onTouchStart(event) {
          if (event.touches.length >= 2) {
            event.preventDefault();
            const touches = Array.from(event.touches).map((touch) => ({
              x: touch.clientX,
              y: touch.clientY
            }));
            send({ type: "PINCH_START", touches });
          }
        }
        function onTouchMove(event) {
          if (event.touches.length >= 2) {
            event.preventDefault();
            const touches = Array.from(event.touches).map((touch) => ({
              x: touch.clientX,
              y: touch.clientY
            }));
            send({ type: "PINCH_MOVE", touches });
          }
        }
        function onTouchEnd(event) {
          if (event.touches.length < 2) {
            send({ type: "PINCH_END" });
          }
        }
        return (0, import_utils.callAll)(
          (0, import_dom_query.addDomEvent)(viewportEl, "touchstart", onTouchStart, { passive: false }),
          (0, import_dom_query.addDomEvent)(viewportEl, "touchmove", onTouchMove, { passive: false }),
          (0, import_dom_query.addDomEvent)(viewportEl, "touchend", onTouchEnd)
        );
      }
    }
  }
});
var getBoundingRect = (el) => {
  const rect = el.getBoundingClientRect();
  return {
    width: rect.width,
    height: rect.height,
    top: rect.top,
    left: rect.left,
    right: rect.right,
    bottom: rect.bottom
  };
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  machine
});
