// @zag-js/cascade-select/dist/cascade-select.connect.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/cascade-select.connect.ts
var cascade_select_connect_exports = {};
__export(cascade_select_connect_exports, {
  connect: () => connect
});
module.exports = __toCommonJS(cascade_select_connect_exports);
var import_dom_query = require("@zag-js/dom-query");
var import_popper = require("@zag-js/popper");
var import_utils = require("@zag-js/utils");
var import_cascade_select = require("./cascade-select.anatomy.js");
var import_cascade_select2 = require("./cascade-select.dom.js");
function connect(service, normalize) {
  const { send, context, prop, scope, computed, state } = service;
  const collection = prop("collection");
  const value = context.get("value");
  const open = state.hasTag("open");
  const focused = state.matches("focused");
  const highlightedIndexPath = context.get("highlightedIndexPath");
  const highlightedValue = context.get("highlightedValue");
  const currentPlacement = context.get("currentPlacement");
  const currentPlacementSide = currentPlacement ? (0, import_popper.getPlacementSide)(currentPlacement) : void 0;
  const disabled = prop("disabled") || context.get("fieldsetDisabled");
  const interactive = computed("isInteractive");
  const valueAsString = computed("valueAsString");
  const highlightedItems = context.get("highlightedItems");
  const selectedItems = context.get("selectedItems");
  const popperStyles = (0, import_popper.getPlacementStyles)({
    ...prop("positioning"),
    placement: currentPlacement
  });
  const getItemState = (props) => {
    const { item, indexPath, value: itemValue } = props;
    const depth = indexPath ? indexPath.length : 0;
    const highlighted = itemValue.every((v, i) => v === highlightedValue[i]);
    const selected = value.some((v) => (0, import_utils.isEqual)(v, itemValue));
    const children = collection.getNodeChildren(collection.at(indexPath));
    const highlightedChild = children[highlightedIndexPath[depth]];
    const highlightedIndex = highlightedIndexPath[depth];
    return {
      value: itemValue,
      disabled: collection.getNodeDisabled(item),
      highlighted,
      selected,
      hasChildren: collection.isBranchNode(item),
      depth,
      highlightedChild,
      highlightedIndex
    };
  };
  const hasSelectedItems = value.length > 0;
  const empty = value.length === 0;
  return {
    collection,
    open,
    focused,
    multiple: !!prop("multiple"),
    disabled,
    value,
    highlightedValue,
    highlightedItems,
    selectedItems,
    hasSelectedItems,
    empty,
    valueAsString,
    reposition(options = {}) {
      send({ type: "POSITIONING.SET", options });
    },
    focus() {
      import_cascade_select2.dom.getTriggerEl(scope)?.focus({ preventScroll: true });
    },
    setOpen(nextOpen) {
      if (nextOpen === open) return;
      send({ type: nextOpen ? "OPEN" : "CLOSE" });
    },
    setHighlightValue(value2) {
      send({ type: "HIGHLIGHTED_VALUE.SET", value: value2 });
    },
    clearHighlightValue() {
      send({ type: "HIGHLIGHTED_VALUE.CLEAR" });
    },
    setValue(value2) {
      send({ type: "VALUE.SET", value: value2 });
    },
    selectValue(value2) {
      send({ type: "ITEM.SELECT", value: value2 });
    },
    clearValue(value2) {
      if (value2) {
        send({ type: "ITEM.CLEAR", value: value2 });
      } else {
        send({ type: "VALUE.CLEAR" });
      }
    },
    getItemState,
    getRootProps() {
      return normalize.element({
        ...import_cascade_select.parts.root.attrs,
        id: import_cascade_select2.dom.getRootId(scope),
        dir: prop("dir"),
        "data-disabled": (0, import_dom_query.dataAttr)(disabled),
        "data-readonly": (0, import_dom_query.dataAttr)(prop("readOnly")),
        "data-invalid": (0, import_dom_query.dataAttr)(prop("invalid")),
        "data-state": open ? "open" : "closed"
      });
    },
    getLabelProps() {
      return normalize.label({
        ...import_cascade_select.parts.label.attrs,
        id: import_cascade_select2.dom.getLabelId(scope),
        dir: prop("dir"),
        htmlFor: import_cascade_select2.dom.getHiddenInputId(scope),
        "data-disabled": (0, import_dom_query.dataAttr)(disabled),
        "data-readonly": (0, import_dom_query.dataAttr)(prop("readOnly")),
        "data-invalid": (0, import_dom_query.dataAttr)(prop("invalid")),
        onClick(event) {
          if (event.defaultPrevented) return;
          if (disabled) return;
          const triggerEl = import_cascade_select2.dom.getTriggerEl(scope);
          triggerEl?.focus({ preventScroll: true });
        }
      });
    },
    getControlProps() {
      return normalize.element({
        ...import_cascade_select.parts.control.attrs,
        dir: prop("dir"),
        id: import_cascade_select2.dom.getControlId(scope),
        "data-disabled": (0, import_dom_query.dataAttr)(disabled),
        "data-focus": (0, import_dom_query.dataAttr)(focused),
        "data-readonly": (0, import_dom_query.dataAttr)(prop("readOnly")),
        "data-invalid": (0, import_dom_query.dataAttr)(prop("invalid")),
        "data-state": open ? "open" : "closed"
      });
    },
    getTriggerProps() {
      return normalize.button({
        ...import_cascade_select.parts.trigger.attrs,
        dir: prop("dir"),
        id: import_cascade_select2.dom.getTriggerId(scope),
        type: "button",
        role: "combobox",
        "aria-controls": import_cascade_select2.dom.getContentId(scope),
        "aria-expanded": open,
        "aria-haspopup": "listbox",
        "aria-labelledby": import_cascade_select2.dom.getLabelId(scope),
        "data-state": open ? "open" : "closed",
        "data-disabled": (0, import_dom_query.dataAttr)(disabled),
        "data-readonly": (0, import_dom_query.dataAttr)(prop("readOnly")),
        "data-invalid": (0, import_dom_query.dataAttr)(prop("invalid")),
        "data-focus": (0, import_dom_query.dataAttr)(focused),
        "data-placement": currentPlacement,
        "data-side": currentPlacementSide,
        "data-placeholder-shown": (0, import_dom_query.dataAttr)(!hasSelectedItems),
        disabled,
        onClick(event) {
          if (event.defaultPrevented) return;
          if (!interactive) return;
          send({ type: "TRIGGER.CLICK" });
        },
        onFocus() {
          send({ type: "TRIGGER.FOCUS" });
        },
        onBlur() {
          send({ type: "TRIGGER.BLUR" });
        },
        onKeyDown(event) {
          if (event.defaultPrevented) return;
          if (!interactive) return;
          const keyMap = {
            ArrowUp() {
              send({ type: "TRIGGER.ARROW_UP" });
            },
            ArrowDown(event2) {
              send({ type: event2.altKey ? "OPEN" : "TRIGGER.ARROW_DOWN" });
            },
            ArrowLeft() {
              send({ type: "TRIGGER.ARROW_LEFT" });
            },
            ArrowRight() {
              send({ type: "TRIGGER.ARROW_RIGHT" });
            },
            Enter() {
              send({ type: "TRIGGER.ENTER" });
            },
            Space() {
              send({ type: "TRIGGER.ENTER" });
            }
          };
          const exec = keyMap[(0, import_dom_query.getEventKey)(event, { dir: prop("dir") })];
          if (exec) {
            exec(event);
            event.preventDefault();
          }
        }
      });
    },
    getClearTriggerProps() {
      return normalize.button({
        ...import_cascade_select.parts.clearTrigger.attrs,
        dir: prop("dir"),
        id: import_cascade_select2.dom.getClearTriggerId(scope),
        type: "button",
        "aria-label": "Clear value",
        hidden: !hasSelectedItems,
        "data-disabled": (0, import_dom_query.dataAttr)(disabled),
        "data-readonly": (0, import_dom_query.dataAttr)(prop("readOnly")),
        "data-invalid": (0, import_dom_query.dataAttr)(prop("invalid")),
        disabled,
        onClick(event) {
          if (event.defaultPrevented) return;
          send({ type: "CLEAR_TRIGGER.CLICK" });
        }
      });
    },
    getPositionerProps() {
      return normalize.element({
        ...import_cascade_select.parts.positioner.attrs,
        dir: prop("dir"),
        id: import_cascade_select2.dom.getPositionerId(scope),
        style: popperStyles.floating
      });
    },
    getContentProps() {
      const highlightedItemId = highlightedValue ? import_cascade_select2.dom.getItemId(scope, highlightedValue.toString()) : void 0;
      return normalize.element({
        ...import_cascade_select.parts.content.attrs,
        id: import_cascade_select2.dom.getContentId(scope),
        role: "listbox",
        "aria-labelledby": import_cascade_select2.dom.getLabelId(scope),
        "aria-activedescendant": highlightedItemId,
        "data-activedescendant": highlightedItemId,
        "data-state": open ? "open" : "closed",
        "aria-multiselectable": prop("multiple"),
        "aria-required": prop("required"),
        "aria-readonly": prop("readOnly"),
        hidden: !open,
        tabIndex: 0,
        onKeyDown(event) {
          if (!interactive) return;
          if (!(0, import_dom_query.isSelfTarget)(event)) return;
          if (event.key === "Tab") {
            const valid = (0, import_dom_query.isValidTabEvent)(event);
            if (!valid) {
              event.preventDefault();
              return;
            }
          }
          const keyMap = {
            ArrowDown() {
              send({ type: "CONTENT.ARROW_DOWN" });
            },
            ArrowUp() {
              send({ type: "CONTENT.ARROW_UP" });
            },
            ArrowRight() {
              send({ type: "CONTENT.ARROW_RIGHT" });
            },
            ArrowLeft() {
              send({ type: "CONTENT.ARROW_LEFT" });
            },
            Home() {
              send({ type: "CONTENT.HOME" });
            },
            End() {
              send({ type: "CONTENT.END" });
            },
            Enter() {
              send({ type: "CONTENT.ENTER" });
            },
            " "() {
              send({ type: "CONTENT.ENTER" });
            }
          };
          const exec = keyMap[(0, import_dom_query.getEventKey)(event, { dir: prop("dir") })];
          if (exec) {
            exec();
            event.preventDefault();
            return;
          }
          if ((0, import_dom_query.isEditableElement)(event.target)) {
            return;
          }
        },
        onPointerMove(event) {
          if (!interactive) return;
          send({ type: "POINTER_MOVE", clientX: event.clientX, clientY: event.clientY, target: event.target });
        }
      });
    },
    getListProps(props) {
      const itemState = getItemState(props);
      return normalize.element({
        ...import_cascade_select.parts.list.attrs,
        id: import_cascade_select2.dom.getListId(scope, itemState.value.toString()),
        dir: prop("dir"),
        "data-depth": itemState.depth,
        "aria-level": itemState.depth,
        role: "group"
      });
    },
    getIndicatorProps() {
      return normalize.element({
        ...import_cascade_select.parts.indicator.attrs,
        id: import_cascade_select2.dom.getIndicatorId(scope),
        dir: prop("dir"),
        "aria-hidden": true,
        "data-state": open ? "open" : "closed",
        "data-disabled": (0, import_dom_query.dataAttr)(disabled),
        "data-readonly": (0, import_dom_query.dataAttr)(prop("readOnly")),
        "data-invalid": (0, import_dom_query.dataAttr)(prop("invalid"))
      });
    },
    getItemProps(props) {
      const { indexPath } = props;
      const itemState = getItemState(props);
      return normalize.element({
        ...import_cascade_select.parts.item.attrs,
        id: import_cascade_select2.dom.getItemId(scope, itemState.value.toString()),
        dir: prop("dir"),
        role: "treeitem",
        "aria-haspopup": itemState.hasChildren ? "menu" : void 0,
        "aria-expanded": itemState.hasChildren ? itemState.highlighted : false,
        "aria-controls": itemState.hasChildren ? import_cascade_select2.dom.getListId(scope, itemState.value.toString()) : void 0,
        "aria-owns": itemState.hasChildren ? import_cascade_select2.dom.getListId(scope, itemState.value.toString()) : void 0,
        "aria-disabled": (0, import_dom_query.ariaAttr)(itemState.disabled),
        "data-value": itemState.value.toString(),
        "data-disabled": (0, import_dom_query.dataAttr)(itemState.disabled),
        "data-highlighted": (0, import_dom_query.dataAttr)(itemState.highlighted),
        "data-selected": (0, import_dom_query.dataAttr)(itemState.selected),
        "data-depth": itemState.depth,
        "data-state": itemState.selected ? "checked" : "unchecked",
        "data-type": itemState.hasChildren ? "branch" : "leaf",
        "data-index-path": indexPath.toString(),
        onDoubleClick() {
          if (itemState.disabled) return;
          send({ type: "CLOSE" });
        },
        onClick(event) {
          if (event.defaultPrevented) return;
          if (!interactive) return;
          if (itemState.disabled) return;
          send({ type: "ITEM.CLICK", value: itemState.value, indexPath });
        },
        onPointerEnter(event) {
          if (!interactive) return;
          if (itemState.disabled) return;
          send({
            type: "ITEM.POINTER_ENTER",
            value: itemState.value,
            indexPath,
            clientX: event.clientX,
            clientY: event.clientY
          });
        },
        onPointerLeave(event) {
          if (!interactive) return;
          if (itemState.disabled) return;
          if (event.pointerType !== "mouse") return;
          const pointerMoved = service.event.previous()?.type.includes("POINTER");
          if (!pointerMoved) return;
          send({
            type: "ITEM.POINTER_LEAVE",
            value: itemState.value,
            indexPath,
            clientX: event.clientX,
            clientY: event.clientY
          });
        }
      });
    },
    getItemTextProps(props) {
      const { item } = props;
      const itemValue = collection.getNodeValue(item);
      const itemState = getItemState(props);
      return normalize.element({
        dir: prop("dir"),
        ...import_cascade_select.parts.itemText.attrs,
        "data-value": itemValue,
        "data-highlighted": (0, import_dom_query.dataAttr)(itemState.highlighted),
        "data-state": itemState.selected ? "checked" : "unchecked",
        "data-disabled": (0, import_dom_query.dataAttr)(itemState.disabled)
      });
    },
    getItemIndicatorProps(props) {
      const { item } = props;
      const itemValue = collection.getNodeValue(item);
      const itemState = getItemState(props);
      return normalize.element({
        ...import_cascade_select.parts.itemIndicator.attrs,
        dir: prop("dir"),
        "data-value": itemValue,
        "data-highlighted": (0, import_dom_query.dataAttr)(itemState.highlighted),
        "data-type": itemState.hasChildren ? "branch" : "leaf",
        "data-state": itemState.selected ? "checked" : "unchecked",
        hidden: !itemState.selected
      });
    },
    getValueTextProps() {
      return normalize.element({
        ...import_cascade_select.parts.valueText.attrs,
        dir: prop("dir"),
        "data-disabled": (0, import_dom_query.dataAttr)(disabled),
        "data-invalid": (0, import_dom_query.dataAttr)(prop("invalid")),
        "data-focus": (0, import_dom_query.dataAttr)(focused)
      });
    },
    getHiddenInputProps() {
      const defaultValue = context.hash("value");
      return normalize.input({
        name: prop("name"),
        form: prop("form"),
        disabled,
        multiple: prop("multiple"),
        required: prop("required"),
        readOnly: prop("readOnly"),
        hidden: true,
        "aria-hidden": true,
        id: import_cascade_select2.dom.getHiddenInputId(scope),
        defaultValue,
        "aria-labelledby": import_cascade_select2.dom.getLabelId(scope)
      });
    }
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  connect
});
