// @zag-js/cascade-select/dist/cascade-select.machine.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/cascade-select.machine.ts
var cascade_select_machine_exports = {};
__export(cascade_select_machine_exports, {
  machine: () => machine
});
module.exports = __toCommonJS(cascade_select_machine_exports);
var import_core = require("@zag-js/core");
var import_dismissable = require("@zag-js/dismissable");
var import_dom_query = require("@zag-js/dom-query");
var import_focus_visible = require("@zag-js/focus-visible");
var import_popper = require("@zag-js/popper");
var import_utils = require("@zag-js/utils");
var import_cascade_select = require("./cascade-select.dom.js");
var import_cascade_select2 = require("./cascade-select.utils.js");
var import_cascade_select3 = require("./cascade-select.collection.js");
var { or, and, not } = (0, import_core.createGuards)();
var machine = (0, import_core.createMachine)({
  props({ props }) {
    return {
      closeOnSelect: true,
      loopFocus: false,
      defaultValue: [],
      defaultHighlightedValue: [],
      defaultOpen: false,
      multiple: false,
      highlightTrigger: "click",
      allowParentSelection: false,
      positioning: {
        placement: "bottom-start",
        gutter: 8,
        ...props.positioning
      },
      ...props,
      collection: props.collection ?? import_cascade_select3.collection.empty()
    };
  },
  context({ prop, bindable }) {
    return {
      value: bindable(() => ({
        defaultValue: prop("defaultValue"),
        value: prop("value"),
        isEqual: import_utils.isEqual,
        hash(value) {
          return value.join(", ");
        }
      })),
      highlightedValue: bindable(() => ({
        defaultValue: prop("defaultHighlightedValue"),
        value: prop("highlightedValue"),
        isEqual: import_utils.isEqual
      })),
      valueIndexPath: bindable(() => {
        const value = prop("value") ?? prop("defaultValue") ?? [];
        const paths = value.map((v) => prop("collection").getIndexPath(v));
        return {
          defaultValue: paths
        };
      }),
      highlightedIndexPath: bindable(() => {
        const value = prop("highlightedValue") ?? prop("defaultHighlightedValue") ?? null;
        return {
          defaultValue: value ? prop("collection").getIndexPath(value) : []
        };
      }),
      highlightedItems: bindable(() => ({
        defaultValue: []
      })),
      selectedItems: bindable(() => ({
        defaultValue: []
      })),
      currentPlacement: bindable(() => ({
        defaultValue: void 0
      })),
      fieldsetDisabled: bindable(() => ({
        defaultValue: false
      })),
      graceArea: bindable(() => ({
        defaultValue: null
      })),
      isPointerInTransit: bindable(() => ({
        defaultValue: false
      }))
    };
  },
  computed: {
    isInteractive: ({ prop }) => !(prop("disabled") || prop("readOnly")),
    valueAsString: ({ prop, context }) => {
      const collection = prop("collection");
      const items = context.get("selectedItems");
      const multiple = prop("multiple");
      const formatMultipleMode = (items2) => collection.stringifyNode(items2.at(-1)) ?? collection.getNodeValue(items2.at(-1));
      const formatSingleMode = (items2) => {
        return items2.map((item) => {
          return collection.stringifyNode(item) ?? collection.getNodeValue(item);
        }).join(" / ");
      };
      const defaultFormatValue = (items2) => items2.map(multiple ? formatMultipleMode : formatSingleMode).join(", ");
      const formatValue = prop("formatValue") ?? defaultFormatValue;
      return formatValue(items);
    }
  },
  initialState({ prop }) {
    const open = prop("open") || prop("defaultOpen");
    return open ? "open" : "idle";
  },
  watch({ context, prop, track, action }) {
    track([() => context.get("value")?.toString()], () => {
      action(["syncInputValue", "dispatchChangeEvent"]);
    });
    track([() => prop("open")], () => {
      action(["toggleVisibility"]);
    });
  },
  on: {
    "VALUE.SET": {
      actions: ["setValue"]
    },
    "VALUE.CLEAR": {
      actions: ["clearValue"]
    },
    "CLEAR_TRIGGER.CLICK": {
      actions: ["clearValue", "focusTriggerEl"]
    },
    "HIGHLIGHTED_VALUE.SET": {
      actions: ["setHighlightedValue"]
    },
    "HIGHLIGHTED_VALUE.CLEAR": {
      actions: ["clearHighlightedValue"]
    },
    "ITEM.SELECT": {
      actions: ["selectItem"]
    },
    "ITEM.CLEAR": {
      actions: ["clearItem"]
    }
  },
  effects: ["trackFormControlState"],
  states: {
    idle: {
      tags: ["closed"],
      on: {
        "CONTROLLED.OPEN": [
          {
            guard: "isTriggerClickEvent",
            target: "open",
            actions: ["setInitialFocus", "highlightFirstSelectedItem"]
          },
          {
            target: "open",
            actions: ["setInitialFocus"]
          }
        ],
        "TRIGGER.CLICK": [
          {
            guard: "isOpenControlled",
            actions: ["invokeOnOpen"]
          },
          {
            target: "open",
            actions: ["invokeOnOpen", "setInitialFocus", "highlightFirstSelectedItem"]
          }
        ],
        "TRIGGER.FOCUS": {
          target: "focused"
        },
        OPEN: [
          {
            guard: "isOpenControlled",
            actions: ["invokeOnOpen"]
          },
          {
            target: "open",
            actions: ["setInitialFocus", "invokeOnOpen"]
          }
        ]
      }
    },
    focused: {
      tags: ["closed"],
      on: {
        "CONTROLLED.OPEN": [
          {
            guard: "isTriggerClickEvent",
            target: "open",
            actions: ["setInitialFocus", "highlightFirstSelectedItem"]
          },
          {
            guard: "isTriggerArrowUpEvent",
            target: "open",
            actions: ["setInitialFocus", "highlightLastItem"]
          },
          {
            guard: or("isTriggerArrowDownEvent", "isTriggerEnterEvent", ""),
            target: "open",
            actions: ["setInitialFocus", "highlightFirstItem"]
          },
          {
            target: "open",
            actions: ["setInitialFocus"]
          }
        ],
        OPEN: [
          {
            guard: "isOpenControlled",
            actions: ["invokeOnOpen"]
          },
          {
            target: "open",
            actions: ["setInitialFocus", "invokeOnOpen"]
          }
        ],
        "TRIGGER.BLUR": {
          target: "idle"
        },
        "TRIGGER.CLICK": [
          {
            guard: "isOpenControlled",
            actions: ["invokeOnOpen"]
          },
          {
            target: "open",
            actions: ["setInitialFocus", "invokeOnOpen", "highlightFirstSelectedItem"]
          }
        ],
        "TRIGGER.ENTER": [
          {
            guard: "isOpenControlled",
            actions: ["invokeOnOpen"]
          },
          {
            target: "open",
            actions: ["setInitialFocus", "invokeOnOpen", "highlightFirstItem"]
          }
        ],
        "TRIGGER.ARROW_UP": [
          {
            guard: "isOpenControlled",
            actions: ["invokeOnOpen"]
          },
          {
            target: "open",
            actions: ["setInitialFocus", "invokeOnOpen", "highlightLastItem"]
          }
        ],
        "TRIGGER.ARROW_DOWN": [
          {
            guard: "isOpenControlled",
            actions: ["invokeOnOpen"]
          },
          {
            target: "open",
            actions: ["setInitialFocus", "invokeOnOpen", "highlightFirstItem"]
          }
        ],
        "TRIGGER.ARROW_LEFT": [
          {
            guard: "isOpenControlled",
            actions: ["invokeOnOpen"]
          },
          {
            target: "open",
            actions: ["invokeOnOpen"]
          }
        ],
        "TRIGGER.ARROW_RIGHT": [
          {
            guard: "isOpenControlled",
            actions: ["invokeOnOpen"]
          },
          {
            target: "open",
            actions: ["invokeOnOpen", "highlightFirstItem"]
          }
        ]
      }
    },
    open: {
      tags: ["open"],
      exit: ["clearHighlightedValue", "scrollContentToTop"],
      effects: ["trackDismissableElement", "trackFocusVisible", "computePlacement", "scrollToHighlightedItems"],
      on: {
        "CONTROLLED.CLOSE": [
          {
            guard: "restoreFocus",
            target: "focused",
            actions: ["focusTriggerEl"]
          },
          {
            target: "idle"
          }
        ],
        CLOSE: [
          {
            guard: "isOpenControlled",
            actions: ["invokeOnClose"]
          },
          {
            guard: "restoreFocus",
            target: "focused",
            actions: ["invokeOnClose", "focusTriggerEl"]
          },
          {
            target: "idle",
            actions: ["invokeOnClose"]
          }
        ],
        "TRIGGER.CLICK": [
          {
            guard: "isOpenControlled",
            actions: ["invokeOnClose"]
          },
          {
            target: "focused",
            actions: ["invokeOnClose", "focusTriggerEl"]
          }
        ],
        "ITEM.CLICK": [
          {
            guard: and("canSelectItem", and("shouldCloseOnSelect", not("multiple")), "isOpenControlled"),
            actions: ["selectItem", "invokeOnClose"]
          },
          {
            guard: and("canSelectItem", and("shouldCloseOnSelect", not("multiple"))),
            target: "focused",
            actions: ["selectItem", "invokeOnClose", "focusTriggerEl"]
          },
          {
            guard: "canSelectItem",
            actions: ["selectItem"]
          },
          {
            // If can't select, at least highlight for click-based highlighting
            actions: ["setHighlightedValue"]
          }
        ],
        "ITEM.POINTER_ENTER": [
          {
            guard: "isHoverHighlight",
            actions: ["setHighlightingForHoveredItem"]
          }
        ],
        "ITEM.POINTER_LEAVE": [
          {
            guard: and("isHoverHighlight", "shouldHighlightOnHover"),
            actions: ["createGraceArea"]
          }
        ],
        POINTER_MOVE: [
          {
            guard: and(
              "isHoverHighlight",
              "hasGraceArea",
              "isPointerOutsideGraceArea",
              "isPointerNotInAnyItem",
              "hasHighlightedValue"
            ),
            actions: ["clearGraceArea"]
          }
        ],
        "GRACE_AREA.CLEAR": [
          {
            guard: "isHoverHighlight",
            actions: ["clearGraceArea"]
          }
        ],
        "CONTENT.HOME": {
          actions: ["highlightFirstItem"]
        },
        "CONTENT.END": {
          actions: ["highlightLastItem"]
        },
        "CONTENT.ARROW_DOWN": [
          {
            guard: or(not("hasHighlightedValue"), and("loop", "isHighlightedLastItem")),
            actions: ["highlightFirstItem"]
          },
          {
            actions: ["highlightNextItem"]
          }
        ],
        "CONTENT.ARROW_UP": [
          {
            guard: or(not("hasHighlightedValue"), and("loop", "isHighlightedFirstItem")),
            actions: ["highlightLastItem"]
          },
          {
            actions: ["highlightPreviousItem"]
          }
        ],
        "CONTENT.ARROW_RIGHT": [
          {
            guard: "canNavigateToChild",
            actions: ["highlightFirstChild"]
          }
        ],
        "CONTENT.ARROW_LEFT": [
          {
            guard: and("isAtRootLevel", "isOpenControlled"),
            actions: ["invokeOnClose", "focusTriggerEl"]
          },
          {
            guard: and("isAtRootLevel", "restoreFocus"),
            target: "focused",
            actions: ["invokeOnClose", "focusTriggerEl"]
          },
          {
            guard: "isAtRootLevel",
            target: "idle",
            actions: ["invokeOnClose"]
          },
          {
            guard: "canNavigateToParent",
            actions: ["highlightParent"]
          }
        ],
        "CONTENT.ENTER": [
          {
            guard: and(
              "canSelectHighlightedItem",
              and("shouldCloseOnSelectHighlighted", not("multiple")),
              "isOpenControlled"
            ),
            actions: ["selectHighlightedItem", "invokeOnClose"]
          },
          {
            guard: and("canSelectHighlightedItem", and("shouldCloseOnSelectHighlighted", not("multiple"))),
            target: "focused",
            actions: ["selectHighlightedItem", "invokeOnClose", "focusTriggerEl"]
          },
          {
            guard: "canSelectHighlightedItem",
            actions: ["selectHighlightedItem"]
          }
        ],
        "POSITIONING.SET": {
          actions: ["reposition"]
        }
      }
    }
  },
  implementations: {
    guards: {
      restoreFocus: ({ event }) => restoreFocusFn(event),
      multiple: ({ prop }) => !!prop("multiple"),
      loop: ({ prop }) => !!prop("loopFocus"),
      isOpenControlled: ({ prop }) => !!prop("open"),
      isTriggerClickEvent: ({ event }) => event.previousEvent?.type === "TRIGGER.CLICK",
      isTriggerArrowUpEvent: ({ event }) => event.previousEvent?.type === "TRIGGER.ARROW_UP",
      isTriggerArrowDownEvent: ({ event }) => event.previousEvent?.type === "TRIGGER.ARROW_DOWN",
      isTriggerEnterEvent: ({ event }) => event.previousEvent?.type === "TRIGGER.ENTER",
      isTriggerArrowRightEvent: ({ event }) => event.previousEvent?.type === "TRIGGER.ARROW_RIGHT",
      hasHighlightedValue: ({ context }) => context.get("highlightedValue").length > 0,
      isHighlightedFirstItem: ({ context }) => context.get("highlightedIndexPath").at(-1) === 0,
      isHighlightedLastItem: ({ prop, context }) => {
        const path = context.get("highlightedIndexPath");
        const itemIndex = path.at(-1);
        if (!itemIndex && itemIndex !== 0) return false;
        const parentIndexPath = path.slice(0, -1);
        const collection = prop("collection");
        const nextSibling = collection.at([...parentIndexPath, itemIndex + 1]);
        return !nextSibling;
      },
      shouldCloseOnSelect: ({ prop, event }) => {
        const collection = prop("collection");
        const node = collection.at(event.indexPath);
        return prop("closeOnSelect") && node && !collection.isBranchNode(node);
      },
      shouldCloseOnSelectHighlighted: ({ prop, context }) => {
        const collection = prop("collection");
        const items = context.get("highlightedItems");
        const node = (0, import_utils.last)(items);
        return prop("closeOnSelect") && node != null && !collection.isBranchNode(node);
      },
      canSelectItem: ({ prop, event }) => {
        const collection = prop("collection");
        const node = collection.at(event.indexPath);
        if (!node) return false;
        return prop("allowParentSelection") || !collection.isBranchNode(node);
      },
      canSelectHighlightedItem: ({ prop, context }) => {
        const collection = prop("collection");
        const node = collection.at(context.get("highlightedIndexPath"));
        if (!node) return false;
        return prop("allowParentSelection") || !collection.isBranchNode(node);
      },
      canNavigateToChild: ({ prop, context }) => {
        const highlightedIndexPath = context.get("highlightedIndexPath");
        if (!highlightedIndexPath.length) return false;
        const collection = prop("collection");
        const node = collection.at(highlightedIndexPath);
        return node && collection.isBranchNode(node);
      },
      canNavigateToParent: ({ context }) => context.get("highlightedIndexPath").length > 1,
      isAtRootLevel: ({ context }) => context.get("highlightedIndexPath").length <= 1,
      isHoverHighlight: ({ prop }) => prop("highlightTrigger") === "hover",
      shouldHighlightOnHover: ({ prop, event }) => {
        const collection = prop("collection");
        const node = collection.at(event.indexPath);
        return node && collection.isBranchNode(node);
      },
      shouldUpdateHighlightedIndexPath: ({ prop, context, event }) => {
        const collection = prop("collection");
        const currentHighlightedIndexPath = context.get("highlightedIndexPath");
        if (!currentHighlightedIndexPath || currentHighlightedIndexPath.length === 0) {
          return false;
        }
        const node = collection.at(event.indexPath);
        if (!node || collection.isBranchNode(node)) {
          return false;
        }
        const indexPath = event.indexPath;
        if (!indexPath) return false;
        const minLength = Math.min(indexPath.length, currentHighlightedIndexPath.length);
        let commonPrefixLength = 0;
        for (let i = 0; i < minLength; i++) {
          if (indexPath[i] === currentHighlightedIndexPath[i]) {
            commonPrefixLength = i + 1;
          } else {
            break;
          }
        }
        return commonPrefixLength > 0 && (commonPrefixLength < currentHighlightedIndexPath.length || commonPrefixLength < indexPath.length);
      },
      hasGraceArea: ({ context }) => {
        return context.get("graceArea") != null;
      },
      isPointerOutsideGraceArea: ({ context, event }) => {
        const graceArea = context.get("graceArea");
        if (!graceArea) return false;
        const point = { x: event.clientX, y: event.clientY };
        return !(0, import_cascade_select2.isPointerInGraceArea)(point, graceArea);
      },
      isPointerNotInAnyItem: ({ event }) => {
        const target = event.target;
        const itemElement = target.closest('[data-part="item"]');
        const contentElement = target.closest('[data-part="content"]');
        return !contentElement || !itemElement && !!contentElement;
      }
    },
    effects: {
      trackFormControlState({ context, scope, prop }) {
        return (0, import_dom_query.trackFormControl)(import_cascade_select.dom.getTriggerEl(scope), {
          onFieldsetDisabledChange(disabled) {
            context.set("fieldsetDisabled", disabled);
          },
          onFormReset() {
            context.set("value", prop("defaultValue") ?? []);
          }
        });
      },
      trackFocusVisible({ scope }) {
        return (0, import_focus_visible.trackFocusVisible)({ root: scope.getRootNode?.() });
      },
      trackDismissableElement({ scope, send, prop }) {
        const contentEl = () => import_cascade_select.dom.getContentEl(scope);
        let restoreFocus = true;
        return (0, import_dismissable.trackDismissableElement)(contentEl, {
          defer: true,
          exclude: [import_cascade_select.dom.getTriggerEl(scope), import_cascade_select.dom.getClearTriggerEl(scope)],
          onFocusOutside: prop("onFocusOutside"),
          onPointerDownOutside: prop("onPointerDownOutside"),
          onInteractOutside(event) {
            prop("onInteractOutside")?.(event);
            restoreFocus = !(event.detail.focusable || event.detail.contextmenu);
          },
          onDismiss() {
            send({ type: "CLOSE", src: "interact-outside", restoreFocus });
          }
        });
      },
      computePlacement({ context, prop, scope }) {
        const triggerEl = () => import_cascade_select.dom.getTriggerEl(scope);
        const positionerEl = () => import_cascade_select.dom.getPositionerEl(scope);
        return (0, import_popper.getPlacement)(triggerEl, positionerEl, {
          ...prop("positioning"),
          onComplete(data) {
            context.set("currentPlacement", data.placement);
          }
        });
      },
      scrollToHighlightedItems({ context, prop, scope }) {
        let cleanups = [];
        const exec = (immediate) => {
          const highlightedValue = context.get("highlightedValue");
          const highlightedIndexPath = context.get("highlightedIndexPath");
          if (!highlightedIndexPath.length) return;
          const modality = (0, import_focus_visible.getInteractionModality)();
          if (modality === "pointer") return;
          const listEls = import_cascade_select.dom.getListEls(scope);
          listEls.forEach((listEl, index) => {
            const itemPath = highlightedIndexPath.slice(0, index + 1);
            const itemEl = import_cascade_select.dom.getItemEl(scope, highlightedValue.toString());
            const scrollToIndexFn = prop("scrollToIndexFn");
            if (scrollToIndexFn) {
              const itemIndexInList = itemPath[itemPath.length - 1];
              scrollToIndexFn({ index: itemIndexInList, immediate, depth: index });
              return;
            }
            const raf_cleanup = (0, import_dom_query.raf)(() => {
              (0, import_dom_query.scrollIntoView)(itemEl, { rootEl: listEl, block: "nearest" });
            });
            cleanups.push(raf_cleanup);
          });
        };
        (0, import_dom_query.raf)(() => {
          (0, import_focus_visible.setInteractionModality)("virtual");
          exec(true);
        });
        const rafCleanup = (0, import_dom_query.raf)(() => exec(true));
        cleanups.push(rafCleanup);
        const contentEl = import_cascade_select.dom.getContentEl(scope);
        const observerCleanup = (0, import_dom_query.observeAttributes)(contentEl, {
          attributes: ["data-activedescendant"],
          callback: () => exec(false)
        });
        cleanups.push(observerCleanup);
        return () => {
          cleanups.forEach((cleanup) => cleanup());
        };
      }
    },
    actions: {
      setValue(params) {
        set.value(params, params.event.value);
      },
      clearValue(params) {
        set.value(params, []);
      },
      setHighlightedValue(params) {
        const { event } = params;
        set.highlightedValue(params, event.value);
      },
      clearHighlightedValue(params) {
        set.highlightedValue(params, []);
      },
      reposition({ context, prop, scope, event }) {
        const positionerEl = () => import_cascade_select.dom.getPositionerEl(scope);
        (0, import_popper.getPlacement)(import_cascade_select.dom.getTriggerEl(scope), positionerEl, {
          ...prop("positioning"),
          ...event.options,
          defer: true,
          listeners: false,
          onComplete(data) {
            context.set("currentPlacement", data.placement);
          }
        });
      },
      selectItem(params) {
        const { context, prop, event } = params;
        const collection = prop("collection");
        const multiple = prop("multiple");
        const value = context.get("value");
        const itemValue = event.value;
        const indexPath = event.indexPath ?? collection.getIndexPath(itemValue);
        const node = collection.at(indexPath);
        const hasChildren = collection.isBranchNode(node);
        if (prop("allowParentSelection")) {
          if (multiple) {
            const filteredValue = value.filter((v) => {
              const shortPath = v.length < itemValue.length ? v : itemValue;
              const longPath = v.length < itemValue.length ? itemValue : v;
              const hasRelation = longPath.slice(0, shortPath.length).every((val, i) => val === shortPath[i]);
              return !hasRelation && !(0, import_utils.isEqual)(v, itemValue);
            });
            set.value(params, [...filteredValue, itemValue]);
          } else {
            set.value(params, [itemValue]);
          }
          if (hasChildren) set.highlightedValue(params, itemValue);
        } else {
          if (hasChildren) {
            if (multiple && value.length > 0) {
              set.value(params, [...value.slice(0, -1), itemValue]);
            } else {
              set.value(params, [itemValue]);
            }
            set.highlightedValue(params, itemValue);
          } else {
            if (multiple) {
              const existingIndex = value.findIndex((path) => (0, import_utils.isEqual)(path, itemValue));
              if (existingIndex >= 0) {
                const newValues = [...value];
                newValues.splice(existingIndex, 1);
                set.value(params, newValues);
              } else {
                set.value(params, [...value, itemValue]);
              }
            } else {
              set.value(params, [itemValue]);
            }
          }
        }
      },
      clearItem(params) {
        const { context, event } = params;
        const value = context.get("value");
        const newValue = value.filter((v) => !(0, import_utils.isEqual)(v, event.value));
        set.value(params, newValue);
      },
      selectHighlightedItem({ context, send }) {
        const indexPath = context.get("highlightedIndexPath");
        const value = context.get("highlightedValue");
        if (value) {
          send({ type: "ITEM.SELECT", value, indexPath });
        }
      },
      highlightFirstItem(params) {
        const { context, prop } = params;
        const collection = prop("collection");
        const highlightedValue = context.get("highlightedValue");
        let parentNode;
        if (!highlightedValue.length) {
          parentNode = collection.rootNode;
        } else {
          const indexPath = context.get("highlightedIndexPath");
          parentNode = collection.getParentNode(indexPath) ?? collection.rootNode;
        }
        const children = collection.getNodeChildren(parentNode);
        const firstChild = children.find((child) => !collection.getNodeDisabled(child));
        if (!firstChild) return;
        const firstValue = collection.getNodeValue(firstChild);
        if (!highlightedValue.length) {
          set.highlightedValue(params, [firstValue]);
        } else {
          const parentPath = highlightedValue.slice(0, -1);
          set.highlightedValue(params, [...parentPath, firstValue]);
        }
      },
      highlightLastItem(params) {
        const { context, prop } = params;
        const collection = prop("collection");
        const highlightedValue = context.get("highlightedValue");
        let parentNode;
        if (!highlightedValue.length) {
          parentNode = collection.rootNode;
        } else {
          const indexPath = context.get("highlightedIndexPath");
          parentNode = collection.getParentNode(indexPath) ?? collection.rootNode;
        }
        const children = collection.getNodeChildren(parentNode);
        const lastChild = children.findLast((child) => !collection.getNodeDisabled(child));
        if (!lastChild) return;
        const lastValue = collection.getNodeValue(lastChild);
        if (!highlightedValue.length) {
          set.highlightedValue(params, [lastValue]);
        } else {
          const parentPath = highlightedValue.slice(0, -1);
          set.highlightedValue(params, [...parentPath, lastValue]);
        }
      },
      highlightNextItem(params) {
        const { context, prop } = params;
        const collection = prop("collection");
        const highlightedValue = context.get("highlightedValue");
        if (!highlightedValue.length) return;
        const indexPath = context.get("highlightedIndexPath");
        const nextSibling = collection.getNextSibling(indexPath);
        if (!nextSibling) return;
        const nextValue = collection.getNodeValue(nextSibling);
        if (highlightedValue.length === 1) {
          set.highlightedValue(params, [nextValue]);
        } else {
          const parentPath = highlightedValue.slice(0, -1);
          set.highlightedValue(params, [...parentPath, nextValue]);
        }
      },
      highlightPreviousItem(params) {
        const { context, prop } = params;
        const collection = prop("collection");
        const highlightedValue = context.get("highlightedValue");
        if (!highlightedValue.length) return;
        const indexPath = context.get("highlightedIndexPath");
        const previousSibling = collection.getPreviousSibling(indexPath);
        if (!previousSibling) return;
        const prevValue = collection.getNodeValue(previousSibling);
        if (highlightedValue.length === 1) {
          set.highlightedValue(params, [prevValue]);
        } else {
          const parentPath = highlightedValue.slice(0, -1);
          set.highlightedValue(params, [...parentPath, prevValue]);
        }
      },
      highlightFirstChild(params) {
        const { context, prop } = params;
        const collection = prop("collection");
        const highlightedValue = context.get("highlightedValue");
        if (!highlightedValue.length) return;
        const indexPath = context.get("highlightedIndexPath");
        const node = collection.getFirstNode(collection.at(indexPath));
        if (!node) return;
        const childValue = collection.getNodeValue(node);
        set.highlightedValue(params, [...highlightedValue, childValue]);
      },
      highlightParent(params) {
        const { context } = params;
        const highlightedValue = context.get("highlightedValue");
        if (!highlightedValue.length) return;
        const parentPath = highlightedValue.slice(0, -1);
        set.highlightedValue(params, parentPath);
      },
      setInitialFocus({ scope }) {
        (0, import_dom_query.raf)(() => {
          const contentEl = import_cascade_select.dom.getContentEl(scope);
          contentEl?.focus({ preventScroll: true });
        });
      },
      focusTriggerEl({ event, scope }) {
        if (!restoreFocusFn(event)) return;
        (0, import_dom_query.raf)(() => {
          const triggerEl = import_cascade_select.dom.getTriggerEl(scope);
          triggerEl?.focus({ preventScroll: true });
        });
      },
      invokeOnOpen({ prop, context }) {
        prop("onOpenChange")?.({ open: true, value: context.get("value") });
      },
      invokeOnClose({ prop, context }) {
        prop("onOpenChange")?.({ open: false, value: context.get("value") });
      },
      toggleVisibility({ send, prop }) {
        if (prop("open") != null) {
          send({ type: prop("open") ? "CONTROLLED.OPEN" : "CONTROLLED.CLOSE" });
        }
      },
      highlightFirstSelectedItem(params) {
        const { context } = params;
        const value = context.get("value");
        if ((0, import_utils.isEmpty)(value)) return;
        const mostRecentSelection = (0, import_utils.last)(value);
        if (mostRecentSelection) {
          set.highlightedValue(params, mostRecentSelection);
        }
      },
      createGraceArea({ context, event, scope }) {
        const value = event.value.toString();
        const triggerElement = import_cascade_select.dom.getItemEl(scope, value);
        if (!triggerElement) return;
        const exitPoint = { x: event.clientX, y: event.clientY };
        const triggerRect = triggerElement.getBoundingClientRect();
        const nextLevelEl = import_cascade_select.dom.getListEl(scope, value);
        if (!nextLevelEl) {
          return;
        }
        const targetRect = nextLevelEl.getBoundingClientRect();
        const graceArea = (0, import_cascade_select2.createGraceArea)(exitPoint, triggerRect, targetRect);
        context.set("graceArea", graceArea);
        setTimeout(() => {
          context.set("graceArea", null);
        }, 300);
      },
      clearGraceArea({ context }) {
        context.set("graceArea", null);
      },
      setHighlightingForHoveredItem(params) {
        const { prop, event } = params;
        const collection = prop("collection");
        const node = collection.at(event.indexPath);
        let newHighlightedValue;
        if (node && collection.isBranchNode(node)) {
          newHighlightedValue = event.value;
        } else {
          newHighlightedValue = event.value.slice(0, -1);
        }
        set.highlightedValue(params, newHighlightedValue);
      },
      syncInputValue({ context, scope }) {
        const inputEl = import_cascade_select.dom.getHiddenInputEl(scope);
        if (!inputEl) return;
        (0, import_dom_query.setElementValue)(inputEl, context.hash("value"));
      },
      dispatchChangeEvent({ scope, context }) {
        (0, import_dom_query.dispatchInputValueEvent)(import_cascade_select.dom.getHiddenInputEl(scope), { value: context.hash("value") });
      },
      scrollContentToTop({ scope, prop }) {
        const scrollToIndexFn = prop("scrollToIndexFn");
        (0, import_dom_query.raf)(() => {
          const contentEl = import_cascade_select.dom.getContentEl(scope);
          const listEls = contentEl?.querySelectorAll('[data-part="list"]');
          listEls?.forEach((listEl, index) => {
            if (scrollToIndexFn) {
              scrollToIndexFn({ index: 0, immediate: true, depth: index });
            } else {
              listEl.scrollTop = 0;
            }
          });
        });
      }
    }
  }
});
var set = {
  value({ context, prop }, value) {
    const collection = prop("collection");
    context.set("value", value);
    const valueIndexPath = value.map((v) => collection.getIndexPath(v));
    context.set("valueIndexPath", valueIndexPath);
    const selectedItems = valueIndexPath.map((indexPath) => {
      return indexPath.map((_, index) => {
        const partialPath = indexPath.slice(0, index + 1);
        return collection.at(partialPath);
      });
    });
    context.set("selectedItems", selectedItems);
    prop("onValueChange")?.({ value, items: selectedItems });
  },
  highlightedValue({ context, prop }, value) {
    const collection = prop("collection");
    context.set("highlightedValue", value);
    const rawPath = value == null ? [] : collection.getIndexPath(value);
    const highlightedIndexPath = rawPath ?? [];
    context.set("highlightedIndexPath", highlightedIndexPath);
    const highlightedItems = highlightedIndexPath.map((_, index) => {
      const partialPath = highlightedIndexPath.slice(0, index + 1);
      return collection.at(partialPath);
    });
    context.set("highlightedItems", highlightedItems);
    prop("onHighlightChange")?.({ highlightedValue: value, highlightedItems });
  }
};
function restoreFocusFn(event) {
  const v = event.restoreFocus ?? event.previousEvent?.restoreFocus;
  return v == null || !!v;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  machine
});
