// @ark-ui/react/dist/components/splitter/index.js
import { splitterAnatomy } from "./splitter.anatomy.js";
import { createSplitterRegistry } from "./create-splitter-registry.js";
import { useSplitterContext } from "./use-splitter-context.js";
import { SplitterContext } from "./splitter-context.js";
import { SplitterPanel } from "./splitter-panel.js";
import { SplitterResizeTrigger } from "./splitter-resize-trigger.js";
import { SplitterResizeTriggerIndicator } from "./splitter-resize-trigger-indicator.js";
import { useSplitter } from "./use-splitter.js";
import { SplitterRoot } from "./splitter-root.js";
import { SplitterRootProvider } from "./splitter-root-provider.js";
import { splitter_exports } from "./splitter.js";
import { layout as getSplitterLayout } from "@zag-js/splitter";
export { splitter_exports as Splitter, SplitterContext, SplitterPanel, SplitterResizeTrigger, SplitterResizeTriggerIndicator, SplitterRoot, SplitterRootProvider, createSplitterRegistry, getSplitterLayout, splitterAnatomy, useSplitter, useSplitterContext };
