// @ark-ui/react/dist/components/splitter/index.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const require_create_splitter_registry = require("./create-splitter-registry.cjs");
const require_use_splitter_context = require("./use-splitter-context.cjs");
const require_splitter_context = require("./splitter-context.cjs");
const require_splitter_panel = require("./splitter-panel.cjs");
const require_splitter_resize_trigger = require("./splitter-resize-trigger.cjs");
const require_splitter_resize_trigger_indicator = require("./splitter-resize-trigger-indicator.cjs");
const require_use_splitter = require("./use-splitter.cjs");
const require_splitter_root = require("./splitter-root.cjs");
const require_splitter_root_provider = require("./splitter-root-provider.cjs");
const require_splitter = require("./splitter.cjs");
let _zag_js_splitter = require("@zag-js/splitter");
Object.defineProperty(exports, "Splitter", {
	enumerable: true,
	get: function() {
		return require_splitter.splitter_exports;
	}
});
exports.SplitterContext = require_splitter_context.SplitterContext;
exports.SplitterPanel = require_splitter_panel.SplitterPanel;
exports.SplitterResizeTrigger = require_splitter_resize_trigger.SplitterResizeTrigger;
exports.SplitterResizeTriggerIndicator = require_splitter_resize_trigger_indicator.SplitterResizeTriggerIndicator;
exports.SplitterRoot = require_splitter_root.SplitterRoot;
exports.SplitterRootProvider = require_splitter_root_provider.SplitterRootProvider;
exports.createSplitterRegistry = require_create_splitter_registry.createSplitterRegistry;
Object.defineProperty(exports, "getSplitterLayout", {
	enumerable: true,
	get: function() {
		return _zag_js_splitter.layout;
	}
});
Object.defineProperty(exports, "splitterAnatomy", {
	enumerable: true,
	get: function() {
		return _zag_js_splitter.anatomy;
	}
});
exports.useSplitter = require_use_splitter.useSplitter;
exports.useSplitterContext = require_use_splitter_context.useSplitterContext;
