// @ark-ui/react/dist/components/splitter/splitter.cjs
const require_runtime = require("../../_virtual/_rolldown/runtime.cjs");
const require_create_splitter_registry = require("./create-splitter-registry.cjs");
const require_splitter_context = require("./splitter-context.cjs");
const require_splitter_panel = require("./splitter-panel.cjs");
const require_splitter_resize_trigger = require("./splitter-resize-trigger.cjs");
const require_splitter_resize_trigger_indicator = require("./splitter-resize-trigger-indicator.cjs");
const require_splitter_root = require("./splitter-root.cjs");
const require_splitter_root_provider = require("./splitter-root-provider.cjs");
let _zag_js_splitter = require("@zag-js/splitter");
//#region src/components/splitter/splitter.ts
var splitter_exports = /* @__PURE__ */ require_runtime.__exportAll({
	Context: () => require_splitter_context.SplitterContext,
	Panel: () => require_splitter_panel.SplitterPanel,
	ResizeTrigger: () => require_splitter_resize_trigger.SplitterResizeTrigger,
	ResizeTriggerIndicator: () => require_splitter_resize_trigger_indicator.SplitterResizeTriggerIndicator,
	Root: () => require_splitter_root.SplitterRoot,
	RootProvider: () => require_splitter_root_provider.SplitterRootProvider,
	createRegistry: () => require_create_splitter_registry.createSplitterRegistry,
	getLayout: () => _zag_js_splitter.layout
});
//#endregion
Object.defineProperty(exports, "splitter_exports", {
	enumerable: true,
	get: function() {
		return splitter_exports;
	}
});
