// @ark-ui/react/dist/components/splitter/splitter.js
import { __exportAll } from "../../_virtual/_rolldown/runtime.js";
import { createSplitterRegistry } from "./create-splitter-registry.js";
import { SplitterContext } from "./splitter-context.js";
import { SplitterPanel } from "./splitter-panel.js";
import { SplitterResizeTrigger } from "./splitter-resize-trigger.js";
import { SplitterResizeTriggerIndicator } from "./splitter-resize-trigger-indicator.js";
import { SplitterRoot } from "./splitter-root.js";
import { SplitterRootProvider } from "./splitter-root-provider.js";
import { layout as getLayout } from "@zag-js/splitter";
//#region src/components/splitter/splitter.ts
var splitter_exports = /* @__PURE__ */ __exportAll({
	Context: () => SplitterContext,
	Panel: () => SplitterPanel,
	ResizeTrigger: () => SplitterResizeTrigger,
	ResizeTriggerIndicator: () => SplitterResizeTriggerIndicator,
	Root: () => SplitterRoot,
	RootProvider: () => SplitterRootProvider,
	createRegistry: () => createSplitterRegistry,
	getLayout: () => getLayout
});
//#endregion
export { splitter_exports };
