// @ark-ui/react/dist/components/segment-group/segment-group.cjs
const require_runtime = require("../../_virtual/_rolldown/runtime.cjs");
const require_segment_group_context = require("./segment-group-context.cjs");
const require_segment_group_indicator = require("./segment-group-indicator.cjs");
const require_segment_group_item = require("./segment-group-item.cjs");
const require_segment_group_item_context = require("./segment-group-item-context.cjs");
const require_segment_group_item_control = require("./segment-group-item-control.cjs");
const require_segment_group_item_hidden_input = require("./segment-group-item-hidden-input.cjs");
const require_segment_group_item_text = require("./segment-group-item-text.cjs");
const require_segment_group_label = require("./segment-group-label.cjs");
const require_segment_group_root = require("./segment-group-root.cjs");
const require_segment_group_root_provider = require("./segment-group-root-provider.cjs");
//#region src/components/segment-group/segment-group.ts
var segment_group_exports = /* @__PURE__ */ require_runtime.__exportAll({
	Context: () => require_segment_group_context.SegmentGroupContext,
	Indicator: () => require_segment_group_indicator.SegmentGroupIndicator,
	Item: () => require_segment_group_item.SegmentGroupItem,
	ItemContext: () => require_segment_group_item_context.SegmentGroupItemContext,
	ItemControl: () => require_segment_group_item_control.SegmentGroupItemControl,
	ItemHiddenInput: () => require_segment_group_item_hidden_input.SegmentGroupItemHiddenInput,
	ItemText: () => require_segment_group_item_text.SegmentGroupItemText,
	Label: () => require_segment_group_label.SegmentGroupLabel,
	Root: () => require_segment_group_root.SegmentGroupRoot,
	RootProvider: () => require_segment_group_root_provider.SegmentGroupRootProvider
});
//#endregion
Object.defineProperty(exports, "segment_group_exports", {
	enumerable: true,
	get: function() {
		return segment_group_exports;
	}
});
