// @chakra-ui/react/dist/cjs/components/splitter/index.cjs
"use strict";
'use strict';

var splitter = require('./splitter.cjs');
var splitter$1 = require('@ark-ui/react/splitter');
var namespace = require('./namespace.cjs');



exports.SplitterContext = splitter.SplitterContext;
exports.SplitterPanel = splitter.SplitterPanel;
exports.SplitterPropsProvider = splitter.SplitterPropsProvider;
exports.SplitterResizeTrigger = splitter.SplitterResizeTrigger;
exports.SplitterResizeTriggerIndicator = splitter.SplitterResizeTriggerIndicator;
exports.SplitterResizeTriggerSeparator = splitter.SplitterResizeTriggerSeparator;
exports.SplitterRoot = splitter.SplitterRoot;
exports.SplitterRootProvider = splitter.SplitterRootProvider;
exports.useSplitterStyles = splitter.useSplitterStyles;
Object.defineProperty(exports, "createSplitterRegistry", {
  enumerable: true,
  get: function () { return splitter$1.createSplitterRegistry; }
});
Object.defineProperty(exports, "getSplitterLayout", {
  enumerable: true,
  get: function () { return splitter$1.getSplitterLayout; }
});
Object.defineProperty(exports, "useSplitter", {
  enumerable: true,
  get: function () { return splitter$1.useSplitter; }
});
Object.defineProperty(exports, "useSplitterContext", {
  enumerable: true,
  get: function () { return splitter$1.useSplitterContext; }
});
exports.Splitter = namespace;
