// @ark-ui/react/dist/components/rating-group/use-rating-group.js
"use client";
import { useEnvironmentContext } from "../../providers/environment/use-environment-context.js";
import { useLocaleContext } from "../../providers/locale/use-locale-context.js";
import { useFieldContext } from "../field/use-field-context.js";
import * as rating from "@zag-js/rating-group";
import { useId } from "react";
import { normalizeProps, useMachine } from "@zag-js/react";
//#region src/components/rating-group/use-rating-group.ts
var useRatingGroup = (props) => {
	const id = useId();
	const { getRootNode } = useEnvironmentContext();
	const { dir } = useLocaleContext();
	const field = useFieldContext();
	const machineProps = {
		id,
		ids: {
			label: field?.ids.label,
			hiddenInput: field?.ids.control
		},
		dir,
		disabled: field?.disabled,
		readOnly: field?.readOnly,
		required: field?.required,
		getRootNode,
		...props
	};
	const service = useMachine(rating.machine, machineProps);
	return rating.connect(service, normalizeProps);
};
//#endregion
export { useRatingGroup };
