// @ark-ui/react/dist/providers/environment/environment-provider.cjs
"use client";
const require_run_if_fn = require("../../utils/run-if-fn.cjs");
const require_use_environment_context = require("./use-environment-context.cjs");
let react = require("react");
let react_jsx_runtime = require("react/jsx-runtime");
let _zag_js_dom_query = require("@zag-js/dom-query");
//#region src/providers/environment/environment-provider.tsx
var EnvironmentProvider = (props) => {
	const { value, children } = props;
	const [spanRef, setSpanRef] = (0, react.useState)();
	const getRootNode = (0, react.useMemo)(() => {
		return () => require_run_if_fn.runIfFn(value) ?? spanRef?.getRootNode() ?? document;
	}, [value, spanRef]);
	return /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(require_use_environment_context.EnvironmentContextProvider, {
		value: (0, react.useMemo)(() => ({
			getRootNode,
			getWindow: () => (0, _zag_js_dom_query.getWindow)(getRootNode()),
			getDocument: () => (0, _zag_js_dom_query.getDocument)(getRootNode())
		}), [getRootNode]),
		children: [children, !value && /* @__PURE__ */ (0, react_jsx_runtime.jsx)("span", {
			hidden: true,
			ref: setSpanRef
		})]
	});
};
//#endregion
exports.EnvironmentProvider = EnvironmentProvider;
