// @zag-js/menu/dist/menu.connect.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/menu.connect.ts
var menu_connect_exports = {};
__export(menu_connect_exports, {
  connect: () => connect
});
module.exports = __toCommonJS(menu_connect_exports);
var import_core = require("@zag-js/core");
var import_dom_query = require("@zag-js/dom-query");
var import_popper = require("@zag-js/popper");
var import_utils = require("@zag-js/utils");
var import_menu = require("./menu.anatomy.js");
var dom = __toESM(require("./menu.dom.js"));
var import_menu2 = require("./menu.utils.js");
function connect(service, normalize) {
  const { context, send, state, computed, prop, scope } = service;
  const open = state.hasTag("open");
  const isSubmenu = context.get("isSubmenu");
  const isTypingAhead = computed("isTypingAhead");
  const composite = prop("composite");
  const currentPlacement = context.get("currentPlacement");
  const currentPlacementSide = currentPlacement ? (0, import_popper.getPlacementSide)(currentPlacement) : void 0;
  const anchorPoint = context.get("anchorPoint");
  const highlightedValue = context.get("highlightedValue");
  const triggerValue = context.get("triggerValue");
  const popperStyles = (0, import_popper.getPlacementStyles)({
    ...prop("positioning"),
    placement: anchorPoint ? "bottom" : currentPlacement
  });
  function getItemState(props) {
    return {
      id: dom.getItemId(scope, props.value),
      disabled: !!props.disabled,
      highlighted: highlightedValue === props.value
    };
  }
  function getOptionItemProps(props) {
    const valueText = props.valueText ?? props.value;
    return { ...props, id: props.value, valueText };
  }
  function getOptionItemState(props) {
    const itemState = getItemState(getOptionItemProps(props));
    return {
      ...itemState,
      checked: !!props.checked
    };
  }
  function getItemProps(props) {
    const { closeOnSelect, valueText, value } = props;
    const itemState = getItemState(props);
    const id = dom.getItemId(scope, value);
    return normalize.element({
      ...import_menu.parts.item.attrs,
      id,
      role: "menuitem",
      "aria-disabled": (0, import_dom_query.ariaAttr)(itemState.disabled),
      "data-disabled": (0, import_dom_query.dataAttr)(itemState.disabled),
      "data-ownedby": dom.getContentId(scope),
      "data-highlighted": (0, import_dom_query.dataAttr)(itemState.highlighted),
      "data-value": value,
      "data-valuetext": valueText,
      onDragStart(event) {
        const isLink = event.currentTarget.matches("a[href]");
        if (isLink) event.preventDefault();
      },
      onPointerMove(event) {
        if (itemState.disabled) return;
        if (event.pointerType !== "mouse") return;
        const target = event.currentTarget;
        if (itemState.highlighted) return;
        const point = (0, import_dom_query.getEventPoint)(event);
        send({ type: "ITEM_POINTERMOVE", id, target, closeOnSelect, point });
      },
      onPointerLeave(event) {
        if (itemState.disabled) return;
        if (event.pointerType !== "mouse") return;
        const pointerMoved = service.event.previous()?.type.includes("POINTER");
        if (!pointerMoved) return;
        const target = event.currentTarget;
        send({ type: "ITEM_POINTERLEAVE", id, target, closeOnSelect });
      },
      onPointerDown(event) {
        if (itemState.disabled) return;
        const target = event.currentTarget;
        send({ type: "ITEM_POINTERDOWN", target, id, closeOnSelect });
      },
      onClick(event) {
        if ((0, import_dom_query.isDownloadingEvent)(event)) return;
        if ((0, import_dom_query.isOpeningInNewTab)(event)) return;
        if (itemState.disabled) return;
        const target = event.currentTarget;
        send({ type: "ITEM_CLICK", target, id, closeOnSelect });
      }
    });
  }
  return {
    highlightedValue,
    open,
    setOpen(nextOpen) {
      const open2 = state.hasTag("open");
      if (open2 === nextOpen) return;
      send({ type: nextOpen ? "OPEN" : "CLOSE" });
    },
    triggerValue,
    setTriggerValue(value) {
      send({ type: "TRIGGER_VALUE.SET", value });
    },
    setHighlightedValue(value) {
      send({ type: "HIGHLIGHTED.SET", value });
    },
    setParent(parent) {
      send({ type: "PARENT.SET", value: parent, id: parent.prop("id") });
    },
    setChild(child) {
      send({ type: "CHILD.SET", value: child, id: child.prop("id") });
    },
    reposition(options = {}) {
      send({ type: "POSITIONING.SET", options });
    },
    addItemListener(props) {
      const node = scope.getById(props.id);
      if (!node) return;
      const listener = () => props.onSelect?.();
      node.addEventListener(dom.itemSelectEvent, listener);
      return () => node.removeEventListener(dom.itemSelectEvent, listener);
    },
    getContextTriggerProps(props = {}) {
      const { value } = props;
      const current = value == null ? false : triggerValue === value;
      const contextTriggerId = dom.getContextTriggerId(scope, value);
      return normalize.element({
        ...import_menu.parts.contextTrigger.attrs,
        dir: prop("dir"),
        id: contextTriggerId,
        "data-ownedby": scope.id,
        "data-value": value,
        "data-current": (0, import_dom_query.dataAttr)(current),
        "data-state": open ? "open" : "closed",
        onPointerDown(event) {
          if (event.pointerType === "mouse") return;
          const point = (0, import_dom_query.getEventPoint)(event);
          send({ type: "CONTEXT_MENU_START", point, value });
        },
        onPointerCancel(event) {
          if (event.pointerType === "mouse") return;
          send({ type: "CONTEXT_MENU_CANCEL" });
        },
        onPointerMove(event) {
          if (event.pointerType === "mouse") return;
          send({ type: "CONTEXT_MENU_CANCEL" });
        },
        onPointerUp(event) {
          if (event.pointerType === "mouse") return;
          send({ type: "CONTEXT_MENU_CANCEL" });
        },
        onContextMenu(event) {
          const point = (0, import_dom_query.getEventPoint)(event);
          const shouldSwitch = open && value != null && !current;
          send({
            type: shouldSwitch ? "TRIGGER_VALUE.SET" : "CONTEXT_MENU",
            point,
            value
          });
          event.preventDefault();
        },
        style: {
          WebkitTouchCallout: "none",
          WebkitUserSelect: "none",
          userSelect: "none"
        }
      });
    },
    getTriggerItemProps(childApi) {
      const triggerProps = childApi.getTriggerProps();
      return (0, import_core.mergeProps)(getItemProps({ value: triggerProps.id }), triggerProps);
    },
    getTriggerProps(props = {}) {
      const { value } = props;
      const current = value == null ? false : triggerValue === value;
      const triggerId = dom.getTriggerId(scope, value);
      return normalize.button({
        ...isSubmenu ? import_menu.parts.triggerItem.attrs : import_menu.parts.trigger.attrs,
        "data-placement": currentPlacement,
        "data-side": currentPlacementSide,
        type: "button",
        dir: prop("dir"),
        id: triggerId,
        // Multi-trigger attributes - only included when value is provided
        ...value != null && {
          "data-ownedby": scope.id,
          "data-value": value,
          "data-current": (0, import_dom_query.dataAttr)(current)
        },
        "data-uid": prop("id"),
        "aria-haspopup": composite ? "menu" : "dialog",
        "aria-controls": dom.getContentId(scope),
        "data-controls": dom.getContentId(scope),
        "aria-expanded": value == null ? open : open && current,
        "data-state": open ? "open" : "closed",
        onPointerMove(event) {
          if (event.pointerType !== "mouse") return;
          const disabled = dom.isTargetDisabled(event.currentTarget);
          if (disabled || !isSubmenu) return;
          const point = (0, import_dom_query.getEventPoint)(event);
          send({ type: "TRIGGER_POINTERMOVE", target: event.currentTarget, point });
        },
        onPointerLeave(event) {
          if (dom.isTargetDisabled(event.currentTarget)) return;
          if (event.pointerType !== "mouse") return;
          if (!isSubmenu) return;
          (0, import_menu2.setParentRoutingLock)(service.refs.get("parent"), true);
          const point = (0, import_dom_query.getEventPoint)(event);
          send({
            type: "TRIGGER_POINTERLEAVE",
            target: event.currentTarget,
            point
          });
        },
        onPointerDown(event) {
          if (dom.isTargetDisabled(event.currentTarget)) return;
          if ((0, import_dom_query.isContextMenuEvent)(event)) return;
          event.preventDefault();
        },
        onClick(event) {
          if (event.defaultPrevented) return;
          if (dom.isTargetDisabled(event.currentTarget)) return;
          const shouldSwitch = open && value != null && !current;
          send({
            type: shouldSwitch ? "TRIGGER_VALUE.SET" : "TRIGGER_CLICK",
            target: event.currentTarget,
            value
          });
        },
        onBlur() {
          send({ type: "TRIGGER_BLUR" });
        },
        onFocus() {
          send({ type: "TRIGGER_FOCUS" });
        },
        onKeyDown(event) {
          if (event.defaultPrevented) return;
          const keyMap = {
            ArrowDown() {
              send({ type: "ARROW_DOWN", value });
            },
            ArrowUp() {
              send({ type: "ARROW_UP", value });
            },
            Enter() {
              send({ type: "ARROW_DOWN", src: "enter", value });
            },
            Space() {
              send({ type: "ARROW_DOWN", src: "space", value });
            }
          };
          const key = (0, import_dom_query.getEventKey)(event, {
            orientation: "vertical",
            dir: prop("dir")
          });
          const exec = keyMap[key];
          if (exec) {
            event.preventDefault();
            exec(event);
          }
        }
      });
    },
    getIndicatorProps() {
      return normalize.element({
        ...import_menu.parts.indicator.attrs,
        dir: prop("dir"),
        "data-state": open ? "open" : "closed"
      });
    },
    getPositionerProps() {
      return normalize.element({
        ...import_menu.parts.positioner.attrs,
        dir: prop("dir"),
        id: dom.getPositionerId(scope),
        style: popperStyles.floating
      });
    },
    getArrowProps() {
      return normalize.element({
        id: dom.getArrowId(scope),
        ...import_menu.parts.arrow.attrs,
        dir: prop("dir"),
        style: popperStyles.arrow
      });
    },
    getArrowTipProps() {
      return normalize.element({
        ...import_menu.parts.arrowTip.attrs,
        dir: prop("dir"),
        style: popperStyles.arrowTip
      });
    },
    getContentProps() {
      return normalize.element({
        ...import_menu.parts.content.attrs,
        id: dom.getContentId(scope),
        "aria-label": prop("aria-label"),
        hidden: !open,
        "data-state": open ? "open" : "closed",
        role: composite ? "menu" : "dialog",
        tabIndex: 0,
        dir: prop("dir"),
        "aria-activedescendant": computed("highlightedId") || void 0,
        "aria-labelledby": anchorPoint ? dom.getContextTriggerId(scope, triggerValue ?? void 0) : dom.getTriggerId(scope, triggerValue ?? void 0),
        "data-placement": currentPlacement,
        "data-side": currentPlacementSide,
        onPointerEnter(event) {
          if (event.pointerType !== "mouse") return;
          send({ type: "MENU_POINTERENTER" });
        },
        onKeyDown(event) {
          if (event.defaultPrevented) return;
          if (!(0, import_dom_query.contains)(event.currentTarget, (0, import_dom_query.getEventTarget)(event))) return;
          const target = (0, import_dom_query.getEventTarget)(event);
          const sameMenu = target?.closest("[role=menu]") === event.currentTarget || target === event.currentTarget;
          if (!sameMenu) return;
          if (event.key === "Tab") {
            const valid = (0, import_dom_query.isValidTabEvent)(event);
            if (!valid) {
              event.preventDefault();
              return;
            }
          }
          const keyMap = {
            ArrowDown() {
              send({ type: "ARROW_DOWN" });
            },
            ArrowUp() {
              send({ type: "ARROW_UP" });
            },
            ArrowLeft() {
              send({ type: "ARROW_LEFT" });
            },
            ArrowRight() {
              send({ type: "ARROW_RIGHT" });
            },
            Enter() {
              send({ type: "ENTER" });
            },
            Space(event2) {
              if (isTypingAhead) {
                send({ type: "TYPEAHEAD", key: event2.key });
              } else {
                keyMap.Enter?.(event2);
              }
            },
            Home() {
              send({ type: "HOME" });
            },
            End() {
              send({ type: "END" });
            }
          };
          const key = (0, import_dom_query.getEventKey)(event, { dir: prop("dir") });
          const exec = keyMap[key];
          if (exec) {
            exec(event);
            event.stopPropagation();
            event.preventDefault();
            return;
          }
          if (!prop("typeahead")) return;
          if (!(0, import_dom_query.isPrintableKey)(event)) return;
          if ((0, import_dom_query.isModifierKey)(event)) return;
          if ((0, import_dom_query.isEditableElement)(target)) return;
          send({ type: "TYPEAHEAD", key: event.key });
          event.preventDefault();
        }
      });
    },
    getSeparatorProps() {
      return normalize.element({
        ...import_menu.parts.separator.attrs,
        role: "separator",
        dir: prop("dir"),
        "aria-orientation": "horizontal"
      });
    },
    getItemState,
    getItemProps,
    getOptionItemState,
    getOptionItemProps(props) {
      const { type, disabled, closeOnSelect } = props;
      const option = getOptionItemProps(props);
      const itemState = getOptionItemState(props);
      return {
        ...getItemProps(option),
        ...normalize.element({
          "data-type": type,
          ...import_menu.parts.item.attrs,
          dir: prop("dir"),
          "data-value": option.value,
          role: `menuitem${type}`,
          "aria-checked": !!itemState.checked,
          "data-state": itemState.checked ? "checked" : "unchecked",
          onClick(event) {
            if (disabled) return;
            if ((0, import_dom_query.isDownloadingEvent)(event)) return;
            if ((0, import_dom_query.isOpeningInNewTab)(event)) return;
            const target = event.currentTarget;
            send({ type: "ITEM_CLICK", target, option, closeOnSelect });
          }
        })
      };
    },
    getItemIndicatorProps(props) {
      const itemState = getOptionItemState((0, import_utils.cast)(props));
      const dataState = itemState.checked ? "checked" : "unchecked";
      return normalize.element({
        ...import_menu.parts.itemIndicator.attrs,
        dir: prop("dir"),
        "data-disabled": (0, import_dom_query.dataAttr)(itemState.disabled),
        "data-highlighted": (0, import_dom_query.dataAttr)(itemState.highlighted),
        "data-state": (0, import_utils.hasProp)(props, "checked") ? dataState : void 0,
        hidden: (0, import_utils.hasProp)(props, "checked") ? !itemState.checked : void 0
      });
    },
    getItemTextProps(props) {
      const itemState = getOptionItemState((0, import_utils.cast)(props));
      const dataState = itemState.checked ? "checked" : "unchecked";
      return normalize.element({
        ...import_menu.parts.itemText.attrs,
        dir: prop("dir"),
        "data-disabled": (0, import_dom_query.dataAttr)(itemState.disabled),
        "data-highlighted": (0, import_dom_query.dataAttr)(itemState.highlighted),
        "data-state": (0, import_utils.hasProp)(props, "checked") ? dataState : void 0
      });
    },
    getItemGroupLabelProps(props) {
      return normalize.element({
        ...import_menu.parts.itemGroupLabel.attrs,
        id: dom.getGroupLabelId(scope, props.htmlFor),
        dir: prop("dir")
      });
    },
    getItemGroupProps(props) {
      return normalize.element({
        id: dom.getGroupId(scope, props.id),
        ...import_menu.parts.itemGroup.attrs,
        dir: prop("dir"),
        "aria-labelledby": dom.getGroupLabelId(scope, props.id),
        role: "group"
      });
    }
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  connect
});
