// @zag-js/scroll-area/dist/scroll-area.dom.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/scroll-area.dom.ts
var scroll_area_dom_exports = {};
__export(scroll_area_dom_exports, {
  getContentEl: () => getContentEl,
  getContentId: () => getContentId,
  getCornerEl: () => getCornerEl,
  getRootEl: () => getRootEl,
  getRootId: () => getRootId,
  getScrollbarXEl: () => getScrollbarXEl,
  getScrollbarYEl: () => getScrollbarYEl,
  getThumbXEl: () => getThumbXEl,
  getThumbYEl: () => getThumbYEl,
  getViewportEl: () => getViewportEl,
  getViewportId: () => getViewportId
});
module.exports = __toCommonJS(scroll_area_dom_exports);
var import_dom_query = require("@zag-js/dom-query");
var getRootId = (ctx) => ctx.ids?.root ?? `scroll-area-${ctx.id}`;
var getViewportId = (ctx) => ctx.ids?.viewport ?? `scroll-area-${ctx.id}:viewport`;
var getContentId = (ctx) => ctx.ids?.content ?? `scroll-area-${ctx.id}:content`;
var getRootEl = (ctx) => ctx.getById(getRootId(ctx));
var getViewportEl = (ctx) => ctx.getById(getViewportId(ctx));
var getContentEl = (ctx) => ctx.getById(getContentId(ctx));
var getScrollbarXEl = (ctx) => (0, import_dom_query.query)(getRootEl(ctx), `[data-part=scrollbar][data-orientation=horizontal][data-ownedby="${getRootId(ctx)}"]`);
var getScrollbarYEl = (ctx) => (0, import_dom_query.query)(getRootEl(ctx), `[data-part=scrollbar][data-orientation=vertical][data-ownedby="${getRootId(ctx)}"]`);
var getThumbXEl = (ctx) => (0, import_dom_query.query)(getScrollbarXEl(ctx), `[data-part=thumb][data-orientation=horizontal][data-ownedby="${getRootId(ctx)}"]`);
var getThumbYEl = (ctx) => (0, import_dom_query.query)(getScrollbarYEl(ctx), `[data-part=thumb][data-orientation=vertical][data-ownedby="${getRootId(ctx)}"]`);
var getCornerEl = (ctx) => (0, import_dom_query.query)(getRootEl(ctx), `[data-part=corner][data-ownedby="${getRootId(ctx)}"]`);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getContentEl,
  getContentId,
  getCornerEl,
  getRootEl,
  getRootId,
  getScrollbarXEl,
  getScrollbarYEl,
  getThumbXEl,
  getThumbYEl,
  getViewportEl,
  getViewportId
});
