// @ark-ui/react/dist/components/signature-pad/signature-pad.cjs
const require_runtime = require("../../_virtual/_rolldown/runtime.cjs");
const require_signature_pad_clear_trigger = require("./signature-pad-clear-trigger.cjs");
const require_signature_pad_context = require("./signature-pad-context.cjs");
const require_signature_pad_control = require("./signature-pad-control.cjs");
const require_signature_pad_guide = require("./signature-pad-guide.cjs");
const require_signature_pad_hidden_input = require("./signature-pad-hidden-input.cjs");
const require_signature_pad_label = require("./signature-pad-label.cjs");
const require_signature_pad_root = require("./signature-pad-root.cjs");
const require_signature_pad_root_provider = require("./signature-pad-root-provider.cjs");
const require_signature_pad_segment = require("./signature-pad-segment.cjs");
//#region src/components/signature-pad/signature-pad.ts
var signature_pad_exports = /* @__PURE__ */ require_runtime.__exportAll({
	ClearTrigger: () => require_signature_pad_clear_trigger.SignaturePadClearTrigger,
	Context: () => require_signature_pad_context.SignaturePadContext,
	Control: () => require_signature_pad_control.SignaturePadControl,
	Guide: () => require_signature_pad_guide.SignaturePadGuide,
	HiddenInput: () => require_signature_pad_hidden_input.SignaturePadHiddenInput,
	Label: () => require_signature_pad_label.SignaturePadLabel,
	Root: () => require_signature_pad_root.SignaturePadRoot,
	RootProvider: () => require_signature_pad_root_provider.SignaturePadRootProvider,
	Segment: () => require_signature_pad_segment.SignaturePadSegment
});
//#endregion
Object.defineProperty(exports, "signature_pad_exports", {
	enumerable: true,
	get: function() {
		return signature_pad_exports;
	}
});
