// @ark-ui/react/dist/components/tags-input/index.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const require_use_tags_input_context = require("./use-tags-input-context.cjs");
const require_tags_input_clear_trigger = require("./tags-input-clear-trigger.cjs");
const require_tags_input_context = require("./tags-input-context.cjs");
const require_tags_input_control = require("./tags-input-control.cjs");
const require_tags_input_hidden_input = require("./tags-input-hidden-input.cjs");
const require_tags_input_input = require("./tags-input-input.cjs");
const require_use_tags_input_item_context = require("./use-tags-input-item-context.cjs");
const require_tags_input_item = require("./tags-input-item.cjs");
const require_tags_input_item_context = require("./tags-input-item-context.cjs");
const require_tags_input_item_delete_trigger = require("./tags-input-item-delete-trigger.cjs");
const require_tags_input_item_input = require("./tags-input-item-input.cjs");
const require_tags_input_item_preview = require("./tags-input-item-preview.cjs");
const require_tags_input_item_text = require("./tags-input-item-text.cjs");
const require_tags_input_label = require("./tags-input-label.cjs");
const require_use_tags_input = require("./use-tags-input.cjs");
const require_tags_input_root = require("./tags-input-root.cjs");
const require_tags_input_root_provider = require("./tags-input-root-provider.cjs");
const require_tags_input = require("./tags-input.cjs");
let _zag_js_tags_input = require("@zag-js/tags-input");
Object.defineProperty(exports, "TagsInput", {
	enumerable: true,
	get: function() {
		return require_tags_input.tags_input_exports;
	}
});
exports.TagsInputClearTrigger = require_tags_input_clear_trigger.TagsInputClearTrigger;
exports.TagsInputContext = require_tags_input_context.TagsInputContext;
exports.TagsInputControl = require_tags_input_control.TagsInputControl;
exports.TagsInputHiddenInput = require_tags_input_hidden_input.TagsInputHiddenInput;
exports.TagsInputInput = require_tags_input_input.TagsInputInput;
exports.TagsInputItem = require_tags_input_item.TagsInputItem;
exports.TagsInputItemContext = require_tags_input_item_context.TagsInputItemContext;
exports.TagsInputItemDeleteTrigger = require_tags_input_item_delete_trigger.TagsInputItemDeleteTrigger;
exports.TagsInputItemInput = require_tags_input_item_input.TagsInputItemInput;
exports.TagsInputItemPreview = require_tags_input_item_preview.TagsInputItemPreview;
exports.TagsInputItemText = require_tags_input_item_text.TagsInputItemText;
exports.TagsInputLabel = require_tags_input_label.TagsInputLabel;
exports.TagsInputRoot = require_tags_input_root.TagsInputRoot;
exports.TagsInputRootProvider = require_tags_input_root_provider.TagsInputRootProvider;
Object.defineProperty(exports, "tagsInputAnatomy", {
	enumerable: true,
	get: function() {
		return _zag_js_tags_input.anatomy;
	}
});
exports.useTagsInput = require_use_tags_input.useTagsInput;
exports.useTagsInputContext = require_use_tags_input_context.useTagsInputContext;
exports.useTagsInputItemContext = require_use_tags_input_item_context.useTagsInputItemContext;
