// @ark-ui/react/dist/components/tags-input/tags-input.cjs
const require_runtime = require("../../_virtual/_rolldown/runtime.cjs");
const require_tags_input_clear_trigger = require("./tags-input-clear-trigger.cjs");
const require_tags_input_context = require("./tags-input-context.cjs");
const require_tags_input_control = require("./tags-input-control.cjs");
const require_tags_input_hidden_input = require("./tags-input-hidden-input.cjs");
const require_tags_input_input = require("./tags-input-input.cjs");
const require_tags_input_item = require("./tags-input-item.cjs");
const require_tags_input_item_context = require("./tags-input-item-context.cjs");
const require_tags_input_item_delete_trigger = require("./tags-input-item-delete-trigger.cjs");
const require_tags_input_item_input = require("./tags-input-item-input.cjs");
const require_tags_input_item_preview = require("./tags-input-item-preview.cjs");
const require_tags_input_item_text = require("./tags-input-item-text.cjs");
const require_tags_input_label = require("./tags-input-label.cjs");
const require_tags_input_root = require("./tags-input-root.cjs");
const require_tags_input_root_provider = require("./tags-input-root-provider.cjs");
//#region src/components/tags-input/tags-input.ts
var tags_input_exports = /* @__PURE__ */ require_runtime.__exportAll({
	ClearTrigger: () => require_tags_input_clear_trigger.TagsInputClearTrigger,
	Context: () => require_tags_input_context.TagsInputContext,
	Control: () => require_tags_input_control.TagsInputControl,
	HiddenInput: () => require_tags_input_hidden_input.TagsInputHiddenInput,
	Input: () => require_tags_input_input.TagsInputInput,
	Item: () => require_tags_input_item.TagsInputItem,
	ItemContext: () => require_tags_input_item_context.TagsInputItemContext,
	ItemDeleteTrigger: () => require_tags_input_item_delete_trigger.TagsInputItemDeleteTrigger,
	ItemInput: () => require_tags_input_item_input.TagsInputItemInput,
	ItemPreview: () => require_tags_input_item_preview.TagsInputItemPreview,
	ItemText: () => require_tags_input_item_text.TagsInputItemText,
	Label: () => require_tags_input_label.TagsInputLabel,
	Root: () => require_tags_input_root.TagsInputRoot,
	RootProvider: () => require_tags_input_root_provider.TagsInputRootProvider
});
//#endregion
Object.defineProperty(exports, "tags_input_exports", {
	enumerable: true,
	get: function() {
		return tags_input_exports;
	}
});
