// @ark-ui/react/dist/components/tags-input/use-tags-input.js
"use client";
import { useEnvironmentContext } from "../../providers/environment/use-environment-context.js";
import { useLocaleContext } from "../../providers/locale/use-locale-context.js";
import { useFieldContext } from "../field/use-field-context.js";
import * as tagsInput from "@zag-js/tags-input";
import { useId } from "react";
import { normalizeProps, useMachine } from "@zag-js/react";
//#region src/components/tags-input/use-tags-input.ts
var useTagsInput = (props) => {
	const id = useId();
	const { getRootNode } = useEnvironmentContext();
	const { dir } = useLocaleContext();
	const field = useFieldContext();
	const machineProps = {
		id,
		ids: {
			label: field?.ids.label,
			hiddenInput: field?.ids.control
		},
		dir,
		disabled: field?.disabled,
		invalid: field?.invalid,
		readOnly: field?.readOnly,
		required: field?.required,
		getRootNode,
		...props
	};
	const service = useMachine(tagsInput.machine, machineProps);
	return tagsInput.connect(service, normalizeProps);
};
//#endregion
export { useTagsInput };
