// @chakra-ui/react/dist/esm/components/floating-panel/floating-panel.js
"use strict";
"use client";
import { jsx } from 'react/jsx-runtime';
import { FloatingPanel } from '@ark-ui/react/floating-panel';
import { createSlotRecipeContext } from '../../styled-system/create-slot-recipe-context.js';
import { For } from '../for/for.js';

const {
  withRootProvider,
  withContext,
  useStyles: useFloatingPanelStyles,
  PropsProvider
} = createSlotRecipeContext({ key: "floatingPanel" });
const FloatingPanelRootProvider = withRootProvider(
  FloatingPanel.RootProvider
);
const FloatingPanelRoot = withRootProvider(
  FloatingPanel.Root,
  {
    defaultProps: {
      lazyMount: true,
      unmountOnExit: true,
      minSize: { width: 240, height: 100 }
    }
  }
);
const FloatingPanelPropsProvider = PropsProvider;
const FloatingPanelTrigger = withContext(FloatingPanel.Trigger, "trigger", { forwardAsChild: true });
const FloatingPanelPositioner = withContext(FloatingPanel.Positioner, "positioner", { forwardAsChild: true });
const FloatingPanelContent = withContext(FloatingPanel.Content, "content", { forwardAsChild: true });
const FloatingPanelHeader = withContext(FloatingPanel.Header, "header", { forwardAsChild: true });
const FloatingPanelBody = withContext(FloatingPanel.Body, "body", { forwardAsChild: true });
const FloatingPanelTitle = withContext(FloatingPanel.Title, "title", { forwardAsChild: true });
const FloatingPanelDragTrigger = withContext(FloatingPanel.DragTrigger, "dragTrigger", { forwardAsChild: true });
const FloatingPanelResizeTrigger = withContext(FloatingPanel.ResizeTrigger, "resizeTrigger", { forwardAsChild: true });
const RESIZE_AXES = ["n", "s", "e", "w", "ne", "nw", "se", "sw"];
const FloatingPanelResizeTriggers = (props) => /* @__PURE__ */ jsx(For, { each: props.axes ?? RESIZE_AXES, children: (axis) => /* @__PURE__ */ jsx(FloatingPanelResizeTrigger, { ...props, axis }, axis) });
const FloatingPanelStageTrigger = withContext(FloatingPanel.StageTrigger, "stageTrigger", { forwardAsChild: true });
const FloatingPanelCloseTrigger = withContext(FloatingPanel.CloseTrigger, "closeTrigger", { forwardAsChild: true });
const FloatingPanelControl = withContext(FloatingPanel.Control, "control", { forwardAsChild: true });

export { FloatingPanelBody, FloatingPanelCloseTrigger, FloatingPanelContent, FloatingPanelControl, FloatingPanelDragTrigger, FloatingPanelHeader, FloatingPanelPositioner, FloatingPanelPropsProvider, FloatingPanelResizeTrigger, FloatingPanelResizeTriggers, FloatingPanelRoot, FloatingPanelRootProvider, FloatingPanelStageTrigger, FloatingPanelTitle, FloatingPanelTrigger, useFloatingPanelStyles };
