// @ark-ui/react/dist/components/number-input/number-input.js
import { __exportAll } from "../../_virtual/_rolldown/runtime.js";
import { NumberInputContext } from "./number-input-context.js";
import { NumberInputControl } from "./number-input-control.js";
import { NumberInputDecrementTrigger } from "./number-input-decrement-trigger.js";
import { NumberInputIncrementTrigger } from "./number-input-increment-trigger.js";
import { NumberInputInput } from "./number-input-input.js";
import { NumberInputLabel } from "./number-input-label.js";
import { NumberInputRoot } from "./number-input-root.js";
import { NumberInputRootProvider } from "./number-input-root-provider.js";
import { NumberInputScrubber } from "./number-input-scrubber.js";
import { NumberInputValueText } from "./number-input-value-text.js";
//#region src/components/number-input/number-input.ts
var number_input_exports = /* @__PURE__ */ __exportAll({
	Context: () => NumberInputContext,
	Control: () => NumberInputControl,
	DecrementTrigger: () => NumberInputDecrementTrigger,
	IncrementTrigger: () => NumberInputIncrementTrigger,
	Input: () => NumberInputInput,
	Label: () => NumberInputLabel,
	Root: () => NumberInputRoot,
	RootProvider: () => NumberInputRootProvider,
	Scrubber: () => NumberInputScrubber,
	ValueText: () => NumberInputValueText
});
//#endregion
export { number_input_exports };
